### Overview
This section details how the mitochondrial genomes were assembled.

### Required Input
This step requires the trimmed sequencing data, which was generated in the previous step (1.1c_read_trimming.md).

### Output
The output is an alignment of the complete mitochondrial genomes of each sample.

# Mitochondrial assembly

There are a few different programs to extract mtDNA sequences from shotgun sequencing, we will use [NOVOPlasty](https://github.com/ndierckx/NOVOPlasty).

## Read trimming
As input, NOVOplasty requires unfiltered reads with adapters removed.

Some of our libraries were sequenced in two lanes, so we have two sets of sequencing reads. NOVOPlasty (at time of writing) only accepts a single library (one R1 file (forward reads), and one R2 file (reverse reads)) so if there are multiple runs of the same sample, they will need to be combined. If the library is not a paired-end sequencing library, there are different options for providing the data to NOVOPlasty.

Step 1) If there are multiple runs of the same sample, combine the runs into the same file so that all read 1 and all read 2 files are together. Make sure that you concatenate the read 1 and read 2 files in the same order, since NOVOPlasty will assume that the order of the paired reads is the same between the two files. If you have a lot of data, you might not need all of it, but we will use all of our data since we have enough computational resources.
```bash
#for example:
#cd into the folder containing your data

#concatenate the R1 files together. I had two files: "$prefix"_R1.fastq.gz and "$prefix2"_R1.fastq.gz. Note that you can just use cat to concatenate zipped fastq.gz files.
cat "$prefix"_R1.fastq.gz "$prefix2"_R1.fastq.gz > "$sample"_R1.fastq.gz

#repeat to concatenate the R2 files together.
cat "$prefix"_R2.fastq.gz "$prefix2"_R2.fastq.gz > "$sample"_R2.fastq.gz
```

Step 2)
Once you have your two files (read1 and read2), trim them with Trimmomatic to get rid of the adapter sequences.   
Only include the adapter trimming step (`ILLUMINACLIP:"./extras/$prefix"\_adapters.fa:2:30:7:5:true`) and no other quality trimming.  NOVOPlasty documentation insists on not quality trimming the reads prior to using NOVOPlasty.  

Provide Trimmomatic the path to your R1 and R2 sequence reads, the output names for the files you want to produce (read 1 and read 2, paired and unpaired, and the path to the file containing the correct adapter sequences (for my data, that is the `TruSeq3-PE-2.fa` file that comes with Trimmomatic))  

We already trimmed the reads with trimmomatic (see step 1.1), they are stored in `/home/0_GENOMES1/0_RESEQUENCING_PROJECTS/1_PROCESSED_READS/3_TRIMMED/Stercorariidae/unfltrd_adaptrlss"$sample"_R1.fastq.gz` and `/home/0_GENOMES1/0_RESEQUENCING_PROJECTS/1_PROCESSED_READS/3_TRIMMED/Stercorariidae/unfltrd_adaptrlss"$sample"_R2.fastq.gz`  

## Find a seed
NOVOPlasty requires a seed sequence to start the mtDNA assembly. Ideally it could be a mitochondrial sequence from a very closely related organism, but if that is not available you can use a sequence from a more distant organism. In birds, it may be important to avoid using the control region as the seed, as this region is most difficult to assemble and is duplicated in many species - however, I found that I needed to repeat the assembly process using a seed from breakpoints of the duplicated control region in order to force NOVOPlasty to correctly assemble this region.  

I initially used a full mtDNA genome (FASTA format) of the closest relative from [Genbank nuccore](https://www.ncbi.nlm.nih.gov/nuccore/KM401546.1). This could be the seed from which the program begins the assembly. I placed that downloaded fasta file in `/home/0_GENOMES1/0_mtDNA_GENOMES/Stercorariidae/Stercorarius_maccormicki_mtDNA_KM401546.fasta`. Later, I decided to redo the assemblies after the release of a newer version of NOVOPlasty - when I redid them, I decided to use just the ND1 sequence and it worked just as well. (I chose ND1 to avoid starting the seed near the dulicated control region, as by then I had realized that the control region was duplicated in the Stercorariidae but the duplication was not assembled in the Genbank sequence). In the next batch of sequences, some samples failed to assemble both duplicates, so I needed to repeat NOVOPlasty using a seed from the missing region (see "Troubleshooting" `.md` files in the 1.2_mitogenome_assembly folder for that process).  

```bash
#make a fasta file with the seed sequence
cat > ND1.fa #copy-paste the file contents, then press ctrl-d to finish
```
```
>ANSK_NC_026125_1
ATGACCCACCTCCCTACCCTAGCCTATCTTATCATGTCCCTATCCTATGTAATCCCAATCCTAATTGCCGTAGCATTTCTAACCCTAGTAGAACGAAAAATCTTAAGTTACATGCAAGCCCGAAAAGGTCCAAACATCGTAGGGCCTTTCGGACTATTGCAACCTGTGGCCGATGGAATTAAACTATTCACCAAAGAACCAATCCGCCCATCTACCTCTTCCCCGTTCCTATTCCTCATAACCCCTATGCTAGCTCTTCTTTTAGCACTTACCATTTGAATTCCCCTCCCCCTCCCCTTCTCTCTCACTGACCTTAACTTAGGCCTCCTTTTCCTCCTTGCCATGTCTAGCCTAGCAGTTTACTCAATTTTGTGATCAGGTTGGGCCTCAAACTCAAAGTACGCTCTAATCGGAGCCCTACGGGCAGTAGCACAGACTATTTCCTACGAAGTAACACTAGCTATCATCCTCTTATCCGTAATCATACTTAGCGGGAACTATACCCTAAACACCCTCGCCATCACACAAGAGCCATTATACCTTGTTTTCTCCTCATGACCCCTTGCAATAATGTGATATATTTCGACACTCGCTGAAACAAATCGTGCCCCATTCGACCTCACAGAAGGGGAATCAGAACTAGTGTCGGGCTTTAACGTAGAATACGCTGCTGGACCATTCGCCCTGTTCTTTCTAGCTGAATACGCGAATATCATACTAATAAACACACTAACAGCAATCCTGTTCCTAAACCCAAGCTCACTAAATCTCTCCCAAGAATTATTTCCCATGGTTCTAGCCACAAAAGTCCTATTCCTCTCTTCAGGATTCCTATGAATCCGTGCCTCCTACCCACGATTCCGCTATGATCAACTCATGCACCTGCTCTGAAAAAACTTCCTACCACTAACATTAGCACTGTGCCTCTGACATATCAGTCTACCAATCTGCTACGCAGGCCTCCCTCCTTACTTAAGG
```

## Make config file

NOVOPlasty needs a config file listing various parameters. You will need to edit the location of your reference seed (.fasta) and the adapter-trimmed read 1 and read 2 files. There is a blank template in the NOVOPlasty folder that was downloaded.  
Copy this into your working directory and edit it, or copy it from this repository.  

In my case, I want to run 8 samples. It is possible to create a batch file which will allow all 8 to be run in a single batch. There are instructions [here](https://github.com/ndierckx/NOVOPlasty/wiki/Batch-function). I am not going to run it in a batch since that would run it in serial, and I had strange results when I tried using it (with an older version of NOVOPlasty). I would rather run them myself.  

```bash
cat > ./NOVOPlasty_Stercorariidae.config.txt #copy-paste the file contents, then press ctrl-d to finish
```
```
Project:
-----------------------
Project name          = GRSK_MKP1592
Type                  = mito
Genome Range          = 15000-20000
K-mer                 = 49
Max memory            = 
Extended log          = 0
Save assembled reads  = yes
Seed Input            = ND1.fa
Reference sequence    = 
Variance detection    = 
Chloroplast sequence  = 

Dataset 1:
-----------------------
Read Length           = 151
Insert size           = 250
Platform              = illumina
Single/Paired         = PE
Combined reads        = 
Forward reads         = unfltrd_adaptrlssGRSK_MKP1592_R1.fastq.gz
Reverse reads         = unfltrd_adaptrlssGRSK_MKP1592_R2.fastq.gz

Heteroplasmy:
-----------------------
MAF                   = 
HP exclude list       = 
PCR-free              = 

Optional:
-----------------------
Insert size auto      = yes
Insert Range          = 1.9
Insert Range strict   = 1.3
Use Quality Scores    = no
```
I decided to run these on Niagara, so for that I needed a batch file.  
Here is the batch file:  

```bash
cat > novoplasty.sh #copy-paste the file contents, then press ctrl-d to finish

chmod +x novoplasty.sh
sbatch novoplasty.sh
```
```bash
#!/bin/bash
#SBATCH --nodes=1 #
#SBATCH --ntasks-per-node=1#
#SBATCH --cpus-per-task=40
#SBATCH --mem=0 #set to 0 to use all available memory of each requested node. Advantage, any of the ntasks of each node can use any amount of the memory provided to sum does not exceed thata available to the node
#SBATCH --time=3:30:00
#SBATCH --job-name NOVOPlasty_GRSK_MKP1592
#SBATCH --output=NOVOPLASTY_GRSK_MKP1592

cd /scratch/j/jweir/elsemikk
time perl ./NOVOPlasty/NOVOPlasty4.2.1.pl -c ./NOVOPlasty_Stercorariidae.config.txt
```

## run the next samples

The things that need to change between runs are the name of the sample (and the name of the seed sequence if you are using a different file as the seed). Note that as it is written, NOVOPlasty will look for the sequence files in the current working directory - provide the full path if they are in another folder. I am actually running this on Niagara so I have all my files in the same scratch folder and do not need to change the paths.  

```bash
#change the sample names in the config and batch file, then submit the job
sed -i 's/GRSK_MKP1592/POJA_MKP1559/g' NOVOPlasty_Stercorariidae.config.txt
sed -i 's/GRSK_MKP1592/POJA_MKP1559/g' novoplasty.sh
sbatch novoplasty.sh

#repeat, making new config files for each sample so they can be submitted at the same time
sed 's/POJA_MKP1559/PAJA_B20730/g' NOVOPlasty_Stercorariidae.config.txt > NOVOPlasty_PAJA.config.txt
sed 's/POJA_MKP1559/LTJA_MKP990/g' NOVOPlasty_Stercorariidae.config.txt > NOVOPlasty_LTJA.config.txt
sed 's/POJA_MKP1559/CHSK_MKP2451/g' NOVOPlasty_Stercorariidae.config.txt > NOVOPlasty_CHSK.config.txt
sed 's/POJA_MKP1559/CISK_32/g' NOVOPlasty_Stercorariidae.config.txt > NOVOPlasty_CISK.config.txt
sed 's/POJA_MKP1559/ANSK7/g' NOVOPlasty_Stercorariidae.config.txt > NOVOPlasty_ANSK.config.txt

sed 's/POJA_MKP1559/POJA_4/g; s/POJA_4_R/POJA_4_S10_L001_R/g' NOVOPlasty_Stercorariidae.config.txt > NOVOPlasty_POJA_4.config.txt
sed 's/POJA_MKP1559/POJA_IB2659/g; s/POJA_IB2659_R/POJA_IB2659_S9_L001_R/g' NOVOPlasty_Stercorariidae.config.txt > NOVOPlasty_POJA_IB2659.config.txt
sed 's/POJA_MKP1559/GRSK_MKP1593/g; s/GRSK_MKP1593_R/GRSK_MKP1593_S4_L001_R/g' NOVOPlasty_Stercorariidae.config.txt > NOVOPlasty_GRSK_MKP1593.config.txt
sed 's/POJA_MKP1559/ANSK01/g; s/ANSK01_R/ANSK01_S7_L001_R/g' NOVOPlasty_Stercorariidae.config.txt > NOVOPlasty_ANSK01.config.txt
sed 's/POJA_MKP1559/ANSK8/g; s/ANSK8_R/ANSK8_S8_L001_R/g' NOVOPlasty_Stercorariidae.config.txt > NOVOPlasty_ANSK8.config.txt
sed 's/POJA_MKP1559/CISK2/g; s/CISK2_R/CISK2_S6_L001_R/g' NOVOPlasty_Stercorariidae.config.txt > NOVOPlasty_CISK2.config.txt
sed 's/POJA_MKP1559/CISK55/g; s/CISK55_R/CISK55_S5_L001_R/g' NOVOPlasty_Stercorariidae.config.txt > NOVOPlasty_CISK55.config.txt
sed 's/POJA_MKP1559/PAJA_USNM606730/g; s/PAJA_USNM606730_R/SRR10019934_R/g' NOVOPlasty_Stercorariidae.config.txt > NOVOPlasty_PAJA_USNM606730.config.txt
```

Then, make batch files for each one and submit them.

I tried runing all at once, but that ran out of memory and crashed.
```bash
cat > novoplasty2.sh
chmod +x novoplasty2.sh
sbatch novoplasty2.sh #failed
```
```bash
#!/bin/bash
#SBATCH --nodes=1 #
#SBATCH --ntasks-per-node=1#
#SBATCH --cpus-per-task=40
#SBATCH --mem=0 #set to 0 to use all available memory of each requested node. Advatage, any of the ntasks of each node can use any amount of the memory provided to sum does not exceed thata available to the node
#SBATCH --time=3:30:00
#SBATCH --job-name NOVOPlasty_5samples
#SBATCH --output=NOVOPlasty_5samples

cd /scratch/j/jweir/elsemikk
time perl ./NOVOPlasty/NOVOPlasty4.2.1.pl -c ./NOVOPlasty_PAJA.config.txt &
time perl ./NOVOPlasty/NOVOPlasty4.2.1.pl -c ./NOVOPlasty_LTJA.config.txt &
time perl ./NOVOPlasty/NOVOPlasty4.2.1.pl -c ./NOVOPlasty_CHSK.config.txt &
time perl ./NOVOPlasty/NOVOPlasty4.2.1.pl -c ./NOVOPlasty_CISK.config.txt &
time perl ./NOVOPlasty/NOVOPlasty4.2.1.pl -c ./NOVOPlasty_ANSK.config.txt
```
So, I ran them individually:

```bash
cat > novoplasty3.sh
chmod +x novoplasty3.sh
sbatch novoplasty3.sh
```
```bash
#!/bin/bash
#SBATCH --nodes=1 #
#SBATCH --ntasks-per-node=1#
#SBATCH --cpus-per-task=40
#SBATCH --mem=0 #set to 0 to use all available memory of each requested node. Advatage, any of the ntasks of each node can use any amount of the memory provided to sum does not exceed thata available to the node
#SBATCH --time=3:30:00
#SBATCH --job-name NOVOPlasty_PAJA
#SBATCH --output=NOVOPlasty_PAJA

cd /scratch/j/jweir/elsemikk
time perl ./NOVOPlasty/NOVOPlasty4.2.1.pl -c ./NOVOPlasty_PAJA.config.txt

```
```bash
cat > novoplasty4.sh
chmod +x novoplasty4.sh
sbatch novoplasty4.sh
```
```bash
#!/bin/bash
#SBATCH --nodes=1 #
#SBATCH --ntasks-per-node=1#
#SBATCH --cpus-per-task=40
#SBATCH --mem=0 #set to 0 to use all available memory of each requested node. Advatage, any of the ntasks of each node can use any amount of the memory provided to sum does not exceed thata available to the node
#SBATCH --time=3:30:00
#SBATCH --job-name NOVOPlasty_LTJA
#SBATCH --output=NOVOPlasty_LTJA

cd /scratch/j/jweir/elsemikk
time perl ./NOVOPlasty/NOVOPlasty4.2.1.pl -c ./NOVOPlasty_LTJA.config.txt
```
```bash
cat > novoplasty5.sh
chmod +x novoplasty5.sh
sbatch novoplasty5.sh
```
```bash
#!/bin/bash
#SBATCH --nodes=1 #
#SBATCH --ntasks-per-node=1#
#SBATCH --cpus-per-task=40
#SBATCH --mem=0 #set to 0 to use all available memory of each requested node. Advatage, any of the ntasks of each node can use any amount of the memory provided to sum does not exceed thata available to the node
#SBATCH --time=3:30:00
#SBATCH --job-name NOVOPlasty_CHSK
#SBATCH --output=NOVOPlasty_CHSK

cd /scratch/j/jweir/elsemikk
time perl ./NOVOPlasty/NOVOPlasty4.2.1.pl -c ./NOVOPlasty_CHSK.config.txt
```
```bash
cat > novoplasty6.sh
chmod +x novoplasty6.sh
sbatch novoplasty6.sh
```
```bash
#!/bin/bash
#SBATCH --nodes=1 #
#SBATCH --ntasks-per-node=1#
#SBATCH --cpus-per-task=40
#SBATCH --mem=0 #set to 0 to use all available memory of each requested node. Advatage, any of the ntasks of each node can use any amount of the memory provided to sum does not exceed thata available to the node
#SBATCH --time=3:30:00
#SBATCH --job-name NOVOPlasty_CISK
#SBATCH --output=NOVOPlasty_CISK

cd /scratch/j/jweir/elsemikk
time perl ./NOVOPlasty/NOVOPlasty4.2.1.pl -c ./NOVOPlasty_CISK.config.txt
```
and then repeated for the remainders:
```bash
sed 's/POJA_MKP1559/ANSK01/g; s/Stercorariidae.config.txt/ANSK01.config.txt/g' novoplasty.sh > NOVOPlasty_ANSK01.sh
sed 's/POJA_MKP1559/ANSK8/g; s/Stercorariidae.config.txt/ANSK8.config.txt/g' novoplasty.sh > NOVOPlasty_ANSK8.sh
sed 's/POJA_MKP1559/CISK2/g; s/Stercorariidae.config.txt/CISK2.config.txt/g' novoplasty.sh > NOVOPlasty_CISK2.sh
sed 's/POJA_MKP1559/CISK55/g; s/Stercorariidae.config.txt/CISK55.config.txt/g' novoplasty.sh > NOVOPlasty_CISK55.sh
sed 's/POJA_MKP1559/GRSK_MKP1593/g; s/Stercorariidae.config.txt/GRSK_MKP1593.config.txt/g' novoplasty.sh > NOVOPlasty_GRSK_MKP1593.sh
sed 's/POJA_MKP1559/POJA_4/g; s/Stercorariidae.config.txt/POJA_4.config.txt/g' novoplasty.sh > NOVOPlasty_POJA_4.sh
sed 's/POJA_MKP1559/POJA_IB2659/g; s/Stercorariidae.config.txt/POJA_IB2659.config.txt/g' novoplasty.sh > NOVOPlasty_POJA_IB2659.sh
sed 's/POJA_MKP1559/PAJA_USNM606730/g; s/Stercorariidae.config.txt/PAJA_USNM606730.config.txt/g' novoplasty.sh > NOVOPlasty_PAJA_USNM606730.sh

sbatch NOVOPlasty_ANSK01.sh
sbatch NOVOPlasty_ANSK8.sh
sbatch NOVOPlasty_CISK2.sh
sbatch NOVOPlasty_CISK55.sh
sbatch NOVOPlasty_GRSK_MKP1593.sh
sbatch NOVOPlasty_POJA_4.sh
sbatch NOVOPlasty_POJA_IB2659.sh 
sbatch NOVOPlasty_PAJA_USNM606730.sh
```

### Notes for running on the server
Note: it might seem to be a bit painful to do each sample one-at-a-time on only one thread each, but I have found that a single sample consumes over 100 Gb of memory so unless you have a *lot* of memory I do not think it would be good to do multiple samples in parallel! Even trying to do a single sample on the lab node, I noticed it was still running after 6 hours on the first sample, but had consumed all 94Gb of memory and the 7Gb of swap memory and was limping along with 2%CPU. On the main server, it took 104 mins to do LTJA, 88 mins for GRSK, 111 mins for PAJA (all in serial, 1 thread each). On Niagara, do not run more than 2 on the same node or it will run out of memory.  

After getting your assembly, compress (with `gzip`) (or delete) your input sequence file (assuming you still have the original) if they are not already compressed, because they are very large! I am keeping mine as a fastq.gz file but I will probably delete them as soon as I am done the next heteroplasmy steps.

## Clean the sequences, stitch scaffolds, and align

This is one of my favourite steps, so I decided to do it in a graphical user interface so that I could look at the sequences.  

When you get back the sequences, it may be in a single or multiple contigs. If it is multiple contigs, you will want to merge these, which can be done manually. This requires some prior knowledge about the layout of the mitogenome in your taxa. Each sample will also need to be aligned, and since the fasta sequences might start at different places in the circular sequence you will need to edit them to start at the same spot.  

1) Identify the complete circularized samples (CISK3 and ANSK7 in my case) and the previously-published references (ANSK_NC_026125_1 in my case, although this turned out to actually be incomplete). If you do not have any complete references, it will be a bit more tricky. Also be aware that rearrangements or duplicated regions are possible, especially if your reference is quite distant.  

2) Combine your complete and incomplete mtDNA contigs into a single fasta file. For example:  
```bash
cat Contigs_*.fasta > mitogenomes.fasta
```
Note that the sample names are not in the fasta headers, you have to keep track of this - probably should rename the fasta headers to avoid mistakes.  

2.2) Open the .fasta file in [Mesquite](https://www.mesquiteproject.org/) and save as a new nexus project.  

3) Align all of the sequences using MAFFT (Matrix -> Align multiple sequences -> MAFFT align. Default parameters)  

4) Examine the alignment, and pick a place to be the start and end of the linearized mtDNA circle. I chose the start of the previously published *S. maccormicki*  mitogenome to be the arbitrary "start" of all of them. Cut the beginnings of the entire alignment up to that point, and paste it at the end of the alignment. The mitochondrial genome is a circle so this is ok to do only if your contigs span the whole circle! Otherwise, split the sequence into two separate "taxa" at the chosen breakpoint  

5) Align again using MAFFT  

6) Mannually edit the alignment if needed to correct mistakes made by MAFFT (I found I did not usually need to do this, but sometimes you get an incorrect alignment- if you have duplications in the mtDNA, mafft may align a fragment to the incorrect copy)  

7) If NOVOPlasty provided multiple possible assemblies, compare the differences between each assembly to the other, fully-circularized samples. Hopefully, one of the possibilities will perfectly match the arrangement of the circularized samples and can be selected as the best assembly. For some samples, NOVOPlasty provided two options, that differred in which copy of the duplicated region came first. I selected the arrangements that matched the fully circularized samples. In this case, I could not use a published reference to decide because the published reference contained only a single collapsed copy of the duplication. Using the completed mitogenomes as a scaffold, I was able to identify the overlaps between adjacent contigs and stitch them together into a single complete sequence. Beware that with a duplication, NOVOPlasty may build some chimeric contigs with sequences in the wrong order. Be also aware that numts can be assembled by NOVOPlasty.  

Then, after manually putting together and verifying that all mitogenomes are complete, and exported them as a fasta, with the sample names as fasta headers, I removed the gaps with find+replace "-" (in Textedit because it is on my main computer now). Then I can split them into separate fasta like this:  

```bash
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/1.2_mitogenome_assembly
mkdir drafts #I placed Stercorariidae_draft_mitogenomes.fasta here
cd drafts
/home/0_PROGRAMS/gt-1.5.10-Linux_x86_64-64bit-barebone/bin/gt splitfasta -splitdesc . Stercorariidae_Mitogenomes_Draft.fasta
cat samples.txt | while read sample ; do mv "$sample".fasta "$sample"_mitogenome_draft.fasta ; done #rename to be more meaningful
```
The name of the file will be the name of the fasta header. For me that is the sample names (I did by hand in Mesquite)  
Now we are done the manual editing and graphical interface parts and back to the command line.  

### Confirming the duplicated control region  

It appeared that the Stercorariids had a region of the mitogenome duplicated. To confirm that the duplication was real, I created a copy of the mitogenome that had one copy of the putative duplication removed, and then mapped the sequences of all samples to this mitogenome.  

Here are the possible scenarios to differentiate:  
1) No duplication, just an assembly error: would not expect substantially higher sequence depth in the putative duplicated region  
2) Duplication in the mitogenome: would expect roughly doubled sequencing depth in the duplicated region (when mapping to the duplication as a single collapsed copy)  
3) numt: would expect slightly higher sequencing depth in this region, but this will be scarcely detectable above random noise in the samples that are sequenced from muscle (in which mtDNA is sequenced at much higher depth than nuclear). It may be more detectable in the blood samples where mtDNA and nuDNA were sequenced to similar depths.  

The combination of numts in the Stercorariid nuclear genome and the duplication in their mitogenome made mitogenome assembly difficult and prone to error. To verify the sequences, I will map the raw reads back to the mitogenome and ensure that the most frequent base is indeed the base in the assembly. Note that since the mitogenomes of most samples were sequenced to thousands of X depth and the nuclear genome was sequenced to less than 20x depth, the most frequent base should essentially always be the mitochondrial allele in this case, at least for the muscle samples.  

```bash
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/1.2_mitogenome_assembly/drafts

#index the mitogenome sequnces
cat samples.txt | time parallel '/home/0_PROGRAMS/bwa/bwa index {1}_mitogenome_draft.fasta' #takes 1 second

#map reads to the mitogenomes
mkdir -p mapping_logs
cat samples.txt | time parallel  'time /home/0_PROGRAMS/bwa/bwa mem ./{1}_mitogenome_draft.fasta /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/1_PROCESSED_READS/3_TRIMMED/Stercorariidae/fltrd_adaptrlss*{1}*_R1.fastq.gz /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/1_PROCESSED_READS/3_TRIMMED/Stercorariidae/fltrd_adaptrlss*{1}*_R2.fastq.gz 2> mapping_logs/{1}_mtDNA_bwa.log | time /home/0_PROGRAMS/samtools-1.14/samtools view -S -b -@ 4 -h > {1}_mtDNA_bwa.bam' #fastq.gz files are stored on the main server
#for PAJA
cat samples.txt | time parallel  'time /home/0_PROGRAMS/bwa/bwa mem ./{1}_mitogenome_draft.fasta /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/1_PROCESSED_READS/3_TRIMMED/Stercorariidae/fltrd_adaptrlss*{1}*_R1.fastq.gz 2> mapping_logs/{1}_mtDNA_bwa.log | time /home/0_PROGRAMS/samtools-1.14/samtools view -S -b -@ 4 -h > {1}_mtDNA_bwa.bam' #fastq.gz files are stored on the main server

#summarize sequencing depth
#cat samples.txt | parallel '/home/0_PROGRAMS/bamtools/bin/bamtools stats -in {1}_mtDNA_bwa.bam > {1}_mtDNA_bwa.bamstats.txt'

#sort and index
cat samples.txt | parallel '/home/0_PROGRAMS/samtools-1.14/samtools sort --threads=4 -o {1}_mtDNA_bwa_sorted.bam {1}_mtDNA_bwa.bam'
cat samples.txt | parallel 'time /home/0_PROGRAMS/samtools-1.14/samtools index {1}_mtDNA_bwa_sorted.bam'

#make a list of the samples, then use Angsd to summarize allele counts at each site
cat samples.txt | while read sample ; do echo "$sample"_mtDNA_bwa_sorted.bam > "$sample".filelist ; done
cat samples.txt | parallel 'time /home/0_PROGRAMS/angsd/angsd -doCounts 1 -dumpCounts 3 -out {1}_angsdcount -minQ 20 -doDepth 1 -bam {1}.filelist'
cat samples.txt | parallel 'gunzip {1}_angsdcount*'
cat samples.txt | while read sample ; do paste "$sample"_angsdcount.pos "$sample"_angsdcount.counts > "$sample"_mtDNA.basecounts ; done

#scroll through the file and evaluate evidence for the duplication and for numts

#look at a couple particular sites of interest
grep "5853" *_mtDNA.basecounts
grep "5852" *_mtDNA.basecounts
grep "5871" *_mtDNA.basecounts
grep "5850" *_mtDNA.basecounts

#looks like the numt is in at least pomarinus, skua, chilensis, parasiticus, and longicaudus

#calculate depth in duplication for POJA to compare to average depth
cat POJA*basecounts | tail -n +8861 | head -n 2000 | cut -f 3 | awk '{x+=$1; next} END{print x/NR}'
```

Now I will create a consensus from the basecounts in order to compare it to the draft and see if there are any problematic sites that differ.
```bash
#sinplify basecounts file to match format needed by python script (only 4 columns, header is just the base)
cat samples.txt | parallel 'printf "A\tC\tG\tT\n" > {1}_mtDNA.simplified.basecounts'
cat samples.txt | parallel 'cut -f 4-7 {1}_mtDNA.basecounts | tail -n +2 >> {1}_mtDNA.simplified.basecounts'

cat samples.txt | parallel 'python basecounts2consensus.py --file {1}_mtDNA.simplified.basecounts | sed "s/{.*$/>{1}_consensus/g" > {1}_mtDNA.cons.fa'
```
This is a python script written by Uriel Garcilazo (https://github.com/UGarCil/MapConsensusSequences/blob/main/main.py) to process the basecounts file:
```python
# Program created on 1/22/2022 by Uriel GC
# Takes a list of items where last 4 columns are int, returning a sequence of G,C,T,A

# Modules
import argparse


# Func. def. mapSeq()
# Signature: str -> str
# interp. map columns with frequencies from 
# nucleotides into letters based on index of their headers
def mapSeq(file):
    with open(file, 'r') as file:
        file = file.readlines()
        headers = file[0].strip().split('\t')[-4:]
        content = [x.strip().split('\t') for x in file[1:]]
        content = [x[-4:] for x in content]
        content = [[int(cell) for cell in row] for row in content]
            
                
    sequence = ''
    for row in content:
        # print(row)
        index = row.index(max(row))
        sequence += headers[index]

    print(sequence)
        

if __name__ == "__main__":
    # construct the argument parse and parse the arguments
    ap = argparse.ArgumentParser()
    ap.add_argument("-f", "--file", required=True,
        help="file to process")
    args = vars(ap.parse_args())
    print(args)
    mapSeq(args["file"])
```
Unfortunately, the blood-derived samples could not be analyzed in this way because the mitogenome was sequenced to low depth comparable to the nuclear genome (as there are few mtDNA in blood cells, as oposed to the thousands in muscle cells). This excluded the 3 CISK and 3 ANSK samples. Luckily, the skuas do not seem to share the large numt found in the PAJA/LTJA that caused such difficulty with assembly.

The following samples look the same as their consensus:
* GRSK_MKP1592
* POJA_MKP1559
* POJA_4
* CHSK_MKP2451

The following samples have sequence that differ from consensus:
* GRSK_MKP1593: draft has a Y at pos 10741, 10751, 10800, and 13102, where consensus has C. There are sites within the duplication, where one duplicate has C and the other has T. It is hard to know which copy of the duplicate has which allele. These sites do not seem to be variant in the POJA samples, so we can't use them as a scaffold to decide which copy has which allele, and the variant sites are seperated by a wide stretch of invariant sites, so the reads themselves do not give us any information about which copy belongs to which side (although we could try to use insert size information). Rather than arbitrarily guess, I will leave the ambiguous sites as "Y" even though we know that one side has C and the other has T.
* POJA_IB2659 similarly shows a pair of two variants that can't be phased between the two copies of the duplicate.
* PAJA_USNM606730 had a Y where 1963 aligned reads had T and only 1 aligned read had A and none had C (however, I did not align the entire dataset of reads, only about one quarter). At the homologous site in the other copy of the duplicate, it is also a Y, and both C's and T's align. I will keep both ambiguous.

chr     pos     totDepth        totA    totC    totG    totT
GRSK_MKP1593    10741   10008   0       5689    4       4315
GRSK_MKP1593    10751   9747    2       5537    0       4208
GRSK_MKP1593    10800   9438    0       5083    0       4355
GRSK_MKP1593    13043   10758   2       4766    1       5989
GRSK_MKP1593    13053   11235   1       4973    0       6261
GRSK_MKP1593    13102   10794   1       6409    0       4384

POJA_IB2659     8996    1277    447     830     0       0
POJA_IB2659     10162   521     80      0       441     0
POJA_IB2659     11328   612     123     489     0       0
POJA_IB2659     12494   500     376     4       120     0

PAJA_USNM606730 11669   1964    1       0       0       1963
PAJA_USNM606730	9381	2492	0		541		0		1951

The following samples look like they had a numt in the draft:
* LTJA_MKP990 shows many variants from pos 16386 to pos 17273. eg) at site 14240, the draft has a G, but there are 7761 reads with an A and only 14 with a G - the G is almost certainly from a numt. It looks like the assembler chose the numt reads in this region, so I will replace those SNPs with the mtDNA version (ie, choose the alleles that were sequenced to 17000x coverage instead of 14x coverage). As a further note, the numt sequence appears to be much more similar to the skua sequences by retaining ancestral alleles (the nuclear genome evolves much more slowly than bird mtDNA).
* PAJA_B20730 shows many variants from pos 16282-17378. These appear to be a numt homologous to that in LTJA, as they share some alleles.
Here are the numt fragments:
```
>LTJA_numt_16386-17273
-CATATATATGACCCACCTCCCTACCCTAGCCTATCTTATCATGTCCCTATCCTATGTAATCCCAATCCTAATTGCCGTAGCATTTCTGACCCTAGTAGAACGAAAAATCTTAAGTTACATGCAAGCCCGAAAAGGTCCAAACATCGTAGGGCCTTTCGGACTATTGCAACCTGTGGCCGATGGAATTAAACTATTCACCAAAGAACCAATCCGCCCATCTACCTCTTCCCCATTCCTATTCCTCATAACCCCTATMCTAGCTCTTCTTTTAGCACTTACCATTTGAATTCCCCTCCCCCTCCCCTTCTCTYTCMCTGACCTCAACTTAGGCCTCCTTTTCCTCCTTGCCATGTCTAGCCTAGCAGTTTACTCAATTTTGTGATCAGGTTGGGCCTCAAACTCAAAGTACGCTCTAATCGGAGCCCTGCGGGCAGTAGCACARACTATTTCCTACGAAGTAACATTAGCTATCATCCTCTTATCCGTAATCATAYTTAGCGGGAACTACACCCTAAACACCCTCGCCATCACACAAGARCCATTATACCTTATTTTCTCCTCATGGCCCCTTGCAATAATATGATAYATTTCGACACTCGCTGAAACAAATCGTGCCCCATTCGACCTCACAGAAGGGGAATCAGAACTAGTGTCGGGCTTTAACGTAGAATATGCTGCCGGACCATTCGCCCTGTTCTTTCTAGCTGAATACGCGAATATCATACTAATAAACACACTAACARYAATCCTGTTCCTAAACCCAAGCTCACTAAATCTCTCCCAAGAATTATTTCCCATGGTTCTAGCCACAAAAGTCCTATTCCTCTCTTCAGGATTCCTATGAATCCGTGCCTCCTATCCACGATTCCGCTATGATCAACTCATGCACC
>LTJA_mtDNA_16386-17273
TCACTTACATGACTCACCTCCCTGCCCTGACCTACCTTGTCATATCCTTGTCCTATGCAGTCCCAATCCTAGTTGCCGTAGCATTTCTAACCCTAGTAGAACGAAAAATCTTAAGCTACATACAAGCTCGAAAAGGCCCAAACATCGTAGGACCTTTCGGACTATTACAACCTGTAGCCGATGGAATTAAACTATTCACTAAAGAACCAATCCGTCCATCTACCTCTTCTCCATTCCTATTCCTCATAACCCCCATGCTAGCCCTTCTTTTGGCACTCACCATTTGAATTCCCCTCCCCCTCCCCTTCTCCCTCACCGACCTTAACTTAGGTCTCCTTTTCCTTCTAGCCATGTCCAGCCTAGCAGTTTATTCAATTTTATGATCAGGATGGGCCTCAAACTCAAAGTATGCCCTAATTGGAGCCTTACGGGCAGTAGCACAGACTATCTCTTATGAAGTAACACTAGCTATCATCCTCCTATCCGTAATCATATTCAGCGGGAACTACACCCTAAACACTCTCGCTACCACACAAGAACCATTATACCTTGTCTTCTCCTCATGACCCCTTGCAATAATATGATATATTTCGACACTCGCTGAAACAAACCGTGCCCCATTCGACCTCACAGAAGGAGAGTCAGAACTAGTATCAGGCTTCAACGTAGAGTATGCTGCTGGACCATTTGCCCTATTCTTCCTAGCTGAATACGCAAATATCATACTGATAAACACACTCACAGCAATCCTGTTCCTAAACCCAAGCTCATTGAATCTCCCACAAGAATTATTCCCCATACTCCTAGCCGCAAAAGTCCTATTTCTCTCTTCAGGTTTTTTATGAATCCGTGCCTCCTATCCACGATTCCGCTATGACCAACTCATGCACT
>PAJA_B20730_numt_16282-17378
TGGTAAATGCAAAAGGCTTAAGCCCTTTACCCAGAGGTTCAAATCCTCTCCCTAGCCCCATATATATGACCCACCTCCCTACCCTAGCCTATCTTATCATGTCCCTATCCTATGTAATCCCAATCCTAATTGCCGTAGCATTTCTGACCCTAGTAGAACGAAAAATCTTAAGTTACATGCAAGCCCGAAAAGGTCCAAACATCGTAGGGCCTTTCGGACTATTGCAACCTGTGGCCGATGGAATTAAACTATTCACCAAAGAACCAATCCGCCCATCTACCTCTTCCCCATTCCTATTCCTCATAACCCCTATACTAGCTCTTCTTTTAGCACTTACCATTTGAATTCCCCTCCCCCTCCCCTTCTCTCTCACTGACCTCAACTTAGGCCTCCTTTTCCTCCTTGCCATGTCTAGCCTAGCAGTTTACTCAATTTTGTGATCAGGTTGGGCCTCAAACTCAAAGTACGCTCTAATCGGAGCCCTGCGGGCAGTAGCACARACTATTTCCTACGAAGTAACATTAGCTATCATCCTCTTATCCGTAATCATAYTTAGCGGGAACTACACCCTAAACACCCTCGCCATCACACAAGARCCATTATACCTTATTTTCTCCTCATGGCCCCTTGCAATAATATGATACATTTCGACACTCGCTGAAACAAATCGTGCCCCATTCGACCTCACAGAAGGGGAGTCAGAACTAGTGTCGGGCTTTAACGTAGAATATGCTGCTGGACCATTCGCCCTGTTCTTTCTAGCTGAATACGCGAATATCATACTAATAAACACACTAACAGCAATCCTGTTCCTAAACCCAAGCTCACTAAATCTCTCCCAAGAATTATTTCCCATGGTTCTAGCCACAAAAGTCCTATTCCTCTCTTCAGGATTCCTATGAATCCGTGCCTCCTATCCACGATTCCGCTATGATCAACTCATGCACCTGCTCTGAAAAAACTTCCTACCACTAACATTAGCACTGTGCCTCTGACATATCAGTCTACCAATCTGCTACGCAGGCCTCCCTCCTTACTTAAGGAAATGTGCCTGAACGTAAAGGGTCACTATGATAAAGTGAACATAGAGGTATACC
>PAJA_B20730_mtDNA_16282-17378
CGGTAAATGCAAAAGGCTTAAGCCCTTTACCCAGAGGTTCAAATCCTCTCCCTAGCCTCACTTACATGACTCACCTCCCTGCCCTAACCTACCTTATCATATCCCTATCCTATGTAGTCCCGATCCTAATTGCCGTAGCATTTCTAACCCTAGTAGAACGAAAAATCTTAAGTTATATACAAGCTCGAAAGGGTCCAAATATCGTAGGACCTTTCGGACTATTACAACCTGTGGCTGATGGAATCAAACTATTCACTAAAGAACCAATCCGCCCATCTACCTCTTCTCCATTTCTATTCCTCATAACCCCCATACTAGCCCTTCTTTTGGCACTCACCATTTGAATTCCCCTCCCCCTTCCCTTCTCCCTCACCGATCTTAACTTAGGCCTCCTTTTCCTTCTAGCCATGTCTAGCCTAGCAGTCTACTCAATTTTATGATCAGGATGGGCCTCAAACTCAAAATACGCCTTAATCGGAGCCCTACGGGCAGTAGCGCAGACTATCTCTTATGAAGTGACGCTAGCTATCATCCTCTTATCCGTAATCATACTTAGCGGGAACTACACTCTAAATACTCTTGCTATCACACAAGAACCATTATACCTTATCTTCTCCTCATGACCCCTCGCAATAATATGATATATTTCAACACTCGCTGAAACAAACCGCGCCCCATTCGACCTCACAGAGGGAGAATCAGAACTAGTATCAGGCTTCAACGTAGAGTATGCTGCTGGACCATTTGCCCTATTCTTTTTAGCTGAATACGCAAATATCATACTGATAAATGCACTCACAGCAATCCTGTTCCTAAACCCAAGCTCGCTAAATCTCTCACAAGAATTGTTCCCCATACTCCTGGCCACAAAAATCCTATTTCTCTCTTCAGGCTTTTTATGAATCCGTGCCTCCTATCCACGATTCCGCTACGATCAGCTCATGCACTTACTCTGAAAAAACTTCCTACCACTAACATTAGCACTGTGCCTCTGACATATCAGCCTACCAACTTGCTACGCAGGCCTACCTCCTTACCTAAGGAAATGTGCCTGAACGTAAAGGGTCACTATGATAAAGTGAACATAGAGGTATACT

```

The mapping data is also an opportunity to check the samples to look for variants that match the genotype of other samples, which would indicate contamination. I did not find any evidence of contamination.

# Annotation
To annotate, submit to [Mitos Webserver](http://mitos.bioinf.uni-leipzig.de/index.py). It found all the proteins and tRNAs expected to be in a bird mitochondrion.
