# Identifying chromosomal sex

## Purpose
This page outlines the pipeline I used to determine which samples were male vs female, in order to know which ones have two Z chromosomes and which are hemizygous for the W.  

In birds, females are the heterogametic sex: males are ZZ with two Z chromosomes and females are ZW with one Z and one W. This means that in males the coverage of the Z chromosome should be similar to that of the autosomes (and non-pseudoautosomal regions of W should have no coverage except due to mapping errors), while in females the Z and W chromosomes should have \~50% the coverage of the autosomes. It is important to know the chromosomal sex of each sample in order to interpret patterns and account for possible sources of error/bias that can be introduced by mixing ZZ and ZW samples in certain analyses.  

I wrote a script to infer the sex of an individual based on the idxstats output after mapping. Essentially, the script reads the output of idxstats and looks for a scaffold that has much lower sequencing depth than expected, which is a candidate for the female Z chromosome. You can see the contents of my scripts at the bottom of theis page in the appendix, and also in the folder for section 1.4.  
On the server, I placed the scripts into `/home/0_PROGRAMS/elsemikk_scripts`. To look for the Z chromosome, I plotted read density vs chromosome size. If the sample is a female, there should be a large contig with approximately half the expected coverage.  

You need:  
1) A series of idxstats files  
2) this script that I wrote for R (and its utilities file)  

Go to wherever the idxstats files are and run the script.  

```bash
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/1.3c_mapping/QC_reports
cp /home/0_PROGRAMS/elsemikk_scripts/idx_Utilities.R . #I wrote the script to look for it's utilities file in the working directory, this could be edited
cat ../samples.txt | while read sample ; do Rscript /home/0_PROGRAMS/elsemikk_scripts/Process_idxstats.R idxstats_"$sample".txt 150 >> "$sample".processidx ; done

#test putative males now that we know which scaffold is the Z (note this looks a little backwards because after I figured out which chromosome was Z, I named it chrZ, making it look like it was unnecessary to try figure out which chromosome was Z)
for sample in idxstats_GRSK_MKP1593.txt idxstats_CISK2.txt idxstats_PAJA_USNM606730.txt idxstats_ANSK01.txt idxstats_ANSK7.txt idxstats_POJA_MKP1559.txt ; do Rscript /home/0_PROGRAMS/elsemikk_scripts/testMale.R "$sample" chrZ >> "$sample".testmale; done

#view results
cat *.processidx | grep -E "file being|Cong"
grep "The sequencing depth of the Z is " *.processidx | sed 's/^.*is //g' | sed 's/ the.*$//g'

```
It will say the name of what is being processed. If it says congratulations, then it guessed the Z chromosome (guessed sample is female) and will tell you what scaffold it thinks is the Z chromosome (Note: if there is a W in the assembly then it might guess that the W is the Z and classify males as females). Otherwise it will say nothing.  

It is not perfect, so I need verify by looking at the graphs that it creates:  
1) do all the predicted females predict the same Z chromosome?  
2) verify by looking at the plot that it produces to see if you agree it looks like a male or a female. Does the chrZ appear much lower than expected (below the diagonal line) as in female, or close to the line as in male?  
 
# Results:  

```
ANSK7 Male
CHSK_MKP2451 Female
CISK_32 Female
GRSK_MKP1592 Female
LTJA_MKP990 Female
PAJA_B20730 Female
POJA_MKP1559 Male
```

On a side note, mtDNA tends to get much higher coverage than autosomes... except for CISK and ANSK. Most of the samples were from muscle tissue, but CISK and ANSK came from blood. There are many more mitochondria in muscle than blood, so it makes sense there are fewer mtDNA reads there.  

# Appendix
Here is a record of `Process_idxstats.R`  
```R
#March 5 2020
#Written by Else Mikkelsen
#This script parses the results of Samtools idxstats or other .txt files which contain chromosome names in column 1, chromosome length in column 2, and number of mapped sequencing reads in column 3
#Usage:
#Rscript Process_idxstats.R {name_of_idxstats_file.txt} {(optional) length of each sequencing read}

#Load environment
library(ggplot2)
source("idx_Utilities.R")

#Parse command line arguments
command <- commandArgs(trailingOnly = T)

#end program if there are no command line arguments given and print an error message
if (length(command)==0){
  cat("Error: This scripts requires the path to the idxstats file as an argument\n")
  cat("Usage:\n")
  cat("Rscript Process_idxstats.R {name_of_idxstats_file.txt} {(optional) length of each sequencing read}\n")
  quit()
}

#end program if there are too many command line arguments given and print an error message.
#If the optional second argument is given (length of sequencing reads, measured in basepairs), 
#then assign this value to a variable
#Otherwise, assume a common default value
if (length(command)>1){ #check that there are multiple command line arguments
  command2 <- as.numeric(command[2]) #make the second argument numeric
  readlength <- command2 #assign read length to a variable
  if (!is.numeric(command2)|is.na(command2)){ #check that it was successfully converted to numeric
    cat("Error: second command line argument is not an integer. This script can only process one file at a time.\n")
    cat("The second command line argument should be the integer length of the sequencing reads (optional)\n")
    quit()
  }
} else {
  readlength <- 150 #if no second argument given, assume a default read length
}

#report the read length that will be used
cat("Using a read length of", readlength, "bp.\nAdjust second command line argument if this is not correct.\n")

#end program if there are too many command line arguments given and print an error message
if (length(command)>2){
  cat("Error: This script can only process one file at a time.\n")
  cat("The second command line argument should be the integer length of the sequencing reads (optional)\n")
  quit()
}


#####################
idxstats <- command[1] #get the filename to be used

cat("The file being processed is", idxstats, "\n") #report the sample being analyzed

#First, load the data
idxdata <- process_idx(idxstats)

cat("\n-----------------------------\n")

#check integrity of data - should not be any unmapped reads in the file, but if there are, the program should not quit
if (sum(idxdata$Unmapped_Reads)>0){
  cat("Warning: Did you know that there are unmapped reads in this data?\n")
}
if (nrow(idxdata)==0|!exists("idxdata")){
  cat("Error: This file appears to be empty\n")
  quit() #exit the program if the file could not be read
}
if (sum(idxdata$Mapped_Reads)==0){
  cat("Error: There appear to be no mapped reads in this file\n")
  quit()
}

#report basic stats about the data: number of scaffolds, length of genome, number of reads, sequencing depth, and the average and standard deviation of number of reads per chromosome
reportstats <- stats_idx(idxdata, readlength)
cat("There are", reportstats$Number_Scaffolds, "scaffolds in this dataset, with a total length of", reportstats$Genome, "bp\n")
cat("There are", reportstats$Number_Reads, "mapped reads for this individual, for an average sequencing depth of", reportstats$seq_depth, "X \n")
cat("The mean number of reads per chromosome is", reportstats$Average_Reads, "with a standard deviation of", reportstats$SD_Reads_perscaff, "\n")

#report the correlation and covariance between scaffold length and mapped reads, without removing the Z chromosome
cat("The correlation between scaffold length and mapped reads is", depth.cor(idxdata), "\n")
cat("The covariance between scaffold length and mapped reads is", depth.cov(idxdata), "\n")
cat("\n-----------------------------\n\n")


#create a linear model of the relationship between scaffold length and mapped reads.
linmod <- depthlm(idxdata)
cat("Created a linear model of the relationship between scaffold length and mapped reads:\n")
print(summary(linmod))

#Visualize the data, and plot a line based on the linear model. 
length_vs_reads <- plot.depth(idxdata, linmod)
pdf(paste(idxstats, "plots.pdf", sep = "")) #Save the plot to a pdf.
print(length_vs_reads)

#Guess which chromosome is the Z chromosome, and detect which scaffolds are outliers based on the linear model
putativeZ <- guessZ(idxdata, linmod)
outliers <- outlierscaffs(idxdata, linmod)

#Report the Z chromosome, if the putative Z chromosome is in the set of outliers, then report it to the user. Otherwise, tell them it is probably a male bird
if (putativeZ %in% outliers){
  foundZ <- TRUE
  cat("Congratulations! It looks like the Z chromosome is", putativeZ, "\n")
  if (length(outliers)>1){ #report other outlier scaffolds so that the user can investigate them if needed
    cat("You might also want to investigate all these scaffolds:", outliers[outliers != putativeZ], "\n")
  }
} else {
  foundZ <- FALSE
  cat("I could not find the Z chromosome. My closest guess is", putativeZ, "\n")
  cat("This is probably a male bird (or female mammal)\n")
  if (length(outliers)>0){ #Even though we did not detect the Z chromosome, report any outliers to the user so they can investigate them if needed
    cat("You might want to investigate these scaffolds:", outliers, " because they appear to be outliers\n")
  }
}

cat("\n-----------------------------\n")

#If we found the Z chromosome, report the stats without the Z chromosome
if (foundZ){
  autosomedata <- idxdata[idxdata$Chromosome !=putativeZ,] #create a dataset with the Z removed
  #report the correlation and covariance between scaffold length and mapped reads, after removing the Z chromosome
  cat("The correlation between scaffold length and mapped reads is", depth.cor(autosomedata), "after removing the Z\n")
  cat("The covariance between scaffold length and mapped reads is", depth.cov(autosomedata), "after removing the Z\n")
  #create a linear model of the relationship between scaffold length and mapped reads, without the Z.
  autolinmod <- depthlm(autosomedata)

  #Visualize the data, and plot a line based on the linear model
  cat("A plot of sequencing reads vs chromosome length has been produced (with and without the putative Z, if detected.)\n")
  cat("You may want to inspect it visually to confirm that it follows the expected pattern\n")
  plot.no.Z <- plot.depth(autosomedata, autolinmod)
  print(plot.no.Z) #save plot to the pdf file
  
  #test whether the relative sequencing depth deviates from the expected value
  binomial <- Zbinomial(idxdata, putativeZ, autolinmod)  #do a binomial test
  if (binomial$p.value<0.05){ #check if the result was significant
    cat("\n-----------------------------\n")
    cat("Warning: the depth of coverage deviates significantly from the expected value of half the autosomal sequencing depth\n")
    cat("Binomial test: p=", binomial$p.value, "\n") #report the p value
    if (binomial$estimate>0.5){ #check whether the estimate was higher or lower than expected
      cat("The sequencing depth of the Z is", binomial$estimate, "times the depth of the other chromosomes\n")
      cat("Warning: a very elevated value here could indicate inaccurate read mapping\n")
    }
    if (binomial$estimate<0.5){
      cat("The sequencing depth of the Z is", binomial$estimate, "the depth of the other chromosomes\n")
    }
  }
  if (binomial$p.value>0.05){ #report to the user if the result of the binomial test was not significant
    cat("The relative sequencing depth of the Z (", binomial$estimate, ") does not deviate significantly from expectation (0.5)\n" )
    cat("Binomial test: p=", binomial$p.value, "\n")
  }
}  

#close pdf
dev.off()
```

Here is a record of `idx_Utilities.R`:  
```R
#March 5 2020
#Written by Else Mikkelsen
#This script parses the results of Samtools idxstats or other .txt files which contain chromosome names in column 1, chromosome length in column 2, and number of mapped sequencing reads in column 3
#Usage:
#Rscript Process_idxstats.R {name_of_idxstats_file.txt} {(optional) length of each sequencing read}

process_idx <- function(filename) {
  idxdata <- read.table(filename)
  colnames(idxdata) <- c("Chromosome", "Length", "Mapped_Reads", "Unmapped_Reads")
  return(idxdata)
}

stats_idx <- function(idxdata, readlength) {
  reportstats <- NULL
  reportstats$Number_Scaffolds <- nrow(idxdata)
  reportstats$Number_Reads <- sum(idxdata$Mapped_Reads)
  reportstats$Genome <- sum(idxdata$Length)
  reportstats$Average_Reads_perscaff <- sum(idxdata$Mapped_Reads/nrow(idxdata))
  reportstats$SD_Reads_perscaff <- sd(idxdata$Mapped_Reads/nrow(idxdata))
  reportstats$seq_depth <- (readlength*reportstats$Number_Reads)/reportstats$Genome
  return(reportstats)
}

plot.depth <- function(idxdata, linear.model) { 
  depthplot <- ggplot(data = idxdata, aes(x = Length, y = Mapped_Reads)) + #create a plot
    geom_point(size = 2, color = "#8a2be2", fill = "#008080", shape = 21) +
    theme_classic() + #remove distracting plot features
    labs(x = "Chromosome Length (bp)", y = "Mapped Sequencing Reads",
         title = "Number of sequencing reads for chromosomes of different lengths") + #add labels that are more meaningful
    theme(plot.title = element_text(hjust = 0.5))+  #make title centred
    geom_abline(slope = linear.model$coefficients[2], intercept = linear.model$coefficients[1])
  return(depthplot)
}

#This function calculates the correlation between wine density and residual sugar content
depth.cor <- function(idxdata) {
  correlation <- cor(idxdata$Length, idxdata$Mapped_Reads)
  return(correlation)
}

#This function calculates the covariance between wine density and residual sugar content
depth.cov <- function(idxdata) {
  covariance <- var(idxdata$Length, idxdata$Mapped_Reads)
  return(covariance)
}

#This function creates a linear model between wine density and residual sugar content
depthlm <- function(idxdata) {
  lin.model <- lm(formula = Mapped_Reads ~ Length, data = idxdata) #create linear model
  return(lin.model) #return linear model
}

guessZ <- function(idxdata, linmod) {
  points <- linmod$coefficients[2]*idxdata$Length + linmod$coefficients[1]
  #In females, the Z (or W) chromosome should show the greatest deviation from the expected value of sequencing depth, and it should always be lower than expected
  deviation <- points - idxdata$Mapped_Reads
  names(deviation) <- idxdata$Chromosome
  putativeZ <- names(deviation)[deviation == max(deviation)]
  return(putativeZ)
}


outlierscaffs <- function(idxdata, linmod) {
  #Make sure that this is a statistically significant outlier
  resids <- residuals(linmod)
  names(resids) <-idxdata$Chromosome
  outliers <- names(abs(resids))[abs(resids)>4*sd(resids)&abs(as.numeric(resids))>0.2*idxdata$Mapped_Reads]
  return(outliers)
}

Zbinomial <- function(idxdata, putativeZ, linmod) {
  #Select data for the Z chromosome
  Zdata <- idxdata[idxdata$Chromosome==putativeZ,]
  #Create expected data based on lengths of chromosomes
  points <- linmod$coefficients[2]*idxdata$Length + linmod$coefficients[1]
  #This value is expected to be equal to 0.5:
  #(Zdata$Mapped_Reads/Zdata$Length)/(points[idxdata$Chromosome==putativeZ]/Zdata$Length)
  #We will test whether it deviates significantly
  #To assess significance, we should test if there is a significant deviation from 0.5
  #This is a test of whether, given the sample size, the (# of mapped reads)/(# of expected reads) is significantly different from 0.5
  #This is essentially a binomial test. If we consider a mapped read to be a "success" and the number of expected reads for a chromosome of its size to be the "number of trials" and the expected proportion of "successes" to be 50%:
  binomial <- binom.test(x=Zdata$Mapped_Reads, n=as.integer(points[idxdata$Chromosome==putativeZ]), p=0.5)
  return(binomial)
}

```

Here is a record of `testMale.R`:  
```R
#This file is to be used after Process_idxstats.R to confirm the sex of male birds after the Z chromosome has been identified
#Else Mikkelsen Mar 6 2020
#Load environment
source("idx_Utilities.R")

#Parse command line arguments
command <- commandArgs(trailingOnly = T)


#end program if there are no command line arguments given and print an error message
if (length(command)==0){
  cat("Error: This scripts requires the path to the idxstats file as an argument\n")
  cat("Usage:\n")
  cat("Rscript Process_idxstats.R {name_of_idxstats_file.txt} {(optional) length of each sequencing read}\n")
  quit()
}

idxstats <- command[1] #get the filename to be used

cat("The file being processed is", idxstats, "\n") #report the sample being analyzed

#First, load the data
idxdata <- process_idx(idxstats)

cat("\n-----------------------------\n")

putativeZ <- command[2]

autosomedata <- idxdata[idxdata$Chromosome !=putativeZ,] #create a dataset with the Z removed
#create a linear model of the relationship between scaffold length and mapped reads, without the Z.
autolinmod <- depthlm(autosomedata)

#test whether the relative sequencing depth deviates from the expected value
binomial <- Zbinomial(idxdata, putativeZ, autolinmod)  #do a binomial test
if (binomial$p.value<0.05){ #check if the result was significant
  cat("\n-----------------------------\n")
  cat("Binomial test is significant: p=", binomial$p.value, "\n") #report the p value
  if (binomial$estimate>0.7){ #check whether the estimate was higher or lower than expected
    cat("The sequencing depth of the Z is", binomial$estimate, "times the depth of the other chromosomes\n")
    cat("Yes, this appears to be a male\n")
  }
  if (binomial$estimate<0.7){
    cat("The sequencing depth of the Z is", binomial$estimate, "the depth of the other chromosomes\n")
    cat("I think this is actually a female\n")
    }
}
if (binomial$p.value>0.05){ #report to the user if the result of the binomial test was not significant
  cat("The relative sequencing depth of the Z (", binomial$estimate, ") does not deviate significantly from expectation (0.5)\n" )
  cat("Binomial test: p=", binomial$p.value, "\n")
  cat("This is probably a female")
}
```
