### Overview
This pipeline describes how to call genotypes after mapping resequencing data to a reference genome. I am generating two datasets, dataset "c" mapped to *Stercorarius parasiticus*, and dataset "d" mapped to *Alca torda*. I wrote these procedures in two different pages (1.5c and 1.5d) which describe essentially the exact same procedure for the two datasets.  

### Required Input
This step requires the mapped sequencing data (`.bam` files), which was generated in step `1.3`. These files are too large to be provided in the repository.   

### Output
The output is a VCF file filled with SNPs with called genotypes for each sample.  

# Genotype calling

There are some handy links about variant calling [from samtools](https://samtools.github.io/bcftools/howtos/variant-calling.html) [and evomics](http://evomics.org/learning/population-and-speciation-genomics/2020-population-and-speciation-genomics/first-steps-in-genomic-data-analysis/#ex3.2) and [Joana Meier](https://speciationgenomics.github.io/filtering_vcfs/).  

In this pipeline, we will be calling genotypes using bcftools. After calling genotypes for each sample, we will combine the samples together and filter the dataset. Note that if you have population data consisting of multiple samples from the same population, it can make sense to call genotypes all together so that the genotype caller can use information from one sample to inform it about the presence of a variant site in a different sample. In my case, I do not want this - I am trying to look for subtle signatures of introgression between different lineages, so I do not want the genotype call of one sample to influence the genotype calls of any other sample.  

Firstly, we will need a list of samples. I will use this list throughout the pipeline to loop through the samples and to analyze in parallel. I generally assume that this file is present in any folder that I am running commands in. The names will be used to name output files, and it assumes that your starting data has the same name.  

If you are running these commands for a different project, you will probably want to change the file suffixes. In my pipeline, the input data is `.PAJAbowtie.marked.bam` which means it was aligned to Stercorarius parasiticus with bowtie3 and the duplicates have been marked.  

```bash
cat > samples.txt
```
```
LTJA_MKP990
PAJA_B20730
POJA_MKP1559
POJA_4
POJA_IB2659
GRSK_MKP1592
GRSK_MKP1593
CISK2
CISK55
CISK3
CHSK_MKP2451
ANSK01
ANSK8
Fratercula_arctica
Alca_torda
PAJA_USNM606730
Uria_lomvia
ANSK7

```

## 1) Call SNPs

As a first step, we will call all of the variants in our individual samples (that is, locations where our sample differs from the reference sequence) as well as invariant sites (where our samples do not differ from the reference).  

I am running this command in parallel because samtools mpileup only uses 1 core and is fairly long.  

Note that mpileup automatically ignores reads that are marked as duplicates, unaligned, secondary, or QCfail in the FLAG field. Since we marked duplicates, these will be excluded.  

Now that we have these input files, we can call genotypes:  
```bash
#set up environment
mkdir -p /home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.5c_genotyping
cd /home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.5c_genotyping #on new server

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/1.5c_genotyping
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/1.5c_genotyping #on old server

#call all sites including invariants for each sample
cat samples.txt | parallel 'time /home/0_PROGRAMS/bcftools-1.14/bcftools mpileup -B -C 50 -a AD,DP,SP -Ou -f ../../0_Reference_genome/0_reference/Stercorarius_parasiticus.ref.fa --threads 2 ../1.3c_mapping/merged/{1}.PAJAbowtie.marked.bam | time /home/0_PROGRAMS/bcftools-1.14/bcftools call --threads 2 -f GQ,GP -m -O z - > {1}.PAJAbowtie.allsites.vcf.gz'
#takes a few hours
#index the vcf files
cat samples.txt | parallel time /home/0_PROGRAMS/bcftools-1.14/bcftools index {1}.PAJAbowtie.allsites.vcf.gz #11m

#First, make a text file listing all of the vcf files from each sample that we want to merge. Then, merge all those files into a single vcf file.
ls ./*PAJAbowtie.allsites.vcf.gz > vcf_files.txt #this only works because there are no other files with that suffix yet that I don't want to merge

#####This is the input for step 2.2c, MC1R gene tree#####
time /home/0_PROGRAMS/bcftools-1.14/bcftools merge --threads 24 -m id -l vcf_files.txt --info-rules MQ:avg -O z > Stercorarius.PAJAbowtie.merged.allsites.vcf.gz #took about a day

#index the VCF file
time /home/0_PROGRAMS/bcftools-1.14/bcftools index Stercorarius.PAJAbowtie.merged.allsites.vcf.gz #took 33m27.611s #21m2.483s
conda activate basic #this conda environment has tabix installed
tabix Stercorarius.PAJAbowtie.merged.allsites.vcf.gz 

```
`bcftool mpileup`: creates genotype likelihoods at each position in the genome, with their depth of coverage
* `-q`: minimum mapping quality to allow
* `-B`: avoid false SNPs by disabling probabilistic realignment
* `-b`: "list of input BAM filenames, one per line"
* `-C 50`: 
* `-Ou`: save time by not compressing the VCF file before piping it to the next command
* `-f`: FASTA file of indexed reference sequence
* `/home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorariidae/1_ALIGNMENT_Alctor_0a_1a/"$line"_Alctor_vvsensitive_unique_sorted.bam`: path to the samle's bam file. Expected to be named `{1}_Alctor_vvsensitive_unique_sorted.bam` where {1} is the name as it appears in samples.txt (change the rest as needed)

`bcftools call`: outputs the VCF file with only variant sites
* `--threads`: specifies number of threads that can be used
* `-m`: use the newer multiallelic caller (opposite of `-c`)
* `-c`: returns to the original samtools consensus caller
* `-v`: only output variant sites
* `-O z`: specifices that the output type is a compressed VCF file
`bcftools merge`:
* `--threads 24`: number of threads to use
* `-m id`: tells the program to merge by ID  
* `-l`: the list file which lists the names of all the vcf files to be merged (created above)

**Output**: a compressed, indexed VCF file for each sample containing calls at all sites. 
**Input for**: Filtering. This is also the VCF file that I use as input for generating haplotypes for the MC1R gene trees.


# Assess VCF file and choose filters

There is a nice tutorial about assessing VCF files [here](https://speciationgenomics.github.io/filtering_vcfs/). Before filtering, it is a good idea to assess some quality metrics of the dataset to choose appropriate cutoffs.  
```bash
cd /home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.5c_genotyping #on node

#now let's get some quality control information
mkdir -p VCF_QC

#First, I am checking the transitions/transversion ratio on the full dataset
#note the file I assessed here is not the exact same one from above - this was an older one that was exactly the same except was made without the "--info-rules MQ:avg" flag (so this will only affect the MQ estimates)
/home/0_PROGRAMS/vcftools/bin/vcftools --TsTv-summary --gzvcf Stercorarius.PAJAbowtie.allsites.vcf.gz --out VCF_QC/Stercorarius.PAJAbowtie.tstv &
#Ts/Tv ratio: 2.465
#and per sample
cat samples.txt | parallel 'time /home/0_PROGRAMS/bcftools-1.14/bcftools stats {1}.PAJAbowtie.allsites.vcf.gz > VCF_QC/{1}.bcftools_stats' #18m36.486s-39m14.111s

#Then, I am checking the relatedness of my samples. I hope I don't have close relatives!
#Let's check if any of our samples are likely to be related. Hopefully not!
/home/0_PROGRAMS/vcftools/bin/vcftools --gzvcf Stercorarius.PAJAbowtie.allsites.vcf.gz --relatedness2 --out VCF_QC/Stercorarius.PAJAbowtie.relatedness2 #this will give us a good indication of relatedness. It uses KING inference of relatedness (Manichaikul et al., BIOINFORMATICS 2010 (doi:10.1093/bioinformatics/btq559))

#subset VCF randomly to keep 1% of data
conda activate vcflib #conda install -y -c conda-forge -c bioconda -c defaults vcflib
/home/0_PROGRAMS/bcftools-1.14/bcftools view Stercorarius.PAJAbowtie.allsites.vcf.gz | vcfrandomsample -r 0.01 > VCF_QC/Stercorarius_subset.PAJAbowtie.vcf.gz & #took a few hours, kept 566,364 sites 

#allele frequencies for bialleleic SNPs
/home/0_PROGRAMS/vcftools/bin/vcftools --gzvcf Stercorarius.PAJAbowtie.allsites.vcf.gz --freq2 --out VCF_QC/Stercorarius --max-alleles 2 &
#mean depth of coverage per sample
/home/0_PROGRAMS/vcftools/bin/vcftools --gzvcf Stercorarius.PAJAbowtie.allsites.vcf.gz --depth --out VCF_QC/Stercorarius &
#mean depth of coverage per site
/home/0_PROGRAMS/vcftools/bin/vcftools --gzvcf Stercorarius.PAJAbowtie.allsites.vcf.gz --site-mean-depth --out VCF_QC/Stercorarius &
#site quality scores
/home/0_PROGRAMS/vcftools/bin/vcftools --gzvcf Stercorarius.PAJAbowtie.allsites.vcf.gz --site-quality --out VCF_QC/Stercorarius
#missing data
/home/0_PROGRAMS/vcftools/bin/vcftools --gzvcf Stercorarius.PAJAbowtie.allsites.vcf.gz --missing-indv --out VCF_QC/Stercorarius_ &
#missing data per site
/home/0_PROGRAMS/vcftools/bin/vcftools --gzvcf Stercorarius.PAJAbowtie.allsites.vcf.gz --missing-site --out VCF_QC/Stercorarius_ &
#heterozygosity
/home/0_PROGRAMS/vcftools/bin/vcftools --gzvcf Stercorarius.PAJAbowtie.allsites.vcf.gz --het --out VCF_QC/Stercorarius_ &

```
* `--hist-indel-len`: "This option will generate a histogram file of the length of all indels (including SNPs). It shows both the count and the percentage of all indels for indel lengths that occur at least once in the input file. SNPs are considered indels with length zero. The output file has the suffix ".indel.hist""
* `--het`: "Calculates a measure of heterozygosity on a per-individual basis. Specfically, the inbreeding coefficient, F, is estimated for each individual using a method of moments. The resulting file has the suffix ".het".""
* `--freq2`: Outputs the allele frequency for each site in a file with the suffix ".frq". The second option is used to suppress output of any information about the alleles.
* `--depth`: Generates a file containing the mean depth per individual. This file has the suffix ".idepth".
* `--TsTv-summary`: Calculates a simple summary of all Transitions and Transversions. The output file has the suffix ".TsTv.summary".



# Filter dataset

**Dataset 3: STARBEAST3 dataset**

This dataset will be used for Starbeast3 and for Phylonet.  

This dataset is similar to the first, but specifically aims to reduce the difference between filtering severity of variant and invariant sites, to estimate diversity more accurately and create the alignments for starbeast3. I am imposing the following filters:  
* average sequencing depth < 20 (to remove possible collapsed repeats/duplications)
* mapping quality > 20
* genotype depth at least 5
* autosomes only (no chrZ)

I am also filtering SNPs to keep only biallelic SNPs.  

Note that I did not impose filters for variant quality: this was intentional, as doing so will remove SNPs without affecting invariant sites, and thus affect the ratio of variant:invariant sites which is important when using molecular clocks. I also did not filter for individual genotype quality, as doing so will be biased to remove more heterozygous sites from low-depth samples, which will shorten their inferred branch lengths.  

```bash
cd /home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.5c_genotyping #on new server

#autosomes only

#filter SNPs
time /home/0_PROGRAMS/bcftools-1.14/bcftools view -m2 -M2 -v snps --regions chr1,chr2,chr3,chr4,chr5,chr6,chr7,chr8,chr9,chr10,chr11,chr12,chr13,chr14,chr15,chr16,chr17,chr18,chr19,chr20,chr21,chr22,chr23,chr24,chr25 -e 'AVG(FMT/DP)>20 || MQ<20' ./Stercorarius.PAJAbowtie.merged.allsites.vcf.gz | /home/0_PROGRAMS/bcftools-1.14/bcftools filter -S . -e 'FMT/DP<5' -Oz -o Stercorarius.PAJAbowtie.merged.filteredSNPs.for_pixy.auto.vcf.gz #took 103m7.702s(138m4.914s)

#filter invariant sites
time /home/0_PROGRAMS/bcftools-1.14/bcftools view -m1 -M1 --regions chr1,chr2,chr3,chr4,chr5,chr6,chr7,chr8,chr9,chr10,chr11,chr12,chr13,chr14,chr15,chr16,chr17,chr18,chr19,chr20,chr21,chr22,chr23,chr24,chr25 -e 'AVG(FMT/DP)>20 || MQ<20' ./Stercorarius.PAJAbowtie.merged.allsites.vcf.gz | /home/0_PROGRAMS/bcftools-1.14/bcftools filter -S . -e 'FMT/DP<5' -Oz -o Stercorarius.PAJAbowtie.merged.invariant.for_pixy.auto.vcf.gz #132m46.303s (339m46.948s)

#index the files
time /home/0_PROGRAMS/bcftools-1.14/bcftools index Stercorarius.PAJAbowtie.merged.filteredSNPs.for_pixy.auto.vcf.gz
time /home/0_PROGRAMS/bcftools-1.14/bcftools index Stercorarius.PAJAbowtie.merged.invariant.for_pixy.auto.vcf.gz


#concatenate them (SNP + invariant)
######This is the dataset for generating phylogenetic blocks for Stabeast3######
time /home/0_PROGRAMS/bcftools-1.14/bcftools concat --threads 24 --allow-overlaps Stercorarius.PAJAbowtie.merged.filteredSNPs.for_pixy.auto.vcf.gz Stercorarius.PAJAbowtie.merged.invariant.for_pixy.auto.vcf.gz -O z -o Stercorarius.PAJAbowtie.merged.allsites.for_pixy.auto.vcf.gz #67m56.970s
#index the VCF file
conda activate basic
time tabix Stercorarius.PAJAbowtie.merged.allsites.for_pixy.auto.vcf.gz #8m18.487s
```

# Archive: unused datasets
These datasets were not used in the project, I am keeping them here just for my records.  
**DATASET 1: MODERATE STRINGENCY SNP FILTERS**  
This first dataset I am applying several filters on the SNP dataset. I want biallelic SNPs only. This dataset was for Treemix or Dsuite, but I did not end up actually using it in the final project. I used it to evaluate the impact of reference genome on the Dsuite analysis. Since the use of *Stercorarius parasiticus* as a reference genome appears to bias some trios of the ABBA BABA test, I did not use this dataset.  

Here are some filters that we will apply:  
* biallelic SNPs only  
* site genotype quality 25  
* mapping quality 25  
* sequencing depth 8-15  
* individual depth > 3  
* individual genotype quality > 15  
* not within 10 bp of an indel (where mapping can be more error-prone)  
* missingness < 3/17 (0.176)  
* MAF: no MAF filter for this dataset, it woulf not be apropriate for this sampling scheme (and it can bias ABBA BABA tests).  

```bash
cd /home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.5c_genotyping #on new server

#max 16 (calculated mean depth + 2*sqrt(mean))
time /home/0_PROGRAMS/bcftools-1.14/bcftools view -m2 -M2 -v snps -t chr1,chr2,chr3,chr4,chr5,chr6,chr7,chr8,chr9,chr10,chr11,chr12,chr13,chr14,chr15,chr16,chr17,chr18,chr19,chr20,chr21,chr22,chr23,chr24,chr25 -e '%QUAL<25 || AVG(FMT/DP)<5 || AVG(FMT/DP)>16 || MQ<25 || AC==0 || AC==AN' ./Stercorarius.PAJAbowtie.merged.allsites.vcf.gz | /home/0_PROGRAMS/bcftools-1.14/bcftools filter -S . -e 'FMT/DP<6 | FMT/GQ<20' --SnpGap 10 | /home/0_PROGRAMS/bcftools-1.14/bcftools view -e 'F_MISSING > 0.176' -Oz -o Stercorarius.PAJAbowtie.merged.filteredSNPs.auto.stringent.max16.vcf.gz #took 95m(115m) for Stepar, 83m(94m) for Alctor
time /home/0_PROGRAMS/bcftools-1.14/bcftools index Stercorarius.PAJAbowtie.merged.filteredSNPs.auto.stringent.max16.vcf.gz
```
* `--allow-overlaps`: "First coordinate of the next file can precede last record of the current file." (so the output will need to be sorted first)  

**Dataset 3: STARBEAST3 dataset**  
I also repeated this filtering for chrZ, but this data is not used for any analyses in the final project, as I am basing the Starbeast3 tree only on autosomal data. chrZ had sequencing depth too low in females to get good quality genotype calls (and note that  they should be called as "haploid").    

```bash
#repeat for chromosome Z
#filter the sex chromosomes. You may consider different depth and missingness filters, because they have different expectations due to different ploidy levels in males and females
time /home/0_PROGRAMS/bcftools-1.14/bcftools view -m2 -M2 -v snps --regions chrZ -e 'AVG(FMT/DP)>20 || MQ<20' ./Stercorarius.PAJAbowtie.merged.allsites.vcf.gz | /home/0_PROGRAMS/bcftools-1.14/bcftools filter -S . -e 'FMT/DP<5' -Oz -o Stercorarius.PAJAbowtie.merged.filteredSNPs.for_pixy.chr_Z.vcf.gz #7m4.687s
time /home/0_PROGRAMS/bcftools-1.14/bcftools index Stercorarius.PAJAbowtie.merged.filteredSNPs.for_pixy.chr_Z.vcf.gz

time /home/0_PROGRAMS/bcftools-1.14/bcftools view -m1 -M1 -t chrZ -e 'AVG(FMT/DP)>20 || MQ<20' ./Stercorarius.PAJAbowtie.merged.allsites.vcf.gz | /home/0_PROGRAMS/bcftools-1.14/bcftools filter -S . -e 'FMT/DP<5' -Oz -o Stercorarius.PAJAbowtie.merged.invariant.for_pixy.chr_Z.vcf.gz  #85m17.155s
time /home/0_PROGRAMS/bcftools-1.14/bcftools index Stercorarius.PAJAbowtie.merged.invariant.for_pixy.chr_Z.vcf.gz

time /home/0_PROGRAMS/bcftools-1.14/bcftools concat --threads 24 --allow-overlaps Stercorarius.PAJAbowtie.merged.filteredSNPs.for_pixy.chr_Z.vcf.gz Stercorarius.PAJAbowtie.merged.invariant.for_pixy.chr_Z.vcf.gz -O z -o Stercorarius.PAJAbowtie.merged.allsites.for_pixy.chr_Z.vcf.gz #51m11.101s
conda activate basic
time tabix Stercorarius.PAJAbowtie.merged.allsites.for_pixy.chr_Z.vcf.gz #1m14.729s

```
