### Overview
This pipeline describes how to call genotypes after mapping resequencing data to a reference genome. I am generating two datasets, dataset "c" mapped to *Stercorarius parasiticus*, and dataset "d" mapped to *Alca torda*. I wrote these procedures in two different pages (1.5c and 1.5d) which describe essentially the exact same procedure for the two datasets.

### Required Input
This step requires the mapped sequencing data (`.bam` files), which was generated in step `1.3`.

### Output
The output is a VCF file filled with SNPs with called genotypes for each sample.

# Genotype calling

There are some handy links about variant calling [from samtools](https://samtools.github.io/bcftools/howtos/variant-calling.html) [and evomics](http://evomics.org/learning/population-and-speciation-genomics/2020-population-and-speciation-genomics/first-steps-in-genomic-data-analysis/#ex3.2) and [Joana Meier](https://speciationgenomics.github.io/filtering_vcfs/).  

In this pipeline, we will be calling genotypes using bcftools. After calling genotypes for each sample, we will combine the samples together and filter the dataset. Note that if you have population data consisting of multiple samples from the same population, it can make sense to call genotypes all together so that the genotype caller can use information from one sample to inform it about the presence of a variant site in a different sample. In my case, I do not want this - I am trying to look for subtle signatures of introgression between different lineages, so I do not want the genotype call of one sample to influence the genotype calls of any other sample.  

Firstly, we will need a list of samples. I will use this list throughout the pipeline to loop through the samples and to analyze in parallel. I generally assume that this file is present in any folder that I am running commands in. The names will be used to name output files, and it assumes that your starting data has the same name.  

If you are running these commands for a different project, you will probably want to change the file suffixes. In my pipeline, the input data is `.RAZObowtie.marked.bam` which means it was aligned to Razorbill with bowtie3 and the duplicates have been marked.  

```bash
cat > samples.txt
```
```
LTJA_MKP990
PAJA_B20730
POJA_MKP1559
POJA_4
POJA_IB2659
GRSK_MKP1592
GRSK_MKP1593
CISK2
CISK55
CISK3
CHSK_MKP2451
ANSK01
ANSK8
ANSK7
PAJA_USNM606730
Alca_torda
Fratercula_arctica
Uria_lomvia

```

## 1) Call SNPs

As a first step, we will call all of the variants in our individual samples (that is, locations where our sample differs from the reference sequence) as well as invariant sites (where our samples do not differ from the reference).  

I am running this command in parallel because samtools mpileup only uses 1 core and is fairly long.  

Note that mpileup automatically ignores reads that are marked as duplicates, unaligned, secondary, or QCfail in the FLAG field. Since we marked duplicates, these will be excluded.  

Now that we have these input files, we can call genotypes:  
```bash
#set up environment
mkdir -p /home/0_GENOMES5/Stercorarius/1_Data/1.5d_genotyping
cd /home/0_GENOMES5/Stercorarius/1_Data/1.5d_genotyping #on node

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/1.5d_genotyping
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/1.5d_genotyping

reference=/home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/0_Reference_genome/0.1_scaffolding/GCA_008658365.1/data/GCA_008658365.1/GCA_008658365.1_bAlcTor1_genomic.fna #on main server
#reference=/home/0_GENOMES5/Stercorarius/0_Reference_genome/0.1_scaffolding/GCA_008658365.1/data/GCA_008658365.1/Alca_torda.ref #on node
tagname=RAZObowtie

#call all sites including invariants
cat samples.txt | parallel 'time /home/0_PROGRAMS/bcftools-1.14/bcftools mpileup -B -C 50 -a AD,DP,SP -Ou -f '$reference' --threads 20 ../1.3d_mapping/merged/{1}.'$tagname'.marked.bam | time /home/0_PROGRAMS/bcftools-1.14/bcftools call --threads 20 -f GQ,GP -m -O z - > {1}.'$tagname'.allsites.vcf.gz'
#takes a few hours
#index the vcf file
cat samples.txt | parallel time /home/0_PROGRAMS/bcftools-1.14/bcftools index {1}."$tagname".allsites.vcf.gz #11m

#First, make a text file listing all of the vcf files from each sample that we want to merge. Then, merge all those files into a single vcf file.
ls ./*"$tagname".allsites.vcf.gz > vcf_files.txt
time /home/0_PROGRAMS/bcftools-1.14/bcftools merge --threads 32 -m id -l vcf_files.txt --info-rules MQ:avg -O z > Stercorarius."$tagname".merged.allsites.vcf.gz #took about a day #962m38.603s (1088m24.292s)

time /home/0_PROGRAMS/bcftools-1.14/bcftools index Stercorarius."$tagname".merged.allsites.vcf.gz #took 28m52.873s

```
`bcftool mpileup`: creates genotype likelihoods at each position in the genome, with their depth of coverage  
* `-q`: minimum mapping quality to allow  
* `-B`: avoid false SNPs by disabling probabilistic realignment  
* `-b`: "list of input BAM filenames, one per line"  
* `-C 50`:   
* `-Ou`: save time by not compressing the VCF file before piping it to the next command  
* `-f`: FASTA file of indexed reference sequence  
* `/home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorariidae/1_ALIGNMENT_Alctor_0a_1a/"$line"_Alctor_vvsensitive_unique_sorted.bam`: path to the samle's bam file. Expected to be named `{1}_Alctor_vvsensitive_unique_sorted.bam` where {1} is the name as it appears in samples.txt (change the rest as needed)  

`bcftools call`: outputs the VCF file with only variant sites  
* `--threads`: specifies number of threads that can be used  
* `-m`: use the newer multiallelic caller (opposite of `-c`)  
* `-c`: returns to the original samtools consensus caller  
* `-v`: only output variant sites  
* `-O z`: specifices that the output type is a compressed VCF file  
`bcftools merge`:  
* `--threads 24`: number of threads to use  
* `-m id`: tells the program to merge by ID    
* `-l`: the list file which lists the names of all the vcf files to be merged (created above)  

**Output**: a compressed, indexed VCF file for each sample containing calls at all sites.  
**Input for**: Filtering.  

# Assess VCF file and choose filters  

There is a nice tutorial about assessing VCF files [here](https://speciationgenomics.github.io/filtering_vcfs/). Before filtering, it is a good idea to assess some quality metrics of the dataset to choose appropriate cutoffs. I am not checking every metric on this dataset (e.g., I am assessing relatedness with the data mapped to Stercorarius instead).  
```bash
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/1.5d_genotyping
tagname=RAZObowtie

#now let's get some quality control information
mkdir -p VCF_QC

#First, I am checking the transitions/transversion ratio 
#Ts/Tv ratio: 2.465 mapped to S. parasiticus
#Ts/Tv ratio: 2.324 mapped to Alca
#first, look per sample
cat samples.txt | parallel 'time /home/0_PROGRAMS/bcftools-1.14/bcftools stats {1}.RAZObowtie.allsites.vcf.gz > VCF_QC/{1}.bcftools_stats' #18m36.486s-39m14.111s
#compare substitution types for each sample. They should show similar signatures. A sample with different mutational signatures may have suffered DNA damage or elevated sequencing errors or bias.
touch VCF_QC/signatures.txt
cat samples.txt | while read sample ; do echo "$sample" ; grep "^ST" VCF_QC/"$sample".bcftools_stats | cut -f 4 | paste VCF_QC/signatures.txt - > temp && mv temp VCF_QC/signatures.txt ; done
#assess whole dataset, this produces a number of metrics
/home/0_PROGRAMS/bcftools-1.14/bcftools stats Stercorarius.RAZObowtie.allsites.vcf.gz > VCF_QC/Stercorarius.bcftools_stats

#Then, I am checking the relatedness of my samples. I hope I don't have close relatives!
#Let's check if any of our samples are likely to be related. Hopefully not!
#/home/0_PROGRAMS/vcftools/bin/vcftools --gzvcf "$tagname".Stercorarius.allsites.vcf.gz --relatedness2 --out VCF_QC/Stercorarius.PAJAbowtie.relatedness2 #this will give us a good indication of relatedness. It uses KING inference of relatedness (Manichaikul et al., BIOINFORMATICS 2010 (doi:10.1093/bioinformatics/btq559))

#subset VCF randomly to keep 1% of data
#conda activate vcflib #conda install -y -c conda-forge -c bioconda -c defaults vcflib
#/home/0_PROGRAMS/bcftools-1.14/bcftools view "$tagname".Stercorarius.allsites.vcf.gz | vcfrandomsample -r 0.01 > VCF_QC/Stercorarius_subset.PAJAbowtie.vcf.gz & #took a few hours, kept 566,364 sites 

#allele frequencies for bialleleic SNPs
#/home/0_PROGRAMS/vcftools/bin/vcftools --gzvcf "$tagname".Stercorarius.allsites.vcf.gz --freq2 --out VCF_QC/Stercorarius --max-alleles 2 &
#mean depth of coverage per sample
#/home/0_PROGRAMS/vcftools/bin/vcftools --gzvcf "$tagname".Stercorarius.allsites.vcf.gz --depth --out VCF_QC/Stercorarius &
#mean depth of coverage per site
/home/0_PROGRAMS/vcftools/bin/vcftools --gzvcf Stercorarius."$tagname".allsites.vcf.gz --site-mean-depth --out VCF_QC/Stercorarius &
#site quality scores
#/home/0_PROGRAMS/vcftools/bin/vcftools --gzvcf "$tagname".Stercorarius.allsites.vcf.gz --site-quality --out VCF_QC/Stercorarius &
#missing data
#/home/0_PROGRAMS/vcftools/bin/vcftools --gzvcf "$tagname".Stercorarius.allsites.vcf.gz --missing-indv --out VCF_QC/Stercorarius &
#missing data per site
#/home/0_PROGRAMS/vcftools/bin/vcftools --gzvcf "$tagname".Stercorarius.allsites.vcf.gz --missing-site --out VCF_QC/Stercorarius &
#heterozygosity
#/home/0_PROGRAMS/vcftools/bin/vcftools --gzvcf "$tagname".Stercorarius.allsites.vcf.gz --het --out VCF_QC/Stercorarius &

```
* `--hist-indel-len`: "This option will generate a histogram file of the length of all indels (including SNPs). It shows both the count and the percentage of all indels for indel lengths that occur at least once in the input file. SNPs are considered indels with length zero. The output file has the suffix ".indel.hist"  
* `--het`: "Calculates a measure of heterozygosity on a per-individual basis. Specfically, the inbreeding coefficient, F, is estimated for each individual using a method of moments. The resulting file has the suffix ".het"."  
* `--freq2`: Outputs the allele frequency for each site in a file with the suffix ".frq". The second option is used to suppress output of any information about the alleles.  
* `--depth`: Generates a file containing the mean depth per individual. This file has the suffix ".idepth".  
* `--TsTv-summary`: Calculates a simple summary of all Transitions and Transversions. The output file has the suffix ".TsTv.summary".  

# Filter dataset

Sequencing data is full of bias, especially after mapping! Bias for or against repeats vs single-copy sequence, high GC vs low GC, centromere vs telomere regions, macro vs microchromosome, heterochromatin vs euchromatin, fast-evolving vs conserved, coding vs noncoding, sex chromosome vs autosome, etc. Each individual can also have systematic differences based on DNA quality/degredation, sequencing errors, distance to reference genome, heterozygosity, and sequencing depth that affect genotype calling. Our task is to filter the dataset to maximize the correctness of the data, while keeping enough data to perform analyses and avoiding exacerbating or introducing biases. This is not straightforward: filters that improve correctness of the data can also exacerbate biases between individuals or regions of the genome, and these biases can affect different analyses in different ways. Because of that, I am producing a few different datasets optimized for different analyses that are sensitive to different types of bias.  

**DATASET 1: MODERATE STRINGENCY SNP FILTERS**  
This first dataset I am applying several filters on the SNP dataset. I want biallelic SNPs only. This dataset is made with the ABBA BABA test in mind. For this test, I want very high accuracy SNPs, and quality is more important than quantity. Critically, filters must be careful not to bias any particular samples towards reference/alternate alleles to a greater extent than any other sample, or this could bias the ABBA BABA test (severely).  

Here are some filters that we will apply:  
* biallelic SNPs only  
* site genotype quality 25  
* mapping quality 25  
* sequencing depth 8-15  
* individual depth > 3  
* individual genotype quality > 15  
* not within 10 bp of an indel (where mapping can be more error-prone)  
* missingness < 3/17 (0.176)  
* MAF: no MAF filter for this dataset, it woulf not be apropriate for this sampling scheme (and it can bias ABBA BABA tests).  

```bash
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/1.5d_genotyping

#****** THIS IS THE DATASET FOR DSUITE ABBA BABA TESTS ******
#raising the max depth threshold after calculating mean(var_depth$mean_depth)+(2*(sqrt(mean(var_depth$mean_depth)))) as 16.8
#removing min average depth filter, which does not do anything since I have a more stringent genotype depth filter + missingness filter
time /home/0_PROGRAMS/bcftools-1.14/bcftools view -m2 -M2 -v snps -t chr1,chr2,chr3,chr4,chr5,chr6,chr7,chr8,chr9,chr10,chr11,chr12,chr13,chr14,chr15,chr16,chr17,chr18,chr19,chr20,chr21,chr22,chr23,chr24,chr25 -e '%QUAL<25 || AVG(FMT/DP)>17 || MQ<25 || AC==0 || AC==AN' ./Stercorarius.RAZObowtie.merged.allsites.vcf.gz | /home/0_PROGRAMS/bcftools-1.14/bcftools filter -S . -e 'FMT/DP<6 | FMT/GQ<20' --SnpGap 10 | /home/0_PROGRAMS/bcftools-1.14/bcftools view -e 'F_MISSING > 0.176' -Oz -o Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17.vcf.gz #took 95m(115m) for Stepar, 83m(94m) for Alctor #89m56.273s
time /home/0_PROGRAMS/bcftools-1.14/bcftools index Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17.vcf.gz #0m28.050s
#2,029,048 sites

```

Next, I am producing a very similar dataset for the Dinvestigate genome scan of *f*~~dM~~. It is the same as above, but with no missing data filter, in order to retain enough sites to perform the scan.  
```bash
#remove max missing filter
######This dataset is used for Dinvestigate######
time /home/0_PROGRAMS/bcftools-1.14/bcftools view -m2 -M2 -v snps -t chr1,chr2,chr3,chr4,chr5,chr6,chr7,chr8,chr9,chr10,chr11,chr12,chr13,chr14,chr15,chr16,chr17,chr18,chr19,chr20,chr21,chr22,chr23,chr24,chr25 -e '%QUAL<25 || AVG(FMT/DP)>17 || MQ<25 || AC==0 || AC==AN' ./Stercorarius.RAZObowtie.merged.allsites.vcf.gz | /home/0_PROGRAMS/bcftools-1.14/bcftools filter -S . -e 'FMT/DP<6 | FMT/GQ<20' --SnpGap 10 -Oz -o Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.vcf.gz #took 95m(115m) for Stepar, 83m(94m) for Alctor #89m56.273s
time /home/0_PROGRAMS/bcftools-1.14/bcftools index Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.vcf.gz #
```

### test datasets: impact of missing data and sequencing depth filters  
It is important to understand how filters affect downstream analyses. Are the results robust to the filters? I am most interested in the effects of the missing data filters, and the individual genotype depth filters. I am going to generate test datasets with filters for a maximum missing data of 100% (i.e., no filter), 50%, 40%, 30%, or 10%. I will produce versions with a minimum genotype sequencing depth of 6x or 10x.  
```bash
#try different max missingness filters (while keeping biallelic autosomal SNPs, qual>25, avgdp<17, mq>25, dp>6, gq>20)

#adjust missing filter
/home/0_PROGRAMS/bcftools-1.14/bcftools view -e 'F_MISSING > 0.10' Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.vcf.gz -Oz -o Stercorarius.RAZObowtie.merged.filteredSNPs.auto.10stringent.max17.vcf.gz 
time /home/0_PROGRAMS/bcftools-1.14/bcftools index Stercorarius.RAZObowtie.merged.filteredSNPs.auto.10stringent.max17.vcf.gz 

/home/0_PROGRAMS/bcftools-1.14/bcftools view -e 'F_MISSING > 0.30' Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.vcf.gz -Oz -o Stercorarius.RAZObowtie.merged.filteredSNPs.auto.30stringent.max17.vcf.gz 
time /home/0_PROGRAMS/bcftools-1.14/bcftools index Stercorarius.RAZObowtie.merged.filteredSNPs.auto.30stringent.max17.vcf.gz 

/home/0_PROGRAMS/bcftools-1.14/bcftools view -e 'F_MISSING > 0.40' Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.vcf.gz -Oz -o Stercorarius.RAZObowtie.merged.filteredSNPs.auto.40stringent.max17.vcf.gz 
time /home/0_PROGRAMS/bcftools-1.14/bcftools index Stercorarius.RAZObowtie.merged.filteredSNPs.auto.40stringent.max17.vcf.gz 

/home/0_PROGRAMS/bcftools-1.14/bcftools view -e 'F_MISSING > 0.50' Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.vcf.gz -Oz -o Stercorarius.RAZObowtie.merged.filteredSNPs.auto.50stringent.max17.vcf.gz 
time /home/0_PROGRAMS/bcftools-1.14/bcftools index Stercorarius.RAZObowtie.merged.filteredSNPs.auto.50stringent.max17.vcf.gz 


#try stricter genotype depth filters for each of the above levels of missing data
/home/0_PROGRAMS/bcftools-1.14/bcftools filter -S . -e 'FMT/DP<10' Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.vcf.gz -Oz -o Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.min10.max17.vcf.gz 
time /home/0_PROGRAMS/bcftools-1.14/bcftools index Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.min10.max17.vcf.gz 

/home/0_PROGRAMS/bcftools-1.14/bcftools view -e 'F_MISSING > 0.50' Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.min10.max17.vcf.gz -Oz -o Stercorarius.RAZObowtie.merged.filteredSNPs.auto.50stringent.min10.max17.vcf.gz
time /home/0_PROGRAMS/bcftools-1.14/bcftools index Stercorarius.RAZObowtie.merged.filteredSNPs.auto.50stringent.min10.max17.vcf.gz 

/home/0_PROGRAMS/bcftools-1.14/bcftools view -e 'F_MISSING > 0.40' Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.min10.max17.vcf.gz -Oz -o Stercorarius.RAZObowtie.merged.filteredSNPs.auto.40stringent.min10.max17.vcf.gz
time /home/0_PROGRAMS/bcftools-1.14/bcftools index Stercorarius.RAZObowtie.merged.filteredSNPs.auto.40stringent.min10.max17.vcf.gz 

/home/0_PROGRAMS/bcftools-1.14/bcftools view -e 'F_MISSING > 0.30' Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.min10.max17.vcf.gz -Oz -o Stercorarius.RAZObowtie.merged.filteredSNPs.auto.30stringent.min10.max17.vcf.gz
time /home/0_PROGRAMS/bcftools-1.14/bcftools index Stercorarius.RAZObowtie.merged.filteredSNPs.auto.30stringent.min10.max17.vcf.gz 
```
I am also making a couple more test datasets, I want to see the effect of relaxing the filters to ensure that patterns remain consistent, to ensure the filters are not introducing biases that would affect interpretation of the results.  
```bash
#this command would include all chromosomes
#time /home/0_PROGRAMS/bcftools-1.14/bcftools view -m2 -M2 -v snps -e '%QUAL<20 || AVG(FMT/DP)<5 || AVG(FMT/DP)>15 || MQ<20 || AC==0 || AC==AN' ./RAZObowtie.Stercorarius.allsites.vcf.gz | /home/0_PROGRAMS/bcftools-1.14/bcftools filter -S . -e 'FMT/DP<3 | FMT/GQ<15' --SnpGap 10 | /home/0_PROGRAMS/bcftools-1.14/bcftools view -e 'F_MISSING > 0.176' -Oz -o Stercorarius.RAZObowtie.filteredSNPs.all.vcf.gz #took 95m(115m) for Stepar, 83m(94m) for Alctor
#this command filters for autosomes only
time /home/0_PROGRAMS/bcftools-1.14/bcftools view -m2 -M2 -v snps -t chr1,chr2,chr3,chr4,chr5,chr6,chr7,chr8,chr9,chr10,chr11,chr12,chr13,chr14,chr15,chr16,chr17,chr18,chr19,chr20,chr21,chr22,chr23,chr24,chr25 -e '%QUAL<20 || AVG(FMT/DP)<5 || AVG(FMT/DP)>15 || MQ<20 || AC==0 || AC==AN' ./Stercorarius.RAZObowtie.allsites.vcf.gz | /home/0_PROGRAMS/bcftools-1.14/bcftools filter -S . -e 'FMT/DP<3 | FMT/GQ<15' --SnpGap 10 | /home/0_PROGRAMS/bcftools-1.14/bcftools view -e 'F_MISSING > 0.176' -Oz -o Stercorarius.RAZObowtie.filteredSNPs.auto.vcf.gz #took 95m(115m) for Stepar, 83m(94m) for Alctor
#index the files
time /home/0_PROGRAMS/bcftools-1.14/bcftools index Stercorarius.RAZObowtie.filteredSNPs.auto.vcf.gz


#This dataset is similar, but more stringently filtered for quality & mapping quality (25 vs 20), genotype depth (6 vs 3), genotype quality (20 vs 15)
time /home/0_PROGRAMS/bcftools-1.14/bcftools view -m2 -M2 -v snps -t chr1,chr2,chr3,chr4,chr5,chr6,chr7,chr8,chr9,chr10,chr11,chr12,chr13,chr14,chr15,chr16,chr17,chr18,chr19,chr20,chr21,chr22,chr23,chr24,chr25 -e '%QUAL<25 || AVG(FMT/DP)<5 || AVG(FMT/DP)>15 || MQ<25 || AC==0 || AC==AN' ./Stercorarius.RAZObowtie.allsites.vcf.gz | /home/0_PROGRAMS/bcftools-1.14/bcftools filter -S . -e 'FMT/DP<6 | FMT/GQ<20' --SnpGap 10 | /home/0_PROGRAMS/bcftools-1.14/bcftools view -e 'F_MISSING > 0.176' -Oz -o Stercorarius.RAZObowtie.filteredSNPs.auto.stringent.vcf.gz #took 95m(115m) for Stepar, 83m(94m) for Alctor #89m56.273s
time /home/0_PROGRAMS/bcftools-1.14/bcftools index Stercorarius.RAZObowtie.filteredSNPs.auto.stringent.vcf.gz #0m28.050s

```
I am making a few chrZ datasets to test for Dinvestigate:  

```bash
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/1.5d_genotyping

#this dataset is for the Z chromosome
time /home/0_PROGRAMS/bcftools-1.14/bcftools view -m2 -M2 -v snps -t chrZ:30000000- -e '%QUAL<25 || AVG(FMT/DP)>17 || MQ<25 || AC==0 || AC==AN' ./Stercorarius.RAZObowtie.merged.allsites.vcf.gz | /home/0_PROGRAMS/bcftools-1.14/bcftools filter -S . -e 'FMT/DP<6 | FMT/GQ<20' --SnpGap 10 | /home/0_PROGRAMS/bcftools-1.14/bcftools view -e 'F_MISSING > 0.176' -Oz -o Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.stringent.max17.vcf.gz #took 95m(115m) for Stepar, 83m(94m) for Alctor #89m56.273s
time /home/0_PROGRAMS/bcftools-1.14/bcftools index Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.stringent.max17.vcf.gz #0m28.050s
#2,029,048 sites

#less stringent chrZ filter
time /home/0_PROGRAMS/bcftools-1.14/bcftools view -m2 -M2 -v snps -t chrZ -e '%QUAL<20 || AVG(FMT/DP)>17 || MQ<20 || AC==0 || AC==AN' ./Stercorarius.RAZObowtie.merged.allsites.vcf.gz | /home/0_PROGRAMS/bcftools-1.14/bcftools filter -S . -e 'FMT/DP<3' --SnpGap 10 | /home/0_PROGRAMS/bcftools-1.14/bcftools view -Oz -o Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.lax.max17.vcf.gz #took 95m(115m) for Stepar, 83m(94m) for Alctor #89m56.273s
time /home/0_PROGRAMS/bcftools-1.14/bcftools index Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.lax.max17.vcf.gz #0m28.050s

time /home/0_PROGRAMS/bcftools-1.14/bcftools view -m2 -M2 -v snps -t chrZ -e '%QUAL<25 || AVG(FMT/DP)>17 || MQ<25 || AC==0 || AC==AN' ./Stercorarius.RAZObowtie.merged.allsites.vcf.gz | /home/0_PROGRAMS/bcftools-1.14/bcftools filter -S . -e 'FMT/DP<3' --SnpGap 10 | /home/0_PROGRAMS/bcftools-1.14/bcftools view -Oz -o Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.medstringent.min3.max17.vcf.gz #80m31.554s
time /home/0_PROGRAMS/bcftools-1.14/bcftools index Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.medstringent.min3.max17.vcf.gz 

time /home/0_PROGRAMS/bcftools-1.14/bcftools view -m2 -M2 -v snps -t chrZ -e '%QUAL<25 || AVG(FMT/DP)>17 || MQ<25 || AC==0 || AC==AN' ./Stercorarius.RAZObowtie.merged.allsites.vcf.gz | /home/0_PROGRAMS/bcftools-1.14/bcftools filter -S . -e 'FMT/DP<6' --SnpGap 10 | /home/0_PROGRAMS/bcftools-1.14/bcftools view -Oz -o Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.medstringent.max17.vcf.gz 
time /home/0_PROGRAMS/bcftools-1.14/bcftools index Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.medstringent.max17.vcf.gz #77m52.454s

```
This dataset is to double-check the Dinvestigate result by ensuring interpretation is not changed by changing bioinformatic filters:  
```bash
#This dataset is a test for Dinvestigate
#The difference is that I am including the Z chromosome, lowering qual filter to 20, lowering mq filter to 20, lowering min depth filter to 4, and removing gq filter.
time /home/0_PROGRAMS/bcftools-1.14/bcftools view -m2 -M2 -v snps -t chr1,chr2,chr3,chr4,chr5,chr6,chr7,chr8,chr9,chr10,chr11,chr12,chr13,chr14,chr15,chr16,chr17,chr18,chr19,chr20,chr21,chr22,chr23,chr24,chr25,chrZ -e '%QUAL<20 || AVG(FMT/DP)>17 || MQ<20 || AC==0 || AC==AN' ./Stercorarius.RAZObowtie.merged.allsites.vcf.gz | /home/0_PROGRAMS/bcftools-1.14/bcftools filter -S . -e 'FMT/DP<4' --SnpGap 10 | /home/0_PROGRAMS/bcftools-1.14/bcftools view -e 'F_MISSING > 0.176' -Oz -o Stercorarius.RAZObowtie.merged.filteredSNPs.auto.max17.vcf.gz #took 95m(115m) for Stepar, 83m(94m) for Alctor #89m56.273s
time /home/0_PROGRAMS/bcftools-1.14/bcftools index Stercorarius.RAZObowtie.merged.filteredSNPs.auto.max17.vcf.gz #0m28.050s

#just lowering min depth to 4
time /home/0_PROGRAMS/bcftools-1.14/bcftools view -m2 -M2 -v snps -t chr1,chr2,chr3,chr4,chr5,chr6,chr7,chr8,chr9,chr10,chr11,chr12,chr13,chr14,chr15,chr16,chr17,chr18,chr19,chr20,chr21,chr22,chr23,chr24,chr25,chrZ -e '%QUAL<25 || AVG(FMT/DP)>17 || MQ<25 || AC==0 || AC==AN' ./Stercorarius.RAZObowtie.merged.allsites.vcf.gz | /home/0_PROGRAMS/bcftools-1.14/bcftools filter -S . -e 'FMT/DP<4 | FMT/GQ<20' --SnpGap 10 | /home/0_PROGRAMS/bcftools-1.14/bcftools view -e 'F_MISSING > 0.176' -Oz -o Stercorarius.RAZObowtie.merged.filteredSNPs.auto.min4.max17.vcf.gz #
time /home/0_PROGRAMS/bcftools-1.14/bcftools index Stercorarius.RAZObowtie.merged.filteredSNPs.auto.min4.max17.vcf.gz #0m28.050s

```
