# Scaffolding the *Stercorarius parasiticus* Genome
This pipeline outlines how we scaffolded the *S. parasiticus* genome assembly to pseudochromosome scale based on synteny with *Alca torda*.  

**Input:** reference genome from NCBI  
**Output** reference genome scaffolded into pseudochromosomes and ready to be used for read mapping.  

Currently, there is one Stercorariidae reference genome available:  
* ASM1339691v1, GCA_013396915.1, Stercorarius parasiticus  

There are several genomes available from the sister group, Alcidae, of which one is chromosome-scale:  
* bAlcTor1 primary, GCA_008658365.1, Alca torda  

I am downloading the dataset from NCBI using the NCBI datasets tool.  
```sh
mkdir -p /home/0_GENOMES5/Stercorarius/0_Reference_genome/0.1_scaffolding
cd /home/0_GENOMES5/Stercorarius/0_Reference_genome/0.1_scaffolding

/home/0_PROGRAMS/datasets download genome accession GCA_013396915.1 --filename GCA_013396915.1.zip 
/home/0_PROGRAMS/datasets download genome accession GCA_008658365.1 --filename GCA_008658365.1.zip 

unzip GCA_013396915.1.zip
mv ncbi_dataset GCA_013396915.1
unzip GCA_008658365.1.zip
mv ncbi_dataset GCA_008658365.1
rm GCA_013396915.1.zip GCA_008658365.1.zip #save space

#combine scaffolds into a single file
time for chr in chr1 chr2 chr3 chr4 chr5 chr6 chr7 chr8 chr9 chr10 chr11 chr12 chr13 chr14 chr15 chr16 chr17 chr18 chr19 chr20 chr21 chr22 chr23 chr24 chr25 chrZ chrMT ; do sed "s/>/>$chr /1" GCA_008658365.1/data/GCA_008658365.1/"$chr".fna >> GCA_008658365.1/data/GCA_008658365.1/GCA_008658365.1_bAlcTor1_genomic.fna ; done #0m4.831s
cat GCA_008658365.1/data/GCA_008658365.1/unplaced.scaf.fna >> GCA_008658365.1/data/GCA_008658365.1/GCA_008658365.1_bAlcTor1_genomic.fna

```
# Reference-based scaffolding with RagTag  
Now we have a reference assembly, but it is not chromosome scale. To get there, we will rely on other published genomes. Luckily, there is one chromosome scale reference of Alcidae as of now (Dec 2021). I will use the program RagTag to improve contiguity by aligning my genome to the reference (with minimap2) and joining scaffolds that map unambiguously to the adjacent regions of same chromosome, separated by gaps of N's of estimated sizes. Note: RagTag will not break or alter your existing scaffolds at all. It will only join them together.  

Note on picking the aligner: RagTag can use either minimap2 or nucmer. You can adjust the settings, but when I compared the defaults of minimap2 (took 1m1 real 3m36 user) vs nucmer (took 15m54 real 205m17 user) with my songbird dataset, minimap2 actually gave a more contiguous assembly with 8684 scaffolds, N50=74Mb, while nucmer only joined the biggest scaffolds resulting in 10612 scaffolds, N50=73 Mb. So, the N50 statistics are very similar but the difference is that minimap2 incorporated more of the small scaffolds. In terms of correctness, the minimap2 assembly appears to not show any interchromosomal rearrangements that could be detected by inspecting the jupiter plots, while three supposed tiny translocations from the autosomes to the Z chromosome appeared with the nucmer assembly - these may be errors since they are not supported by any other data. Based on this, I decided to go with minimap2 since it is 1) 15 times faster 2) the default of RagTag 3) gives a more contiguous genome 4) does not give any suspicious interchromosomal rearrangements.  

```bash
#set up a directory
mkdir -p /home/0_GENOMES5/Stercorarius/0_Reference_genome/0.1_scaffolding/ragtag
cd /home/0_GENOMES5/Stercorarius/0_Reference_genome/0.1_scaffolding/ragtag

#get a copy of your genome assembly using soft links to save space
ln -s /home/0_GENOMES5/Stercorarius/0_Reference_genome/0.1_scaffolding/GCA_013396915.1/data/GCA_013396915.1/GCA_013396915.1_ASM1339691v1_genomic.fna .

#get a copy of your reference genome assembly using soft links to save space
ln -s /home/0_GENOMES5/Stercorarius/0_Reference_genome/0.1_scaffolding/GCA_008658365.1/data/GCA_008658365.1/GCA_008658365.1_bAlcTor1_genomic.fna .

#now scaffold further with RagTag, with each available assembly
conda activate ragtag

#Here I am saving all my assembly names as variables to more easily change the pipeline
DRAFT1=GCA_013396915.1_ASM1339691v1_genomic.fna #this is the target assembly to scaffold
ref2=GCA_008658365.1_bAlcTor1_genomic.fna #this is the best, chromosome-scale reference available

#run RagTag
time python /home/0_PROGRAMS/RagTag/ragtag.py scaffold "$ref2" "$DRAFT1" -f 1000 -q 10 -d 100000 -r -g 100 -m 100000 -o "$DRAFT1"_ragtag1_"$ref2" -t 20 --aligner /home/0_PROGRAMS/minimap2-2.17_x64-linux/minimap2 -u 

#I'm using samtools faidx just to be able to view the lengths of scaffolds 
/home/0_PROGRAMS/samtools-1.12/samtools faidx "$DRAFT1"_ragtag1_"$ref2"/ragtag.scaffold.fasta

#check the resulting continguity of the assemblies at each step
#Ideally, the contiguity should be similar to a chromosome-scale assembly (ie, on the order of N50=74 Mb). If it is much lower, that is too bad - maybe you need better references, or better quality (eg sequencing depth) of your draft assembly. If it is much higher, then maybe something went wrong (unless you had a real chromosomal fusion in your taxon).
conda activate abyss #conda install -y -c bioconda abyss #conda activate abyss on new server, was /home/0_PROGRAMS/else_conda/conda/envs/abyss on old server
abyss-fac "$DRAFT1"
abyss-fac "$ref2"
abyss-fac "$DRAFT1"_ragtag1_"$ref2"/ragtag.scaffold.fasta
#see "assessment.md" page for results
```
Ragtag settings  

* the first two arguments are the (not gzipped) reference and the target assembly to scaffold.  
* `-f `: minimum length of unique alignment to consider for scaffolding  
* `-q`: min mapping quality  
* `-d`: syntenic alignments within this distance of each other on the query and merged into one alignment.  
* `-r`: infer gap sizes  
* `-g`: min gap size  
* `-m`: max gap size  
* `-o`: output directory  
* `-t`: number of threads for the mapping step  
* `--aligner`: path to minimap2 (or nucmer)  
* `-j`: txt file listing query scaffolds to skip (leave unplaced)  

**Timing**: Running ragtag with minimap2 took around 1m to 1m30 for each iteration.   

# Rename  
Now that we have the reference genome, I will just clean up the scaffold names a bit for aesthetics.  
```bash
mkdir -p /home/0_GENOMES5/Stercorarius/0_Reference_genome/0_reference
cd /home/0_GENOMES5/Stercorarius/0_Reference_genome/0_reference

/home/0_PROGRAMS/seqkit grep -r -p "chr[1-9]" ../0.1_scaffolding/ragtag/"$DRAFT1"_ragtag1_"$ref2"/ragtag.scaffold.fasta | /home/0_PROGRAMS/seqkit sort --by-name --natural-order | sed 's/_RagTag//g' > Stercorarius_parasiticus.ref.fa
/home/0_PROGRAMS/seqkit grep -r -p "chrZ" ../0.1_scaffolding/ragtag/"$DRAFT1"_ragtag1_"$ref2"/ragtag.scaffold.fasta | sed 's/_RagTag//g' >> Stercorarius_parasiticus.ref.fa
/home/0_PROGRAMS/seqkit grep -r -p "chrMT" ../0.1_scaffolding/ragtag/"$DRAFT1"_ragtag1_"$ref2"/ragtag.scaffold.fasta | sed 's/_RagTag//g' >> Stercorarius_parasiticus.ref.fa
/home/0_PROGRAMS/seqkit grep -v -r -p "chr" ../0.1_scaffolding/ragtag/"$DRAFT1"_ragtag1_"$ref2"/ragtag.scaffold.fasta | /home/0_PROGRAMS/seqkit sort -l --reverse - | sed 's/_RagTag//g' >> Stercorarius_parasiticus.ref.fa
```

# Index  
Now I will index the reference genome to be used for downstream mapping steps.  
```bash
cd /home/0_GENOMES5/Stercorarius/0_Reference_genome/0_reference

#index the reference
time /home/0_PROGRAMS/samtools-1.14/samtools faidx Stercorarius_parasiticus.ref.fa #0m4.733s

#Before we can map our sequences, we must prepare the reference genome for Bowtie2:
time /home/0_PROGRAMS/bowtie2-2.4.4-linux-x86_64/bowtie2-build Stercorarius_parasiticus.ref.fa Stercorarius_parasiticus.ref #39m56.672s

#or bwa
time /home/0_PROGRAMS/bwa/bwa index Stercorarius_parasiticus.ref.fa #25m29.456s

```