


First, I would like to assess the genome annotations. These can  be downloaded from the ftp site at NCBI:

```bash

#proteins
wget https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/013/396/915/GCA_013396915.1_ASM1339691v1/GCA_013396915.1_ASM1339691v1_translated_cds.faa.gz
#RNA
wget https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/013/396/915/GCA_013396915.1_ASM1339691v1/GCA_013396915.1_ASM1339691v1_cds_from_genomic.fna.gz
```

# Assembly contiguity


```bash
/home/0_PROGRAMS/bbmap/stats.sh /home/0_GENOMES5/Stercorarius/0_Reference_genome/0_reference/Stercorarius_parasiticus.ref.fa -Xmx20g 
/home/0_PROGRAMS/bbmap/stats.sh /home/0_GENOMES5/Stercorarius/0_Reference_genome/0.1_scaffolding/GCA_008658365.1/data/GCA_008658365.1/GCA_008658365.1_bAlcTor1_genomic.fna -Xmx20g 

```
Here are the results of `bbmap stats.sh` for Stercorarius:
```
A	C	G	T	N	IUPAC	Other	GC	GC_stdev
0.2895	0.2103	0.2105	0.2897	0.0178	0.0000	0.0000	0.4208	0.1142

Main genome scaffold total:         	5916
Main genome contig total:           	41919
Main genome scaffold sequence total:	1172.165 MB
Main genome contig sequence total:  	1151.357 MB  	1.775% gap
Main genome scaffold N/L50:         	5/81.736 MB
Main genome contig N/L50:           	3126/107.044 KB
Main genome scaffold N/L90:         	14/34.47 MB
Main genome contig N/L90:           	11477/25.161 KB
Max scaffold length:                	214.282 MB
Max contig length:                  	737.599 KB
Number of scaffolds > 50 KB:        	51
% main genome in scaffolds > 50 KB: 	99.35%


Minimum 	Number        	Number        	Total         	Total         	Scaffold
Scaffold	of            	of            	Scaffold      	Contig        	Contig  
Length  	Scaffolds     	Contigs       	Length        	Length        	Coverage
--------	--------------	--------------	--------------	--------------	--------
    All 	         5,916	        41,919	 1,172,164,891	 1,151,356,498	  98.22%
    100 	         5,916	        41,919	 1,172,164,891	 1,151,356,498	  98.22%
    250 	         4,761	        40,764	 1,171,909,723	 1,151,101,330	  98.22%
    500 	         3,082	        38,998	 1,171,308,648	 1,150,512,611	  98.22%
   1 KB 	         1,559	        36,987	 1,170,235,845	 1,149,504,063	  98.23%
 2.5 KB 	           687	        35,635	 1,168,846,680	 1,148,380,804	  98.25%
   5 KB 	           312	        34,697	 1,167,568,794	 1,147,388,545	  98.27%
  10 KB 	           158	        34,113	 1,166,534,819	 1,146,562,032	  98.29%
  25 KB 	            72	        33,632	 1,165,322,010	 1,145,584,192	  98.31%
  50 KB 	            51	        33,387	 1,164,569,238	 1,145,012,214	  98.32%
 100 KB 	            43	        33,185	 1,164,013,636	 1,144,693,154	  98.34%
 250 KB 	            30	        32,768	 1,162,183,230	 1,143,565,156	  98.40%
 500 KB 	            28	        32,614	 1,161,471,981	 1,142,971,936	  98.41%
   1 MB 	            25	        32,245	 1,159,479,667	 1,141,530,346	  98.45%
 2.5 MB 	            24	        31,657	 1,157,123,321	 1,139,542,553	  98.48%
   5 MB 	            23	        30,977	 1,153,958,656	 1,136,970,696	  98.53%
  10 MB 	            16	        26,659	 1,099,729,789	 1,085,901,356	  98.74%
  25 MB 	            14	        25,649	 1,073,785,387	 1,060,347,807	  98.75%
  50 MB 	             7	        17,825	   788,487,196	   778,432,115	  98.72%
 100 MB 	             3	        10,894	   501,727,704	   495,778,043	  98.81%

```
and Alca torda:
```
A	C	G	T	N	IUPAC	Other	GC	GC_stdev
0.2866	0.2132	0.2134	0.2868	0.0030	0.0000	0.0000	0.4267	0.0787

Main genome scaffold total:         	95
Main genome contig total:           	322
Main genome scaffold sequence total:	1178.493 MB
Main genome contig sequence total:  	1174.931 MB  	0.302% gap
Main genome scaffold N/L50:         	4/84.527 MB
Main genome contig N/L50:           	23/18.221 MB
Main genome scaffold N/L90:         	14/34.323 MB
Main genome contig N/L90:           	78/3.374 MB
Max scaffold length:                	215.872 MB
Max contig length:                  	48.161 MB
Number of scaffolds > 50 KB:        	56
% main genome in scaffolds > 50 KB: 	99.90%


Minimum 	Number        	Number        	Total         	Total         	Scaffold
Scaffold	of            	of            	Scaffold      	Contig        	Contig  
Length  	Scaffolds     	Contigs       	Length        	Length        	Coverage
--------	--------------	--------------	--------------	--------------	--------
    All 	            95	           322	 1,178,492,568	 1,174,931,441	  99.70%
  10 KB 	            95	           322	 1,178,492,568	 1,174,931,441	  99.70%
  25 KB 	            82	           308	 1,178,215,599	 1,174,654,572	  99.70%
  50 KB 	            56	           281	 1,177,319,460	 1,173,758,929	  99.70%
 100 KB 	            44	           263	 1,176,524,500	 1,172,966,965	  99.70%
 250 KB 	            31	           237	 1,174,312,523	 1,170,760,664	  99.70%
 500 KB 	            28	           230	 1,173,180,414	 1,169,630,148	  99.70%
   1 MB 	            25	           217	 1,170,952,296	 1,167,409,107	  99.70%
 2.5 MB 	            25	           217	 1,170,952,296	 1,167,409,107	  99.70%
   5 MB 	            23	           195	 1,165,155,097	 1,161,665,047	  99.70%
  10 MB 	            16	           154	 1,109,021,400	 1,105,993,260	  99.73%
  25 MB 	            14	           150	 1,082,648,368	 1,079,693,360	  99.73%
  50 MB 	             7	           114	   796,932,213	   794,395,638	  99.68%
 100 MB 	             3	            69	   506,433,921	   505,632,688	  99.84%

```

Here are the results of `abyss-fac` for the starting assembly, Alca torda reference, and RagTag-scaffolded Stercorarius:

| n     | n:500 | L50 | min   | N75      | N50      | N25      | E-size   | max      | sum      | name                                                                                                        |
|-------|-------|-----|-------|----------|----------|----------|----------|----------|----------|-------------------------------------------------------------------------------------------------------------|
| 11280 | 7660  | 507 | 500   | 343805   | 648067   | 1154041  | 852404   | 4190736  | 1.15E+09 | GCA_013396915.1_ASM1339691v1_genomic.fna                                                                    |
| 95    | 95    | 4   | 13591 | 4.39E+07 | 8.37E+07 | 1.65E+08 | 1.05E+08 | 2.16E+08 | 1.18E+09 | GCA_008658365.1_bAlcTor1_genomic.fna                                                                        |
| 5916  | 2884  | 4   | 500   | 4.38E+07 | 8.15E+07 | 1.61E+08 | 1.03E+08 | 2.12E+08 | 1.15E+09 | GCA_013396915.1_ASM1339691v1_genomic.fna_ragtag1_GCA_008658365.1_bAlcTor1_genomic.fna/ragtag.scaffold.fasta |


# Jupiter plot
```bash
#set up your environment
#conda activate circos #my conda environment has circos, via 'install -y -c bioconda circos'
conda activate /home/0_PROGRAMS/else_conda/conda/envs/circos #running on Troglodytes node in a temp folder

#requires samtools and minimap2 to be in the PATH so that JupiterPlot can use them
export PATH=/home/0_PROGRAMS/samtools-1.14:$PATH
export PATH=/home/0_PROGRAMS/minimap2-2.23_x64-linux:$PATH

#if you have too many scaffolds, circos by default will have an error and not want to plot them. You must change this parameter in the housekeeping.conf that belongs to circos (it ended up in my conda env in the etc folder) like this to exceed how many scaffolds you have:
sed -i 's/max_ideograms.*$/max_ideograms        = 10000/g' /home/0_PROGRAMS/else_conda/conda/envs/circos/etc/housekeeping.conf

#lets compare the Stercrarius assembly to Alca torda
mkdir -p /home/0_GENOMES5/Stercorarius/0_Reference_genome/0.5_assessment/jupiter
cd /home/0_GENOMES5/Stercorarius/0_Reference_genome/0.5_assessment/jupiter
ln -s /home/0_GENOMES5/Stercorarius/0_Reference_genome/0.1_scaffolding/GCA_008658365.1/data/GCA_008658365.1/GCA_008658365.1_bAlcTor1_genomic.fna .
ln -s /home/0_GENOMES5/Stercorarius/0_Reference_genome/0.1_scaffolding/ragtag/GCA_013396915.1_ASM1339691v1_genomic.fna_ragtag1_GCA_008658365.1_bAlcTor1_genomic.fna/ragtag.scaffold.fasta .

time /home/0_PROGRAMS/JupiterPlot/jupiter name=Stercorarius_Alca ng=99 ref=./GCA_008658365.1_bAlcTor1_genomic.fna fa=./ragtag.scaffold.fasta

#to make the plot more aesthetically with fewer gaps you can fiddle with the maxGap parameter
time /home/0_PROGRAMS/JupiterPlot/jupiter name=Stercorarius_Alca_pretty ng=99 ref=./GCA_008658365.1_bAlcTor1_genomic.fna fa=./ragtag.scaffold.fasta maxGap=500000 t=8 &

#You can also check synteny with a model organism, like Zebra Finch, to better know which chromosomes are which
wget https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/008/822/105/GCF_008822105.2_bTaeGut2.pat.W.v2/GCF_008822105.2_bTaeGut2.pat.W.v2_genomic.fna.gz
gunzip GCF_008822105.2_bTaeGut2.pat.W.v2_genomic.fna.gz

time /home/0_PROGRAMS/JupiterPlot/jupiter name=Stercorarius_Taeniopygia ng=99 ref=GCF_008822105.2_bTaeGut2.pat.W.v2_genomic.fna fa=ragtag.scaffold.fasta m=400000 minBundleSize=10000 maxGap=200000 t=24

#test a different zebra finch assembly to make sure it's consistent
wget https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/003/957/565/GCF_003957565.2_bTaeGut1.4.pri/GCF_003957565.2_bTaeGut1.4.pri_genomic.fna.gz
gunzip GCF_003957565.2_bTaeGut1.4.pri_genomic.fna.gz

time /home/0_PROGRAMS/JupiterPlot/jupiter name=Stercorarius_Taeniopygia2 ng=99 ref=GCF_003957565.2_bTaeGut1.4.pri_genomic.fna fa=ragtag.scaffold.fasta m=400000 minBundleSize=10000 maxGap=200000 t=8 #10m51.120s (32m59.726s)

#save space by removing large files when you are done with them
rm ./*.sam
```
* `ng`: what proportion of the genome to plot? ng=99 will plot the scaffolds that make up 99% of the reference genome (skipping the smallest scaffolds to avoid too many meaningless tiny lines)
* `ref`: reference genome assembly to compare to (should probably be the more contiguous of the pair)
* `fa`: query genome of interest to compare to the reference
* `name`: prefix to name the output files

**Results**: it looks like there are several very tiny interchromosomal rearrangements visible on the Jupiter plot, but overall the pseudochromosomes of Stercorarius map very closely to the assembled chromosomes of Alca. Of course, we cannot know whether the synteny is truly this high, and whether the several small interchromosomal rearrangements are real (and whether they should be larger) without scaffolding a Stercorarius genome *de novo*, but this assembly will be sufficient for our purposes.
