### Overview
This section details how a maximum likelihood phylogeny was constructed for the Stercorariidae mitogenomes.

### Required Input
This step requires the complete mitochondrial genome sequences, stored in the file `Stercorariidae_MitogenomesCorrectedSinglecopy_withOutgroups.mafft.afa`.

### Output
The output is a phylogeny of the mitogenomes, in which branch lengths represent substitutions per site. This file is stored in `Stercorariidae_mtDNA.tre`.


# Maximum Likelihood Phylogeny of mitogenome

Before running the time-calibrated BEAST phylogeny, I am making a much faster maximum likelihood phylogeny. This is necessary to confirm the basal split within the Stercorariidae, since I am going to root the BEAST phylogeny using this basal split. I am rooting it based on the deepest split within Stercorariidae, rather than the divergence to Alcidae, since the sister of Stercorariidae (Alcidae) is quite distant and I prefer not to include it in the BEAST analysis due to the possibility for variation in substitution rate over such vast timescales (c. 35 Mya) and because a plot of nuclear divergence vs mitochondrial divergence shows some evidence of saturation in mitochondrial distances: the plot is curved such that the mitochondrial distance between Alcidae and Stercorariidae is lower than expected based on the slope of mtDNA vs nuclear divergence within Stercorariidae. This is to be expected due to the high mutation rate of mtDNA in birds, and strong sequence conservation of mtDNA genes.

## Prepare alignment
First, we need an outgroup. I selected three of the Alcid mitogenomes that are currently (Jul 2020) available on NCBI:
1) Alca torda `Alca torda isolate 2006_11_07_Graehlm_vinge_78 mitochondrion, complete sequence, whole genome shotgun sequence` (CM018102.1), which seems to have assembled the entire mitogenome including both repeats
2) `NC_045517.1 Aethia cristatella voucher IN4608 mitochondrion, complete genome` which seems to have assembled the second repeat and not the first (assuming that Aethia really does retain two repeats in its mitogenome)
3) `NC_029328.1 Synthliboramphus wumizusume mitochondrion, complete genome` which also seems to have assembled the second repeat (rather than the first), but I am not positive since the telltale `CGGGC` sequence that differentiates repeat 2 from repeat 1 doesn't align very well to the other samples.

I did not include Cassin's Auklet because it is labelled "unverified" and because the branch lengths were very long when I made preliminary trees for all the Charadriiformes sequences last year, making me concerned that it contains numts or a faster rate of evolution. I also did not include the Great Auk because it was assembled from ancient DNA so could possibly contain ancient DNA damage (although it is probably a good assembly) and I did not include the *Uria* since it has a stretch at the repeat that does not align well and I am concerned that the assembly is not correct. I did not include `NC_007978.1 Synthliboramphus antiquus mitochondrion, complete genome` because I only wanted one *Synthliboramphus* and arbitrarily chose the other one.

To ensure the circular genomes start at the same point, I first aligned them with the Stercorariidae, and then cut/pasted the trailing sequences at the start of the alignment to the end of the alignment. 

Next, we need to trim the sequences. In the Stercorariidae (and also in their sister Alcidae), I found that the mitogenome contains a large duplication, and so I removed one copy of the duplicated region. Since this region appears to evolve under concerted evolution (the two copies are identical within an individual despite differences between individuals), it could be considered to not represent independantly evolving sites. Including both copies could be seen to be a form of pseudoreplication, so I am excluding one copy. This is the structure of the duplication:
1) identical duplicate 5' end 8860-10197 in POJA_4 (8860-10196 in GRSK_MKP1593) (8864-10203 in the Stercorarius alignment)
2) unique middle with run of C's 10198-10431 in POJA_4 (10197-10441 in GRSK_MKP1593) (10204-10450 in the Stercorarius alignment)
3) identical duplicate 3' end 10432-11167 in POJA_4 (10442-11179 in GRSK_MKP1593) (10451-11219 in the Stercorarius alignment)
4) short unique block at breakpoint between two copies of duplication 11164-11192 in POJA_4 (11220-11244 in the Stercorarius alignment)
5) identical duplicate 5' end 11193-12531 in POJA_4 (11245-12585 in the Stercorarius alignment)
6) unique middle with run of C's 12532-12652 in POJA_4 (12586-12720 in the Stercorarius alignment)
7) identical duplicate 3' end 12653-13470 in POJA_4 (-13482 in GRSK_MKP1593) (12721-13571 in Stercorarius alignment)
Then it goes back to single-copy DNA.


I removed sections 1-4 for the purposes of the phylogeny (even though it's single copy, I removed block 4 since it was not assembled in auklets/murrelets)

Finally, we need to align the sequences. I am using mafft since it is quite quick and accurate with these types of sequences. After aligning, I gave a quick scan by eye to ensure that everything looked reasonable. 
```bash
mkdir -p /home/0_GENOMES5/Stercorarius/2_Phylogeny/2.1.1_mtDNA_ML_phylogeny
cd /home/0_GENOMES5/Stercorarius/2_Phylogeny/2.1.1_mtDNA_ML_phylogeny #on Troglodytes
cd /home/1_Else/0_GENOMES5/Stercorarius/2_Phylogeny/2.1.1_mtDNA_ML_phylogeny #on new server

#align using mafft
time /home/0_PROGRAMS/mafft-7.490-with-extensions/bin/mafft --auto --thread 24 Stercorariidae_MitogenomesCorrectedSinglecopy_withOutgroups.fasta > Stercorariidae_MitogenomesCorrectedSinglecopy_withOutgroups.mafft.afa
#Strategy:
# FFT-NS-2 

```
**Timing**: 2.3 seconds to align. 
Output of terminal:
```
nthread = 24
nthreadpair = 24
nthreadtb = 16
ppenalty_ex = 0
stacksize: 8192 kb
generating a scoring matrix for nucleotide (dist=200) ... done
Gap Penalty = -1.53, +0.00, +0.00



Making a distance matrix ..

There are 30 ambiguous characters.
    1 / 19 (thread    0)
done.

Constructing a UPGMA tree (efffree=0) ... 
   10 / 19
done.

Progressive alignment 1/2... 
STEP    17 / 18 (thread    2) f
Reallocating..done. *alloclen = 34732
STEP    18 / 18 (thread    0) f
done.

Making a distance matrix from msa.. 
    0 / 19 (thread    1)
done.

Constructing a UPGMA tree (efffree=1) ... 
   10 / 19
done.

Progressive alignment 2/2... 
STEP    15 / 18 (thread   14) f
Reallocating..done. *alloclen = 34724
STEP    18 / 18 (thread    4) f
done.

disttbfast (nuc) Version 7.490
alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0
16 thread(s)


Strategy:
 FFT-NS-2 (Fast but rough)
 Progressive method (guide trees were built 2 times.)

If unsure which option to use, try 'mafft --auto input > output'.
For more information, see 'mafft --help', 'mafft --man' and the mafft page.

The default gap scoring scheme has been changed in version 7.110 (2013 Oct).
It tends to insert more gaps into gap-rich regions than previous versions.
To disable this change, add the --leavegappyregion option.


real  0m2.653s
user  0m4.293s
```

## Build Phylogeny
With our alignment, we can now make a phylogeny. I am using IQtree2 to build a maximum likelihood phylogeny with 1000 bootstrap replicates.

```bash
#first get a quick first look
time /home/0_PROGRAMS/iqtree-2.1.3-Linux/bin/iqtree2 -s Stercorariidae_MitogenomesCorrectedSinglecopy_withOutgroups.mafft.afa -nt AUTO
#1m31.789s

#now build a tree more thoroughly, with 1000 bootstrap replicates
time /home/0_PROGRAMS/iqtree-2.1.3-Linux/bin/iqtree2 -s Stercorariidae_MitogenomesCorrectedSinglecopy_withOutgroups.mafft.afa -nt AUTO -b 1000 -redo
#2652 parsimony-informative, 1664 singleton sites, 12698 constant sites
#Best-fit model: TIM2+F+I+G4 chosen according to BIC
#took 92m23.342s (268m43.486s)
```
**Results:**

Quick tree (without bootstrap replicates):
```
(LTJA_MKP990_c:0.0352172361,(PAJA_B20730_c:0.0010656727,PAJA_USNM606730:0.0012542487):0.0435739681,(((((POJA_MKP1559:0.0014084558,POJA_4:0.0002555472):0.0002285070,POJA_IB2659:0.0008752304):0.0016221608,(GRSK_MKP1592:0.0001351507,GRSK_MKP1593:0.0001749906):0.0034832656):0.0063295620,(((((CISK55:0.0000010124,CISK3:0.0000010124):0.0003668152,CHSK_MKP2451:0.0004558753):0.0000623214,ANSK_NC_026125_1_FromNCBI_NotComplete:0.0005529177):0.0007206361,((ANSK8:0.0008087760,ANSK01:0.0006163417):0.0003934980,ANSK7:0.0006592636):0.0002048613):0.0003921604,CISK2:0.0007880889):0.0049305575):0.0458711626,((Alca_torda:0.1134354916,NC_029328.1:0.1144968937):0.0239570532,NC_045517.1:0.1076701477):0.1135525519):0.0277267624);
```
Consensus tree:  
```
(LTJA_MKP990:0.0350055976,(PAJA_B20730_c:0.0010659849,PAJA_USNM606730:0.0012502796)100:0.0433761172,(((((POJA_MKP1559:0.0014065817,POJA_4:0.0002550831)91:0.0002265251,POJA_IB2659:0.0008756760)99:0.0016215553,(GRSK_MKP1592:0.0001349209,GRSK_MKP1593:0.0001747268)100:0.0034752072)100:0.0063064174,(((((CISK55:0.0000028057,CISK3:0.0000028057)99:0.0003656041,CHSK_MKP2451:0.0004546291)63:0.0000620663,ANSK_NC_026125_1:0.0005522057)100:0.0007193737,((ANSK8:0.0008075267,ANSK01:0.0006153681)97:0.0003919807,ANSK7:0.0006583913)89:0.0002043385)86:0.0003921981,CISK2:0.0007874825)99:0.0049272764)100:0.0455767372,((Alca_torda:0.1126628399,NC_029328.1:0.1137493198)100:0.0241287072,NC_045517.1:0.1069286572)100:0.1126836952)100:0.0276365921);

```
ML tree with 1000 bootstrap replicates:  
```
(LTJA_MKP990:0.0350076127,(PAJA_B20730_c:0.0010649311,PAJA_USNM606730:0.0012509385)100:0.0433705382,(((((POJA_MKP1559:0.0014066293,POJA_4:0.0002550826)91:0.0002266395,POJA_IB2659:0.0008752432)99:0.0016218945,(GRSK_MKP1592:0.0001356846,GRSK_MKP1593:0.0001746169)100:0.0034735157)100:0.0063043952,(((((CISK55:0.0000009995,CISK3:0.0000009995)99:0.0003661667,CHSK_MKP2451:0.0004553602)63:0.0000627442,ANSK_NC_026125_1e:0.0005525740)100:0.0007200231,((ANSK8:0.0008081503,ANSK01:0.0006152919)97:0.0003920620,ANSK7:0.0006575157)89:0.0002044640)86:0.0003907636,CISK2:0.0007882157)99:0.0049240013)100:0.0455723188,((Alca_torda:0.1126390474,NC_029328.1:0.1137214403)100:0.0241339700,NC_045517.1:0.1068977649)100:0.1126400549)100:0.0276346793);
```

Note that the three southern lineages (CHSK, CISK, ANSK) are not reciprocally monophyletic.

## Prepare Figure

I'm making the figure in Figtree. Open the treefile -> name the labels "bootstrap". Check "node labels" -> Display -> bootstrap. Line Weight -> 3. Tip labels -> Font -> 12; Times New Roman italic. Selected the basal Alcidae branch -> Reroot. Trees -> Rooting: "Midpoint". Then I rotated some nodes for aesthetics, to make the tree easier to interpret (to make it faster at a glance to see which groups are not monophyletic). Saved as `Stercorariidae_mtDNA.tre`. Export pdf -> `Stercorariidae_mtDNA.tre.pdf`
Then, I added icons of the species in Illustrator, formatted the names, and removed any bootstrap labels that were higher than 97% support (leaving only 4 less certain nodes with labels).

I also made a nexus format file to upload to TreeBase (Stercorariidae_Mitogenomes_ML_phylogeny_TreeBase.nex), following their instructions for generating the nexus file in Mesquite.
