## Overview
This section details how the MC1R gene trees were made.

### Required Input
This step requires the referece genome, and the data mapped to the reference. I am using data in the VCF file `Stercorarius.PAJAbowtie.merged.allsites.vcf.gz` generated in step 1.5c. This VCF file is too large to store on the Github repository, so the analysis can instead be replicated starting from the file `Stercorarius.PAJAbowtie.MC1R.vcf.gz`, which is the VCF file subsetted to just the region around MC1R.

### Output
The output contains alignments of the MC1R gene and flanking sequences, and phylogenies for these alignments. The sequence alignment for MC1R is stored in `alignment/MC1R.POJA4_haplotypes_corrected.fasta` (our samples only) and `alignment/MC1R.POJA4_haplotypes_corrected.NCBI_MC1R.mafft.afa` (our samples and shorter sequences from NCBI). The corresponding phylogenies are stored in `MC1R.POJA4_haplotypes_corrected.fasta.treefile` and `MC1R.POJA4_haplotypes_corrected.NCBI_MC1R.mafft.N.afa.treefile`.


In the paper "The genetic basis and enigmatic origin of melanic polymorphism in pomarine skuas (Stercorarius pomarinus)" by Janssen and Mundy (2017) there is an interesting result: a dark-morph Pomarine Jaeger MC1R haplotype comes out very close to the "Catharacta" skuas, while the light morph comes out closer to S. parasiticus/longicaudus in a haplotype network. Could the light morph have introgressed from the small jaegers into the Pomarine Jaeger? Could the dark morph have introgressed from the Catharacta skuas into the Pomarine Jaeger? Is it an ancient balanced polymorphism?

Here is a quote from this paper:

>"Introgression has been inferred among skuas [9,33] and is quite common in seabirds in general [34]. If the polymorphism did arise by introgression then, given the close relationship between pomarine and great skua alleles at other loci, the most likely direction of introgression is of the pale MC1R allele into the pomarine lineage. This hypothesis is supported by the signal of directional selection of pale alleles in the haplotype network... MC1R haplotypes of the long-tailed skua are the most closely related to the pale allele in the pomarine skua, and these skuas have a pale phenotype. However, although the breeding ranges of pomarine and long-tailed skuas overlap, no hybrids or breeding attempts between the species have been reported, and they have substantial ecological and biological differences [7,8]. Overall therefore, ancestral polymorphism and introgres-sion of a pale MC1R allele are both plausible mechanisms for the origin of the MC1R polymorphism in pomarine skuas but each have certain weaknesses. Genome-wide data on genetic variation would be very useful in discriminating between these hypotheses and is an exciting prospect for the future"

Now we have genome-wide data on the pale-morph S. pomarinus, dark-morph S. pomarinus, and the Catharacta skuas, so lets take a broader look.

Here, I am interested in extracting the MC1R (melanocortin 1 receptor) sequence of the samples, just to take a look at what we can find.

First, we need to find out where MC1R is in the reference genome, whose gene annotation is not yet released. This is a simple task with BLAST.

Firstly, we must obtain some MC1R sequences to use as a reference. I will obtain MC1R orthologs on [NCBI](https://www.ncbi.nlm.nih.gov/gene/4157/ortholog/?scope=117570). MC1R has no introns, which will make it easier to extract.

This is what NCBI says about MC1R:
>"This intronless gene encodes the receptor protein for melanocyte-stimulating hormone (MSH). The encoded protein, a seven pass transmembrane G protein coupled receptor, controls melanogenesis. Two types of melanin exist: red pheomelanin and black eumelanin. Gene mutations that lead to a loss in function are associated with increased pheomelanin production, which leads to lighter skin and hair color. Eumelanin is photoprotective but pheomelanin may contribute to UV-induced skin damage by generating free radicals upon UV radiation. Binding of MSH to its receptor activates the receptor and stimulates eumelanin synthesis. This receptor is a major determining factor in sun sensitivity and is a genetic risk factor for melanoma and non-melanoma skin cancer. Over 30 variant alleles have been identified which correlate with skin and hair color, providing evidence that this gene is an important component in determining normal human pigment variation. [provided by RefSeq, Jul 2008]"

I will take this sequence for *Charadrius vociferus*, which is in the same order (Charadriiformes), and is the closest (along with *Calidris pugnax*) annotated genome in Refseq at the moment (Feb 2020). It should certainly be close enough to locate MC1R in the auk.

```bash
#make a working directory and enter it
mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/2_Phylogeny/2.2c_MC1R_GeneTree #on main server
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/2_Phylogeny/2.2c_MC1R_GeneTree #on main server

#download or copy-paste a fasta sequence of the reference MC1R gene
cat > Chavoc_MC1R.fasta #Charadrius vociferus MC1R protein sequence
```
```
>XP_009888634.1 PREDICTED: melanocyte-stimulating hormone receptor [Charadrius vociferus]
MSTLAPLRLLREPWNASEGNQSNATAGAGSAWCQGLNIPNELFLTLGLVSLVENLLVVAAILKNRNLHSP
TYYFICCLAVSDMLVSVSNLVETLFMLLMEHGLLVIHASIVRHMDNVIDMLICSSIVSSLSFLGVIAVDR
YITIFYALRYHSIMTLQRAVVTMASVWLASTISSTIFITYYRNNAILLCLIGFFLFMLVLMLVLYIHMFA
LARHHLCSISSQQKQPTIYRSSSLKGAITLTILLGVFFICWGPFFFHLILIVTCPTNPFCTCFFSYFNLF
LILIICNSVVDPLIYVFRSQELRRTLREVVLCSW
```

# Step 1: Locate sequences

Firstly, the reference genome must be prepared with a database for BLAST with the command `makeblastdb`.
```bash
/home/0_PROGRAMS/ncbi-blast-2.12.0+/bin/makeblastdb -in ../../0_Reference_genome/0_reference/Stercorarius_parasiticus.ref.fa -dbtype nucl #took 22s
```

Then, blast the MC1R protein sequence to the genome to find the best match. Since it is using a protein as the query and nucleotide as the database, we will use tblastn. The query sequence is 314 AA long, so I am hoping for a hit that is also 314 AA long.

Note: if you are dealing with many proteins/many genomes, there are better ways of extracting sequences. For example, see a tutorial [here](https://github.com/mmatschiner/tutorials/tree/master/ortholog_detection#q2). A simple method is sufficient for me as I only have one protein that I specifically want to see.
```bash
/home/0_PROGRAMS/ncbi-blast-2.12.0+/bin/tblastn -query Chavoc_MC1R.fasta -db ../../0_Reference_genome/0_reference/Stercorarius_parasiticus.ref.fa -outfmt "6 qseqid sseqid evalue bitscore pident qstart qend sstart send qseq sseq" > MC1R2Stercorarius_parasiticus.tsv
```
**Results**: MC1R: very straightforward, the top hit has 95.9% identity (e-value=\~0.0 (as low as detectable), bitscore=582 (very high)) and it covers the entire query sequence from residue 1 to 314. 

```
XP_009888634.1  chr7    0.0     582     95.860  1       314     45887798        45888739        MSTLAPLRLLREPWNASEGNQSNATAGAGSAWCQGLNIPNELFLTLGLVSLVENLLVVAAILKNRNLHSPTYYFICCLAVSDMLVSVSNLVETLFMLLMEHGLLVIHASIVRHMDNVIDMLICSSIVSSLSFLGVIAVDRYITIFYALRYHSIMTLQRAVVTMASVWLASTISSTIFITYYRNNAILLCLIGFFLFMLVLMLVLYIHMFALARHHLCSISSQQKQPTIYRSSSLKGAITLTILLGVFFICWGPFFFHLILIVTCPTNPFCTCFFSYFNLFLILIICNSVVDPLIYVFRSQELRRTLREVVLCSW      MSTLAPLHLLREPWNASEGNQSNTTAGAGGPWCQGLNIPNELFLTLGLVSLVENLLVVAAILKNRNLHSPTYYFICCLAVSDMLVSVSNLVETLFMLLMEHGVLVIHASIVRHMDNVIDMLICSSIMSSLSFLGVIAVDRYITIFYALRYHSIMTLQRAVVTMASVWLASTVSSTIFITYYRNNAILLCLIGFFLFVLVLMLVLYIHMFILSRHHLCSISSQQKQPTIYRSSSLKGTITLTILLGVFFICWGPFFFHLILIVTCPTNPFCTCFFSYFNLFLILIICNSVVNPLIYVFRSQELRRMLREVVLCSW
XP_009888634.1  chr7    2.66e-07        57.0    28.426  128     306     2543596 2544156 SSLSFLGVIAVDRYITIFYALRYHSIMTLQRAVVTMASVW----------------LASTISSTIFITYYRNNAILLCLIGFFLFMLVLMLVLYIHMFALARHHLCSISSQ-QKQPTIYRSSSLKGAITLTILLGVFFICWGPF-FFHLILIVTCPTNPFCTCFFSYFNLFLILIICNSVVDPLIYVFRSQELRRTL   ASVCSLLAITVDRYLSLYNALTYHSERTLGFTCAMVLLMWLLCLGVGLLPLLGWNCLRDQSACSILRPVTKDNAAVLAV--TFLLLFALMMQLYLQICKIAFRHAQQIAVQHQFIATAQATSTRKGLSTLSLILGTFALCWIPFAIYSLVADSSYPV--------VYTYSLALPATCNSLINPIIYAFRNPDIQKSL

```
(note, when I used auk as the reference, I had 98.408% identity (e-value=\~0.0 (as low as detectable), bitscore=595 (very high)), the sequence is located on scaffold CM018082.1 from 46,292,412 to 46,293,353 bp (942 bp). The next-best hit has a bit score of only 57.4 and 28% identity, which is much worse than the best hit, so there is no reasonable worry about paralogs here.)

# Step 2: Phase sequences

Now, we would like to extract nucleotide sequences from these identified regions to get a fasta file containing the sequences of all species. Each sample has two copies of MC1R, and the two copies can have different evolutionary histories. We need to disentangle the two haplotypes in order to reliably construct the tree - particularly since we have samples heterozygous for the dark vs pale morph alleles. I am using Shapeit since it can take advantage of the information contained by the raw sequencing reads - since we have small sample sizes, phasing would otherwise be unreliable.

# Phasing

To run shapeit, we need an input bamlist file:
```bash
cat > bamlist
```
This file has three columns: 1) sample name 2) absolute path to bam file 3) chromosome of interest
```
LTJA_MKP990 /home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.3c_mapping/merged/LTJA_MKP990.PAJAbowtie.marked.bam chr7
PAJA_B20730 /home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.3c_mapping/merged/PAJA_B20730.PAJAbowtie.marked.bam chr7
PAJA_USNM606730 /home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.3c_mapping/merged/PAJA_USNM606730.PAJAbowtie.marked.bam chr7
POJA_MKP1559 /home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.3c_mapping/merged/POJA_MKP1559.PAJAbowtie.marked.bam chr7
POJA_4 /home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.3c_mapping/merged/POJA_4.PAJAbowtie.marked.bam chr7
POJA_IB2659 /home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.3c_mapping/merged/POJA_IB2659.PAJAbowtie.marked.bam chr7
GRSK_MKP1592 /home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.3c_mapping/merged/GRSK_MKP1592.PAJAbowtie.marked.bam chr7
GRSK_MKP1593 /home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.3c_mapping/merged/GRSK_MKP1593.PAJAbowtie.marked.bam chr7
CISK2 /home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.3c_mapping/merged/CISK2.PAJAbowtie.marked.bam chr7
CISK55 /home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.3c_mapping/merged/CISK55.PAJAbowtie.marked.bam chr7
CISK_3 /home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.3c_mapping/merged/CISK3.PAJAbowtie.marked.bam chr7
CHSK_MKP2451 /home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.3c_mapping/merged/CHSK_MKP2451.PAJAbowtie.marked.bam chr7
ANSK7 /home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.3c_mapping/merged/ANSK7.PAJAbowtie.marked.bam chr7
ANSK01 /home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.3c_mapping/merged/ANSK01.PAJAbowtie.marked.bam chr7
ANSK8 /home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.3c_mapping/merged/ANSK8.PAJAbowtie.marked.bam chr7
Alca_torda /home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.3c_mapping/merged/Alca_torda.PAJAbowtie.marked.bam chr7
```
Now we run shapeit:
```bash
cd /home/1_Else/0_GENOMES5/Stercorarius/2_Phylogeny/2.2c_MC1R_GeneTree 

#First, extract our region of interest from the vcf
/home/0_PROGRAMS/bcftools-1.14/bcftools view -r chr7:45877798-45898739 ../../1_Data/1.5c_genotyping/Stercorarius.PAJAbowtie.merged.allsites.vcf.gz -e 'MQ<20' -O z > Stercorarius.PAJAbowtie.MC1R.vcf.gz #kept 16113 sites
/home/0_PROGRAMS/bcftools-1.14/bcftools index Stercorarius.PAJAbowtie.MC1R.vcf.gz

#quickly look at mapping quality scores to see if filtering should be done
zgrep -v "^#" Stercorarius.PAJAbowtie.MC1R.vcf.gz | cut -f 8 | sed 's/^.*MQ=//g; s/;.*$//g' | sort | less

#phasing only applies to heterozygous sites, so keep only sites where there are heterozygotes to save time
zgrep "^#" Stercorarius.PAJAbowtie.MC1R.vcf.gz > Stercorarius.PAJAbowtie.MC1R.heterozygotes.vcf
/home/0_PROGRAMS/bcftools-1.14/bcftools view -v snps -m 2 -M 2 Stercorarius.PAJAbowtie.MC1R.vcf.gz | grep -v "^#" | grep -e "0/1" -e "1/0" >> Stercorarius.PAJAbowtie.MC1R.heterozygotes.vcf

#now, extract phase informative reads in this region
time /home/0_PROGRAMS/extractPIRs.v1.r68.x86_64/extractPIRs --bam bamlist --vcf Stercorarius.PAJAbowtie.MC1R.heterozygotes.vcf --out Stercorarius.PAJAbowtie.MC1R.heterozygotes.PIRsList --base-quality 20 --read-quality 20 #took 1s

#finally, phase the vcf
time /home/0_PROGRAMS/shapeit.v2.904.3.10.0-693.11.6.el7.x86_64/bin/shapeit -assemble --input-vcf Stercorarius.PAJAbowtie.MC1R.heterozygotes.vcf --input-pir Stercorarius.PAJAbowtie.MC1R.heterozygotes.PIRsList -O Stercorarius.PAJAbowtie.MC1R.heterozygotes.shapeit.phased --force

#and convert the output back to VCF format
/home/0_PROGRAMS/shapeit.v2.904.3.10.0-693.11.6.el7.x86_64/bin/shapeit -convert --input-haps Stercorarius.PAJAbowtie.MC1R.heterozygotes.shapeit.phased --output-vcf Stercorarius.PAJAbowtie.MC1R.heterozygotes.shapeit.phased.vcf
bgzip Stercorarius.PAJAbowtie.MC1R.heterozygotes.shapeit.phased.vcf
/home/0_PROGRAMS/bcftools-1.14/bcftools index Stercorarius.PAJAbowtie.MC1R.heterozygotes.shapeit.phased.vcf.gz

#combine the SNPs back together with the invariants
zgrep -v "0/1" Stercorarius.PAJAbowtie.MC1R.vcf.gz | grep -v "INDEL" | bgzip > Stercorarius.PAJAbowtie.MC1R.noheterozygotes.vcf.gz
/home/0_PROGRAMS/bcftools-1.14/bcftools index Stercorarius.PAJAbowtie.MC1R.noheterozygotes.vcf.gz
/home/0_PROGRAMS/bcftools-1.14/bcftools concat --allow-overlaps Stercorarius.PAJAbowtie.MC1R.noheterozygotes.vcf.gz Stercorarius.PAJAbowtie.MC1R.heterozygotes.shapeit.phased.vcf.gz -O z > Stercorarius.PAJAbowtie.MC1R.heterozygotes.shapeit.phased.all.vcf.gz 
/home/0_PROGRAMS/bcftools-1.14/bcftools index Stercorarius.PAJAbowtie.MC1R.heterozygotes.shapeit.phased.all.vcf.gz 


```
Now we have a VCF file, but we need to be aware that all the missing data at the SNP sites has been imputed - I do not want that! I will have to replace it with missing data downstream.

# Step 3: generate haplotypes

Now we will generate the haplotype sequences for our samples.

```bash
#generate haplotype 1
parallel --colsep " " 'time /home/0_PROGRAMS/samtools-1.14/samtools faidx ../../0_Reference_genome/0_reference/Stercorarius_parasiticus.ref.fa chr7:45877798-45898739 | /home/0_PROGRAMS/bcftools-1.14/bcftools consensus --sample {1} --haplotype 1 --absent N --missing N Stercorarius.PAJAbowtie.MC1R.heterozygotes.shapeit.phased.all.vcf.gz | sed "s/>.*$/>{1}.1/g" > {1}.1.PAJAbowtie.MC1R.fa' ::: LTJA_MKP990 PAJA_B20730 PAJA_USNM606730 POJA_MKP1559 POJA_4 POJA_IB2659 GRSK_MKP1592 GRSK_MKP1593 CISK2 CISK55 CISK_3 CHSK_MKP2451 ANSK7 ANSK01 ANSK8 Alca_torda

#generate haplotype 2
parallel --colsep " " '/home/0_PROGRAMS/samtools-1.14/samtools faidx ../../0_Reference_genome/0_reference/Stercorarius_parasiticus.ref.fa chr7:45877798-45898739 | /home/0_PROGRAMS/bcftools-1.14/bcftools consensus --sample {1} --haplotype 2 --absent N --missing N Stercorarius.PAJAbowtie.MC1R.heterozygotes.shapeit.phased.all.vcf.gz | sed "s/>.*$/>{1}.2/g" > {1}.2.PAJAbowtie.MC1R.fa' ::: LTJA_MKP990 PAJA_B20730 PAJA_USNM606730 POJA_MKP1559 POJA_4 POJA_IB2659 GRSK_MKP1592 GRSK_MKP1593 CISK2 CISK55 CISK_3 CHSK_MKP2451 ANSK7 ANSK01 ANSK8 Alca_torda

```
We are not quite done! Will still have to take out the imputed genotypes. I will do this with bedtools:
1) make a bed file listing all the snp sites that have missing data for a given sample
2) use bedtools to mask these sites in the fasta files.
Remember:
* bed files are *zero-based* for the start position. Meaning, if you want to mask the 10th nucleotide, you need to call it "9" in the bed start site
* bed files are *one-based* for the end position, Meaning, if you want to mask the 10th nucleotide, you need to call it "9" in the bed end site
So, this bed entry would mask just the tenth nucleotide: `chromosome1 9 10`


```bash
#generate a bed file listing all the positions where a sample has a missing genotype
#this command views the VCF file for one particular sample - now all the sites either have 0% or 100% missing data. Exclude the sites with less than 50% missing data (so, exclude all the sites with a genotype call for that sample) and then use awk to turn that VCF file of missing sites into a bed file listing all of the missing sites for that sample. Remember to subtract one from the start position to match the BED zero-based system!
mkdir -p phasing
parallel /home/0_PROGRAMS/bcftools-1.14/bcftools view --samples {1} Stercorarius.PAJAbowtie.MC1R.heterozygotes.vcf '|' /home/0_PROGRAMS/bcftools-1.14/bcftools view --exclude \''F_MISSING<0.5'\'  '|' awk \''$0 !~ /^#/ {print $1":",$2-1,$2}'\' '>' phasing/{1}.PAJAbowtie.MC1R.mask.bed ::: LTJA_MKP990 PAJA_B20730 PAJA_USNM606730 POJA_MKP1559 POJA_4 POJA_IB2659 GRSK_MKP1592 GRSK_MKP1593 CISK2 CISK55 CISK_3 CHSK_MKP2451 ANSK7 ANSK01 ANSK8 Alca_torda
#make bed files tab-delimited
parallel 'sed -i "s/ /\t/g" phasing/{1}.PAJAbowtie.MC1R.mask.bed' ::: LTJA_MKP990 PAJA_B20730 PAJA_USNM606730 POJA_MKP1559 POJA_4 POJA_IB2659 GRSK_MKP1592 GRSK_MKP1593 CISK2 CISK55 CISK_3 CHSK_MKP2451 ANSK7 ANSK01 ANSK8 Alca_torda

#mask the fasta files by replacing these sites with N's. This will undo the imputation by Shapeit.
mkdir -p haplotypes
#first the first haplotype
parallel 'time /home/0_PROGRAMS/bedtools maskfasta -fi {1}.1.PAJAbowtie.MC1R.fa -bed phasing/{1}.PAJAbowtie.MC1R.mask.bed -fo haplotypes/{1}.1.PAJAbowtie.masked.fa' ::: LTJA_MKP990 PAJA_B20730 PAJA_USNM606730 POJA_MKP1559 POJA_4 POJA_IB2659 GRSK_MKP1592 GRSK_MKP1593 CISK2 CISK55 CISK_3 CHSK_MKP2451 ANSK7 ANSK01 ANSK8 Alca_torda
#then the second haplotype
parallel 'time /home/0_PROGRAMS/bedtools maskfasta -fi {1}.2.PAJAbowtie.MC1R.fa -bed phasing/{1}.PAJAbowtie.MC1R.mask.bed -fo haplotypes/{1}.2.PAJAbowtie.masked.fa' ::: LTJA_MKP990 PAJA_B20730 PAJA_USNM606730 POJA_MKP1559 POJA_4 POJA_IB2659 GRSK_MKP1592 GRSK_MKP1593 CISK2 CISK55 CISK_3 CHSK_MKP2451 ANSK7 ANSK01 ANSK8 Alca_torda

#put the sample name in the fasta header
parallel 'sed -i "s/>.*$/>{1}.1/g" haplotypes/{1}.1.PAJAbowtie.masked.fa' ::: LTJA_MKP990 PAJA_B20730 PAJA_USNM606730 POJA_MKP1559 POJA_4 POJA_IB2659 GRSK_MKP1592 GRSK_MKP1593 CISK2 CISK55 CISK_3 CHSK_MKP2451 ANSK7 ANSK01 ANSK8 Alca_torda
parallel 'sed -i "s/>.*$/>{1}.2/g" haplotypes/{1}.2.PAJAbowtie.masked.fa' ::: LTJA_MKP990 PAJA_B20730 PAJA_USNM606730 POJA_MKP1559 POJA_4 POJA_IB2659 GRSK_MKP1592 GRSK_MKP1593 CISK2 CISK55 CISK_3 CHSK_MKP2451 ANSK7 ANSK01 ANSK8 Alca_torda

#Combine all samples together
mkdir -p alignment
#cat chromosomes | parallel 'cat haplotypes/{1}.*.'$tagname'.lowconfmasked.fa > haplotypes/{1}.fa'
#make a copy in which the fasta entries have a single line per sequence, not multiline fasta
cat haplotypes/*.PAJAbowtie.masked.fa |  awk '/^>/ {printf("%s%s\t",(N>0?"\n":""),$0);N++;next;} {printf("%s",$0);} END {printf("\n");}' | tr "\t" "\n" > alignment/MC1R.fasta #16m3.865s (6m35.168s)

```

Now we have an alignment of "phased" haplotypes for each of our samples at MC1R and flanking regions. The first 10 kb are flanking DNA upstream of MC1R, and the start codon starts at position 10,001. The stop codon is at 10,943-10,945 and 10,946-20,942 is flanking DNA downstream of MC1R.

# Examine results

Now, we can open the alignment in a sequence viewer, such as Aliview or Mesquite. If the alignment looks very gappy, it could be filtered such as with Gblocks or TrimAl, but in my case the CDS alignment is beautiful - no gaps, it all looks well aligned. This is expected since out fasta sequences in the first place are based on being mapped to the same reference genome. There are no internal stop codons in the coding sequence. The problem is that there are very few SNPs in the coding region! To build a phylogeny we will need to rely on signal from the flanking regions. Note that we are making the assumption that the MC1R region shares the same phylogenetic history as the flanking regions. This is likely reasonable for the closest flanking regions, at least at distances that we are concerned with. 

The MC1R gene of the Pomarine Jaeger has an interesting feature in that dark-morph and pale-morph birds have different haplotypes which have been identified as discussed earlier. To determine whether I have a pale-morph or dark-morph bird, I downloaded the fasta sequences from the previous Janssen and Mundy (2017) paper and added them to the alignment `Stercorariidae_MC1R.mafft.afa` (simply copy-pasted the fasta entry into Aliview, and then I just realigned within Aliviewer using MUSCLE). All but one of the *S. pomarinus* sequence came out within the pale-morph haplotypes, and when I checked the nonsynonymous changes that had been described between their pale and dark morph haplotypes, all but one of our sequences had the pale morph alleles. One of the birds - chosen because it had dark-morph phenotypes - is heterozygous, carrying one dark morph and one light morph haplotype.

The dark-morph individual, POJA_4.2 appears to be heterozygous for the pale and dark haplotypes. Note that POJA_IB2659 is a juvenile that was chosen because the museum specimen looked relatively dark, but it appears that it in fact is homozygous for the pale morph allele.

POJA_4.2 matches the "Catharacta" alleles at these positions of the alignment:

* 9443
* 9478
* 9746
* 9799
* 9877
* 10,034 (nonsynonymous, dark morph is derived)
* 10,046 (nonsynonymous, dark morph is derived)
* 10,084 (synonymous, light morph is derived)
* at 10,211, POJA_4 is homozygous for a derived nonsynonymous allele fixed in all other pomarinus
* 10,238 (nonsynonymous, dark morph is derived)
* 10,348 (synonymous, light morph derived)
* at 10,455, POJA_4 is homozygous ancestral while all Catharacta carry a derived nonsybonymous allele. 
* 10,545 (nonsynonymous, derived in dark)
* 10,588 (synonymous, derived in dark)
* 10,688 (nonsynonymous, derived in dark)
* 10,841 (nonsynonymous, derived in dark)
* 10,887-10,878 (nonsynonymous, derived in dark)
* 10,965 (synonymous, derived in pale)
* 10,970 
* 10,983 (MKP1559 also heterozygous)
* 11,148
* 11,217

The phasing switches such that POJA_4.1 matches the "Catharacta" alleles at:
* 9221
* 9215
This could be a phase switch error, or perhaps that is the boundary of the haplotype, and there was a recombination event between 9221 and 9443 at some point.
Also at 6567, but seems seems like a separate tract.
At 9055, it is homozygous for a derived allele shared with all Catharacta.

I can't be positive whether the haplotype at 9221-9215 is the same (unbroken by recombination) haplotype as 9443.

I trimmed the alignment to stretch from 9443-11217 bp, encompasing the region where the dark morph haplotype shares alleles with "Catharacta", and ending at the last of those sites before what I presume to be recombination breakpoints.

# Mannually verify genotypes
Since POJA_4 is heterozygous for the light and dark morph haplotypes, any genotype errors that miss calling a heterozygous site (eg, due to low sequencing coverage or uneven allele balance) could lead to errors that will inflate branch lengths of one or the other allele. This is less of an issue for other samples that have low heterozygosity in this region, but being heterozygous for the divergent light and dark allele makes this an important issue, especially since we are interested in understanding the different phylogenetic positions of the light vs dark allele.

I opened the bam file (sequences aligned to the reference genome) in IGV (Integrated Genomics Viewer) and mannually looked over the 1775 bp region. I made sure that every allele that was called as being different from the reference genome (*S. parasiticus*) was indeed present in the raw reads, and that every site that had multiple reads differing from the reference was called in the genotypes. I found four sites where the genotype caller was too conservative and missed a heterozygous genotype. In all four cases, POJA_4 had been given the reference allele, which happenned to match the pale morph *S. pomarinus*. The alternate allele matched *S. antarcticus* in all cases. Making these 4 changes greatly decreased the branch length of the dark morph *S. pomarinus*, leaving only a 1 bp difference to *S. antarcticus* (rather than 5 bp).

To summarise, here are the 5 genotype calls that were different from *S. antarcticus*:
* 9: 3 reads have T, 4 reads have C. I'm surprised it was called homozygous C! The T's are linked to G at position 1, so I am changing POJA_4.2 position 9 to T.
* 336: 4 reads have T, 10 reads have C. It was called homozygous C. The T's are linked to A at position 357 and 304, so I will change POJA_4.2 at position 336 to T.
* 769: confirmed that this allele is linked to both haplotypes, homozygous (present on reads with both states for a linked SNP), and covered by 15 reads, so unlikely to be actually heterozygous.
* 966: this looks to be a genotyping error. In the aligned reads of the bam file, 6 show T and 10 show C. I am quite surprised that the genotype was called as homozygous C! The T is on the same reads that have C at position 906. I will make this correction, and change POJA_4.2 to T at position 966. 
* 1013: 10 reads have G, 5 have A. The A's are linked to the T's of position 966. Clearly another genotyping error - I will correct in to A at position 1013 of POJA_4.2.

After that double-checking, POJA_4.2 is in fact only 1 bp away from the CISK haplotypes!

The corrected alignment is saved n `alignment/MC1R.POJA4_haplotypes_corrected.fasta`.

# Phylogenetic tree

Now that we have our alignment, we can build a phylogenetic tree.
```bash
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/2_Phylogeny/2.2c_MC1R_GeneTree 

#first run a quick ML analysis
#first testing the uncorrected dataset (this dataset has genptyping errors in POJA_4)
/home/0_PROGRAMS/iqtree-2.1.3-Linux/bin/iqtree2 -s alignment/MC1R.POJA4_haplotypes.fasta -nt AUTO -redo
#(ANSK01.1:0.0000010000,ANSK01.2:0.0000010000,(((ANSK7.1:0.0000010000,CISK2.2:0.0000010000):0.0000010000,((ANSK8.1:0.0000010000,ANSK8.2:0.0000010000):0.0005609271,((ANSK7.2:0.0000010000,((CISK55.1:0.0000010000,CISK2.1:0.0000010000):0.0000010000,POJA_4.2:0.0028432985):0.0005593590):0.0005511946,((CHSK_MKP2451.1:0.0005628470,CHSK_MKP2451.2:0.0000010000):0.0004373899,((((POJA_IB2659.1:0.0000010000,(POJA_IB2659.2:0.0017159287,POJA_MKP1559.1:0.0011419930):0.0000010000):0.0000010000,POJA_4.1:0.0005697762):0.0000010000,POJA_MKP1559.2:0.0005691461):0.0029238898,(LTJA_MKP990.1:0.0040948394,(LTJA_MKP990.2:0.0005841075,((((PAJA_B20730.1:0.0000010000,PAJA_B20730.2:0.0005756396):0.0011541314,PAJA_USNM606730.2:0.0000010000):0.0000010000,PAJA_USNM606730.1:0.0017415924):0.0017290721,(Alca_torda.1:0.0019900503,Alca_torda.2:0.0000010000):0.0316905994):0.0011970845):0.0000010000):0.0030015129):0.0086093833):0.0029952263):0.0005727394):0.0000010000):0.0000010000,((GRSK_MKP1592.1:0.0000010000,GRSK_MKP1592.2:0.0000010000):0.0000010000,GRSK_MKP1593.1:0.0005591130):0.0005600241):0.0000010000);

#then the mannually corrected haplotypes. The branch length on POJA_4.2 is much shorter than the uncorrected dataset, after genotyping errors were fixed.
/home/0_PROGRAMS/iqtree-2.1.3-Linux/bin/iqtree2 -s alignment/MC1R.POJA4_haplotypes_corrected.fasta -nt AUTO -redo
#(PAJA_USNM606730.2:0.0000010020,(((((((((((((CISK_3.2:0.0000000000,CISK2.1:0.0000000000):0.0000000000,CISK_3.1:0.0000000000):0.0000000000,CISK55.2:0.0000000000):0.0000010020,CISK55.1:0.0000010020):0.0000010020,POJA_4.2:0.0005614570):0.0005603585,ANSK7.2:0.0000010020):0.0005548097,((ANSK01.1:0.0000010020,ANSK01.2:0.0000010020):0.0000010020,((ANSK7.1:0.0000010020,CISK2.2:0.0000010020):0.0000010020,((ANSK8.1:0.0000010020,ANSK8.2:0.0000010020):0.0005615057,(((GRSK_MKP1592.1:0.0000000000,GRSK_MKP1593.2:0.0000000000):0.0000010020,GRSK_MKP1592.2:0.0000010020):0.0000010020,GRSK_MKP1593.1:0.0005601846):0.0005608464):0.0000010020):0.0000010020):0.0005697776):0.0029467450,(CHSK_MKP2451.1:0.0005628980,CHSK_MKP2451.2:0.0000010020):0.0004732195):0.0084153909,((((POJA_4.1:0.0005682063,POJA_IB2659.2:0.0017093011):0.0000010020,POJA_IB2659.1:0.0000010020):0.0000010020,POJA_MKP1559.1:0.0011379294):0.0000010020,POJA_MKP1559.2:0.0005675545):0.0028958051):0.0029538791,LTJA_MKP990.1:0.0040553031):0.0000021033,LTJA_MKP990.2:0.0005780392):0.0011114991,(Alca_torda.1:0.0019688336,Alca_torda.2:0.0000010020):0.0299680117):0.0017871349,PAJA_USNM606730.1:0.0017275178):0.0000010020,(PAJA_B20730.1:0.0000010020,PAJA_B20730.2:0.0005721086):0.0011468138);


#now add bootstrap replicates to the analysis
/home/0_PROGRAMS/iqtree-2.1.3-Linux/bin/iqtree2 -s alignment/MC1R.POJA4_haplotypes_corrected.fasta -nt AUTO -b 1000 -redo
#(PAJA_USNM606730.2:0.0000021203,((((((((((((CISK_3.2:0.0000021203,CISK55.2:0.0000021203)10:0.0000021203,(CISK55.1:0.0000021203,CISK_3.1:0.0000021203)13:0.0000021203)6:0.0000021203,CISK2.1:0.0000021203)20:0.0000021203,POJA_4.2:0.0005627654)42:0.0005615368,ANSK7.2:0.0000021203)56:0.0005547183,((((ANSK01.1:0.0000021203,ANSK01.2:0.0000021203)89:0.0000021203,(ANSK7.1:0.0000021203,CISK2.2:0.0000021203)28:0.0000021203)18:0.0000021203,(ANSK8.1:0.0000021203,ANSK8.2:0.0000021203)67:0.0005628084)36:0.0000021203,(((GRSK_MKP1592.1:0.0000021203,GRSK_MKP1593.2:0.0000021203)20:0.0000021203,GRSK_MKP1592.2:0.0000021203)37:0.0000021203,GRSK_MKP1593.1:0.0005613416)43:0.0005620737)60:0.0005725515)99:0.0029677805,(CHSK_MKP2451.1:0.0005642369,CHSK_MKP2451.2:0.0000021203)98:0.0004662799)100:0.0085003548,(((POJA_4.1:0.0005700498,POJA_IB2659.1:0.0000021203)25:0.0000021203,(POJA_IB2659.2:0.0017156223,POJA_MKP1559.1:0.0011419914)16:0.0000021203)34:0.0000027411,POJA_MKP1559.2:0.0005684580)72:0.0029128511)94:0.0029751955,LTJA_MKP990.1:0.0040788054)92:0.0000021203,LTJA_MKP990.2:0.0005807308)95:0.0011481723,(Alca_torda.1:0.0019803709,Alca_torda.2:0.0000021203)100:0.0307211851)73:0.0017654603,PAJA_USNM606730.1:0.0017363048)47:0.0000021203,(PAJA_B20730.1:0.0000021203,PAJA_B20730.2:0.0005747040)90:0.0011513801);

```
Note that the output is unrooted - we will have to mannually root it using *Alca torda* as the outgroup.

Note that CISK55, CISK_3, ANSK01, ANSK8, and GRSK_MKP1592 have no heterozygous sites. Also note that CISK55 and CISK_3 are identical, and so is one of CISK2's haplotypes. One of GRSK_MKP1593's haplotypes is identical to GRSK_MKP1592.

# Add previously-published MC1R haplotypes
Now I would like to add more context by adding the previously-published dataset.
I placed these previously-published sequences into `NCBI_MC1R.fasta`.
```bash
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/2_Phylogeny/2.2c_MC1R_GeneTree 

#combine our samples with NCBI haplotypes
cat alignment/MC1R.POJA4_haplotypes_corrected.fasta NCBI_MC1R.fasta > alignment/MC1R.POJA4_haplotypes_corrected.NCBI_MC1R.fasta

#align sequences
time /home/0_PROGRAMS/mafft-7.490-with-extensions/bin/mafft --auto --thread 24 alignment/MC1R.POJA4_haplotypes_corrected.NCBI_MC1R.fasta > alignment/MC1R.POJA4_haplotypes_corrected.NCBI_MC1R.mafft.afa

#make a quick ML phylogeny
/home/0_PROGRAMS/iqtree-2.1.3-Linux/bin/iqtree2 -s alignment/MC1R.POJA4_haplotypes_corrected.NCBI_MC1R.mafft.afa -nt AUTO -redo

#convert gaps to N's
sed 's/-/N/g' alignment/MC1R.POJA4_haplotypes_corrected.NCBI_MC1R.mafft.afa > alignment/MC1R.POJA4_haplotypes_corrected.NCBI_MC1R.mafft.N.afa

#add bootstrap
/home/0_PROGRAMS/iqtree-2.1.3-Linux/bin/iqtree2 -s alignment/MC1R.POJA4_haplotypes_corrected.NCBI_MC1R.mafft.N.afa -nt AUTO -b 100 -redo
#(PAJA_USNM606730.2:0.0000028636,((((((((((((((CISK_3.2:0.0000025259,CISK_3.1:0.0000025259)15:0.0000025259,(CISK2.1:0.0000025259,CISK55.2:0.0000025259)13:0.0000025259)6:0.0000025259,CISK55.1:0.0000025259)28:0.0000025259,MG515660.1:0.0000025319)14:0.0000025259,(POJA_4.2:0.0000025259,MG515679.1:0.0000025259)55:0.0005787131)37:0.0005768173,ANSK7.2:0.0000020706)47:0.0005658959,(((((ANSK01.1:0.0000025259,ANSK01.2:0.0000025259)76:0.0000025259,(ANSK7.1:0.0000025259,CISK2.2:0.0000025259)27:0.0000025259)24:0.0000025259,(ANSK8.1:0.0000022156,ANSK8.2:0.0000029570)67:0.0005786754)28:0.0000025259,((((GRSK_MKP1592.1:0.0000025259,(GRSK_MKP1592.2:0.0000025259,GRSK_MKP1593.2:0.0000025259)21:0.0000025259)44:0.0000025259,MG515658.1:0.0000025259)21:0.0000025259,(GRSK_MKP1593.1:0.0000021968,MG515659.1:0.0000020125)67:0.0005764188)39:0.0005763677,AY521219.1:0.0000025259)27:0.0000011863)21:0.0000025259,MG515661.1:0.0000025259)60:0.0005943077)94:0.0031382559,(CHSK_MKP2451.1:0.0005809121,CHSK_MKP2451.2:0.0000025259)96:0.0004250887)100:0.0091040270,((((((POJA_4.1:0.0005896584,POJA_IB2659.1:0.0000020062)22:0.0000025259,((POJA_IB2659.2:0.0008855201,MG515680.1:0.0009700856)46:0.0008943471,(POJA_MKP1559.2:0.0000022523,(MG515675.1:0.0000025819,MG515676.1:0.0009646908)27:0.0000025259)28:0.0005896528)6:0.0000025259)2:0.0000025259,(MG515662.1:0.0009679395,MG515678.1:0.0009675981)2:0.0000025259,MG515663.1:0.0000025259,(MG515664.1:0.0009675688,(MG515673.1:0.0009678749,MG515677.1:0.0019446774)2:0.0000025259)2:0.0000025259,(MG515666.1:0.0009678891,MG515672.1:0.0019562373)1:0.0000025259,MG515674.1:0.0009741166)2:0.0000025259,((POJA_MKP1559.1:0.0000025696,MG515671.1:0.0029337407)32:0.0011787710,MG515670.1:0.0009679791)4:0.0000025259)1:0.0000025259,(MG515667.1:0.0000020500,(MG515668.1:0.0000029300,MG515669.1:0.0009686522)55:0.0009715432)30:0.0009705729)24:0.0009848814,MG515665.1:0.0009733012)42:0.0020645834)82:0.0031390714,(LTJA_MKP990.1:0.0021334151,(((KF268977.1:0.0000026886,KF268982.1:0.0009807701)21:0.0000025259,KF268981.1:0.0009804258)22:0.0000025259,KF268979.1:0.0029702228)32:0.0009566697)35:0.0021664989)15:0.0000025259,(((LTJA_MKP990.2:0.0005990739,KF268978.1:0.0000025259)11:0.0000010507,(((KF268980.1:0.0019759173,KF268987.1:0.0019766468)22:0.0000025259,KF268985.1:0.0039896011)15:0.0000025259,KF268986.1:0.0019767189)42:0.0009869955)5:0.0000010691,(AY521218.1:0.0000026917,(KF268983.1:0.0000028406,KF268984.1:0.0009810907)63:0.0019824927)13:0.0000020703)12:0.0000012880)61:0.0011200896,(Alca_torda.1:0.0020965230,Alca_torda.2:0.0000029579)100:0.0363783342)43:0.0013503221,((AY521217.1:0.0000025259,KF268971.1:0.0009687820)42:0.0000025259,(KF268968.1:0.0000025259,KF268969.1:0.0009717148)47:0.0000025259)64:0.0019566731)21:0.0005764066,(((PAJA_USNM606730.1:0.0000025259,KF268970.1:0.0000025259)36:0.0000025259,KF268976.1:0.0009733052)52:0.0018066127,AY521214.1:0.0000025524)11:0.0000029852)1:0.0000025259,(((((((PAJA_B20730.1:0.0000021016,PAJA_B20730.2:0.0005977332)48:0.0000025259,((AY521215.1:0.0000025259,KF268974.1:0.0009810634)13:0.0000025259,(KF268964.1:0.0000025259,(KF268965.1:0.0009769505,KF268966.1:0.0009772953)6:0.0000025259)7:0.0000025259)17:0.0000025259)61:0.0011920313,KF268967.1:0.0029489459)10:0.0000021502,((KF268963.1:0.0000029149,KF268973.1:0.0009741234)58:0.0009741319,KF268972.1:0.0009777305)11:0.0000025259)3:0.0000025259,KF268975.1:0.0009741683)2:0.0000025259,KF268961.1:0.0000025259)3:0.0000025259,(AY521216.1:0.0000025600,KF268962.1:0.0000026769)45:0.0009440001)3:0.0000024200);

#more bootstraps
time /home/0_PROGRAMS/iqtree-2.1.3-Linux/bin/iqtree2 -s alignment/MC1R.POJA4_haplotypes_corrected.NCBI_MC1R.mafft.N.afa -nt AUTO -b 1000
#(PAJA_USNM606730.2:0.0000025259,(((((((((((((((CISK_3.2:0.0000025259,CISK55.2:0.0000025259)12:0.0000025259,CISK_3.1:0.0000025259)6:0.0000025259,(CISK55.1:0.0000025259,CISK2.1:0.0000025259)10:0.0000025259)26:0.0000025259,MG515660.1:0.0000025259)10:0.0000025259,(POJA_4.2:0.0000025259,MG515679.1:0.0000021675)61:0.0005801798)29:0.0000025259,ANSK7.2:0.0005775814)36:0.0005641435,(((((ANSK01.1:0.0000025259,ANSK01.2:0.0000025259)76:0.0000025259,(ANSK7.1:0.0000025259,CISK2.2:0.0000025259)28:0.0000025259)22:0.0000025259,(ANSK8.1:0.0000023679,ANSK8.2:0.0000026772)68:0.0005801415)19:0.0000025259,MG515661.1:0.0000025259)17:0.0005780900,(((((GRSK_MKP1592.1:0.0000025259,GRSK_MKP1593.2:0.0000025259)21:0.0000025259,GRSK_MKP1592.2:0.0000025259)37:0.0000025259,MG515658.1:0.0000025259)13:0.0000025259,(GRSK_MKP1593.1:0.0000022092,MG515659.1:0.0000024674)64:0.0005778717)28:0.0000025259,AY521219.1:0.0000025259)19:0.0000025259)57:0.0005977770)93:0.0037976439,(CHSK_MKP2451.1:0.0005824561,CHSK_MKP2451.2:0.0000025259)94:0.0003846288)100:0.0091768552,(((((((POJA_4.1:0.0005915005,POJA_IB2659.1:0.0000025259)26:0.0000025259,((POJA_IB2659.2:0.0008883610,MG515680.1:0.0009732989)51:0.0008969090,(POJA_MKP1559.2:0.0000025259,(MG515675.1:0.0000025259,MG515676.1:0.0009679043)38:0.0000025259)40:0.0005915129)10:0.0000025259)5:0.0000026880,((POJA_MKP1559.1:0.0000022730,MG515671.1:0.0029417479)32:0.0011832513,MG515670.1:0.0009710594)2:0.0000025259)0:0.0000025259,(MG515662.1:0.0009709311,MG515673.1:0.0009709395)1:0.0000025259,(MG515664.1:0.0009706223,MG515674.1:0.0009772098)1:0.0000025259,(MG515666.1:0.0009709417,MG515672.1:0.0019624362)1:0.0000025259,(MG515677.1:0.0019509655,MG515678.1:0.0009706289)1:0.0000025259)0:0.0000025259,MG515663.1:0.0000025259)2:0.0000029862,(MG515667.1:0.0000029100,(MG515668.1:0.0000022251,MG515669.1:0.0009715672)62:0.0009744755)31:0.0009733880)27:0.0009850679,MG515665.1:0.0009778631)57:0.0020707738)83:0.0031471846,(LTJA_MKP990.1:0.0021409966,((KF268977.1:0.0000029997,KF268982.1:0.0009837505)17:0.0000025259,(KF268979.1:0.0029793315,KF268981.1:0.0009833313)16:0.0000025259)30:0.0009594654)32:0.0021708458)13:0.0000025259,((((LTJA_MKP990.2:0.0005953741,AY521218.1:0.0000025259)13:0.0000025259,KF268978.1:0.0000025259)9:0.0000025259,(((KF268980.1:0.0019819166,KF268985.1:0.0040019232)19:0.0000025259,KF268987.1:0.0019827239)17:0.0000025259,KF268986.1:0.0019827482)44:0.0009899957)9:0.0000025259,(KF268983.1:0.0000026885,KF268984.1:0.0009840104)64:0.0019909978)14:0.0000012834)63:0.0011254887,(Alca_torda.1:0.0021030801,Alca_torda.2:0.0000021524)100:0.0365451439)46:0.0013523981,((AY521217.1:0.0000025259,KF268971.1:0.0001972834)41:0.0007722094,(KF268968.1:0.0000028681,KF268969.1:0.0009746534)45:0.0000025259)57:0.0019629545)20:0.0005755236,AY521214.1:0.0000025589)7:0.0000025259,(((((PAJA_USNM606730.1:0.0000023505,KF268970.1:0.0000025259)39:0.0000025259,KF268976.1:0.0009762277)51:0.0018100136,KF268975.1:0.0009771220)10:0.0000025259,(KF268961.1:0.0000025259,(((KF268963.1:0.0000029152,KF268973.1:0.0009770644)52:0.0009770697,KF268972.1:0.0009806814)11:0.0000025259,KF268967.1:0.0029578939)2:0.0000025259)2:0.0000025259)4:0.0000025259,(AY521216.1:0.0000023512,KF268962.1:0.0000025867)42:0.0009468226)3:0.0000025259)15:0.0000025259,(PAJA_B20730.1:0.0000021564,(((((PAJA_B20730.2:0.0005868267,KF268965.1:0.0009799212)16:0.0000025259,KF268966.1:0.0009802997)10:0.0000025259,KF268974.1:0.0009866423)10:0.0000025259,KF268964.1:0.0000025259)22:0.0000025259,AY521215.1:0.0000024493)41:0.0000025259)63:0.0012028759);

```
* `-b`: regular non-parametric bootstrap


Now I am transferring the phylogeny onto my personal computer and openning it with Figtree to format it for publication.
* It will ask what to name the labels. I am naming them "bootstrap"
* select the outgroup, Alca torda, and click "Reroot"
* Check Node labels and choose bootstrap, font size 10
* Under appearance, increase line weight to 3
Then I am openning it in Illustrator to add species labels, colour, and taxon images

## Organize files once done, to be easier to browse in storage
```bash
mv *PAJAbowtie.MC1R.fa haplotypes
gzip Stercorarius.PAJAbowtie.MC1R.heterozygotes.vcf #compress file to save space
mv shapeit_* phasing
mv Stercorarius.PAJAbowtie.MC1R.heterozygotes.PIRsList phasing
gzip -9 alignment/MC1R.POJA4_haplotypes_corrected.fasta.log
gzip -9 alignment/MC1R.POJA4_haplotypes_corrected.fasta.boottrees
gzip -9 alignment/MC1R.POJA4_haplotypes_corrected.NCBI_MC1R.mafft.N.afa.log
gzip -9 alignment/MC1R.POJA4_haplotypes_corrected.NCBI_MC1R.mafft.N.afa.boottrees
```

# Appendix: Density of heterozygous sites
If the bird heterozygous for the dark and light morph allele is carrrying two deeply diverged haplotypes, I expect it to have much higher heterozygosity in that region. In this analysis, I am observing the density of heterozygous sites amongst the three *S. pomarinus* samples to see whether the dark bird has higher heterozygosity than the pale birds, and whether heterozygosity in the putative introgressed region is much higher than flanking regions.
```bash
#extract location of heterozygous site
for sample in POJA_MKP1559 POJA_4 POJA_IB2659 ; do /home/0_PROGRAMS/bcftools-1.14/bcftools view --samples "$sample"  Stercorarius.PAJAbowtie.MC1R.heterozygotes.vcf | grep -v "^#" | cut -f 2,10 | grep -e "0/1" -e "1/0" | cut -f 1 > "$sample".heterozygotes ; done
for sample in POJA_MKP1559 POJA_4 POJA_IB2659 ; do echo "$sample" ; cat "$sample".heterozygotes ; done

```
```
POJA_MKP1559
45878080
45878315
45878898
45879230
45879825
45880954
45882383
45882755
45882903
45882933
45882957
45883040
45883048
45883313
45883629
45884119
45884298
45884409
45884410
45886007
45886295
45886297
45886350
45886629
45887596
45888514
45888780
45891045
45891148
45891463
45892487
45892516
45895906
45897232
45897783
45897786
45897959
POJA_4
45879737
45880920
45881220
45881286
45882161
45883313
45884119
45884298
45884364
45886295
45886297
45886367
45886693
45886892
45886991
45887012
45887018
45887240
45887275
45887470
45887543
45887596
45887674
45887831
45887843
45887881
45888035
45888145
45888342
45888385
45888485
45888638
45888684
45888685
45888762
45888767
45888780
45888945
45889014
45891007
45891045
45891166
45891463
45896234
45897232
45897769
POJA_IB2659
45878315
45879163
45879947
45880036
45880631
45882933
45882957
45883021
45883048
45883313
45884119
45884517
45884859
45884894
45886277
45888118
45888205
45888944
45890795
45890972
45891007
45891045
45891166
45891463
45891532
45892120
45892948
45896476
45897485
45897508
45897769
45897786
45897931
```
POJA_4 does have a strikingly elevated heterozygosity at position chr7:45886295-45889014 compared to other samples and surrounding regions.

# Appendix: dark POJA haplotype
For reference, here is the published dark-morph S. pomarinus haplotype
```
>MG515679.1 Stercorarius pomarinus haplotype 18 melancortin 1 receptor (MC1R) gene, partial cds
TCCATCACCCTGGGCTGGCGGCGTGTGGCTATGGGGACGCTGGGCAGGGCTGGCCAGGAGGATGGCTGAG
ACACTGGAGTCCCAGCAGGCACGCGTCACCCCTGGCACATCCCCAGGCAGTGGGACTCCCTGTCCCCAGT
TCCTCAGCATCCTCTCGGCCTTGCCATCAGGCGCTGCGGGAGCGCTGGCAGGGTGGGCTGGGGGCCTGGG
GAATCTACGGGCACTGCAAGGGGCCCCGGGGGCTCAGCCCCCAGCTGGGGGGGGCCTGGGGTTGAGGTGG
GGGCCATGTCGACGCTGGCCCCCCTGCGCCTGCTGCGCAAGCCCTGGAACACCAGTGAGGGCAACCAGAG
CAACACCACGGCCGGGGCCGGCGGCCCCTGGTGCCAGGGGCTCAACATCCCCAACGAGCTCTTCCTCACG
CTGGGGCTGGTGAGCCTGGTGGAGAACCTGCTGGTGGTGGCTGCCATCCTGAAGAACAGGAACCTGCACT
CGCCCTCGTACTACTTCATCTGCTGCCTGGCCATCTCCGACATGCTGGTGAGCGTCAGCAACCTGGTGGA
GACGCTCTTCATGCTGCTGATGGAGCACGGTGTGCTGGTGATCCACGCCAGCATCGTCCGCCACATGGAT
AACGTCATCGACATGCTCATCTGCAGCTCCATCGTGTCCTCCCTTTCCTTCCTGGGGGTCATTGCTGTGG
ACCGCTACATCACCATCTTCTACGCCCTGCGCTACCACAACATCATGACGCTGCAACGGGCTGTGGTGAC
CATGGCCAGTGTCTGGCTGGCCAGCACCGTCTCCAGCACCATCTTCATCACCTACTACCACAATAACGCC
ATCCTCCTCTGCCTCATCGGCTTCTTCCTCTTTGTGCTGGTCCTCATGCTGGTGCTCTACATCCACATGT
TCATCCTGTCCCGCCACCACCTCTGCAGCATCTCCAGCCAGCAGAAGCAGCCCACCATCTACTGCAGCAG
CAGCCTGAAGGGAACCATCACGCTCACCATCCTCCTGGGAGTCTTCTTCATCTGCTGGGGGCCCTTCTTC
TTCCACCTCATCCTCATCGTCACCTGCCCCACAAACCCCTTCTGCACCTGCTTCTTCAGCTACTTCAACC
TCTTCTTCATC
```
