## Overview
This page describes how to build a species tree using gene trees with Astral. First, calculate maximum likelihood gene trees for many loci across the genome and provide these to Astral to create a species phylogeny. Then, it calculates concordance factors for each node.

### Required Input
The input is a set of sequence alignments in nexus format, produced in the phylogenetic blocks section (`1.6.2c_phylogenetic_blocks`). Ideally the blocks should be short enough to not contain recombination but long enough to resolve a gene tree. In the repository these blocks are placed in a zipped folder that will need to be uncompressed before the code can run. The gene tree sets are provided in `*.trees` files.  

### Output
The output consists of maximum likelihood gene trees for each chunk of the genome, and a species tree built from these gene trees. I tested several different variations of the dataset to ensure that it was robust to my filters: the versions used in the paper are `species.Astralblocks.unlinked.partitioned.tre` (autosomal) and `species.Astralblocks.unlinked.partitioned.chrZ.nochrW.tre` (Z chromosome).

# Astral species tree estimation

Here, I will produce maximum likelihood phylogenies for single chunks of the genome ("gene trees"), where each chunk is small enough that hopefully there has been limited recombination within it. Then I will analyze these trees in a coalescent framework to try to reconstruct the species tree. Note that there is a trade-off between using a smaller block size to avoid recombination, and using a larger block size to have more signal needed to generate a ML tree. 

The program we are using is ASTRAL, which is able to handle incomplete lineage sorting. It is based on optimizing a species tree by finding the tree with the greatest agreement with quartets in the gene trees that are given as input.

There is a tutorial [here](https://github.com/mmatschiner/tutorials/tree/master/ml_species_tree_inference) and the ASTRAL tutorial [here](https://github.com/smirarab/ASTRAL/blob/master/astral-tutorial.md). There is another tutorial [here](https://tandy.cs.illinois.edu/astral-Mirarab-tutorial-v2.pdf) as well.

As input, Astral needs gene trees that are unrooted. Branch lengths do not matter.

I previously divided the genome into 5 kb chunks. Now, we will make maximum likelihood gene trees for each chunk. I am not doing bootstrap replicates because ASTRAL can estimate branch support without them. This will save some time because we have thousands of alignments to analyze. It takes a couple seconds per alignment, so doing them in parallel (1 thread each on 24 threads) means you can do several hundred per minute.

## Create gene trees

Here is the pipeline to generate the gene trees for each chunk.
```bash
#setup directory
#mkdir -p /home/1_Else/0_GENOMES5/Stercorarius/1_Data/2.5c_phylogeny_ASTRAL
#cd /home/1_Else/0_GENOMES5/Stercorarius/1_Data/2.5c_phylogeny_ASTRAL
mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/2.5c_phylogeny_ASTRAL
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/2.5c_phylogeny_ASTRAL

#create a folder to hold results for our particular dataset
mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/2.5c_phylogeny_ASTRAL/dataset1
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/2.5c_phylogeny_ASTRAL/dataset1

#get a list of all the alignments to analyze
#make a list of the chunks to loop through
find /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/1.6.2c_phylogenetic_blocks/strict/blocks/ | tail -n +2 > chunks.txt #removes the first line because it is just the directory name #fast
#note that there are too many files to list with wildcards, so the following type of command would not work:
#find blocks/*.nex > chunks.txt
wc -l chunks.txt 
#There are 223915 loci in this dataset

#remove Fratercula and Uria from the dataset - we are using Alca torda as the outgroup
mkdir -p alignments
time cat chunks.txt | sed 's|/home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/1.6.2c_phylogenetic_blocks/strict/blocks/||g' | parallel 'grep -v -e "Uria" -e "Fratercula" /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/1.6.2c_phylogenetic_blocks/strict/blocks/{1} | sed "s/ntax=18/ntax=16/g" > alignments/{1}'

#Generate maximum likelihood gene trees for each alignment
time cat chunks.txt | sed 's|/home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/1.6.2c_phylogenetic_blocks/strict/blocks/||g' | parallel time /home/0_PROGRAMS/iqtree-2.1.3-Linux/bin/iqtree2 -s alignments/{1}
#944m57.960 (27545m44.372s)



```
iqtree2 settings:
* `-B 1000`: would add to perform 1000 bootstrap replicates for each alignment.
* `-s`: alignment to create a tree from
* `-o`: specify the outgroup so that the tree is rooted (not needed for Astral)

**Timing**: Each chunk took around 0.5-3 seconds with 8 samples, and so the whole thing finished in 410m45s (8499m user time). Good thing to do it in parallel or it would take a week to finish. With 117 samples, it took 30s-1m30 per alignment. 16m5.285s (1637m11.947s) for 1673 trees. Took 919m17.769s (80927m4.308s). Took 393m2.926s (30256m12.073s) for 17 species.

The maximum likelihood trees will be produced in `*.nex.treefile`. Concatenate them all into a single file for Astral.  
To better understand the dataset, I am going to make several different subsets of the data to analyze with Astral:
1) All loci (223887 loci)
2) Autosomal loci (208189 loci)
3) Autosomal loci with less than 300 bp missing in any Stercorariid (117576 loci)
4) Autosomal loci separated by at least 5 kb, with less than 300 bp missing in any Stercorariid (58747 loci)
5) chrZ loci (15698 loci)
6) chrZ loci with less than 300 bp missing in any Stercorariid (3373 loci)
7) chrZ loci separated by at least 5 kb, with less than 300 bp missing in any Stercorariid (1680 loci)
8) chrZ loci outside pseudoautosomal region (9959 loci)
9) chrZ loci outside pseudoautosomal region with less than 300 bp missing in any Stercorariid (1507 loci)
10) chrZ loci outside pseudoautosomal region separated by at least 5 kb, with less than 300 bp missing in any Stercorariid (754 loci)

This will allow me to know how filtering for linkage and missing data affected my results, and to understand how patterns vary between autosomal and chrZ regions.
```bash
#all trees
time cat chunks.txt | sed 's|/home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/1.6.2c_phylogenetic_blocks/strict/blocks/||g' | while read line ; do cat alignments/"$line".treefile >> ml_best.trees ; done
wc -l ml_best.trees
#223887

#all autosomal trees (exclude chrZ)
time cat chunks.txt | sed 's|/home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/1.6.2c_phylogenetic_blocks/strict/blocks/||g' | grep -v "chrZ" | while read line ; do cat alignments/"$line".treefile >> ml_best.autosomal.trees ; done
wc -l ml_best.autosomal.trees
#208189

#Autosomal trees that have less than 300 bp missing in any Stercorariid
time cat ./Astralblocks.txt | grep -v "chrZ" | while read line ; do cat alignments/"$line".treefile >> ml_best.Astralblocks.trees ; done #4m16.130s
wc -l ml_best.Astralblocks.trees
#117576

#trees that have less than 300 bp missing in any Stercorariid AFTER being pruned to keep only every second locus (to reduce linkage)
time cat ./Astralblocks.unlinked.txt | grep -v "chrZ" | while read line ; do cat alignments/"$line".treefile >> ml_best.Astralblocks.unlinked.trees ; done
wc -l ml_best.Astralblocks.unlinked.trees
#58747


#Also make datasets for chrZ
#all chrZ loci
time cat chunks.txt | sed 's|/home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/1.6.2c_phylogenetic_blocks/strict/blocks/||g' | grep "chrZ" | while read line ; do cat alignments/"$line".treefile >> ml_best.chrZ.trees ; done
wc -l ml_best.chrZ.trees
#15698

#chrZ loci that have less than 300 bp missing in any Stercorariid
time cat ./Astralblocks.txt | grep "chrZ" | while read line ; do cat alignments/"$line".treefile >> ml_best.Astralblocks.chrZ.trees ; done 
wc -l ml_best.Astralblocks.chrZ.trees
#3373

#chrZ trees that have less than 300 bp missing in any Stercorariid AFTER being pruned to keep only every second locus (to reduce linkage)
time cat ./Astralblocks.unlinked.txt | grep "chrZ" | while read line ; do cat alignments/"$line".treefile >> ml_best.Astralblocks.unlinked.chrZ.trees ; done
wc -l ml_best.Astralblocks.unlinked.chrZ.trees
#1680


#Also make datasets for chrZ, excluding the first 30 Mb (which has the pseudoautosomal region)
#only chrZ. I am removing the first 30 Mb of Z where the pseudoauosomal region is

time cat chunks.txt | sed 's|/home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/1.6.2c_phylogenetic_blocks/strict/blocks/||g' | grep "chrZ" | awk '{split($1,a,"_"); if (a[2] >= 30000000) { print } }' | while read line ; do cat alignments/"$line".treefile >> ml_best.chrZ.nochrW.trees ; done
wc -l ml_best.chrZ.nochrW.trees
#9959

time cat ./Astralblocks.txt | grep "chrZ" | awk '{split($1,a,"_"); if (a[2] >= 30000000) { print } }' | while read line ; do cat alignments/"$line".treefile >> ml_best.Astralblocks.chrZ.nochrW.trees ; done 
wc -l ml_best.Astralblocks.chrZ.nochrW.trees
#1507

cat ./Astralblocks.unlinked.txt | grep "chrZ" | awk '{split($1,a,"_"); if (a[2] >= 30000000) { print } }' | while read line ; do cat alignments/"$line".treefile >> ml_best.Astralblocks.unlinked.chrZ.nochrW.trees ; done
wc -l ml_best.Astralblocks.unlinked.chrZ.nochrW.trees
#754

```

## Run Astral

Astral is based on the multispecies coalescent framework, and so it needs multiple sequences per population to estimate branchlengths. It is possible to estimate topologies while treating each individual as a lineage, as long as we are aware that this greatly violates the assumptions of the model. To tell Astral which individuals belong to the same population, we will specify a mapping file. Each line lists the name of the population, a colon, and then a comma-separated list of samples belonging to that population.
```bash
cat > Astral_SpeciesMappingFile
````
```
longicaudus:LTJA_MKP990
parasiticus:PAJA_B20730,PAJA_US-M606730
pomarinus:POJA_MKP1559,POJA_4,POJA_IB2659
skua:GRSK_MKP1592,GRSK_MKP1593
antarcticus:CISK2,CISK55,CISK3
chilensis:CHSK_MKP2451
maccormicki:A-SK7,A-SK01,A-SK8
Alcidae:Alca_torda
```

Then we run Astral.  
```bash
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/2.5c_phylogeny_ASTRAL

#first we run without specifying species identity, treating each sample as a "species", as an exploration of the data and to ensure that each of our population would be recovered as monophyletic.

#all data
time java -jar /home/0_PROGRAMS/Astral/astral.5.7.3.jar -i ml_best.trees -o species.tre 2> astral_species.log #344m56.859s

#autosomal datasets
time java -jar /home/0_PROGRAMS/Astral/astral.5.7.3.jar -i ml_best.autosomal.trees -o species.autosomal.tre 2> astral_species.autosomal.log #
time java -jar /home/0_PROGRAMS/Astral/astral.5.7.3.jar -i ml_best.Astralblocks.trees -o species.Astralblocks.tre 2> astral_species.Astralblocks.log #175m57.668s
time java -jar /home/0_PROGRAMS/Astral/astral.5.7.3.jar -i ml_best.Astralblocks.unlinked.trees -o species.Astralblocks.unlinked.tre 2> astral_species.Astralblocks.unlinked.log #57m40.350s

#chrZ datasets
time java -jar /home/0_PROGRAMS/Astral/astral.5.7.3.jar -i ml_best.chrZ.trees -o species.chrZ.tre 2> astral_species.chrZ.log #
time java -jar /home/0_PROGRAMS/Astral/astral.5.7.3.jar -i ml_best.Astralblocks.chrZ.trees -o species.Astralblocks.chrZ.tre 2> astral_species.Astralblocks.chrZ.log #175m57.668s
time java -jar /home/0_PROGRAMS/Astral/astral.5.7.3.jar -i ml_best.Astralblocks.unlinked.chrZ.trees -o species.Astralblocks.unlinked.chrZ.tre 2> astral_species.Astralblocks.unlinked.chrZ.log #

#chrZ, no pseudoautosomal region
time java -jar /home/0_PROGRAMS/Astral/astral.5.7.3.jar -i ml_best.chrZ.nochrW.trees -o species.chrZ.nochrW.tre 2> astral_species.chrZ.nochrW.log #
time java -jar /home/0_PROGRAMS/Astral/astral.5.7.3.jar -i ml_best.Astralblocks.chrZ.nochrW.trees -o species.Astralblocks.chrZ.nochrW.tre 2> astral_species.Astralblocks.chrZ.nochrW.log #175m57.668s
time java -jar /home/0_PROGRAMS/Astral/astral.5.7.3.jar -i ml_best.Astralblocks.unlinked.chrZ.nochrW.trees -o species.Astralblocks.unlinked.chrZ.nochrW.tre 2> astral_species.Astralblocks.unlinked.chrZ.nochrW.log #




#now we run while specifying species identity with the mapping file
#all data
time java -jar /home/0_PROGRAMS/Astral/astral.5.7.3.jar -i ml_best.trees --namemapfile Astral_SpeciesMappingFile -o species.partitioned.tre --reps 100 2> astral_species.partitioned.log #202m

#autosomal datasets
time java -jar /home/0_PROGRAMS/Astral/astral.5.7.3.jar -i ml_best.autosomal.trees --namemapfile Astral_SpeciesMappingFile -o species.partitioned.autosomal.tre --reps 100 2> astral_species.partitioned.autosomal.log #277m33.241s
time java -jar /home/0_PROGRAMS/Astral/astral.5.7.3.jar -i ml_best.Astralblocks.trees --namemapfile Astral_SpeciesMappingFile -o species.Astralblocks.partitioned.tre --reps 100 2> astral_species.Astralblocks.partitioned.log #140m17.505s
time java -jar /home/0_PROGRAMS/Astral/astral.5.7.3.jar -i ml_best.Astralblocks.unlinked.trees --namemapfile Astral_SpeciesMappingFile -o species.Astralblocks.unlinked.partitioned.tre --reps 100 2> astral_species.Astralblocks.unlinked.partitioned.log #64m20.301s

#chrZ datasets
time java -jar /home/0_PROGRAMS/Astral/astral.5.7.3.jar -i ml_best.chrZ.trees --namemapfile Astral_SpeciesMappingFile -o species.partitioned.chrZ.tre --reps 100 2> astral_species.partitioned.chrZ.log #277m33.241s
time java -jar /home/0_PROGRAMS/Astral/astral.5.7.3.jar -i ml_best.Astralblocks.chrZ.trees --namemapfile Astral_SpeciesMappingFile -o species.Astralblocks.partitioned.chrZ.tre --reps 100 2> astral_species.Astralblocks.partitioned.chrZ.log #140m17.505s
time java -jar /home/0_PROGRAMS/Astral/astral.5.7.3.jar -i ml_best.Astralblocks.unlinked.chrZ.trees --namemapfile Astral_SpeciesMappingFile -o species.Astralblocks.unlinked.partitioned.chrZ.tre --reps 100 2> astral_species.Astralblocks.unlinked.partitioned.chrZ.log #64m20.301s

#chrZ, no pseudoautosomal
time java -jar /home/0_PROGRAMS/Astral/astral.5.7.3.jar -i ml_best.chrZ.nochrW.trees --namemapfile Astral_SpeciesMappingFile -o species.partitioned.chrZ.nochrW.tre --reps 100 2> astral_species.partitioned.chrZ.nochrW.log #277m33.241s
time java -jar /home/0_PROGRAMS/Astral/astral.5.7.3.jar -i ml_best.Astralblocks.chrZ.nochrW.trees --namemapfile Astral_SpeciesMappingFile -o species.Astralblocks.partitioned.chrZ.nochrW.tre --reps 100 2> astral_species.Astralblocks.partitioned.chrZ.nochrW.log #140m17.505s
time java -jar /home/0_PROGRAMS/Astral/astral.5.7.3.jar -i ml_best.Astralblocks.unlinked.chrZ.nochrW.trees --namemapfile Astral_SpeciesMappingFile -o species.Astralblocks.unlinked.partitioned.chrZ.nochrW.tre --reps 100 2> astral_species.Astralblocks.unlinked.partitioned.chrZ.nochrW.log #64m20.301s

```
* `-i`: concatented file of maximum likelihood trees
* `-b`: "The -b option tells ASTRAL that bootstrapping needs to be performed. Following -b is the name of a file that contains the file path of gene tree bootstrap files, one line per gene. Thus, the input file is not a file full of trees. It's a file full of paths of files full of trees."
* `-o`: output file name
* ` [(-a|--namemapfile) <mapping file>]`: "a file containing the mapping between names in gene tree and names in the species tree. The mapping file has one line per species, with one of two formats:
        species: gene1,gene2,gene3,gene4
        species 4 gene1 gene2 gene3 gene4"

From the ASTRAL documentation:
>What to use: The most important tree is the tree outputted at the end; this is the ASTRAL tree on main input trees, with support values drawn based on bootstrap replicates. Our analyses have shown this tree to be better than the consensus tree in most cases.


Next, these trees can be visualized, for example in Figtree. the tree should be rerooted as ASTRAL does not root the tree. Note that the internal branch lengths are estimated (in coalescent units) but terminal branch lengths leading to a single sample are **not** estimated. Terminal branchlengths are estimated for populations with more than one haplotype.

Remember that these trees are **unrooted**, we will need to root them with Alcidae as the outgroup in order to make sense of them.

## Results for treating each sample as a separate lineage

In the full dataset, *S. pomarinus* is placed as monophyletic, and *S. chilensis* is placed sister to *S. antarcticus*, and all nodes have full support. In the autosomal dataset, all nodes also have full support... but now *S. pomarinus* is not placed monophyletic, and *S. maccormicki* is placed sister to *S. antarcticus*. The unlinked autosomal dataset is similar to the Astralblocks autosomal dataset, but now the node placing *S. antarcticus* and *S. maccormicki* together has only 0.74 support.

This is interesting. From looking at genome scans, it seems that very few parts of the genome have undergone lineage sorting in *S. pomarinus* (ie., sequence divergence is lower between two *S. pomarinus* than it is between *S. pomarinus* and a Catharacta). It seems that the filtering may have been somewhat biased against these parts of the genome that have undergone sorting in *S. pomarinus*. Removing chrZ sequences may be a large part of this pattern, as chrZ undergoes lineage sorting more quickly than autosomal loci in general, due to a smaller effective population size.

*S. pomarinus* is recovered as monophyletic by the chrZ datasets and *S. chilensis* is placed sister to *S. antarcticus*. 

```
# All data
species.tre #POJA monophyletic, chilensis sister to antarcticus
(Alca_torda,((LTJA_MKP990,(PAJA_B20730,PAJA_US-M606730)1:7.684919322567162)1:1.8138131184908137,((POJA_MKP1559,(POJA_4,POJA_IB2659)1:0.020924699860759125)1:0.020225925129960565,((GRSK_MKP1592,GRSK_MKP1593)1:0.3866658945177509,((A-SK7,(A-SK01,A-SK8)1:0.02630917928583699)1:0.25012668096840485,(CHSK_MKP2451,(CISK55,(CISK3,CISK2)1:0.10345398537910017)1:0.3857620136015808)1:0.03258689802555164)1:0.05110689336414165)1:0.7341583620176271)1:7.63078654343352):0.0); 


# Autosomal data
species.autosomal.tre #POJA not monophyletic, maccormicki sister to antarcticus
(Alca_torda,((LTJA_MKP990,(PAJA_B20730,PAJA_US-M606730)1:8.250832558605424)1:1.7651824574958221,(POJA_MKP1559,((POJA_4,POJA_IB2659)1:0.026140619955225645,((GRSK_MKP1592,GRSK_MKP1593)1:0.3726432848470574,(CHSK_MKP2451,((A-SK7,(A-SK01,A-SK8)1:0.023159787959077825)1:0.22401918674462531,(CISK55,(CISK3,CISK2)1:0.10527738499388356)1:0.39453860173472255)1:0.030241826097979426)1:0.04509865173926645)1:0.6856751486943805)1:0.014392508169382817)1:7.674311006166837):0.0); 

species.Astralblocks.tre #POJA not monophyletic, maccormicki sister to antarcticus
(Alca_torda,((LTJA_MKP990,(PAJA_B20730,PAJA_US-M606730)1:8.473040800731436)1:1.7257008933908222,(POJA_MKP1559,((POJA_4,POJA_IB2659)1:0.027935047180616086,((GRSK_MKP1592,GRSK_MKP1593)1:0.3904957995120633,(CHSK_MKP2451,((A-SK7,(A-SK01,A-SK8)1:0.02041768024617985)1:0.22999119434030202,(CISK55,(CISK3,CISK2)1:0.1031555569422771)1:0.4109803016244416)1:0.031071256056973)1:0.049566978224489674)1:0.6750060642746167)1:0.01594534286846934)1:7.75557930313865):0.0); 

species.Astralblocks.unlinked #POJA not monophyletic, maccormicki sister to antarcticus
(Alca_torda,((LTJA_MKP990,(PAJA_B20730,PAJA_US-M606730)1:8.00469820117675)1:1.723561834536233,(POJA_MKP1559,((POJA_4,POJA_IB2659)1:0.027001624324884514,((GRSK_MKP1592,GRSK_MKP1593)1:0.39094226801746085,(CHSK_MKP2451,((CISK55,(CISK2,CISK3)1:0.10031993181074395)1:0.41071970606777597,(A-SK7,(A-SK01,A-SK8)1:0.02205092485689999)1:0.2298748950809826)0.74:0.03034652771631089)1:0.05055241425692841)1:0.6762585308768301)1:0.016582325256478)1:7.835685252028271):0.0); 


# chrZ data
species.chrZ.tre #POJA monophyletic, CISK & CHSK sister
(Alca_torda,((LTJA_MKP990,(PAJA_B20730,PAJA_US-M606730)1:5.747792010293216)1:2.8935777323626124,((POJA_MKP1559,(POJA_4,POJA_IB2659)1:0.037128502436055)1:0.25344775240289996,((A-SK7,(A-SK01,A-SK8)1:0.06900442124989373)1:0.4958897504136514,((GRSK_MKP1592,GRSK_MKP1593)1:0.5447856588033687,(CHSK_MKP2451,((CISK3,CISK2)1:0.07954120913766802,CISK55)1:0.43683168949031614)1:0.08270538911346569)0.84:0.05716704526784505)1:1.4682304940069397)1:6.916934449766795):0.0); 

species.Astralblocks.chrZ.tre #POJA monophyletic, CISK & CHSK sister
(Alca_torda,((LTJA_MKP990,(PAJA_B20730,PAJA_US-M606730)1:7.718389154997753)1:2.445389596434002,((POJA_MKP1559,(POJA_4,POJA_IB2659)0.97:0.030145244741234586)1:0.14055632024798514,((GRSK_MKP1592,GRSK_MKP1593)1:0.5727585557123375,((CHSK_MKP2451,(CISK55,(CISK3,CISK2)1:0.116909938734883)1:0.46244483572717626)1:0.0997498654222675,(A-SK7,(A-SK01,A-SK8)1:0.04473678791151648)1:0.4771820693196418)1:0.09611841767938682)1:1.0324093160020138)1:6.809533401611065):0.0); 

species.Astralblocks.unlinked.chrZ.tre #POJA monophyletic, CISK & CHSK sister
(Alca_torda,((LTJA_MKP990,(PAJA_B20730,PAJA_US-M606730)1:7.021679025300523)1:2.3967062120161806,((POJA_MKP1559,(POJA_IB2659,POJA_4)0.9:0.031705891201212394)1:0.1415804697702994,((GRSK_MKP1593,GRSK_MKP1592)1:0.5516696231660181,((A-SK7,(A-SK01,A-SK8)1:0.060382979390284475)1:0.4812504575354247,((CISK55,(CISK2,CISK3)1:0.11654255702704684)1:0.4629494490111479,CHSK_MKP2451)1:0.11171630796087395)1:0.0834349559378503)1:0.987679665668543)1:7.021679025300523):0.0); 


# chrZ (nonpseudoautosomal) data
species.chrZ.nochrW.tre #POJA monophyletic, CISK & CHSK sister
(Alca_torda,((LTJA_MKP990,(PAJA_B20730,PAJA_US-M606730)1:7.146308894755901)1:3.148085707280873,((POJA_MKP1559,(POJA_4,POJA_IB2659)1:0.03477278852735777)1:0.3508738881314375,((GRSK_MKP1592,GRSK_MKP1593)1:0.6928275857051635,((A-SK7,(A-SK01,A-SK8)1:0.09366417654582086)1:0.4250168256197785,(CHSK_MKP2451,(CISK3,(CISK55,CISK2)1:0.06594822607498534)1:0.4370875027009923)1:0.07670203853561529)1:0.0603990798045831)1:1.4506072431528476)1:7.015299743490164):0.0); 

species.Astralblocks.chrZ.nochrW.tre #POJA monophyletic, CISK & CHSK sister
(Alca_torda,((LTJA_MKP990,(PAJA_B20730,PAJA_US-M606730)1:6.913074440459677)1:2.7541913571000656,((POJA_4,(POJA_IB2659,POJA_MKP1559)0.43:0.007629520610783906)1:0.26430580320704056,(((A-SK7,(A-SK01,A-SK8)1:0.08945356205439314)1:0.4616829050192995,(CHSK_MKP2451,(CISK55,(CISK2,CISK3)1:0.06719456519568731)1:0.5160059059988721)1:0.11327563631624708)1:0.1021644475262497,(GRSK_MKP1593,GRSK_MKP1592)1:0.7312847366972762)1:1.2421267631828403)1:6.913074440459677):0.0); 

species.Astralblocks.unlinked.chrZ.nochrW.tre #POJA monophyletic, CISK & CHSK sister
(Alca_torda,((LTJA_MKP990,(PAJA_B20730,PAJA_US-M606730)1:6.221252641140822)1:2.6659045796514462,((POJA_4,(POJA_IB2659,POJA_MKP1559)0.45:0.00926335342740126)1:0.2675765000945348,((GRSK_MKP1592,GRSK_MKP1593)1:0.7200777326912281,((A-SK7,(A-SK01,A-SK8)1:0.09774819775163013)1:0.48818108777812835,((CISK55,(CISK3,CISK2)0.96:0.0624825616953873)1:0.5331479792038871,CHSK_MKP2451)1:0.13217977134887013)0.98:0.07805103894146795)1:1.1990966641099376)1:6.221252641140822):0.0); 
```

## Results for assigning samples to populations

Now, let's look at the 'Species' tree.  

All three base datasets (all gene trees, Astraltrees, or thinned Astraltrees) and are nearly identical, but some parts of the tree differ depending on which loci we look at. 

When only autosomal loci are looked at, *S. maccormicki* is placed sister to *S. antarcticus*. 

When examining chrZ and removing the first 30 Mb (where chrW reads map in females), all three datasets place *S. chilensis* sister to *S. antarcticus*. Notably, the internal branch length is longer - suggsting more genetic drift or other sources of lineage sorting - as expected for chrZ. When we don't remove the psuedoautosomal region, the position of *S. maccormicki* and *S. skua* are flipped (with 0.84 support) in the full dataset. I suspect that this is because the *S. maccormicki* samples are only 1/3 female, while the other three are 1/2, 1/1, and 2/3 female. Female samples have many chrW reads mapping to the pseudoautosomal region, which would cause them to group together to the exclusion of male samples that lack these chrW sequences. This is therefore not a valid phylogeny as it is based on alignments that are mixed between orthologs and paralogs.

If we include all the trees, including both autosomal and chrZ, *S. chilensis* is placed sister to *S. antarcticus*, suggesting that the chrZ data tips the balance over the signal present in the autosomes.

Recall that our other results have indicated that *S. antarcticus* has had a lot of introgression from *S. chilensis* and/or *S. maccormicki* (and may best be considered as a hybrid lineage with ancestry from both branches), so it is not surprising that the relationship is unstable, but it is interesting that the filtering process flipped which relationship was best supported. Note that the process of thinning to reduce linkage had no impact on the topology, and negligible impact on estimated branch lengths. Note also that the branch lengths are very comparable between the filtered and unfiltered dataset, despite the flip in topology.

Notice that the uncertain node has 100% support in the filtered and unfiltered dataset - despite them supporting different topologies! We cannot simply trust the support values, especially when using so much data. Note that the contentious node has 74% support in the unlinked dataset - a bit more reasonable for a dataset with a lot of discordance.

```
# all data
#species.partitioned.tre #chilensis sister to antarcticus
(antarcticus:0.42546839156672683,((maccormicki:0.2576813382882886,(skua:0.3866658945177509,(pomarinus:0.028398376273275176,(Alcidae,(parasiticus:7.684919322567162,longicaudus)1:1.8138131184908137)1:7.63078654343352)1:0.7341583620176271)1:0.05110689336414165)1:0.03258689802555164,chilensis):0.0); 

# Autosomal data
#species.partitioned.autosomal.tre #maccormicki sister to antarcticus
(antarcticus:0.43735865979118266,(maccormicki:0.23000494138411912,((skua:0.3726432848470574,(pomarinus:0.012654719189174537,(Alcidae,(parasiticus:8.250832558605424,longicaudus)1:1.7651824574958221)1:7.696627232681659)1:0.702198678524129)1:0.04509865173926645,chilensis)1:0.030241826097979426):0.0); 

#species.Astralblocks.partitioned.tre #maccormicki sister to antarcticus
(antarcticus:0.4542759929563439,(maccormicki:0.2351328586913995,((skua:0.3904957995120633,(pomarinus:0.01663998613803205,(Alcidae,(parasiticus:8.473040800731436,longicaudus)1:1.7257008933908222)1:7.690677330206883)1:0.6931515849784127)1:0.049566978224489674,chilensis)1:0.031071256056973):0.0); 

#species.Astralblocks.unlinked.partitioned.tre #maccormicki sister to antarcticus
(antarcticus:0.45371612269011635,(maccormicki:0.23507190798777355,((skua:0.39094226801746085,(pomarinus:0.01713279848921311,(Alcidae,(parasiticus:8.00469820117675,longicaudus)1:1.723561834536233)1:7.788404490826047)1:0.6954886009850313)1:0.05055241425692841,chilensis)0.74:0.03034652771631089):0.0); 


# chrZ data
species.partitioned.chrZ.tre #chilensis sister to antarcticus, then to skua!
(antarcticus:0.4682216647953282,(chilensis,(skua:0.5447856588033687,(maccormicki:0.5203900728428702,(pomarinus:0.2646268741889442,(Alcidae,(parasiticus:5.747792010293216,longicaudus)1:2.8935777323626124)1:6.916934449766795)1:1.4682304940069397)0.84:0.05716704526784505)1:0.08270538911346569):0.0); 

species.Astralblocks.partitioned.chrZ.tre #chilensis sister to antarcticus
(antarcticus:0.5043549683022397,(chilensis,(maccormicki:0.4970201215387717,(skua:0.5727585557123375,(pomarinus:0.1551837551438314,(Alcidae,(parasiticus:7.718389154997753,longicaudus)1:2.445389596434002)1:6.809533401611065)1:1.0324093160020138)1:0.09611841767938682)1:0.0997498654222675):0.0); 

species.Astralblocks.unlinked.partitioned.chrZ.tre #chilensis sister to antarcticus
(antarcticus:0.5005671001858917,((maccormicki:0.4976273626449655,(skua:0.5516696231660181,(pomarinus:0.15923109365264876,(Alcidae,(parasiticus:7.021679025300523,longicaudus)1:2.3967062120161806)1:7.021679025300523)1:0.987679665668543)1:0.0834349559378503)1:0.11171630796087395,chilensis):0.0); 


# chrZ (nonpseudoautosomal) data
species.partitioned.chrZ.nochrW.tre #chilensis sister to antarcticus
(antarcticus:0.4683984924645092,(chilensis,(maccormicki:0.46077149951753593,(skua:0.6928275857051635,(pomarinus:0.3592480545719488,(Alcidae,(parasiticus:7.146308894755901,longicaudus)1:3.148085707280873)1:7.015299743490164)1:1.4506072431528476)1:0.0603990798045831)1:0.07670203853561529):0.0); 

species.Astralblocks.partitioned.chrZ.nochrW.tre #chilensis sister to antarcticus
(antarcticus:0.5409404123990403,(chilensis,(maccormicki:0.5052933814418213,(skua:0.7312847366972762,(pomarinus:0.25896964955605445,(Alcidae,(parasiticus:6.913074440459677,longicaudus)1:2.7541913571000656)1:6.913074440459677)1:1.2421267631828403)1:0.1021644475262497)1:0.11327563631624708):0.0); 

species.Astralblocks.unlinked.partitioned.chrZ.nochrW.tre #chilensis sister to antarcticus
(antarcticus:0.5444037140211387,(chilensis,(maccormicki:0.5250295308172289,(skua:0.7200777326912281,(pomarinus:0.2574828938660886,(Alcidae,(parasiticus:6.221252641140822,longicaudus)1:2.6659045796514462)1:6.221252641140822)1:1.1990966641099376)0.98:0.07805103894146795)1:0.13217977134887013):0.0); 

```

To conclude, the results that we are reporting are:  
1) Phylogeny based on non-pseudoautosomal chrZ, filtered to reduce linkage and remove loci with too much missing data (species.Astralblocks.unlinked.partitioned.chrZ.nochrW.tre) `(antarcticus:0.5444037140211387,(chilensis,(maccormicki:0.5250295308172289,(skua:0.7200777326912281,(pomarinus:0.2574828938660886,(Alcidae,(parasiticus:6.221252641140822,longicaudus)1:2.6659045796514462)1:6.221252641140822)1:1.1990966641099376)0.98:0.07805103894146795)1:0.13217977134887013):0.0);`
2) Phylogeny based on autosomes, filtered to reduce linkage and remove loci with too much missing data (species.Astralblocks.unlinked.partitioned.tre) `(antarcticus:0.45371612269011635,(maccormicki:0.23507190798777355,((skua:0.39094226801746085,(pomarinus:0.01713279848921311,(Alcidae,(parasiticus:8.00469820117675,longicaudus)1:1.723561834536233)1:7.788404490826047)1:0.6954886009850313)1:0.05055241425692841,chilensis)0.74:0.03034652771631089):0.0);`

# Quartet scoring
To better understand the structure of our data, we can estimate quartet scores. There is documentation about this [here](https://github.com/smirarab/ASTRAL/blob/master/astral-tutorial.md#extensive-branch-annotations). Astral estimates quartet scores given a species tree and the set of gene trees. 
```bash
#chrZ data
time /home/0_PROGRAMS/Astral/astral.5.7.3.jar -q species.Astralblocks.unlinked.partitioned.chrZ.nochrW.tre -i ml_best.Astralblocks.unlinked.chrZ.nochrW.trees -t 16 --namemapfile Astral_SpeciesMappingFile -o species.Astralblocks.unlinked.partitioned.chrZ.nochrW.scored.tre #took 2 seconds
mv freqQuad.csv species.Astralblocks.unlinked.partitioned.chrZ.nochrW.freqQuad.tsv

#autosomal data
time /home/0_PROGRAMS/Astral/astral.5.7.3.jar -q species.Astralblocks.unlinked.partitioned.tre -i ml_best.Astralblocks.unlinked.trees -t 16 --namemapfile Astral_SpeciesMappingFile -o species.Astralblocks.unlinked.partitioned.scored.tre #took 22 mins
mv freqQuad.csv species.Astralblocks.unlinked.partitioned.freqQuad.tsv

```
astral settings:  
* `[(-q|--score-tree) <score species trees>] score the provided species tree and exit`
* `[(-t|--branch-annotate) <branch annotation level>]
        How much annotations should be added to each branch: 0, 1, or 2. 
        0: no annotations. 
        1: only the quartet support for the main resolution. 
        2: full annotation (quartet support, quartet frequency, and posterior
        probability for all three alternatives, plus total number of quartets
        around the branch and effective number of genes).
        3 (default): only the posterior probability for the main resolution.
        4: three alternative posterior probabilities.
        8: three alternative quartet scores.
        16/32: hidden commands useful to create a file called freqQuad.csv.
        10: p-values of a polytomy null hypothesis test. (default: 3)`

This reveals some important patterns. On chrZ (ronrecombining), 41.6% of quartets support placing CHSK with CISK, while 32.0 support placing ANSK with CISK - a moderate difference. On autosomal loci, however, there is almost no difference; 35.33% of gene trees support CISK,ANSK vs 35.31% that support CISK,CHSK!

I will add these normalized quartet scores to the nodes of the phylogeny in the figure.  

# Making the figure
I am making the figure using Figtree and Illustrator.
First, I open `species.Astralblocks.unlinked.partitioned.tre` in Figtree (`(antarcticus:0.45371612269011635,(maccormicki:0.23507190798777355,((skua:0.39094226801746085,(pomarinus:0.01713279848921311,(Alcidae,(parasiticus:8.00469820117675,longicaudus)1:1.723561834536233)1:7.788404490826047)1:0.6954886009850313)1:0.05055241425692841,chilensis)0.74:0.03034652771631089):0.0);`)
* Click on the branch for "Alcidae" -> "Reroot"
* Check "Node Labels" -> Display: "Label"
* File -> Export PDF 
Then I edit aesthetics in Illustrator.

I then repeated with `species.Astralblocks.unlinked.partitioned.chrZ.nochrW.tre` (`(antarcticus:0.5444037140211387,(chilensis,(maccormicki:0.5250295308172289,(skua:0.7200777326912281,(pomarinus:0.2574828938660886,(Alcidae,(parasiticus:6.221252641140822,longicaudus)1:2.6659045796514462)1:6.221252641140822)1:1.1990966641099376)0.98:0.07805103894146795)1:0.13217977134887013):0.0); `).

# Visualizing gene trees as a densitree

This section is to create a densitree plot visualizing all the gene trees together. This is a nice way to visualize congruence (or lack thereof) between gene trees. First, trees should be rooted and scaled or the plot will be a mess. Note that scaling to a particular root age is only valid if the root is old enough that there is not too much real variance between the real root age of gene trees. At 36.6 million years, I am not expecting deep coalescence or introgression to be significantly affecting the age of gene trees between Alcidae and Stercorariidae. There will be some real variation, but I am not expecting any gene trees to coalesce millions of years more recently than others, so the proportional differences between loci will be small. Note we are rescaling to the same root age because some loci may evolve at very different rates, and so plotting them on the same scale (in terms of substitutions per site) creates a very messy figure that is extremely difficult to interpret due to a dense tangle of overlapping lines.

I am doing these edits in R:
```bash
#set working directory
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/2.5c_phylogeny_ASTRAL/dataset1
#run R
R
```
In this script, I am using the R package `ape` in order to root the gene trees and force them to be ultrametric (ie all branch tips reach the same point at the present) with the root scaled to 36.6 million years ago. This will be much nicer to visualize. 
Note that phangorn is not installed on the main server, so I needed to run the midpoint rooting on the new server, but the old server is much much faster at running the chronos loop, so I ended up saving the phangorm output, transferring it over to the old server, and then reloading it over there to run chronos.
```R
#set working directory
setwd("/home/1_Else/0_GENOMES5/Stercorarius/1_Data/2.5c_phylogeny_ASTRAL/dataset1")
#load the phangorn library for midpoint rooting
library(phangorn)
#load the ape library for chronos
library(ape)


#first run the Z chromosome dataset

#load trees from the ML tree set produced above
treeset <- ape::read.tree("ml_best.Astralblocks.unlinked.chrZ.nochrW.trees")
#midpoint root the trees #I am doing this rather than with the ape root() function so that the branch length is not zero between the split of the small jaegers from skuas, and the root.
treeset <- midpoint(treeset)
#if you want to subset the data, you could do it like this:
#treeset<-sample(treeset,size=1000)

#make a time tree
timetreeset<- treeset #initialize the multiphylo
for(i in 1:length(treeset)){
	calibration <- makeChronosCalib(treeset[[i]], node="root", age.min=36.6, age.max=36.6)
	treetemp <- try(chronos(treeset[[i]], lambda = 1, model = "discrete", calibration = calibration, control = chronos.control() ),TRUE)
	if(isTRUE(class(treetemp)=="try-error")) { next } else {timetreeset[[i]] <- treetemp }
}

#write the trees to a file
write.tree(timetreeset, file = "timetreeset.ml_best.Astralblocks.unlinked.chrZ.nochrW.midpointed.trees", append = FALSE, digits = 10, tree.names = FALSE) #saving intermediate results so I can transfer it between servers


#now repeat for the autosomal dataset

#load trees from the ML tree set produced above
treeset <- ape::read.tree("ml_best.Astralblocks.unlinked.trees")
#midpoint root the trees #I am doing this rather than with the ape root() function so that the branch length is not zero between the split of the small jaegers from skuas, and the root.
treeset <- midpoint(treeset)
#if you want to subset the data, you could do it like this:
#treeset<-sample(treeset,size=1000)

write.tree(treeset, file = "ml_best.Astralblocks.unlinked.midpointed.trees", append = FALSE, digits = 10, tree.names = FALSE) #saving intermediate results so I can transfer it between servers


#I have to switch between servers now

treeset <- ape::read.tree("ml_best.Astralblocks.unlinked.midpointed.trees") #reload the saved data on the old server

#make a time tree
timetreeset<- treeset #initialize the multiphylo
for(i in 1:length(treeset)){
	calibration <- makeChronosCalib(treeset[[i]], node="root", age.min=36.6, age.max=36.6)
	treetemp <- try(chronos(treeset[[i]], lambda = 1, model = "discrete", calibration = calibration, control = chronos.control() ),TRUE)
	if(isTRUE(class(treetemp)=="try-error")) { next } else {timetreeset[[i]] <- treetemp }
}

#write the trees to a file
write.tree(timetreeset, file = "timetreeset.ml_best.Astralblocks.unlinked.midpointed.trees", append = FALSE, digits = 10, tree.names = FALSE)
#resulted in 56,005
```
That actually took a couple days to run last time, using one thread. I was not in a rush so let it go in the background but if in a rush you could subset the data (do we really need 200,000 trees to just visualize this dataset?) or you could make it go in parallel (would take a couple hours rather than days). When I ran it the next time, it took a couple hours, but not days. The old server was much faster than the new server!

## Visualize in Densitree
Now that we have our set of rooted gene trees, visualize the gene trees in Densitree:
```bash
#first, get rid of the trees that failed to be scaled properly
#save the current results so we don't rewrite the original file
mv timetreeset.ml_best.Astralblocks.unlinked.midpointed.trees timetreeset.ml_best.Astralblocks.unlinked.midpointed.trees.raw #227728 trees
#get rid of any trees that don't have the root at 36.6 My, for which the script failed
grep "36\.6" timetreeset.ml_best.Astralblocks.unlinked.midpointed.trees.raw > timetreeset.ml_best.Astralblocks.unlinked.midpointed.trees #224894 trees since some failed to be made ultrametric #36.6 is the age of the root, so select all trees that have this number in them.
grep -v "Alca" timetreeset.ml_best.Astralblocks.unlinked.midpointed.trees | wc -l #check if there are lines without the outgroup - should be none!
#0

#select 1% of the dataset to plot to be easier to visualize if needed to get a quick glimpse
perl -ne 'print if (rand() < .01)' timetreeset.ml_best.Astralblocks.unlinked.midpointed.trees > timetreeset.ml_best.Astralblocks.unlinked.midpointed.subsampled.trees


#Now visualize with Densitree, see below for the steps within Densitree. I am opening it on my personal laptop.


#this is how you can open it on the server, but I am using my personal laptop
#conda activate java8
#/home/0_PROGRAMS/beast/bin/densitree -c 1.0 -i 1.0 -j 0 -w 4 -v 1 -f 100 -t 2 -b 200000 timetreeset.trees

#Since it is so fast, I want to verify that my timetree scaling would not have affected the results inferred by Astral
time java -jar /home/0_PROGRAMS/Astral/astral.5.7.3.jar -i timetreeset.trees -o timetreeset.tre 2> astral_species.timetreeset.log
#No, the results are identical in topology and branch lengths are essentially identical (eg 1.4932 vs 1.4929).
```
* `-b 0`: no burnin (use all gene trees for plotting)
* `-j'`: amount of jitter in trees

Once loaded in Densitree, I edit the appearance like this:
* Edit -> Shuffle -> Manual -> LTJA_MKP990 PAJA_B20730 PAJA_US-M606730 CHSK_MKP2451 CISK2 CISK55 CISK3 A-SK7 A-SK01 A-SK8 GRSK_MKP1592 GRSK_MKP1593 POJA_MKP1559 POJA_4 POJA_IB2659 Alca_torda  
#LTJA_MKP990 PAJA_B20730 CISK_32 CHSK_MKP2451 ANSK7 GRSK_MKP1592 POJA_MKP1559 Urilom
(that rearranges the tips to match the order displayed in other figures)
* Burn in -> 0  
(these are not MCMC runs but separate gene trees, so do not discard any as burn-in)
* Label -> hide  
(I am using species name rather than sample names in the publication so will draw them in manually)
* style -> Triangle  
(At first I favoured style steep, but changed my mind that triangle showed the data more clearly)
* Line color -> color by metadata
(At first I favoured colour by clade but changed my mind that the colour did not add any additional insight and was only distracting and drew to much attention to the clade highlighted in red). I decided that the default (which draws the most common and second most common etc trees in different colours) was actually very difficult to interpret since there are so many trees it is impossible to see any single gene tree clearly, so you could only really see the other gene trees in the branch towards the root and not in the blur of the skuas where the discordance actually ocurrs. Colour by metadata forces everything to be blue since there is no metadata.
* File -> export -> `timetreeset.densitree.blue.triable.png`, png -> save
I tried also saving as an svg (Standard Vector Graphics File) file format since generally vectors are better to work with when making figures since it does not lose resolution. However when I tried openning the resulting huge svg file on my computer, it would not display and caused my programs to crash so I went for the much easier-to-work-with png format which is still high resolution for publications.


# Expected gene tree topologies
We can actually calculate how much ILS to expect based on the branch lengths given by Astral. These branch lengths are in coalescent units. The proportion of gene trees that are expected to NOT coalesce along a branch are e^(-[branch length]), and these ILS trees should be evenly distributed amongst the 3 topologies (EXCEPT for deviations caused by ancestral population structure or introgression). For example, on Excel, if the branch length is in cell `D2` then the function to calculate the expected frequency of a discordant topology is `=(EXP(D2*-1)/3)`.

Warning: Astral branch lengths are not always accurate, especially since they rely on estimated gene trees which can have high rates of error. Also, since Astral estimates branch-lengths based on the proportion of discordant trees, it is somewhat circular to use its branch lengths to calculate the expected proportion of discordant trees.


# Appendix: IQtree2 Concordance analysis

We can also calculate concordance factors for each node. This requires us to pick a tree. Interpreting these gene tree concordance factors is a little less straightforward since they will not have three topologies that sum to 1 - since we have now sample multiple individuals per species, there will be many gene trees in which a species is not monophyletic, and will not cleanly contribute to any of the three alternative species topologies.

We provide iqtree2 with the trees that contain all samples (instead of populations)
```bash
#autosomal tree, all data
/home/0_PROGRAMS/iqtree-2.1.3-Linux/bin/iqtree2 -t species.Astralblocks.unlinked.tre --gcf ml_best.Astralblocks.unlinked.trees --scf 100 -seed 47532 -p alignments --prefix ml_best.Astralblocks.unlinked.gene_concordance

#autosomal tree, unlinked autosomal data
#make a folder to hold just the alignments for scf analysis, then copy the alignments of interest there
mkdir -p ml_best.Astralblocks.unlinked
cat ./Astralblocks.unlinked.txt | grep -v "chrZ" | while read line ; do cp alignments/"$line" ml_best.Astralblocks.unlinked/"$line" ; done
#run scf analysis on this subset of alignments
/home/0_PROGRAMS/iqtree-2.1.3-Linux/bin/iqtree2 -t species.Astralblocks.unlinked.tre --gcf ml_best.Astralblocks.unlinked.trees --scf 100 -seed 47532 -p ml_best.Astralblocks.unlinked --prefix ml_best.Astralblocks.unlinked.gene_concordance


#chrZ tree gene tree concordance
/home/0_PROGRAMS/iqtree-2.1.3-Linux/bin/iqtree2 -t species.Astralblocks.unlinked.chrZ.nochrW.tre --gcf ml_best.Astralblocks.unlinked.chrZ.nochrW.trees --prefix ml_best.Astralblocks.unlinked.chrZ.nochrW.gene_concordance

#chrZ tree gene tree concordance + chrZ site concordance
#make a folder to hold just the alignments for scf analysis, then copy the alignments of interest there
mkdir -p ml_best.Astralblocks.unlinked.chrZ.nochrW
cat ./Astralblocks.unlinked.txt | grep "chrZ" | awk '{split($1,a,"_"); if (a[2] >= 30000000) { print } }' | while read line ; do cp alignments/"$line" ml_best.Astralblocks.unlinked.chrZ.nochrW/"$line" ; done
#run scf analysis on this subset of alignments
/home/0_PROGRAMS/iqtree-2.1.3-Linux/bin/iqtree2 -t species.Astralblocks.unlinked.chrZ.nochrW.tre --gcf ml_best.Astralblocks.unlinked.chrZ.nochrW.trees --scf 100 -seed 47532 -p ml_best.Astralblocks.unlinked.chrZ.nochrW --prefix ml_best.Astralblocks.unlinked.chrZ.nochrW.gene_concordance

```

I am not using these results in the publication, as it is not straightforward to quantify the discordance for trees when species are not monophyletic. Instead, the quartet scores output by Astral fill this task.


# Appendix: Astral with phased datasets

Before moving on, I want to verify my results with one more dataset. I will use the phased dataset based on genotype calls. I did not using this dataset because the filters were too stringent to retain data for the non-pseudoautosomal chrZ region, but I still want to run Astral on the autosomal data to enssure that the results are robust to choice of dataset (unphased, angsd genotype likelihoods vs phased, bcftools genotype calls).
```bash
mkdir -p /home/1_Else/0_GENOMES5/Stercorarius/2_Phylogeny/2.5c_phylogeny_ASTRAL/dataset_phased
cd /home/1_Else/0_GENOMES5/Stercorarius/2_Phylogeny/2.5c_phylogeny_ASTRAL/dataset_phased

cp /home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.6.1c_phylogenetic_blocks/filtered_haplotypes/filtered_blocks.txt . #53200 loci
#place the chosen alignments into a folder
mkdir -p filtered_blocks_alignments
#remove Alca and Fratercula
cat filtered_blocks.txt | cut -f 1 | while read block ; do /home/0_PROGRAMS/seqkit grep -v -p "Uria" -p "Fratercula" --use-regexp /home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.6.1c_phylogenetic_blocks/aligned/"$block".fasta > ./filtered_blocks_alignments/"$block".Alcaout.fasta ; done
#make phylogenies
time cat filtered_blocks.txt | cut -f 1 | parallel --jobs 70 time /home/0_PROGRAMS/iqtree-2.1.3-Linux/bin/iqtree2 -s filtered_blocks_alignments/{1}.Alcaout.fasta #357m39.094s real 19370m20.393s user

#repeat for chrZ (Note that these are ALL from the pseudoautosomal region)
cp /home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.6.1c_phylogenetic_blocks/filtered_haplotypes_chrZ/chrZ_filtered_blocks.txt . #197 loci
cat chrZ_filtered_blocks.txt | cut -f 1 | while read block ; do /home/0_PROGRAMS/seqkit grep -v -p "Uria" -p "Fratercula" --use-regexp /home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.6.1c_phylogenetic_blocks/aligned_chrZ/"$block".fasta > ./filtered_blocks_alignments/"$block".Alcaout.fasta ; done
time cat chrZ_filtered_blocks.txt | cut -f 1 | parallel --jobs 70 time /home/0_PROGRAMS/iqtree-2.1.3-Linux/bin/iqtree2 -s filtered_blocks_alignments/{1}.Alcaout.fasta #357m39.094s real 19370m20.393s user

```
The maximum likelihood trees will be produced in `*.treefile`. Concatenate them all for Astral.

```bash
#all trees except chrZ (redundant, there were no chrZ trees in this dataset to begin with)
time cat filtered_blocks.txt | cut -f 1 | grep -v "chrZ" | while read line ; do cat filtered_blocks_alignments/"$line".Alcaout.fasta.treefile >> ml_best.autosomal.trees ; done
wc -l ml_best.autosomal.trees
#48960

#all chrZ trees (all of these are from the first 30 Mb, where chrW sequences also align)
time cat chrZ_filtered_blocks.txt | cut -f 1 | while read line ; do cat filtered_blocks_alignments/"$line".Alcaout.fasta.treefile >> ml_best.chrZ.trees ; done
wc -l ml_best.chrZ.trees
#171

```
Since we have multiple individuals per species, I will specify a mapping file:
```bash
cat > Astral_SpeciesMappingFile
````
```
longicaudus:LTJA_MKP990.1,LTJA_MKP990.2
parasiticus:PAJA_B20730.1,PAJA_USNM606730.1,PAJA_B20730.2,PAJA_USNM606730.2
pomarinus:POJA_MKP1559.1,POJA_4.1,POJA_IB2659.1,POJA_MKP1559.2,POJA_4.2,POJA_IB2659.2
skua:GRSK_MKP1592.1,GRSK_MKP1593.1,GRSK_MKP1592.2,GRSK_MKP1593.2
antarcticus:CISK2.1,CISK55.1,CISK_3.1,CISK2.2,CISK55.2,CISK_3.2
chilensis:CHSK_MKP2451.1,CHSK_MKP2451.2
maccormicki:ANSK7.1,ANSK01.1,ANSK8.1,ANSK7.2,ANSK01.2,ANSK8.2
Alcidae:Alca_torda.1,Alca_torda.2
```
Then I run Astral
```bash
#without specifying species identity, treating each sample as a "species"
#autosomal, treating individual haplotypes as populations
time java -jar /home/0_PROGRAMS/Astral/astral.5.7.3.jar -i ml_best.autosomal.trees -o species.autosomal.tre 2> astral_species.autosomal.log #
#crashed after a 3 days due to power outage, did not finish

#autosomal
time java -jar /home/0_PROGRAMS/Astral/astral.5.7.3.jar -i ml_best.autosomal.trees --namemapfile Astral_SpeciesMappingFile -o species.partitioned.autosomal.tre --reps 100 2> astral_species.partitioned.autosomal.log #372m58.594s
#(skua:0.2430679184258866,((pomarinus:0.04421049985520899,(Alcidae:9.280908622905825,(parasiticus:4.503318213231763,longicaudus:2.359864789954984)1:1.3882448794113076)1:3.135991617491465)1:0.5551908406721346,((antarcticus:0.3187411610243539,maccormicki:0.18923911894852138)1:0.034783384172420224,chilensis:0.20748662971631232)1:0.031998928757699954):0.0); 

#chrZ (pseudoautosomal region)
time java -jar /home/0_PROGRAMS/Astral/astral.5.7.3.jar -i ml_best.chrZ.trees -o species.chrZ.tre 2> astral_species.chrZ.log #
#(ANSK01.1,((ANSK7.1,ANSK7.2)0.6:0.04217615878165698,((ANSK01.2,ANSK8.2)0.56:0.035395137338493844,(ANSK8.1,(((CISK2.1,CISK_3.1)0.79:0.07316236186379217,((CISK_3.2,CISK2.2)0.98:0.14925714944575988,(CISK55.2,CISK55.1)0.77:0.0950662311569002)0.45:0.019631856994536928)1:0.31437978257011095,((((GRSK_MKP1593.2,GRSK_MKP1592.2)0.97:0.13276045114823612,(GRSK_MKP1593.1,GRSK_MKP1592.1)0.54:0.03362592807554536)0.99:0.15734827006296886,(((POJA_4.1,POJA_4.2)0.94:0.12700405822388575,((POJA_IB2659.1,POJA_IB2659.2)0.93:0.11513263703133578,(POJA_MKP1559.1,POJA_MKP1559.2)0.64:0.04986268679067076)0.85:0.08683351835628372)0.65:0.05861295955094287,((Alca_torda.1,Alca_torda.2)1:4.742029368705287,((LTJA_MKP990.1,LTJA_MKP990.2)1:2.6625878270254533,(PAJA_B20730.2,(PAJA_USNM606730.1,(PAJA_B20730.1,PAJA_USNM606730.2)0.5:0.030326940006608554)0.73:0.06920053424338249)1:3.6180992720528824)1:1.4278433640327635)1:3.181937330374341)1:0.46123817165102715)0.91:0.1120141427200831,(CHSK_MKP2451.1,CHSK_MKP2451.2)1:0.5085269112420707)0.73:0.07436267562745279)0.98:0.14131911343929043)0.42:0.015663174464573937)0.77:0.07789170662908157):0.0); 

```
With the phased dataset, results are very similar to the unphased dataset. The autosomal data places *S. antarcticus* sister to *S. maccormicki*. Branch lengths are also quite similar, but the phased dataset is able to estimate branchlengths for the root (*Alca torda*) and for *S. chilensis*. However, the phased dataset did not retain data for the nonrecombining region of chrZ, so we cannot compare chrZ to autosomes. For that reason, I chose to report the unphased results, with the reassurance that the autosomal topology was unaffected by phasing.
