## Overview
This page describes how to build a species tree from a set of sequence alignments using StarBeast3. 

### Required Input
A set of (phased) sequence alignments from blocks throuhout the genome. Ideally the blocks should be short enough to not contain recombination but long enough to resolve a gene tree. I used 5 kb blocks, which are stored in the folder `strictgenes.1`.

### Output
A species tree. The species tree is found in the file `species.16runs.100MILL.tree`

We used Starbeast3 v1.0.4 in Beast v2.6.7. We used Metropolis-coupled MCMC implemented in the CoupledMCMC v1.0.2 package, with one cold chain and two hot chains to improve mixing. We used a HKY site model with one gamma categories (i.e., no gamma heterogeneity) while estimating the proportion of invariant sites. This site model was suggested by the author of StarBeast3 in order to allow the program to run more quickly (it would take more than twice as long to run with 4 gamma categories)

To calibrate the age of the nodes, we constrained the divergence between Alcidae and Stercorariidae to match a previous estimate for the age of this node. We ran the analysis for 100,000,000 generations, sampling every 5000 generations and discarding the first 20% as burn-in. We repeated the analysis with 16 iterations starting from different random seeds to verify that the runs converged.

# Get alignments

The input data for StarBeast3 is a set of sequence alignments. I built a dataset of phased haplotype blocks in step 1.6.1c, and I will be using these here. Since StarBeast3 is so computationally intensive, I am only using about 100 blocks even though we have thousands of blocks available. Unfortunately, analyzing the full genome in a full multispecies coalescent model is not yet feasible.

```bash
#mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/2_Phylogeny/2.6c_Starbeast3_phylogeny
#cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/2_Phylogeny/2.6c_Starbeast3_phylogeny

mkdir -p /home/1_Else/0_GENOMES5/Stercorarius/2_Phylogeny/2.6c_Starbeast3_phylogeny
cd /home/1_Else/0_GENOMES5/Stercorarius/2_Phylogeny/2.6c_Starbeast3_phylogeny

#first, get a list of our loci
#only the strict dataset
cp /home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.6.1c_phylogenetic_blocks/missingness/strictpassed.txt . #107 trees

#place the chosen alignments into a folder
mkdir -p strictgenes.1
cat strictpassed.txt | cut -f 1 | while read block ; do cp /home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.6.1c_phylogenetic_blocks/aligned/"$block".fasta ./strictgenes.1 ; done

#remove Alca and Fratercula
cat strictpassed.txt | cut -f 1 | while read block ; do /home/0_PROGRAMS/seqkit grep -v -p "Uria" -p "Fratercula" --use-regexp ./strictgenes.1/"$block".fasta > ./strictgenes.1/"$block".Alcaout.fasta ; done

```


# build the .xml file

BEAST analyses can be quite complicated, with many different parameters that would be far too long to input on the command line. Instead, BEAST uses an .xml file, which includes the data as well as specifcations for the analysis. This file can be very large, and so BEAST comes with a program (Beauti) to generate the xml file using a graphical user interface. It is possible to write scripts to generate xml files, but to keep up with updates, it may be simpler to use Beauti. Since I can't use the gui on the server, I am running this on my personal laptop. Working in beauti with about 100 genes is easy, but with 1000 it becomes painfully slow (almost an hour between mouse clicks), so with extremely large datasets a script could be handy.

Now, we need to set up the analysis. First, load the starbeast3 template for Beauti:
`File -> Template -> Starbeast3`
If that template is not an option, Starbeast3 might not be installed. You can check in `File -> Manage Packages`

Next, load all of the alignment data (nexus or fasta files) to analyze.
`File -> "Import Alignment" -> [select all of the alignments] -> Open`
`all are nucleotide`

### Partitions tab
Now we are in the "Partitions" tab. Since our loci are assumedly genetically unlinked, we will not link the trees. If we had mitochondrial loci, we would need to link them. There is also the option to link clock or site models. I am not linking site models because the parameters in the site model are quite fast to estimate, so we might as well allow the different loci to have different models (and it is true that different regions of the genome have different base frequencies, etc). I am not linking clock models based on advise from Jordan Douglas and in the starbeast tutorial: "`By default, each gene tree is associated with its own clock rate, and these rates have a prior distribution with a mean of 1. The substitution rate of a branch in a gene tree is equal to its clock rate multiplied by the clock rate of the species tree (configured in the "Clock Model" tab). This model works quite well and accounts for different genetic loci being exposed to unique selective pressures. When there are many genes trees, the clock rate estimates average out to ~1.0, and therefore are able to be estimated without affecting the estimated species tree height.
If gene tree clock models are linked, then they will share their clock rate. If all clock models are linked, then all gene trees will share the same clock rate, and this will likely introduce convergence issues, because the parameter is non-identifiable with the tree heights. We therefore advise against linking all clock models.`"

### Taxon sets tab
Now we need to ensure that BEAST knows which samples belong to the same population. Go to the "Taxon Sets" tab. Here, it lists all the samples on the left, and their guessed populations on the right. Unless you have nicely formatted sample names, the populations guesses are probably not right. You can mannually correct it, or use a text file that provides the correct population assignments. I am using my samples.txt file which has two tab separated columns: the first has the sample names, and the second has the corresponding population name for each sample (see above).
`Taxon Sets -> Guess -> read from file -> Browse -> samples.txt -> Open -> ok`
Here are the contents of `samples.txt`:
```
ANSK01.1	maccormicki
ANSK01.2	maccormicki
ANSK7.1	maccormicki
ANSK7.2	maccormicki
ANSK8.1	maccormicki
ANSK8.2	maccormicki
CHSK_MKP2451.1	chilensis
CHSK_MKP2451.2	chilensis
CISK2.1	antarcticus
CISK2.2	antarcticus
CISK_3.1	antarcticus
CISK_3.2	antarcticus
CISK55.1	antarcticus
CISK55.2	antarcticus
GRSK_MKP1592.1	skua
GRSK_MKP1592.2	skua
GRSK_MKP1593.1	skua
GRSK_MKP1593.2	skua
LTJA_MKP990.1	longicaudus
LTJA_MKP990.2	longicaudus
PAJA_B20730.1	parasiticus
PAJA_B20730.2	parasiticus
PAJA_USNM606730.1	parasiticus
PAJA_USNM606730.2	parasiticus
POJA_4.1	pomarinus
POJA_4.2	pomarinus
POJA_IB2659.1	pomarinus
POJA_IB2659.2	pomarinus
POJA_MKP1559.1	pomarinus
POJA_MKP1559.2	pomarinus
Uria_lomvia.1	alcid
Uria_lomvia.2	alcid
Alca_torda.1	alcid
Alca_torda.2	alcid
```
(note I have both *Uria* and *Alca* listed as "alcid" - that is for if I wanted to run tests with *Uria* as the alcid, it would of course not be appropriate to include both as the same time and call them a single population. I only have data for *Alca* in the alignments, not *Uria*)

### Gene ploidy tab
All my alignments are from autosomal loci, so I will leave all of the ploidy options set to 2.0. In general, Z/X chromosome loci have ploidy of 1.5, and W/Y chromosome has 0.5.

### Site model tab
Next I will set up the model in the Site Model tab. I will use a HKY model so that transitions and transversions can have different rates, and I will use 1 gamma category and estimate the proportion of invariant sites: I am doing this due to recommendations of the authors, in order to speed the analysis so that it will eventually reach stationary; using 4 gamma categories would be much slower. I specify this model for the first locus/partition and then copy it to all the other partitions. 
* `Subst Model: HKY`
* `Gamma Category Count: 1`
* `Select all other partitions -> Clone from [name of first partition]`

### Clock model tab
Now we specify the clock model. Since we want our branch lengths in units of million years, rather than substitutions per site, we need to estimate a clock rate. Before switching to this tab, we need to disable the "automatic set clock rate"
* `Mode -> uncheck "automatic set clock rate`
Since the Stercorariids are closely related and appear to have equal genetic divergence from their outgroup (Alcidae), I will assume that there all lineages have equal substitution rates, and so I will use a strict clock model, which is much more computationally efficient. In some cases this may not be biologically reasonable, so you should evaluate your data beforehand to see whether it is reasonable to assume a strict clock. Note I am starting the clock rate at 0.001 to start the model somewhere in the ballpark of a reasonable estimate (based on my estimate of 4.6% divergence between Alcidae and Stercorariidae over 36.6 My), but this number will be estimated during the run.
* `Clock Model: Species tree strict clock`
* `Clock.rate: 0.001`
* `check "estimate" for Clock.rate`

### Priors tab
Now I will set up the priors. I am going to use a birth death model. preliminary runs with the Calibrated Yule Model produced the same topology but could not reach a higher ESS score for two of the parameters. There has almost certainly been a lot of extinction in the Stercorariid phylogeny (and, we have only one Alcid!).
* `Tree.t:Species: Birth Death Model`
Now we set up the prior for the clock model. I am choosing a distribution that puts most of the distribution around a reasonable value (0.001) but allows for some larger values. It ended up estimating a value around 0.0005.
* `SpeciesTreeStrictClockRate: log 0` Starting value: 0.001 (set in clock model tab).
Click the dropdown arrow beside `SpeciesTreeStrictClockRate`:
* `M=0.002`
* `S=2.0`
* `check "Mean in real space"`
Next we add a calibration based on prior knowledge about the divergence between Alcidae and Stercorariidae. We are going to constrain the root of the tree to be 36.6 My by placing a very tight prior distribution around this parameter. Scroll all the way to the bottom and click `+ Add Prior`
* `"+Add Prior" -> MRCA prior -> Tree.t:species -> (select all but outgroup) -> >> -> ok` You also need to give a name ("Taxon set label"), I called it "root"
* check "monophyletic" for the new prior. Log normal distribution. Mean = 36.6, S = 0.001, check "mean in real space", check "use originate". NOTE: before running, I changed the distribution to have a mean of 34.7 and standard deviation of 0.008. This corresponds to  the estimate from Smith & Clark (2015).

### MCMC tab
Finally, go to the MCMC tab. I am going to make it store the results more often, so that I can see the progress go by faster, and possibly to increase ESS more quickly by saving states more often.
* I will start out with the default 10,000,000 generations.
* `Store every: 5000`
* `Tracelog: Log every: 5000`
* `speciesTreeLogger: Log Every: 5000`
* `screenlog: log Every: 5000`

Now it is ready to save.
`File -> Save As -> strictgenes_g1.bd.xml`

## Edit xml file

There was a bug in the xml file ParallelMCMCTreeOperator, in which it did not contain the plate tag that lists all the gene in the alignment, but this can be fixed manually (bug fix by Jordan Douglas):  
```bash
#to get a list of the gene names in the xml file, could use this command:
grep 'id="chr' strictgenes_g1.bd.xml | sed 's/id="//g; s/"//g' | tr "\n" "," | sed 's/,$//g'
#then copy-paste this list in between the quotes where it says range="PASTE_LIST_OF_LOCI_HERE"
#sed 's|<operator id="ParallelMCMCTreeOperator" spec="starbeast3.operators.ParallelMCMCTreeOperator" chainCoverage="1.0" learning="false" nregression="50" otherState="@state" runtime="1000.0" speciesTree="@Tree.t:Species" targetCPU="0.0" weight="1.0">|<operator id="ParallelMCMCTreeOperator" spec="starbeast3.operators.ParallelMCMCTreeOperator" chainCoverage="1.0" learning="false" nregression="50" otherState="@state" runtime="1000.0" speciesTree="@Tree.t:Species" targetCPU="0.0" weight="1.0">\n\t<plate var="n" range="PASTE_LIST_OF_LOCI_HERE">\n\t\t<distribution id="ParallelMCMCTreeOperatorLikelihood.$(n)" spec="starbeast3.operators.ParallelMCMCTreeOperatorTreeDistribution" geneprior="@treePrior.t:$(n)" tree="@Tree.t:$(n)" treelikelihood="@treeLikelihood.$(n)"/>\n\t</plate>|g' INPUT.xml > OUTPUT.xml

#fix bug in the xml file
sed 's|<operator id="ParallelMCMCTreeOperator" spec="starbeast3.operators.ParallelMCMCTreeOperator" chainCoverage="1.0" learning="false" nregression="50" otherState="@state" runtime="1000.0" speciesTree="@Tree.t:Species" targetCPU="0.0" weight="1.0">|<operator id="ParallelMCMCTreeOperator" spec="starbeast3.operators.ParallelMCMCTreeOperator" chainCoverage="1.0" learning="false" nregression="50" otherState="@state" runtime="1000.0" speciesTree="@Tree.t:Species" targetCPU="0.0" weight="1.0">\n\t<plate var="n" range="chr3:034505001-034509999,chr6:39800001-39804999,chr18:5590001-5594999,chr3:046040001-046044999,chr8:07165001-07169999,chr21:6340001-6344999,chr13:33465001-33469999,chr3:034600001-034604999,chr1:056410001-056414999,chr20:0285001-0289999,chr5:53455001-53459999,chr5:41895001-41899999,chr6:31280001-31284999,chr11:35580001-35584999,chr22:5835001-5839999,chr11:20020001-20024999,chr5:59615001-59619999,chr9:17755001-17759999,chr11:31915001-31919999,chr14:11930001-11934999,chr20:6615001-6619999,chr6:10635001-10639999,chr3:034125001-034129999,chr15:02850001-02854999,chr17:1580001-1584999,chr2:022665001-022669999,chr1:100420001-100424999,chr10:41015001-41019999,chr5:25420001-25424999,chr13:22055001-22059999,chr13:22335001-22339999,chr4:47180001-47184999,chr1:202240001-202244999,chr20:0170001-0174999,chr9:22220001-22224999,chr18:3400001-3404999,chr10:29530001-29534999,chr21:5330001-5334999,chr3:043140001-043144999,chr6:29945001-29949999,chr9:17715001-17719999,chr13:22170001-22174999,chr22:5890001-5894999,chr17:0265001-0269999,chr22:5870001-5874999,chr17:2400001-2404999,chr2:035615001-035619999,chr8:34255001-34259999,chr4:33270001-33274999,chr1:056145001-056149999,chr13:32150001-32154999,chr6:38390001-38394999,chr20:6075001-6079999,chr3:034200001-034204999,chr12:31755001-31759999,chr21:3395001-3399999,chr15:12275001-12279999,chr9:43220001-43224999,chr13:01285001-01289999,chr13:02400001-02404999,chr21:5385001-5389999,chr1:056455001-056459999,chr13:33555001-33559999,chr3:034710001-034714999,chr1:055905001-055909999,chr11:30925001-30929999,chr3:034015001-034019999,chr8:02650001-02654999,chr9:28870001-28874999,chr15:02965001-02969999,chr16:6580001-6584999,chr8:16985001-16989999,chr6:11510001-11514999,chr11:16625001-16629999,chr13:33440001-33444999,chr12:18030001-18034999,chr8:43315001-43319999,chr9:41735001-41739999,chr9:22230001-22234999,chr5:31395001-31399999,chr5:24580001-24584999,chr20:7285001-7289999,chr13:34450001-34454999,chr6:12315001-12319999,chr3:024040001-024044999,chr1:088135001-088139999,chr12:10435001-10439999,chr22:5880001-5884999,chr8:02625001-02629999,chr13:00125001-00129999,chr4:25350001-25354999,chr15:00400001-00404999,chr13:05340001-05344999,chr4:33165001-33169999,chr11:02860001-02864999,chr19:3730001-3734999,chr1:037340001-037344999,chr1:100175001-100179999,chr6:38455001-38459999,chr24:1810001-1814999,chr11:17975001-17979999,chr13:04880001-04884999,chr14:00235001-00239999,chr20:7225001-7229999,chr3:044950001-044954999,chr4:26310001-26314999,chr3:004705001-004709999">\n\t\t<distribution id="ParallelMCMCTreeOperatorLikelihood.$(n)" spec="starbeast3.operators.ParallelMCMCTreeOperatorTreeDistribution" geneprior="@treePrior.t:$(n)" tree="@Tree.t:$(n)" treelikelihood="@treeLikelihood.$(n)"/>\n\t</plate>|g' strictgenes_g1.bd.xml > strictgenes_g1_bd_fixed.xml

```

Next, I am editing the xml file so that it will use coupledMCMC to traverse posterior parameter space for efficiently. I will use three chains (two hot, one cold). I am also editing the weights of a few operators based on recommendations from preliminary runs of StarBeast3 (these recommendations are printed at the end of a run)
```bash
#deltaTemperature="0.025" defines the temperature difference between the chain n and chain n-1. This value should be changed such that the acceptance probability of a swap is between 0.25 and 0.6
#chains="2" defines the number of parallel chains that are run. The first chain is the one that explores the posterior just like a normal MCMC chain. All other chains are what's called heated. This means that MCMC moves of those chains have a higher probability of being accepted. While these heated chains don't explore the posterior properly, they can be used to propose new states to the one cold chain.
sed 's|<run id="mcmc" spec="MCMC" chainLength="100000000" storeEvery="5000">|<run id="mcmc" spec="beast.coupledMCMC.CoupledMCMC" chainLength="100000000" storeEvery="5000" deltaTemperature="0.025" chains="3" resampleEvery="10000">|g' ./strictgenes_g1_bd_fixed.xml > strictgenes_g1_bd_coupledmcmc_fixed.xml

#also, follow recommendations about the operators from the last round
sed -i 's|<operator id="strictClockRateScaler" spec="BactrianScaleOperator" parameter="@SpeciesTreeStrictClockRate" rootOnly="true" scaleFactor="0.7" upper="10.0" weight="3.0"/>|<operator id="strictClockRateScaler" spec="BactrianScaleOperator" parameter="@SpeciesTreeStrictClockRate" rootOnly="true" scaleFactor="0.015" upper="10.0" weight="3.0"/>|g; s|<operator id="TreeRootScaler.t:Species" spec="BactrianScaleOperator" rootOnly="true" scaleFactor="0.03" tree="@Tree.t:Species" upper="10.0" weight="3.0"/>|<operator id="TreeRootScaler.t:Species" spec="BactrianScaleOperator" rootOnly="true" scaleFactor="0.002" tree="@Tree.t:Species" upper="10.0" weight="3.0"/>|g; s|<operator id="BactrianNodeOperator.t:Species" spec="BactrianNodeOperator" tree="@Tree.t:Species" weight="3.0"/>|<operator id="BactrianNodeOperator.t:Species" spec="BactrianNodeOperator" tree="@Tree.t:Species" scaleFactor="0.067" weight="3.0"/>|g' strictgenes_g1_bd_coupledmcmc_fixed.xml

#no point printing screenlog if we are not running in a visible terminal, so make it log much less frequently
sed -i 's/logger id="screenlog" spec="Logger" logEvery="5000"/logger id="screenlog" spec="Logger" logEvery="500000"/g' strictgenes_g1_bd_coupledmcmc_fixed.xml

```

# Run StarBeast3

StarBeast3 is quite computationally intensive, so I am runnning it on Niagara. This way, I can run several iterations at the same time. This will allow me to ensure that the runs converge to the same result (they are not probably getting stuck at a local optimum), and to combine the results of these independant runs to reach a higher effective sample size in a shorter amount of time, so that it takes weeks instead of months.

First, I set up a directory and bring my xml file here

# Birth death model, round 1

```bash
mkdir $SCRATCH/strictgenes_g1.bd.34My
cd $SCRATCH/strictgenes_g1.bd.34My

#get xml file here
cp ../strictgenes_g1.bd/strictgenes_g1_bd_coupledmcmc_fixed.xml .
sed 's|<parameter id="RealParameter.754" spec="parameter.RealParameter" estimate="false" name="M">36.6</parameter>|<parameter id="RealParameter.754" spec="parameter.RealParameter" estimate="false" name="M">34.7</parameter>|g; s|<parameter id="RealParameter.755" spec="parameter.RealParameter" estimate="false" lower="0.0" name="S" upper="5.0">0.001</parameter>|<parameter id="RealParameter.755" spec="parameter.RealParameter" estimate="false" lower="0.0" name="S" upper="5.0">0.008</parameter>|g' ../strictgenes_g1.bd/strictgenes_g1_bd_coupledmcmc_fixed.xml > strictgenes_g1_bd_34My_coupledmcmc_fixed.xml

#make folders to hold the results
for replicate in run1 run2 run3 run4 run5 run6 run7 run8 run9 run10 run11 run12 run13 run14 run15 run16 ; do mkdir "$replicate" ; done

```

I am running 4 replicates at a time, such that each run can use 10 of the 40 available cores. For 16 replicates, that is 4 batches. Each batch is run with a batch file:  

```bash
cat > strictgenes_g1_coupledmcmc_fixed_1to4round1.sh
chmod +x strictgenes_g1_coupledmcmc_fixed_1to4round1.sh
```
```bash
#!/bin/bash
#SBATCH --nodes=1 #
#SBATCH --ntasks-per-node=4
#SBATCH --cpus-per-task=10
#SBATCH --mem=0 #set to 0 to use all available memory of each requested node. Advatage, any of the ntasks of each node can use any amount of the memory provided to sum does not exceed thata available to the node
#SBATCH --time=23:59:00
#SBATCH --job-name 1to4round1
#SBATCH --output=strictgenes_g1_coupledmcmc_fixed_1to4round1.txt 

module load CCEnv
module load StdEnv
module load java
module load beagle-lib
export LD_LIBRARY_PATH=/home/j/jweir/elsemikk/tools/beagle-lib/lib:$LD_LIBRARY_PATH

cd $SCRATCH/strictgenes_g1.bd.34My

#I used random.org to generate 8 random numbers for the seed

cd $SCRATCH/strictgenes_g1.bd.34My/run1
$HOME/tools/BEAST_v2.6.7/beast/bin/beast -seed 2120 -threads 10 ../strictgenes_g1_bd_34My_coupledmcmc_fixed.xml &

cd $SCRATCH/strictgenes_g1.bd.34My/run2
$HOME/tools/BEAST_v2.6.7/beast/bin/beast -seed 9818 -threads 10 ../strictgenes_g1_bd_34My_coupledmcmc_fixed.xml &

cd $SCRATCH/strictgenes_g1.bd.34My/run3
$HOME/tools/BEAST_v2.6.7/beast/bin/beast -seed 3933 -threads 10 ../strictgenes_g1_bd_34My_coupledmcmc_fixed.xml &

cd $SCRATCH/strictgenes_g1.bd.34My/run4
$HOME/tools/BEAST_v2.6.7/beast/bin/beast -seed 4763 -threads 10 ../strictgenes_g1_bd_34My_coupledmcmc_fixed.xml &

wait

```
```bash
cat > strictgenes_g1_coupledmcmc_fixed_5to8round1.sh
chmod +x strictgenes_g1_coupledmcmc_fixed_5to8round1.sh
```
```bash
#!/bin/bash
#SBATCH --nodes=1 #
#SBATCH --ntasks-per-node=4
#SBATCH --cpus-per-task=10
#SBATCH --mem=0 #set to 0 to use all available memory of each requested node. Advatage, any of the ntasks of each node can use any amount of the memory provided to sum does not exceed thata available to the node
#SBATCH --time=23:59:00
#SBATCH --job-name 5to8round1
#SBATCH --output=strictgenes_g1_coupledmcmc_fixed_5to8round1.txt 

module load CCEnv
module load StdEnv
module load java
module load beagle-lib
export LD_LIBRARY_PATH=/home/j/jweir/elsemikk/tools/beagle-lib/lib:$LD_LIBRARY_PATH

cd $SCRATCH/strictgenes_g1.bd.34My

#I used random.org to generate 8 random numbers for the seed

cd $SCRATCH/strictgenes_g1.bd.34My/run5
$HOME/tools/BEAST_v2.6.7/beast/bin/beast -seed 6448 -threads 10 ../strictgenes_g1_bd_34My_coupledmcmc_fixed.xml &

cd $SCRATCH/strictgenes_g1.bd.34My/run6
$HOME/tools/BEAST_v2.6.7/beast/bin/beast -seed 5596 -threads 10 ../strictgenes_g1_bd_34My_coupledmcmc_fixed.xml &

cd $SCRATCH/strictgenes_g1.bd.34My/run7
$HOME/tools/BEAST_v2.6.7/beast/bin/beast -seed 2363 -threads 10 ../strictgenes_g1_bd_34My_coupledmcmc_fixed.xml &

cd $SCRATCH/strictgenes_g1.bd.34My/run8
$HOME/tools/BEAST_v2.6.7/beast/bin/beast -seed 5303 -threads 10 ../strictgenes_g1_bd_34My_coupledmcmc_fixed.xml &

wait

```

```bash
cat > strictgenes_g1_coupledmcmc_fixed_9to12round1.sh
chmod +x strictgenes_g1_coupledmcmc_fixed_9to12round1.sh
```
```bash
#!/bin/bash
#SBATCH --nodes=1 #
#SBATCH --ntasks-per-node=4
#SBATCH --cpus-per-task=10
#SBATCH --mem=0 #set to 0 to use all available memory of each requested node. Advatage, any of the ntasks of each node can use any amount of the memory provided to sum does not exceed thata available to the node
#SBATCH --time=23:59:00
#SBATCH --job-name 9to12round1
#SBATCH --output=strictgenes_g1_coupledmcmc_fixed_9to12round1.txt 

module load CCEnv
module load StdEnv
module load java
module load beagle-lib
export LD_LIBRARY_PATH=/home/j/jweir/elsemikk/tools/beagle-lib/lib:$LD_LIBRARY_PATH

cd $SCRATCH/strictgenes_g1.bd.34My

#I used random.org to generate 8 random numbers for the seed

cd $SCRATCH/strictgenes_g1.bd.34My/run9
$HOME/tools/BEAST_v2.6.7/beast/bin/beast -seed 8032 -threads 10 ../strictgenes_g1_bd_34My_coupledmcmc_fixed.xml &

cd $SCRATCH/strictgenes_g1.bd.34My/run10
$HOME/tools/BEAST_v2.6.7/beast/bin/beast -seed 3583 -threads 10 ../strictgenes_g1_bd_34My_coupledmcmc_fixed.xml &

cd $SCRATCH/strictgenes_g1.bd.34My/run11
$HOME/tools/BEAST_v2.6.7/beast/bin/beast -seed 8503 -threads 10 ../strictgenes_g1_bd_34My_coupledmcmc_fixed.xml &

cd $SCRATCH/strictgenes_g1.bd.34My/run12
$HOME/tools/BEAST_v2.6.7/beast/bin/beast -seed 7683 -threads 10 ../strictgenes_g1_bd_34My_coupledmcmc_fixed.xml &

wait

```
```bash
cat > strictgenes_g1_coupledmcmc_fixed_13to16round1.sh
chmod +x strictgenes_g1_coupledmcmc_fixed_13to16round1.sh
```

```bash
#!/bin/bash
#SBATCH --nodes=1 #
#SBATCH --ntasks-per-node=4
#SBATCH --cpus-per-task=10
#SBATCH --mem=0 #set to 0 to use all available memory of each requested node. Advatage, any of the ntasks of each node can use any amount of the memory provided to sum does not exceed thata available to the node
#SBATCH --time=23:59:00
#SBATCH --job-name 13to16round1
#SBATCH --output=strictgenes_g1_coupledmcmc_fixed_13to16round1.txt 
module load CCEnv
module load StdEnv
module load java
module load beagle-lib
export LD_LIBRARY_PATH=/home/j/jweir/elsemikk/tools/beagle-lib/lib:$LD_LIBRARY_PATH

cd $SCRATCH/strictgenes_g1.bd.34My

#I used random.org to generate 8 random numbers for the seed

cd $SCRATCH/strictgenes_g1.bd.34My/run13
$HOME/tools/BEAST_v2.6.7/beast/bin/beast -seed 3183 -threads 10 ../strictgenes_g1_bd_34My_coupledmcmc_fixed.xml &

cd $SCRATCH/strictgenes_g1.bd.34My/run14
$HOME/tools/BEAST_v2.6.7/beast/bin/beast -seed 6201 -threads 10 ../strictgenes_g1_bd_34My_coupledmcmc_fixed.xml &

cd $SCRATCH/strictgenes_g1.bd.34My/run15
$HOME/tools/BEAST_v2.6.7/beast/bin/beast -seed 6598 -threads 10 ../strictgenes_g1_bd_34My_coupledmcmc_fixed.xml &

cd $SCRATCH/strictgenes_g1.bd.34My/run16
$HOME/tools/BEAST_v2.6.7/beast/bin/beast -seed 2596 -threads 10 ../strictgenes_g1_bd_34My_coupledmcmc_fixed.xml &

wait


```
Submit the jobs.
```bash
sbatch strictgenes_g1_coupledmcmc_fixed_13to16round1.sh
sbatch strictgenes_g1_coupledmcmc_fixed_1to4round1.sh
sbatch strictgenes_g1_coupledmcmc_fixed_5to8round1.sh
sbatch strictgenes_g1_coupledmcmc_fixed_9to12round1.sh
```

Assess run
```bash
cd $SCRATCH/strictgenes_g1.bd.34My

#save a copy in case anything goes wrong
mkdir backup_round1
cp -r ./* backup_round1

#before we can restart, need to ensure that all logs ended at the same state
for replicate in run1 run2 run3 run4 run5 run6 run7 run8 run9 run10 run11 run12 run13 run14 run15 run16 ; do echo "$replicate" ; tail -n 1 "$replicate"/*trees | grep STATE | sed 's/= .*$//g' | grep "5000 " ; done
#for those whose tree states end in 5000, they will be one state ahead of the gene tree logs and this will make them crash. Remove the last state from the species tree log and main log. This happens because I set the species tree and main log to log twice as frequently as the gene tree logs.
for replicate in run1 run2 run5 run8 run9 run10 run11 run12  ; do mv "$replicate"/species.trees "$replicate"/species.trees.old ; head -n -1 "$replicate"/species.trees.old > "$replicate"/species.trees ; mv "$replicate"/starbeast3.log "$replicate"/starbeast3.log.old ; head -n -1 "$replicate"/starbeast3.log.old > "$replicate"/starbeast3.log ; done

#prepare batch files for round 2

#modify name of output in the bash command, and add the resume flag to the beast command
sed 's/round1/round2/g; s/-threads 10/-threads 10 -resume/g' strictgenes_g1_coupledmcmc_fixed_13to16round1.sh > strictgenes_g1_coupledmcmc_fixed_13to16round2.sh
sed 's/round1/round2/g; s/-threads 10/-threads 10 -resume/g' strictgenes_g1_coupledmcmc_fixed_1to4round1.sh > strictgenes_g1_coupledmcmc_fixed_1to4round2.sh
sed 's/round1/round2/g; s/-threads 10/-threads 10 -resume/g' strictgenes_g1_coupledmcmc_fixed_5to8round1.sh > strictgenes_g1_coupledmcmc_fixed_5to8round2.sh
sed 's/round1/round2/g; s/-threads 10/-threads 10 -resume/g' strictgenes_g1_coupledmcmc_fixed_9to12round1.sh > strictgenes_g1_coupledmcmc_fixed_9to12round2.sh


```
Then run round 2
```bash
sbatch strictgenes_g1_coupledmcmc_fixed_13to16round2.sh
sbatch strictgenes_g1_coupledmcmc_fixed_1to4round2.sh
sbatch strictgenes_g1_coupledmcmc_fixed_5to8round2.sh
sbatch strictgenes_g1_coupledmcmc_fixed_9to12round2.sh
````

Two iterations failed because their gene trees were not all the same state! We can fix that by removing the entries that ended one state ahead of the others.

```bash
#remove all generations past 25830000 from the log and species tree files for run13
ls run13/chr*.trees | while read locus ; do grep -v "^tree STATE_25840000" "$locus" > "$locus"temp ; done
ls run13/chr*.trees | while read locus ; do mv -f "$locus"temp "$locus" ; done
grep -v "^tree STATE_25840000" run13/species.trees > run13/species.treestemp ; mv -f run13/species.treestemp run13/species.trees
grep -v "^25840000\t" run13/starbeast3.log > run13/starbeast3.logtemp ; mv -f run13/starbeast3.logtemp run13/starbeast3.log

#remove all generations past STATE_26450000 from the log and species tree files for run13
ls run15/chr*.trees | while read locus ; do grep -v "^tree STATE_26460000" "$locus" > "$locus"temp ; done
ls run15/chr*.trees | while read locus ; do mv -f "$locus"temp "$locus" ; done
grep -v "^tree STATE_26460000" run15/species.trees > run15/species.treestemp ; mv -f run15/species.treestemp run15/species.trees
grep -v "^26460000\t" run15/starbeast3.log > run15/starbeast3.logtemp ; mv -f run15/starbeast3.logtemp run15/starbeast3.log

#remove last line of species trees and logs that end on a multiple of 5000
for replicate in run13 run14 run15 run16 ; do echo "$replicate" ; tail -n 1 "$replicate"/*trees | grep STATE | sed 's/= .*$//g' | grep "5000 " ; done
for replicate in run13 run15 ; do mv "$replicate"/species.trees "$replicate"/species.trees.old ; head -n -1 "$replicate"/species.trees.old > "$replicate"/species.trees ; mv "$replicate"/starbeast3.log "$replicate"/starbeast3.log.old ; head -n -1 "$replicate"/starbeast3.log.old > "$replicate"/starbeast3.log ; done
for replicate in run13 run15 ; do mv "$replicate"/starbeast3.log "$replicate"/starbeast3.log.old ; head -n -1 "$replicate"/starbeast3.log.old > "$replicate"/starbeast3.log ; done

```

```bash
cat > strictgenes_g1_coupledmcmc_fixed_1315catchupround2.sh
chmod +x strictgenes_g1_coupledmcmc_fixed_1315catchupround2.sh
```

```bash
#!/bin/bash
#SBATCH --nodes=1 #
#SBATCH --ntasks-per-node=4
#SBATCH --cpus-per-task=10
#SBATCH --mem=0 #set to 0 to use all available memory of each requested node. Advatage, any of the ntasks of each node can use any amount of the memory provided to sum does not exceed thata available to the node
#SBATCH --time=23:59:00
#SBATCH --job-name 1315catchupround2
#SBATCH --output=strictgenes_g1_coupledmcmc_fixed_1315catchupround2.txt 
module load CCEnv
module load StdEnv
module load java
module load beagle-lib
export LD_LIBRARY_PATH=/home/j/jweir/elsemikk/tools/beagle-lib/lib:$LD_LIBRARY_PATH

cd $SCRATCH/strictgenes_g1.bd.34My

#I used random.org to generate 8 random numbers for the seed

cd $SCRATCH/strictgenes_g1.bd.34My/run13
$HOME/tools/BEAST_v2.6.7/beast/bin/beast -seed 3183 -threads 10 -resume ../strictgenes_g1_bd_34My_coupledmcmc_fixed.xml &

cd $SCRATCH/strictgenes_g1.bd.34My/run15
$HOME/tools/BEAST_v2.6.7/beast/bin/beast -seed 6598 -threads 10 -resume ../strictgenes_g1_bd_34My_coupledmcmc_fixed.xml &

wait

```
```bash
sbatch strictgenes_g1_coupledmcmc_fixed_1315catchupround2.sh
```

## run 3

Assess run
```bash
cd $SCRATCH/strictgenes_g1.bd.34My

#save a copy in case anything goes wrong
mkdir backup_round2
mv backup_round1 backup_round2
cp -r ./* backup_round2

#before we can restart, need to ensure that all logs ended at the same state
for replicate in run1 run2 run3 run4 run5 run6 run7 run8 run9 run10 run11 run12 run13 run14 run15 run16 ; do echo "$replicate" ; tail -n 1 "$replicate"/*trees | grep STATE | sed 's/= .*$//g' | grep "5000 " ; done
#for those whose tree states end in 5000, they will be one state ahead of the gene tree logs and this will make them crash. Remove the last state from the species tree log and main log. This happens because I set the species tree and main log to log twice as frequently as the gene tree logs.
for replicate in run1 run3 run4 run5 ; do mv "$replicate"/species.trees "$replicate"/species.trees.old ; head -n -1 "$replicate"/species.trees.old > "$replicate"/species.trees ; mv "$replicate"/starbeast3.log "$replicate"/starbeast3.log.old ; head -n -1 "$replicate"/starbeast3.log.old > "$replicate"/starbeast3.log ; done

#prepare batch files for round 2

#modify name of output in the bash command, and add the resume flag to the beast command
sed 's/round2/round3/g' strictgenes_g1_coupledmcmc_fixed_13to16round2.sh > strictgenes_g1_coupledmcmc_fixed_13to16round3.sh
sed 's/round2/round3/g' strictgenes_g1_coupledmcmc_fixed_1to4round2.sh > strictgenes_g1_coupledmcmc_fixed_1to4round3.sh
sed 's/round2/round3/g' strictgenes_g1_coupledmcmc_fixed_5to8round2.sh > strictgenes_g1_coupledmcmc_fixed_5to8round3.sh
sed 's/round2/round3/g' strictgenes_g1_coupledmcmc_fixed_9to12round2.sh > strictgenes_g1_coupledmcmc_fixed_9to12round3.sh


```
Then run round 3
```bash
sbatch strictgenes_g1_coupledmcmc_fixed_13to16round3.sh
sbatch strictgenes_g1_coupledmcmc_fixed_1to4round3.sh
sbatch strictgenes_g1_coupledmcmc_fixed_5to8round3.sh
sbatch strictgenes_g1_coupledmcmc_fixed_9to12round3.sh
````

Periodically check what iteration each one has reached, and get a preview of the species trees

```bash
cd $SCRATCH/strictgenes_g1.bd.34My
tail -n 1 run*/species.trees

time $HOME/tools/BEAST_v2.6.7/beast/bin/logcombiner -log run1/species.trees -log run2/species.trees -log run3/species.trees -log run4/species.trees -log run5/species.trees -log run6/species.trees -log run7/species.trees -log run8/species.trees -log run9/species.trees -log run10/species.trees -log run11/species.trees -log run12/species.trees -log run13/species.trees -log run14/species.trees -log run15/species.trees -log run16/species.trees -o prelim.trees -b 75 #4 seconds

time $HOME/tools/BEAST_v2.6.7/beast/bin/treeannotator -heights mean -burnin 0 prelim.trees prelim.tree #1 second


```

## run 4
Repeat for the next run.

Assess run
```bash
cd $SCRATCH/strictgenes_g1.bd.34My

#save a copy in case anything goes wrong
mkdir backup_round3
mv backup_round2 backup_round3
cp -r ./* backup_round3

#before we can restart, need to ensure that all logs ended at the same state
for replicate in run1 run2 run3 run4 run5 run6 run7 run8 run9 run10 run11 run12 run13 run14 run15 run16 ; do echo "$replicate" ; tail -n 1 "$replicate"/*trees | grep STATE | sed 's/= .*$//g' | grep "5000 " ; done
#for those whose tree states end in 5000, they will be one state ahead of the gene tree logs and this will make them crash. Remove the last state from the species tree log and main log. This happens because I set the species tree and main log to log twice as frequently as the gene tree logs.
for replicate in run2 run3 run4 run7 run8 run12 run14 run15 run16 ; do mv "$replicate"/species.trees "$replicate"/species.trees.old ; head -n -1 "$replicate"/species.trees.old > "$replicate"/species.trees ; mv "$replicate"/starbeast3.log "$replicate"/starbeast3.log.old ; head -n -1 "$replicate"/starbeast3.log.old > "$replicate"/starbeast3.log ; done

#prepare batch files for round 2

#modify name of output in the bash command, and add the resume flag to the beast command
sed 's/round2/round4/g' strictgenes_g1_coupledmcmc_fixed_13to16round2.sh > strictgenes_g1_coupledmcmc_fixed_13to16round4.sh
sed 's/round2/round4/g' strictgenes_g1_coupledmcmc_fixed_1to4round2.sh > strictgenes_g1_coupledmcmc_fixed_1to4round4.sh
sed 's/round2/round4/g' strictgenes_g1_coupledmcmc_fixed_5to8round2.sh > strictgenes_g1_coupledmcmc_fixed_5to8round4.sh
sed 's/round2/round4/g' strictgenes_g1_coupledmcmc_fixed_9to12round2.sh > strictgenes_g1_coupledmcmc_fixed_9to12round4.sh

```
Then run round 4
```bash
sbatch strictgenes_g1_coupledmcmc_fixed_13to16round4.sh
sbatch strictgenes_g1_coupledmcmc_fixed_1to4round4.sh
sbatch strictgenes_g1_coupledmcmc_fixed_5to8round4.sh
sbatch strictgenes_g1_coupledmcmc_fixed_9to12round4.sh
````

Now, all have reached 100 million states except run 13 and run 15. Run those.

```bash
cd $SCRATCH/strictgenes_g1.bd.34My
#save a copy in case anything goes wrong
mkdir backup_round4
mv backup_round3 backup_round4
cp -r ./* backup_round4

#before we can restart, need to ensure that all logs ended at the same state
for replicate in run13 run15 ; do echo "$replicate" ; tail -n 1 "$replicate"/*trees | grep STATE | sed 's/= .*$//g' | grep "5000 " ; done
#for those whose tree states end in 5000, they will be one state ahead of the gene tree logs and this will make them crash. Remove the last state from the species tree log and main log. This happens because I set the species tree and main log to log twice as frequently as the gene tree logs.
for replicate in run15 ; do mv "$replicate"/species.trees "$replicate"/species.trees.old ; head -n -1 "$replicate"/species.trees.old > "$replicate"/species.trees ; mv "$replicate"/starbeast3.log "$replicate"/starbeast3.log.old ; head -n -1 "$replicate"/starbeast3.log.old > "$replicate"/starbeast3.log ; done

```
```bash
cat > strictgenes_g1_coupledmcmc_fixed_1315catchupround4.sh
chmod +x strictgenes_g1_coupledmcmc_fixed_1315catchupround4.sh
```

```bash
#!/bin/bash
#SBATCH --nodes=1 #
#SBATCH --ntasks-per-node=4
#SBATCH --cpus-per-task=10
#SBATCH --mem=0 #set to 0 to use all available memory of each requested node. Advatage, any of the ntasks of each node can use any amount of the memory provided to sum does not exceed thata available to the node
#SBATCH --time=23:59:00
#SBATCH --job-name 1315catchupround4
#SBATCH --output=strictgenes_g1_coupledmcmc_fixed_1315catchupround4.txt 
module load CCEnv
module load StdEnv
module load java
module load beagle-lib
export LD_LIBRARY_PATH=/home/j/jweir/elsemikk/tools/beagle-lib/lib:$LD_LIBRARY_PATH

cd $SCRATCH/strictgenes_g1.bd.34My

#I used random.org to generate 8 random numbers for the seed

cd $SCRATCH/strictgenes_g1.bd.34My/run13
$HOME/tools/BEAST_v2.6.7/beast/bin/beast -seed 3183 -threads 10 -resume ../strictgenes_g1_bd_34My_coupledmcmc_fixed.xml &

cd $SCRATCH/strictgenes_g1.bd.34My/run15
$HOME/tools/BEAST_v2.6.7/beast/bin/beast -seed 6598 -threads 10 -resume ../strictgenes_g1_bd_34My_coupledmcmc_fixed.xml &

wait

```
```bash
sbatch strictgenes_g1_coupledmcmc_fixed_1315catchupround4.sh
```

After completing 4 24-hour runs per replicate (16 days total wall time used on 40-core sever), I verified that each replicate reached at least 100,000,000 samples.
```bash
tail -n 1 ru*/s*trees
```
```
==> run10/species.trees <==
tree STATE_104780000 = ((((((1[&pop=0.3532001490259313]:0.26136877825349436,4[&pop=0.5192123602082321]:0.26136877825349436)[&pop=0.283033816119272]:0.023032870770469283,3[&pop=0.719985843205117]:0.28440164902396364)[&pop=1.81793734681259]:0.03570900213340755,2[&pop=0.8298629424475829]:0.3201106511573712)[&pop=0.5740763090020932]:0.69920791310684,6[&pop=11.971565445033447]:1.0193185642642113)[&pop=0.8926800452912187]:9.759790010095898,(5[&pop=3.2418752414119623]:7.859295726658283,8[&pop=1.23251344847277]:7.859295726658283)[&pop=0.8065610416580828]:2.9198128477018255)[&pop=0.9714463015343389]:24.13564613752292,7[&pop=4.269972682837855]:34.91475471188303)[&pop=5.290219014971771];

==> run11/species.trees <==
tree STATE_103860000 = ((((((1[&pop=0.4069097960328669]:0.29754446623907765,4[&pop=0.6648369366564002]:0.29754446623907765)[&pop=0.9142114560138931]:2.7856355083089035E-4,2[&pop=0.4701568706662765]:0.29782302978990854)[&pop=0.6327685387448222]:0.03535770192088944,3[&pop=0.73726323744419]:0.333180731710798)[&pop=0.5648864532533637]:0.6783031081746593,6[&pop=13.837911109483352]:1.0114838398854573)[&pop=0.7891257408826051]:9.853408681142497,(5[&pop=3.071495274481424]:8.071498729220645,8[&pop=1.2790129292278158]:8.071498729220645)[&pop=0.7496225200458873]:2.7933937918073095)[&pop=0.8215547222571552]:23.520701883109254,7[&pop=4.518217204095341]:34.38559440413721)[&pop=5.31611763285048];

==> run12/species.trees <==
tree STATE_104295000 = ((((((1[&pop=0.310033897258154]:0.2500498802163217,4[&pop=0.6829425134293622]:0.2500498802163217)[&pop=0.8963475833084701]:0.0598956124963117,3[&pop=0.7897874194338815]:0.3099454927126334)[&pop=1.0269225399029762]:0.018418503962237986,2[&pop=0.9933686692678123]:0.3283639966748714)[&pop=0.5887535396335222]:0.6461815481881026,6[&pop=16.768066892777604]:0.974545544862974)[&pop=0.9002274460703471]:9.5100098267883,(5[&pop=3.423301429180739]:7.443883320886296,8[&pop=1.197762715459647]:7.443883320886296)[&pop=0.9181237608216452]:3.040672050764978)[&pop=0.8579432029132322]:24.01549537100403,7[&pop=3.6528506933243623]:34.500050742655304)[&pop=4.542088127970627];

==> run13/species.trees <==
tree STATE_126515000 = ((((((1[&pop=0.37231603163251237]:0.26334566039905544,4[&pop=0.5179836079315788]:0.26334566039905544)[&pop=0.38058603744131686]:0.02771922373495206,2[&pop=0.9456681934056514]:0.2910648841340075)[&pop=2.981693785787923]:0.0436432108900226,3[&pop=0.8945788125902358]:0.3347080950240301)[&pop=0.4831049280572622]:0.6194870757305501,6[&pop=15.420870003233006]:0.9541951707545802)[&pop=0.8102869909259038]:9.74602131004471,(5[&pop=3.3982445349312043]:7.623981580272548,8[&pop=1.1396449264099573]:7.623981580272548)[&pop=0.663362432681699]:3.0762349005267433)[&pop=0.6631846345582485]:23.974955354894796,7[&pop=3.5551018754104584]:34.67517183569409)[&pop=4.810711224439938];

==> run14/species.trees <==
tree STATE_125820000 = (((((1[&pop=0.49856155327614704]:0.3134352132728361,4[&pop=0.6329630973773889]:0.3134352132728361)[&pop=0.6291249544213691]:0.05506404635385759,(2[&pop=1.1031083391068122]:0.3084251220558565,3[&pop=0.9148530765213956]:0.3084251220558565)[&pop=1.1692427557457914]:0.06007413757083718)[&pop=0.5594807699718608]:0.7339391017066708,6[&pop=15.750205497732614]:1.1024383613333644)[&pop=0.9938716895106909]:10.5185055613362,(5[&pop=3.6096543215626964]:8.152956986031205,8[&pop=1.1777481974551185]:8.152956986031205)[&pop=1.382873276376757]:3.4679869366383596)[&pop=0.9409939017063896]:23.582858945064363,7[&pop=3.7411810053235155]:35.20380286773393)[&pop=8.057951795374217];

==> run15/species.trees <==
tree STATE_125455000 = ((((((1[&pop=0.3347421077027999]:0.23703711852253542,4[&pop=0.5319744750656777]:0.23703711852253542)[&pop=0.31530613588488043]:0.041245601976194896,2[&pop=0.6899091946102313]:0.2782827204987303)[&pop=1.1688637567363582]:0.026003966852101934,3[&pop=0.798909831897651]:0.30428668735083225)[&pop=0.5668331100906356]:0.6796615618057811,6[&pop=14.271647288016208]:0.9839482491566133)[&pop=0.8145829187550029]:9.728366700202207,(5[&pop=2.370825132127475]:7.613741963311046,8[&pop=1.0108924518610276]:7.613741963311046)[&pop=0.9844945168305064]:3.098572986047774)[&pop=0.6064917243909735]:24.1428963084091,7[&pop=4.215870342838838]:34.85521125776792)[&pop=4.9567307352351495];

==> run16/species.trees <==
tree STATE_123540000 = ((((((1[&pop=0.36139048783380345]:0.2806056974552705,4[&pop=0.490528262649116]:0.2806056974552705)[&pop=0.48420321961110824]:0.031781510856105644,2[&pop=0.6077829450586371]:0.3123872083113761)[&pop=0.7466627748880821]:0.01885434863701846,3[&pop=0.9347899565018198]:0.3312415569483946)[&pop=0.5833479112840299]:0.7090781737898657,6[&pop=15.579076021513954]:1.0403197307382603)[&pop=0.9466725106384102]:10.285614176351917,(5[&pop=3.2604691422456233]:8.023551035630568,8[&pop=1.177751218048283]:8.023551035630568)[&pop=0.8127710777163948]:3.302382871459608)[&pop=0.7180582434371664]:22.910076058127004,7[&pop=4.0524704001917655]:34.23600996521718)[&pop=6.719484842167371];

==> run1/species.trees <==
tree STATE_103225000 = (((((1[&pop=0.31462258702759166]:0.2738419043365852,4[&pop=0.6304726804314444]:0.2738419043365852)[&pop=0.5535680250935934]:0.03629705326271243,(2[&pop=0.6720635970263727]:0.30691497205126517,3[&pop=0.9209617405374595]:0.30691497205126517)[&pop=9.605987236572286]:0.0032239855480324375)[&pop=0.552646194264606]:0.6861118072072021,6[&pop=17.206807500103505]:0.9962507648064998)[&pop=0.9894392177573744]:9.869285772755461,(5[&pop=2.889454027730153]:8.09518096747722,8[&pop=1.029587856508307]:8.09518096747722)[&pop=0.9736798870506382]:2.7703555700847424)[&pop=0.8489783274556365]:23.668156203689126,7[&pop=3.1687580991661326]:34.53369274125109)[&pop=6.488480297544756];

==> run2/species.trees <==
tree STATE_102885000 = ((((((1[&pop=0.35011598474788513]:0.24995690143041616,4[&pop=0.46939615994388845]:0.24995690143041616)[&pop=1.2071021154636403]:0.05179164124017571,2[&pop=0.5473573200645818]:0.30174854267059187)[&pop=1.5786021917828201]:0.040385022191504194,3[&pop=0.8919829947134811]:0.34213356486209606)[&pop=0.5181674226722879]:0.6141422800797525,6[&pop=17.742256539991597]:0.9562758449418485)[&pop=0.8691486859345854]:9.356064807669169,(5[&pop=3.328638587346271]:6.914037590027794,8[&pop=1.150608099970481]:6.914037590027794)[&pop=1.2781450527741225]:3.3983030625832225)[&pop=0.7673335518367093]:24.054444739660948,7[&pop=3.567878667412566]:34.366785392271964)[&pop=5.3892478073466785];

==> run3/species.trees <==
tree STATE_103005000 = ((((((1[&pop=0.4403512790400701]:0.27467449377281616,4[&pop=0.6064683111828225]:0.27467449377281616)[&pop=0.3722950444687408]:0.06184489493051326,2[&pop=0.8217870893671497]:0.3365193887033294)[&pop=1.6354773217703147]:0.012526956358496544,3[&pop=0.7859909687440384]:0.34904634506182597)[&pop=0.5717096040839091]:0.7398578099063713,6[&pop=19.452085095998573]:1.0889041549681973)[&pop=0.8968995636897313]:10.238047644977467,(5[&pop=3.065572025445604]:7.960656350685438,8[&pop=1.1994717275910727]:7.960656350685438)[&pop=0.9620586808402266]:3.3662954492602264)[&pop=0.612656360077344]:23.755347467014193,7[&pop=4.280676526539892]:35.08229926695986)[&pop=5.498140622291325];

==> run4/species.trees <==
tree STATE_103735000 = ((((((1[&pop=0.3733127601691411]:0.2675471490999099,4[&pop=0.516599690189208]:0.2675471490999099)[&pop=0.3811434310016098]:0.02807135487861928,2[&pop=0.9330794074853814]:0.2956185039785292)[&pop=4.378315591306692]:0.010974901758651456,3[&pop=0.7474231010604718]:0.30659340573718064)[&pop=0.6251686706659069]:0.7111867945278378,6[&pop=19.09290639668598]:1.0177802002650185)[&pop=1.0378958954840811]:9.86228202273099,(5[&pop=3.1674296975789313]:7.6926576301547716,8[&pop=1.2561945280954556]:7.6926576301547716)[&pop=1.2068468490272382]:3.1874045928412382)[&pop=0.8230286035873128]:23.73780812283394,7[&pop=4.250232320253856]:34.61787034582995)[&pop=7.700476879718503];

==> run5/species.trees <==
tree STATE_103705000 = ((((((1[&pop=0.44119584874635914]:0.2863390759474359,4[&pop=0.6366739326068657]:0.2863390759474359)[&pop=0.35663065964349755]:0.04695329469914339,2[&pop=0.6801223028794063]:0.3332923706465793)[&pop=1.7900287018287302]:0.03673871053478339,3[&pop=0.8406118993094769]:0.3700310811813627)[&pop=0.5934015252314977]:0.6705877912092011,6[&pop=13.840901907672787]:1.0406188723905638)[&pop=0.9244728496191575]:9.721350813729346,(5[&pop=3.8181469515170368]:7.752592759439397,8[&pop=1.2400602867717219]:7.752592759439397)[&pop=1.0705117878009016]:3.0093769266805133)[&pop=0.9303514638393603]:23.67380704787427,7[&pop=4.408608215261155]:34.43577673399418)[&pop=6.417004790953915];

==> run6/species.trees <==
tree STATE_104490000 = (((((1[&pop=0.40481383010398336]:0.25667380840036924,4[&pop=0.671638333934732]:0.25667380840036924)[&pop=0.5249084930354865]:0.07748109097388,(2[&pop=0.717980478729725]:0.2944725900069212,3[&pop=0.6658243210800634]:0.2944725900069212)[&pop=2.187873303604808]:0.03968230936732803)[&pop=0.6114094344189663]:0.7153004216977187,6[&pop=16.375298013817424]:1.049455321071968)[&pop=0.8490286133112395]:9.778347386803969,(5[&pop=3.4644357514008988]:7.901909064992349,8[&pop=1.2500179590752745]:7.901909064992349)[&pop=0.9887025159558361]:2.9258936428835867)[&pop=0.8193772132493359]:23.615660974627435,7[&pop=4.052243443922719]:34.44346368250337)[&pop=5.800219101434322];

==> run7/species.trees <==
tree STATE_103585000 = ((((((1[&pop=0.37474223307324994]:0.26340810393155967,4[&pop=0.6141509861598964]:0.26340810393155967)[&pop=0.8351893105697904]:0.05153873190768671,2[&pop=0.5480910550118748]:0.3149468358392464)[&pop=0.6505551813667889]:0.02741811632646657,3[&pop=1.0680066747167671]:0.34236495216571294)[&pop=0.551357898796667]:0.7160230869807233,6[&pop=10.392671368253374]:1.0583880391464362)[&pop=0.8912737437773234]:10.109034001540063,(5[&pop=2.6614934316145655]:8.19873341036757,8[&pop=1.140273324617473]:8.19873341036757)[&pop=0.7040554434796008]:2.9686886303189297)[&pop=0.7881651924111538]:23.002510885161062,7[&pop=4.342179044883317]:34.16993292584756)[&pop=5.1496773414983075];

==> run8/species.trees <==
tree STATE_102950000 = ((((((1[&pop=0.3577665012999911]:0.27318047753066926,4[&pop=0.6136693898204263]:0.27318047753066926)[&pop=0.26396711100612097]:0.020074258692982516,3[&pop=0.7008278677728105]:0.2932547362236518)[&pop=0.7138970516017029]:0.018576504109788117,2[&pop=0.6600342811680374]:0.3118312403334399)[&pop=0.59939679142182]:0.6400178656656809,6[&pop=12.45043554593465]:0.9518491059991208)[&pop=0.868429791998642]:9.377717969313139,(5[&pop=2.5681551309689254]:7.590185898692825,8[&pop=1.0836428255357407]:7.590185898692825)[&pop=0.7768258331017445]:2.7393811766194354)[&pop=0.8263870946962648]:24.77033158244516,7[&pop=3.936351961578473]:35.09989865775742)[&pop=4.523613409605559];

==> run9/species.trees <==
tree STATE_102885000 = ((((((1[&pop=0.3469286076604514]:0.25722820578375977,4[&pop=0.49398079783376003]:0.25722820578375977)[&pop=0.6058634051492866]:0.04335969222292746,2[&pop=0.5768314948425499]:0.3005878980066872)[&pop=1.1251219722528918]:0.007429449474111172,3[&pop=0.6416356998558753]:0.3080173474807984)[&pop=0.5740751405139193]:0.6838450468792945,6[&pop=7.134590482664593]:0.9918623943600928)[&pop=0.8657202801181841]:9.510045739520764,(5[&pop=3.4593812769023313]:7.840603314271665,8[&pop=1.0887947285993353]:7.840603314271665)[&pop=0.9916892056091111]:2.6613048196091915)[&pop=0.8467581472182731]:24.066832259146405,7[&pop=3.3755491716760964]:34.56874039302726)[&pop=5.402731111318417];
```
All 16 runs have reached more than 100,000,000 generations.

# Combine runs
Now we are ready to combine the replicates.

Firstly, I will trim all runs to 100,000,000 generations. Since I had to restart the runs many times (due to the 24 hour time limit on Niagara), each run ran for slightly different numbers of generations. This does not really matter, but to make reporting of the methods more clean, I am going to chop off any excess generations past 100,000,000 so that all runs are of uniform length.
```bash
#remove all generations past 100000000 from the log and species tree files
for replicate in run1 run2 run3 run4 run5 run6 run7 run8 run9 run10 run11 run12 run13 run15 run14 run16 ; do echo "$replicate" ; sed -n '/^100005000\t/q;p' "$replicate"/starbeast3.log > "$replicate"/starbeast3.100MILL.log ; done
for replicate in run1 run2 run3 run4 run5 run6 run7 run8 run9 run10 run11 run12 run13 run15 run14 run16 ; do echo "$replicate" ; sed -n '/^tree STATE_100005000/q;p' "$replicate"/species.trees > "$replicate"/species.100MILL.trees ; done


```

Now, I am opening the `starbeast3.100MILL.log` files in Tracer to examine the runs. Most parameters in all runs reached ESS>200, but posterior, speciescoalescent, and tree.t:Species.treeLength all have ESS generally below 100. To reach ESS>200 for the parameters of interest, we can combine the results of our independant runs, now that we verified that they converged onto the same area of parameter space (if any had converged onto a different local optimum, we would not be able to combine them)

Most runs appear to reach stationary after about 10,000,000 samples. run8 had a big spike in the posterior at around state 18,000,000 that brings the ESS far down. run14 also has a moderate spike around 19,000,000. Removing the first 20,000,000 samples (20%) as burn-in will remove these problematic spikes, although this is more than the burnin necessary for the other runs.

To combine BEAST runs, we can use the program `logcombiner` that comes in the BEAST package. We will specify a burnin of 20,000,000 generations, as this looked appropriate for all of the runs when examining the traces in Tracer (after 15,000,000 generations, the runs appear to stabilize, making the trace look like a "fuzzy" caterpillar with no discernible trajectory).

```bash
#combine all runs, log files
/Applications/BEAST_2.6.6/bin/logcombiner -log run1/starbeast3.100MILL.log -log run2/starbeast3.100MILL.log -log run3/starbeast3.100MILL.log -log run4/starbeast3.100MILL.log -log run5/starbeast3.100MILL.log -log run6/starbeast3.100MILL.log -log run7/starbeast3.100MILL.log -log run8/starbeast3.100MILL.log -log run9/starbeast3.100MILL.log -log run10/starbeast3.100MILL.log -log run11/starbeast3.100MILL.log -log run12/starbeast3.100MILL.log -log run13/starbeast3.100MILL.log -log run14/starbeast3.100MILL.log -log run15/starbeast3.100MILL.log -log run16/starbeast3.100MILL.log -o 16runs.100MILL.log -b 20  #note the path is to version 2.6.6, but the program is updated to 2.6.7

#repeat for tree files
time /Applications/BEAST_2.6.6/bin/logcombiner -log run1/species.100MILL.trees -log run2/species.100MILL.trees -log run3/species.100MILL.trees -log run4/species.100MILL.trees -log run5/species.100MILL.trees -log run6/species.100MILL.trees -log run7/species.100MILL.trees -log run8/species.100MILL.trees -log run9/species.100MILL.trees -log run10/species.100MILL.trees -log run11/species.100MILL.trees -log run12/species.100MILL.trees -log run13/species.100MILL.trees -log run14/species.100MILL.trees -log run15/species.100MILL.trees -log run16/species.100MILL.trees -o species.16runs.100MILL.trees -b 20 #8 seconds #note the path is to version 2.6.6, but the program is updated to 2.6.7

#generate a maximum clade credibility tree from the posterior distribution of species trees
time /Applications/BEAST_2.6.7/bin/treeannotator -heights mean -burnin 0 species.16runs.100MILL.trees species.16runs.100MILL.tree #1 second

```
Since we sampled every 5000 states for 100,000,000 generations, each run has 20,001 samples (the extra "1" is for the starting tree). I am discarding the first 20%.

Unfortunately, the log file is too large for me to open in tracer, as tracer loads it into memory and the file is over 6 Gb. To get around this, I will make a downsampled file to look at (but we are not downsampling the data when producing the trees, this is just to look at to ensure the runs converged and ESS scores are high enough to stop). I will downsample to keep every 50,000 generations instead of every 5000 generations.
```bash
time /Applications/BEAST_2.6.6/bin/logcombiner -log run1/starbeast3.100MILL.log -log run2/starbeast3.100MILL.log -log run3/starbeast3.100MILL.log -log run4/starbeast3.100MILL.log -log run5/starbeast3.100MILL.log -log run6/starbeast3.100MILL.log -log run7/starbeast3.100MILL.log -log run8/starbeast3.100MILL.log -log run9/starbeast3.100MILL.log -log run10/starbeast3.100MILL.log -log run11/starbeast3.100MILL.log -log run12/starbeast3.100MILL.log -log run13/starbeast3.100MILL.log -log run14/starbeast3.100MILL.log -log run15/starbeast3.100MILL.log -log run16/starbeast3.100MILL.log -o 16runs.100MILL.50k.log -b 20 -resample 50000 #note the path is to version 2.6.6, but the program is updated to 2.6.7
```

 Posterior has ESS=11949, likelihood ESS=16,898 and prior ESS=8407. The TreeDistanceNJ.t and TreeDistanceUPGMA.t parameters for several genes did not converge, but those are not parameters of interest.

# Plot result
 Here is the tree generated from combining the 16 runs (`species.16runs.100MILL.tree`)
```
#NEXUS

Begin taxa;
	Dimensions ntax=8;
		Taxlabels
			alcid
			antarcticus
			chilensis
			longicaudus
			maccormicki
			parasiticus
			pomarinus
			skua
			;
End;
Begin trees;
	Translate
		   1 alcid,
		   2 antarcticus,
		   3 chilensis,
		   4 longicaudus,
		   5 maccormicki,
		   6 parasiticus,
		   7 pomarinus,
		   8 skua
;
tree TREE1 = (1[&height=2.2864446831496598E-15,height_95%_HPD={0.0,7.105427357601002E-15},height_median=0.0,height_range={0.0,1.4210854715202004E-14},length=34.698252257166956,length_95%_HPD={34.1514920666459,35.23923788224488},length_median=34.697729939106836,length_range={33.37983378979391,36.08607778654373},pop=3.973381653621933,pop_95%_HPD={3.181966796693219,4.799049460923579},pop_median=3.9466152058402004,pop_range={2.5989456347682585,6.315063088648485}]:34.69825225716696,(((((2[&height=2.300821172787953E-15,height_95%_HPD={0.0,7.105427357601002E-15},height_median=0.0,height_range={0.0,2.1316282072803006E-14},length=0.26557388152325245,length_95%_HPD={0.22225875806632445,0.3096026986789653},length_median=0.2649357172546729,length_range={0.16137652939899993,0.5947010542072988},pop=0.5725599949065479,pop_95%_HPD={0.45396037876147133,0.696107840651964},pop_median=0.5683234433150859,pop_range={0.34142995624101347,0.9736463279795826}]:0.2646388893125972,8[&height=2.2990726808049178E-15,height_95%_HPD={0.0,7.105427357601002E-15},height_median=0.0,height_range={0.0,2.1316282072803006E-14},length=0.2657925469450577,length_95%_HPD={0.22164931040793334,0.3106891276522745},length_median=0.2651341246069663,length_range={0.16137652939899993,0.4312405561577748},pop=0.39302803754791,pop_95%_HPD={0.3032154952365416,0.490221557899821},pop_median=0.38943918875538613,pop_range={0.22387123591211547,0.7583002822992195}]:0.2646388893125972)[&height=0.2646388893125995,height_95%_HPD={0.22297637648418345,0.3086266664172328},height_median=0.2643310595439772,height_range={0.16137652939899993,0.3885458349990216},length=0.046449103141739466,length_95%_HPD={0.006418350350557489,0.08842873685742347},length_median=0.04439183042327244,length_range={1.1154638173138665E-6,0.1725327180456162},pop=0.5691179701418098,pop_95%_HPD={0.19588479743513637,1.003824197503789},pop_median=0.5127868982895775,pop_range={0.09164758439806066,145.35367744225027},posterior=0.953116211486782]:0.03793391659042994,3[&height=2.2924950204877833E-15,height_95%_HPD={0.0,7.105427357601002E-15},height_median=0.0,height_range={0.0,2.1316282072803006E-14},length=0.3125074078589988,length_95%_HPD={0.25716262611204854,0.3665649396068176},length_median=0.3123900726674691,length_range={0.19086020182925978,0.4513497403814313},pop=0.7699161142804138,pop_95%_HPD={0.39561097442370524,1.2469747469735184},pop_median=0.7238918318070429,pop_range={0.2645985035341224,4.762344938562417}]:0.30257280590302715)[&height=0.30257280590302943,height_95%_HPD={0.2577082755454825,0.34783075996486446},height_median=0.3026004271292848,height_range={0.21211409687472127,0.41101486306469326},length=0.029517571112082044,length_95%_HPD={1.0957994049931585E-6,0.06820565956798674},length_median=0.026207011080735754,length_range={1.0957994049931585E-6,0.12835544413044175},pop=1.4328438532805676,pop_95%_HPD={0.2057078325777838,3.5853426185966555},pop_median=0.9917032884188539,pop_range={0.1232451202121663,196.91308579890415},posterior=0.5169950315605275]:0.02968898862712327,5[&height=2.290219205525737E-15,height_95%_HPD={0.0,7.105427357601002E-15},height_median=0.0,height_range={0.0,1.4210854715202004E-14},length=0.3216835292875983,length_95%_HPD={0.2752932732168034,0.36799223284115357},length_median=0.3218589153793445,length_range={0.1960460143629632,0.43502718232591775},pop=0.8528614227726962,pop_95%_HPD={0.6758224138719313,1.046136207244444},pop_median=0.8460367353105295,pop_range={0.5345426479477215,1.4352618254122913}]:0.3322617945301504)[&height=0.3322617945301527,height_95%_HPD={0.29173030646306586,0.3724928651431796},height_median=0.3312767749216512,height_range={0.25494512669796165,0.5947010542072988},length=0.660848150053334,length_95%_HPD={0.5732501975831994,0.7524557520149742},length_median=0.6598149850589188,length_range={0.04916595094203302,0.8904064984162119},pop=0.5565737839132807,pop_95%_HPD={0.476248832846616,0.6418169132509037},pop_median=0.5554883766153793,pop_range={0.0869651345777227,0.768866260661158},posterior=1.0]:0.6608481500533427,7[&height=2.2957699737258502E-15,height_95%_HPD={0.0,7.105427357601002E-15},height_median=0.0,height_range={0.0,1.4210854715202004E-14},length=0.9931099445834932,length_95%_HPD={0.898485470999276,1.092834389264496},length_median=0.9910286517990237,length_range={0.5316268452976516,1.2563086734601754},pop=15.906991120362818,pop_95%_HPD={8.971767892028993,24.66470290165131},pop_median=15.092952628902964,pop_range={6.290820562723268,261.94288323086755}]:0.9931099445834931)[&height=0.9931099445834954,height_95%_HPD={0.898485470999276,1.092834389264496},height_median=0.9910286517990272,height_range={0.5316268452976516,1.2563086734601754},length=9.72344872427138,length_95%_HPD={9.059476107429298,10.403882230507946},length_median=9.711658015187831,length_range={8.384754927251095,11.534370075099147},pop=0.9021761063662361,pop_95%_HPD={0.7903072702110471,1.0131226674538907},pop_median=0.9000685001884012,pop_range={0.6419278385940629,1.4948431982713604},posterior=1.0]:9.723448724271037,(4[&height=2.287027513810672E-15,height_95%_HPD={0.0,7.105427357601002E-15},height_median=0.0,height_range={0.0,2.1316282072803006E-14},length=7.744069732083711,length_95%_HPD={7.008614334955414,8.501790518987029},length_median=7.7361669871536645,length_range={6.066025750218277,9.69300410202819},pop=3.1149491229607706,pop_95%_HPD={2.4132222462300716,3.8847950623361185},pop_median=3.0838303971071213,pop_range={1.9443226639444553,5.490239159188758}]:7.744069732083711,6[&height=2.287027513810672E-15,height_95%_HPD={0.0,7.105427357601002E-15},height_median=0.0,height_range={0.0,2.1316282072803006E-14},length=7.744069732083711,length_95%_HPD={7.008614334955414,8.501790518987029},length_median=7.7361669871536645,length_range={6.066025750218277,9.69300410202819},pop=1.1544927160867138,pop_95%_HPD={0.9988073962728304,1.315008088606562},pop_median=1.1510244530198372,pop_range={0.8472024835374767,1.5857350944050994}]:7.744069732083711)[&height=7.744069732083713,height_95%_HPD={7.008614334955421,8.501790518987029},height_median=7.736166987153668,height_range={6.066025750218277,9.69300410202819},length=2.97248893677097,length_95%_HPD={2.2866135556481986,3.661289284382331},length_median=2.9694676076061235,length_range={1.506711921182898,4.766882311148482},pop=0.9091598799188038,pop_95%_HPD={0.5777368516003213,1.26697023297603},pop_median=0.8953961483505141,pop_range={0.31135221356002896,2.119135369265493},posterior=1.0]:2.9724889367708185)[&height=10.716558668854532,height_95%_HPD={10.007303612623517,11.451960864277957},height_median=10.703549395464211,height_range={9.31898348496459,12.631609713025817},length=23.981693588313416,length_95%_HPD={23.183366988832095,24.782854103575346},length_median=23.990707113226556,length_range={21.989991702628416,25.659009886826297},pop=0.8474350634730043,pop_95%_HPD={0.560554378473931,1.1621678511875737},pop_median=0.8373436163015655,pop_range={0.3082858904609658,1.7292226371239832},posterior=1.0]:23.98169358831243)[&height=34.69825225716696,height_95%_HPD={34.15149206664591,35.23923788224488},height_median=34.697729939106836,height_range={33.37983378979391,36.08607778654374},length=0.0,pop=5.406111462828704,pop_95%_HPD={3.614281466565643,7.322542953207705},pop_median=5.328086053120273,pop_range={2.2179200953022358,11.863482666255852},posterior=1.0]:0.0;
End;
```

I am opening it in FigTree v1.4.4 to look at it.
* Node labels -> height
* Node bars -> check -> `height_95%_HDP`
Then, I rotated branches to match the order that I want to display the taxa in the figure (minimize the number of crossing lines when I plot the mtDNA tree on top of it)
* Tip labels -> pop (need to keep track of taxa names if changing to pop size)
File -> save -> `strictgenes_g1_bd_coupledmcmc_fixed.16runs.100MILL.nodebars.tre`
File -> "Export PDF" -> `strictgenes_g1_bd_coupledmcmc_fixed.16runs.100MILL.nodebars.pdf`

Then, I am opening the file in Illustrator to edit the aesthetics. I am adding circles whose area is proportional to the estimated population sizes `width_in_cm = sqrt(popsize/pi)`