### Overview
This pipeline describes making a phylogenetic network with Phylonet.

### Required Input
This step requires a dataset of sequence alignments, generated in step 1.6.1. These are phased haplotype sequences. The set of genetrees for these alignments (input for Phylonet) is stored in `ml_best.phased.midpointed.trees`.

### Output
The output is a phylogenetic network. The best network is in the file `ml_best.phased.midpointed.partitioned.3.rep3.x100.results`

When gene flow between lineages is common, it can complicate inference of the species tree, and can in fact lead to incorrect inferences of the species tree when not taken into account (even under the multispecies coalescent model, such as in Astral). Gene flow can be taken into account during tree inference, but it is computationally expensive with current programs. One popular program is PhyloNet, which we will use here. It can take gene trees as input and generate a species tree with a given number of reticulations. I will not be able to use all 227,000 gene trees in my dataset due to computational limitations. Instead, this is my plan: run phylonet on the smaller dataset of phased haplotypes, allowing 0, 1, 2, or 3 reticulation events.

Some things to keep in mind:
"Calling the reticulation event hybridization and drawing the network as in Fig. 16(a)
or calling the event hybrid speciation and drawing the network as in Fig. 16(b) is not
a distinction made by PhyloNet nor is it an aspect that is inferable from data. Such a
distinction can be made after the network is inferred, and using some knowledge that is
â€œexternal" to the inferred network topology."

There is a Phylonet tutorial [here](https://wiki.rice.edu/confluence/pages/viewpage.action?pageId=39500205) and a very useful article [here](https://www.biorxiv.org/content/biorxiv/early/2019/08/24/746362.full.pdf).


# Phylonet
## Prepare gene trees

First, we need to make maximum likelihood gene trees for each alignment. I have a list of alignments that passed my previous filters in step 1.6.1c.

I am removing the Alcidae samples to speed computation, because Phylonet is quite computationally intensive and including the Alcids would slow computation without adding insight (I am not interested in looking for introgression between Alcidae and Stercorariidae, which would be biologically implausible to have happenned in any recent time)
```bash
cd /home/1_Else/0_GENOMES5/Stercorarius/2_Phylogeny/2.8c_phylonet

#get alignments
#make directory to hold alignments
mkdir -p genes
mkdir -p genes_jaegerout
#obtain my list of alignments that passed quality control
cp /home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.6.1c_phylogenetic_blocks/missingness/passed.txt .
#for each alignment, remove the Alcids.
cat passed.txt | cut -f 1 | parallel 'cat /home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.6.1c_phylogenetic_blocks/aligned/{1}.fasta | /home/0_PROGRAMS/seqkit grep -v -p "Uria" -p "Fratercula" --use-regexp > ./genes/{1}.fasta' #I kept a dataset with Alca
cat passed.txt | cut -f 1 | parallel 'cat genes/{1}.fasta | /home/0_PROGRAMS/seqkit grep -v -p "Alca" --use-regexp > ./genes_jaegerout/{1}.fasta' #a dataset without Alca

#make ML gene trees for each alignment
cat passed.txt | cut -f 1 | parallel '/home/0_PROGRAMS/iqtree-2.1.3-Linux/bin/iqtree2 -s ./genes_jaegerout/{1}.fasta'
#put the ML gene trees together into a single file.
cat passed.txt | cut -f 1 | while read gene ; do cat ./genes_jaegerout/"$gene".fasta.treefile >> genes_jaegerout/ml_best.phased.trees ; done

```
The trees need to be rooted. A quick way is to use midpoint rooting, which is not always the best option, but in this case we already have a pretty good idea of the relationships at the root of the tree, and midpoint rooting works well - it put the root between (*S. parasiticus*,*S. longicaudus*) and the *S. pomarinus*+*Catharacta* clade, which is almost certainly the correct rooting given results of all other analyses. I mannually verified the rooting of the first 100 trees, and only 1/100 placed the root differently (with *S. parasiticus* sister to *S. pomarinus*+*Catharacta*), which would not be unexpected for low levels of ILS or gene tree error and is consistent with expected levels of discordance based on the Astral-III quartet scores.

```R
#load ape to read the tree
library(ape)
#load the phangorn library for midpoint rooting
library(phangorn)

#load trees from the ML tree set produced above
treeset <- ape::read.tree("genes_jaegerout/ml_best.phased.trees")
#midpoint root the trees
treeset <- midpoint(treeset)
#save the results
write.tree(treeset, file = "genes_jaegerout/ml_best.phased.midpointed.trees", append = FALSE, digits = 10, tree.names = FALSE) #

```

Now we have a dataset of 9899 gene trees for all the Stercorariids. Gene trees are from the phased dataset generated for Starbeast3.

### Run phylonet
Now, we can run phylonet on this dataset. First, we prepare a nexus file that contains the phylonet command, then we give this nexus file to phylonet.
```bash
cd /home/1_Else/0_GENOMES5/Stercorarius/2_Phylogeny/2.8c_phylonet

#make a nexus file
#write nexus header
printf "#NEXUS\n\nBEGIN TREES;\n\n" > genes_jaegerout/ml_best.phased.midpointed.nex
#add gene trees to the nexus file
printf '%s\n' {1..9899} | sed 's/^/Tree gt/g; s/$/ = /g' > temp
cat genes_jaegerout/ml_best.phased.midpointed.trees | paste -d "" temp - >> genes_jaegerout/ml_best.phased.midpointed.nex
#add footer and phylonet command to the nexus file.
printf "\nEND;\n\nBEGIN PHYLONET;\n\nInferNetwork_MPL (all) 3 -pl 24;\n\nEND;" >> genes_jaegerout/ml_best.phased.midpointed.nex

#add partitioning to specify which samples belong to which species. This is our template file which we can edit as needed
sed 's/^InferNetwork_MPL.*$/InferNetwork_MPL (all) 2 -a <maccormicki:ANSK01.1,ANSK01.2,ANSK7.1,ANSK7.2,ANSK8.1,ANSK8.2;chilensis:CHSK_MKP2451.1,CHSK_MKP2451.2;antarcticus:CISK2.1,CISK2.2,CISK_3.1,CISK_3.2,CISK55.1,CISK55.2;skua:GRSK_MKP1592.1,GRSK_MKP1592.2,GRSK_MKP1593.1,GRSK_MKP1593.2;longicaudus:LTJA_MKP990.1,LTJA_MKP990.2;parasiticus:PAJA_B20730.1,PAJA_B20730.2,PAJA_USNM606730.1,PAJA_USNM606730.2;pomarinus:POJA_4.1,POJA_4.2,POJA_IB2659.1,POJA_IB2659.2,POJA_MKP1559.1,POJA_MKP1559.2> -pl 24;/g' genes_jaegerout/ml_best.phased.midpointed.nex > genes_jaegerout/ml_best.phased.midpointed.partitioned.2.nex

#Note: I initially tried running with the -po option, but it ran for days, never finishing, using 120 cores, then crashed. It seems to be too computationally heavy for this dataset.

#Now we generate a nex file for each of 0-4 reticulation edges, and add 100 runs. make copies of the nexus file for 0-4 reticulation edges. then run phylonet.

#0 reticulation edges
sed 's/^InferNetwork_MPL.*$/InferNetwork_MPL (all) 0 -x 100 -a <maccormicki:ANSK01.1,ANSK01.2,ANSK7.1,ANSK7.2,ANSK8.1,ANSK8.2;chilensis:CHSK_MKP2451.1,CHSK_MKP2451.2;antarcticus:CISK2.1,CISK2.2,CISK_3.1,CISK_3.2,CISK55.1,CISK55.2;skua:GRSK_MKP1592.1,GRSK_MKP1592.2,GRSK_MKP1593.1,GRSK_MKP1593.2;longicaudus:LTJA_MKP990.1,LTJA_MKP990.2;parasiticus:PAJA_B20730.1,PAJA_B20730.2,PAJA_USNM606730.1,PAJA_USNM606730.2;pomarinus:POJA_4.1,POJA_4.2,POJA_IB2659.1,POJA_IB2659.2,POJA_MKP1559.1,POJA_MKP1559.2> -pl 24;/g' genes_jaegerout/ml_best.phased.midpointed.partitioned.2.nex > genes_jaegerout/ml_best.phased.midpointed.partitioned.0.x100.nex
java -jar /home/0_PROGRAMS/PhyloNet_3.8.2.jar genes_jaegerout/ml_best.phased.midpointed.partitioned.0.x100.nex 24 > genes_jaegerout/ml_best.phased.midpointed.partitioned.0.x100.results

#1 reticulation edge
sed 's/^InferNetwork_MPL.*$/InferNetwork_MPL (all) 1 -x 100 -a <maccormicki:ANSK01.1,ANSK01.2,ANSK7.1,ANSK7.2,ANSK8.1,ANSK8.2;chilensis:CHSK_MKP2451.1,CHSK_MKP2451.2;antarcticus:CISK2.1,CISK2.2,CISK_3.1,CISK_3.2,CISK55.1,CISK55.2;skua:GRSK_MKP1592.1,GRSK_MKP1592.2,GRSK_MKP1593.1,GRSK_MKP1593.2;longicaudus:LTJA_MKP990.1,LTJA_MKP990.2;parasiticus:PAJA_B20730.1,PAJA_B20730.2,PAJA_USNM606730.1,PAJA_USNM606730.2;pomarinus:POJA_4.1,POJA_4.2,POJA_IB2659.1,POJA_IB2659.2,POJA_MKP1559.1,POJA_MKP1559.2> -pl 24;/g' genes_jaegerout/ml_best.phased.midpointed.partitioned.2.nex > genes_jaegerout/ml_best.phased.midpointed.partitioned.1.x100.nex
time java -jar /home/0_PROGRAMS/PhyloNet_3.8.2.jar genes_jaegerout/ml_best.phased.midpointed.partitioned.1.x100.nex 24 > genes_jaegerout/ml_best.phased.midpointed.partitioned.1.x100.results #447m22.072s (400m42.085s)

#2 reticulation edges
sed 's/^InferNetwork_MPL.*$/InferNetwork_MPL (all) 2 -x 100 -a <maccormicki:ANSK01.1,ANSK01.2,ANSK7.1,ANSK7.2,ANSK8.1,ANSK8.2;chilensis:CHSK_MKP2451.1,CHSK_MKP2451.2;antarcticus:CISK2.1,CISK2.2,CISK_3.1,CISK_3.2,CISK55.1,CISK55.2;skua:GRSK_MKP1592.1,GRSK_MKP1592.2,GRSK_MKP1593.1,GRSK_MKP1593.2;longicaudus:LTJA_MKP990.1,LTJA_MKP990.2;parasiticus:PAJA_B20730.1,PAJA_B20730.2,PAJA_USNM606730.1,PAJA_USNM606730.2;pomarinus:POJA_4.1,POJA_4.2,POJA_IB2659.1,POJA_IB2659.2,POJA_MKP1559.1,POJA_MKP1559.2> -pl 24;/g' genes_jaegerout/ml_best.phased.midpointed.partitioned.2.nex > genes_jaegerout/ml_best.phased.midpointed.partitioned.2.x100.nex
java -jar /home/0_PROGRAMS/PhyloNet_3.8.2.jar genes_jaegerout/ml_best.phased.midpointed.partitioned.2.x100.nex 24 > genes_jaegerout/ml_best.phased.midpointed.partitioned.2.x100.results

#3 reticulation edges
sed 's/^InferNetwork_MPL.*$/InferNetwork_MPL (all) 3 -x 100 -a <maccormicki:ANSK01.1,ANSK01.2,ANSK7.1,ANSK7.2,ANSK8.1,ANSK8.2;chilensis:CHSK_MKP2451.1,CHSK_MKP2451.2;antarcticus:CISK2.1,CISK2.2,CISK_3.1,CISK_3.2,CISK55.1,CISK55.2;skua:GRSK_MKP1592.1,GRSK_MKP1592.2,GRSK_MKP1593.1,GRSK_MKP1593.2;longicaudus:LTJA_MKP990.1,LTJA_MKP990.2;parasiticus:PAJA_B20730.1,PAJA_B20730.2,PAJA_USNM606730.1,PAJA_USNM606730.2;pomarinus:POJA_4.1,POJA_4.2,POJA_IB2659.1,POJA_IB2659.2,POJA_MKP1559.1,POJA_MKP1559.2> -pl 24;/g' genes_jaegerout/ml_best.phased.midpointed.partitioned.2.nex > genes_jaegerout/ml_best.phased.midpointed.partitioned.3.x100.nex
time java -jar /home/0_PROGRAMS/PhyloNet_3.8.2.jar genes_jaegerout/ml_best.phased.midpointed.partitioned.3.x100.nex 24 > genes_jaegerout/ml_best.phased.midpointed.partitioned.3.x100.results #952m59.849s (912m2.528s)

#4 reticulation edges
sed 's/^InferNetwork_MPL.*$/InferNetwork_MPL (all) 4 -x 100 -a <maccormicki:ANSK01.1,ANSK01.2,ANSK7.1,ANSK7.2,ANSK8.1,ANSK8.2;chilensis:CHSK_MKP2451.1,CHSK_MKP2451.2;antarcticus:CISK2.1,CISK2.2,CISK_3.1,CISK_3.2,CISK55.1,CISK55.2;skua:GRSK_MKP1592.1,GRSK_MKP1592.2,GRSK_MKP1593.1,GRSK_MKP1593.2;longicaudus:LTJA_MKP990.1,LTJA_MKP990.2;parasiticus:PAJA_B20730.1,PAJA_B20730.2,PAJA_USNM606730.1,PAJA_USNM606730.2;pomarinus:POJA_4.1,POJA_4.2,POJA_IB2659.1,POJA_IB2659.2,POJA_MKP1559.1,POJA_MKP1559.2> -pl 24;/g' genes_jaegerout/ml_best.phased.midpointed.partitioned.2.nex > genes_jaegerout/ml_best.phased.midpointed.partitioned.4.x100.nex
time java -jar /home/0_PROGRAMS/PhyloNet_3.8.2.jar genes_jaegerout/ml_best.phased.midpointed.partitioned.4.x100.nex 24 > genes_jaegerout/ml_best.phased.midpointed.partitioned.4.x100.results #1148m45.621s (1292m47.183s)

#replicate once more, in case we got stuck at a local maximum
java -jar /home/0_PROGRAMS/PhyloNet_3.8.2.jar genes_jaegerout/ml_best.phased.midpointed.partitioned.0.x100.nex 24 > genes_jaegerout/ml_best.phased.midpointed.partitioned.0.rep2.x100.results
java -jar /home/0_PROGRAMS/PhyloNet_3.8.2.jar genes_jaegerout/ml_best.phased.midpointed.partitioned.1.x100.nex 24 > genes_jaegerout/ml_best.phased.midpointed.partitioned.1.rep2.x100.results
java -jar /home/0_PROGRAMS/PhyloNet_3.8.2.jar genes_jaegerout/ml_best.phased.midpointed.partitioned.2.x100.nex 24 > genes_jaegerout/ml_best.phased.midpointed.partitioned.2.rep2.x100.results
java -jar /home/0_PROGRAMS/PhyloNet_3.8.2.jar genes_jaegerout/ml_best.phased.midpointed.partitioned.3.x100.nex 24 > genes_jaegerout/ml_best.phased.midpointed.partitioned.3.rep2.x100.results
time java -jar /home/0_PROGRAMS/PhyloNet_3.8.2.jar genes_jaegerout/ml_best.phased.midpointed.partitioned.4.x100.nex 24 > genes_jaegerout/ml_best.phased.midpointed.partitioned.4.rep2.x100.results #1278m27.073s

#replicate once more, in case we got stuck at a local maximum
time java -jar /home/0_PROGRAMS/PhyloNet_3.8.2.jar genes_jaegerout/ml_best.phased.midpointed.partitioned.0.x100.nex 24 > genes_jaegerout/ml_best.phased.midpointed.partitioned.0.rep3.x100.results #143m28.617s
time java -jar /home/0_PROGRAMS/PhyloNet_3.8.2.jar genes_jaegerout/ml_best.phased.midpointed.partitioned.1.x100.nex 24 > genes_jaegerout/ml_best.phased.midpointed.partitioned.1.rep3.x100.results #531m14.790s
time java -jar /home/0_PROGRAMS/PhyloNet_3.8.2.jar genes_jaegerout/ml_best.phased.midpointed.partitioned.2.x100.nex 24 > genes_jaegerout/ml_best.phased.midpointed.partitioned.2.rep3.x100.results #824m22.815s
java -jar /home/0_PROGRAMS/PhyloNet_3.8.2.jar genes_jaegerout/ml_best.phased.midpointed.partitioned.3.x100.nex 24 > genes_jaegerout/ml_best.phased.midpointed.partitioned.3.rep3.x100.results
time java -jar /home/0_PROGRAMS/PhyloNet_3.8.2.jar genes_jaegerout/ml_best.phased.midpointed.partitioned.4.x100.nex 24 > genes_jaegerout/ml_best.phased.midpointed.partitioned.4.rep3.x100.results #1149m53.899s


#that was not the best way to code it, for the future, I should have done this:
#parallel 'time java -jar /home/0_PROGRAMS/PhyloNet_3.8.2.jar genes_jaegerout/ml_best.phased.midpointed.partitioned.4.x100.nex 4 > genes_jaegerout/ml_best.phased.midpointed.partitioned.4.rep{1}.x100.results' ::: 1 2 3
#parallel 'time java -jar /home/0_PROGRAMS/PhyloNet_3.8.2.jar genes_jaegerout/ml_best.phased.midpointed.partitioned.3.x100.nex 4 > genes_jaegerout/ml_best.phased.midpointed.partitioned.3.rep{1}.x100.results' ::: 1 2 3
#parallel 'time java -jar /home/0_PROGRAMS/PhyloNet_3.8.2.jar genes_jaegerout/ml_best.phased.midpointed.partitioned.2.x100.nex 4 > genes_jaegerout/ml_best.phased.midpointed.partitioned.2.rep{1}.x100.results' ::: 1 2 3
#parallel 'time java -jar /home/0_PROGRAMS/PhyloNet_3.8.2.jar genes_jaegerout/ml_best.phased.midpointed.partitioned.1.x100.nex 4 > genes_jaegerout/ml_best.phased.midpointed.partitioned.1.rep{1}.x100.results' ::: 1 2 3
#parallel 'time java -jar /home/0_PROGRAMS/PhyloNet_3.8.2.jar genes_jaegerout/ml_best.phased.midpointed.partitioned.0.x100.nex 4 > genes_jaegerout/ml_best.phased.midpointed.partitioned.0.rep{1}.x100.results' ::: 1 2 3

#read the results 
cd genes_jaegerout
grep -A 2 "Inferred Network #1" ml_best.phased.midpointed.partitioned.*.x100.results | grep -v "Inferred Network #1" | sed 's/^.*Total/Total/g; s/results-/results\n/g'

```
Phylonet command:
* `InferNetwork_MPL`: "Infers a phylogenetic network from gene trees under maximum pseudo-likelihood."
* `(all)`: use all gene trees in the nexus file, instead of specifying which ones
* `3`: number of reticulations to allow
* `-pl 4`: number of threads to use for computation
* `-x 100`: "The number of runs of the search. Default value is 10." in 'The evolutionary history of bears is characterized by gene flow across species' they used 50.

Results:
* With no reticulations, it consistently places chilensis with antarcticus, and matches the expected topology from Astral.
* With one reticulation, it is harder to interpret. It puts antarcticus as the first split within "Catharacta" (the same reticulation in both runs, but they differ in which "Catharacta" are sister - chilensis and maccormicki, or skua and maccormicki).
* With two reticulations, antarcticus is a hybrid of chilensis, maccormicki, and skua

Here are all the networks. Note the best network is `ml_best.phased.midpointed.partitioned.3.rep3.x100.results`
```
ml_best.phased.midpointed.partitioned.0.rep2.x100.results
((parasiticus:1.0,longicaudus:1.0):2.27122442701069,((skua:1.0,(maccormicki:1.0,(antarcticus:1.0,chilensis:1.0):0.03572405846225172):0.04511442089577701):0.6628957033803604,pomarinus:1.0):5.906693481561525);
Total log probability: -9979171.257233238
--
ml_best.phased.midpointed.partitioned.0.rep3.x100.results
((pomarinus:1.0,(skua:1.0,(maccormicki:1.0,(antarcticus:1.0,chilensis:1.0):0.03544609480919304):0.0467108011859693):0.6589226152042542):5.906696453886863,(parasiticus:1.0,longicaudus:1.0):2.2712244270108695);
Total log probability: -9979174.134682542
--
ml_best.phased.midpointed.partitioned.0.x100.results
((parasiticus:1.0,longicaudus:1.0):2.2712244270106896,(pomarinus:1.0,(((antarcticus:1.0,chilensis:1.0):0.03554379690846529,maccormicki:1.0):0.046716517158801305,skua:1.0):0.6588706429656966):5.906733089739576);
Total log probability: -9979173.731270386
--
ml_best.phased.midpointed.partitioned.1.rep2.x100.results
((((antarcticus:1.0,((chilensis:1.0,(skua:1.0,maccormicki:1.0):0.0011774181844964955):0.0011774181844964955)#H1:0.8077014951050601::0.14975867068476262):1.4879247682521373,#H1:0.0011774181844964955::0.8502413293152373):0.6065575059796623,pomarinus:1.0):5.906694690486041,(longicaudus:1.0,parasiticus:1.0):2.2712244270108695);
Total log probability: -9964564.479226695
--
ml_best.phased.midpointed.partitioned.1.rep3.x100.results
((longicaudus:1.0,parasiticus:1.0):2.2712244270108695,(((antarcticus:1.0,((skua:1.0,(maccormicki:1.0,chilensis:1.0):0.025531858417379726):0.0011774181844964955)#H1:1.476556449488378::0.11693078742231044):5.926587139417571,#H1:0.0011774181844964955::0.8830692125776896):0.60584478338471,pomarinus:1.0):5.906694357492815);
Total log probability: -9963625.338256733
--
ml_best.phased.midpointed.partitioned.1.x100.results
((parasiticus:1.0,longicaudus:1.0):2.271224427010869,(((antarcticus:1.0,(((chilensis:1.0,maccormicki:1.0):0.026753622120033432,skua:1.0):0.022318907079184457)#H1:0.01599827063490876::0.38915497822751066):0.9136790927872083,#H1:0.0690196245003215::0.6108450217724893):0.46693329357564384,pomarinus:1.0):5.906692341704081);
Total log probability: -9963590.007483179
--
ml_best.phased.midpointed.partitioned.2.rep2.x100.results
((parasiticus:1.0,longicaudus:1.0):2.2712244270108695,(((((((skua:1.0)#H2:1.0::0.30492682443119795,(chilensis:1.0)#H1:1.0::0.2732813159701633):0.0011774181844964955,antarcticus:1.0):0.33385261102646757,maccormicki:1.0):0.09249806586859832,#H1:1.0::0.7267186840298367):0.05461594535637179,#H2:1.0::0.695073175568802):0.583713828000823,pomarinus:1.0):5.9066984733732735);
Total log probability: -9963121.436344834
--
ml_best.phased.midpointed.partitioned.2.rep3.x100.results
((((((((maccormicki:1.0)#H1:1.0::0.47575127362530956,(skua:1.0)#H2:1.0::0.7740326812752832):0.03977542565894311,antarcticus:1.0):0.13343961029583284,chilensis:1.0):0.14705291713490998,#H1:1.0::0.5242487263746904):0.443071042263456,#H2:1.0::0.22596731872471676):0.18015748981961205,pomarinus:1.0):5.906706583032203,(parasiticus:1.0,longicaudus:1.0):2.271224427010513);
Total log probability: -9962745.538725965
--
ml_best.phased.midpointed.partitioned.2.x100.results
((((skua:1.0,((antarcticus:1.0)#H2:1.0::0.6559302381632088)#H1:1.0::0.8722990697697526):0.16148406824201222,((maccormicki:1.0,#H1:1.0::0.12770093023024748):5.939321169815863,(chilensis:1.0,#H2:1.0::0.3440697618367912):0.32925103936690286):0.03459657821510344):0.6235733975111527,pomarinus:1.0):5.906693752145476,(parasiticus:1.0,longicaudus:1.0):2.271224427010867);
Total log probability: -9963246.096916482
--
ml_best.phased.midpointed.partitioned.3.rep2.x100.results
((longicaudus:1.0,parasiticus:1.0):2.2712244270108695,(((((((((maccormicki:1.0)#H3:1.0::0.25789450165413524,(skua:1.0)#H1:1.0::0.3753128280028648):0.08616365875586751,antarcticus:1.0):0.289014186872938,chilensis:1.0):0.1077876795531794,#H3:1.0::0.7421054983458648):0.018362294625332787)#H2:0.0011774181844964955::0.3207308112988215,#H1:1.0::0.6246871719971352):1.2325194809434066,#H2:0.6336756001649044::0.6792691887011785):0.28446211725721576,pomarinus:1.0):5.906692197275008);
Total log probability: -9962740.292335892
--
ml_best.phased.midpointed.partitioned.3.rep3.x100.results
((((((((antarcticus:1.0,(skua:1.0)#H2:1.0::0.6560297706637683):0.0011774181844964955,(chilensis:1.0)#H1:1.0::0.5432468985029395):0.14960458231800056,maccormicki:1.0):0.1540191212400932,#H1:1.0::0.4567531014970605):0.17249492098347122,#H2:1.0::0.34397022933623167):0.2648670430336287,(pomarinus:1.0)#H3:1.0::0.8619363521391367):4.348225129705804,#H3:1.0::0.13806364786086334):5.906794351572711,(parasiticus:1.0,longicaudus:1.0):2.2712244270108695);
Total log probability: -9949127.653184531
--
ml_best.phased.midpointed.partitioned.3.x100.results
(((((((antarcticus:1.0,(skua:1.0)#H2:1.0::0.8423839912398144):0.0011774181844964955,(maccormicki:1.0)#H1:1.0::0.5660663504994714):0.10707631476538301,chilensis:1.0):0.15797467195699488,#H1:1.0::0.4339336495005286):0.5810648339573792,#H2:1.0::0.15761600876018556):0.028235886741119154,pomarinus:1.0):5.906693759997153,(longicaudus:1.0,parasiticus:1.0):2.271224427010869);
Total log probability: -9963071.373117227
--
ml_best.phased.midpointed.partitioned.4.rep2.x100.results
((longicaudus:1.0,parasiticus:1.0):2.2712244270108695,((((maccormicki:1.0,(((skua:1.0)#H2:1.0::0.7797878147482133,antarcticus:1.0):0.11454888223988964)#H1:0.0011774181844964955::0.3011071643186969):0.2528025044602495,(chilensis:1.0,#H1:0.0011774181844964955::0.6988928356813031):0.11340032346284744):0.45940573730435486,#H2:1.0::0.22021218525178665):0.21894209099360173,pomarinus:1.0):5.906767367016588);
Total log probability: -9962671.020711029
--
ml_best.phased.midpointed.partitioned.4.rep3.x100.results
(((pomarinus:1.0,((skua:1.0)#H2:1.0::0.5684039201324551)#H1:1.0::0.036695232193858285):0.5333794583785624,(((((antarcticus:1.0,(#H2:1.0::0.4315960798675449)#H4:1.0::0.8119693390664352):0.18804296398126877)#H3:0.24573512381355547::0.6319599669323652,chilensis:1.0):0.1296003291791889,((maccormicki:1.0,#H3:0.07826544971177253::0.36804003306763483):0.18788282243861143,#H1:1.0::0.9633047678061417):0.00475277085178716):0.65406213518083,#H4:1.0::0.1880306609335648):0.027931584387564424):5.906692566431537,(longicaudus:1.0,parasiticus:1.0):2.2712244270108695);
Total log probability: -9962224.30027509
--
ml_best.phased.midpointed.partitioned.4.x100.results
(((((((((((maccormicki:1.0)#H3:1.0::0.17810160543232134,antarcticus:1.0):0.0011774181844964955,((skua:1.0)#H2:1.0::0.9629042589375438)#H4:1.0::0.21166121408483424):0.040207455027638846,#H2:1.0::0.03709574106245617):0.41785299312769875,chilensis:1.0):0.09827839303494451,#H3:1.0::0.8218983945676787):0.05266312378505531,#H4:1.0::0.7883387859151658):0.716285971824363)#H1:0.0011774181844964955::0.8433426663198563,pomarinus:1.0):5.912439595260167,#H1:0.0011774181844964955::0.15665733368014367):5.903223928679281,(parasiticus:1.0,longicaudus:1.0):2.271224427010867);
Total log probability: -9950612.70727069
```

# Visualize
We can visualize the results in IcyTree - go to [icytree](https://icytree.org/) and press `e` to copy paste the network. I find the Icytree layout much harder to look at than Dendroscope though.
To visualize in Dendroscope, Dendroscope has to be installed. Then, we have to remove the inheritance probabilities (i.e., remove the numbers that follow the double colons `::`)  
Here is our best network formatted for Dendroscope:  
`((((((((antarcticus:1.0,(skua:1.0)#H2:1.0):0.0011774181844964955,(chilensis:1.0)#H1:1.0):0.14960458231800056,maccormicki:1.0):0.1540191212400932,#H1:1.0):0.17249492098347122,#H2:1.0):0.2648670430336287,(pomarinus:1.0)#H3:1.0):4.348225129705804,#H3:1.0):5.906794351572711,(parasiticus:1.0,longicaudus:1.0):2.2712244270108695);`

To see branch lengths, go to Layout -> "Draw rectangular phylogram"

I saved an image of the best network as `ml_best.phased.midpointed.partitioned.3.rep3.x100.results.svg`, as well as a png.

The second best network (ml_best.phased.midpointed.partitioned.4.x100.results) has k=4 and shows the same reticulations, but places two reticulations between *S. skua* and *S. antarcticus*, instead of 1: `(((((((((((maccormicki:1.0)#H3:1.000017810160543232134,antarcticus:1.0):0.0011774181844964955,((skua:1.0)#H2:1.00009629042589375438)#H4:1.000021166121408483424):0.040207455027638846,#H2:1.000003709574106245617):0.41785299312769875,chilensis:1.0):0.09827839303494451,#H3:1.00008218983945676787):0.05266312378505531,#H4:1.00007883387859151658):0.716285971824363)#H1:0.00117741818449649550008433426663198563,pomarinus:1.0):5.912439595260167,#H1:0.001177418184496495500015665733368014367):5.903223928679281,(parasiticus:1.0,longicaudus:1.0):2.271224427010867);
`

# Select the best network

To choose the best network, we can look for the networks with the highest (least negative) log probability. However, more complicated models have the potential to overfit the model, and receive a higher log probability due to having more parameters that can fit the data more precisely, but they may not be the best model to choose. How do we choose whether adding a parameter was a significantly better fit than a simpler model? There are a few methods, particularly BIC, AIC, and AICc.

AIC is simple. First, you need to count the number of parameters in your model (k), and calculate its likelihood (L). `AIC = 2k - 2*ln(L)`

In phylonet, the number of parameters for AIC is usually set to the number of branches plus number of inheritance probabilities being estimated. A tree with 7 taxa has 12 branches (ntaxa + ntaxa-2). 

We can also go a step further and use bias-corrected AIC, AICc, to account for sample size. In our case, we have a sample of 9899 gene trees, so AICc is almost identical to AIC, but it does not hurt to calculate AICc anyways. To calculate AICc, it is `AICc = 2k*(n/(n-K-1)) - 2*ln(L)`. Notice that `(n/(n-K-1))` converges towards 1 as n increases, and thus AICc converges to AIC as sample sizes get very large.

Let's get the log probabilities to calculate AICc with:
```bash
cd /home/1_Else/0_GENOMES5/Stercorarius/2_Phylogeny/2.8c_phylonet/genes_jaegerout
grep -A 2 "Inferred Network #1" ml_best.phased.midpointed.partitioned.*.x100.results | grep "Total" | sed 's/^.*: //g'

```
```
-9979171.257233238
-9979174.134682542
-9979173.731270386
-9964564.479226695
-9963625.338256733
-9963590.007483179
-9963121.436344834
-9962745.538725965
-9963246.096916482
-9962740.292335892
-9949127.653184531
-9963071.373117227
-9962671.020711029
-9962224.30027509
-9950612.70727069
```
In our case, the network with the greatest log likelihood (-9949127.653) is so much higher than the second best, that the difference in AICc is very large (more than 2000 AIC units). The network with the best log likelihood thus seems to be the best model that we have.

Our best network has *k*=3 reticulations: between *S. antarcticus* and *S. skua*, *S. antarcticus* and *S. chilensis*, and between *S. pomarinus* and the ancestor of all *Catharacta* skuas. Keep in mind that the designation of which branch is the reticulation branch and which branch is the species tree branch is somewhat arbitrary - it could be just as valid to draw the tree such that *S. chilensis* is sister to *S. antarcticus* with introgression from *S. maccormicki*.

