# ABBA BABA Test with Dsuite

### Overview
This section details the pipeline for the ABBA BABA tests of introgession.  

### Required Input
This step requires the VCF file of called genotypes (`Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17.vcf.gz`) that was made in step `1.5d`.

### Output
The output is the results of the ABBA BABA test, as well as *f*d, and *f*dM.


## Pipeline
To test for introgression, I will use an ABBA BABA test. [Dsuite](https://github.com/millanek/Dsuite) is a convenient program that can perform an ABBA BABA test from a VCF file. There is a tutorial [here](https://github.com/mmatschiner/tutorials/tree/master/analysis_of_introgression_with_snp_data).  

It performs an [ABBA-BABA test](https://science.sciencemag.org/content/328/5979/710.full), starting with just VCF files. This is very nice as many other programs require the data to be transformed into custom formats.  
It also provides a jackknife p-value.  

It needs at least 4 species: 2 are sisters (P1 and P2), a third is the one that possible introgressed with them (P3), and a fourth is an outgroup (P4). All the species should be together in a vcf.gz (or vcf) file  

First, we need a sample file that specifies what the outgroup is. This sample file has the sample names in column 1, and the population name in column 2, with the outgroup labelled "Outgroup" (and any samples to exclude would be labelled "xxx").  
```bash
mkdir -p  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/AlcaUria_out
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/AlcaUria_out 
cat > samples_ID.txt
```
```
LTJA_MKP990 longicaudus
PAJA_B20730 parasiticus
PAJA_USNM606730 parasiticus
POJA_MKP1559 pomarinus
POJA_4 pomarinus
POJA_IB2659 pomarinus
GRSK_MKP1592 skua
GRSK_MKP1593 skua
CISK2 antarcticus
CISK55 antarcticus
CISK_3 antarcticus
CHSK_MKP2451 chilensis
ANSK7 maccormicki
ANSK01 maccormicki
ANSK8 maccormicki
Fratercula_arctica xxx
Alca_torda Outgroup
Uria_lomvia Outgroup
```
(Note that there is a *Fratercula* in the file, but it is not used for any analysis in this project).  
```bash
#make tab separated
sed -i 's/ /\t/g' samples_ID.txt
```
To run Dsuite, we need our VCF file. I will use the one that has been filtered for:  

* Base call quality   
* Depth of coverage  
* Mapping quality  
* SNPs only (no indels or invariants)  

I am not filtering for missing data/minor allele count because the ABBA BABA test will only use SNPs that have data at all the 4 pops for each trio analyzed, so it is unnecessary. Beware that filtering for minor allele frequency may introduce bias if the allele frequency spectrum has been shifted due to demographic changes in one of the lineages.  

I am bringing a copy of this VCF file here. Check whether you are happy with the sample names, in some datasets their names are the full path to their bam file which I do not want. Mine match the names in `samples_ID.txt`.  
```bash
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA

ln -s ../../1_Data/1.5d_genotyping/Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17.vcf.gz . #1,621,635 sites

#I also investigated other test datasets to better understand how different filters might lead to different biases, to ensure that interpretation of the results is not being driven by choice of bioinformatic filters. For details of each dataset, see section 1.5d
ln -s ../../1_Data/1.5d_genotyping/Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.stringent.max17.vcf.gz .
ln -s ../../1_Data/1.5d_genotyping/Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.vcf.gz #17,038,613 sites
ln -s ../../1_Data/1.5d_genotyping/Stercorarius.RAZObowtie.merged.filteredSNPs.auto.10stringent.max17.vcf.gz #503,024 sites
ln -s ../../1_Data/1.5d_genotyping/Stercorarius.RAZObowtie.merged.filteredSNPs.auto.30stringent.max17.vcf.gz #2,905,495 sites
ln -s ../../1_Data/1.5d_genotyping/Stercorarius.RAZObowtie.merged.filteredSNPs.auto.40stringent.max17.vcf.gz #4,186,763 sites
ln -s ../../1_Data/1.5d_genotyping/Stercorarius.RAZObowtie.merged.filteredSNPs.auto.50stringent.max17.vcf.gz #5,356,608 sites
ln -s ../../1_Data/1.5d_genotyping/Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.min10.max17.vcf.gz #17,038,613 sites
ln -s ../../1_Data/1.5d_genotyping/Stercorarius.RAZObowtie.merged.filteredSNPs.auto.30stringent.min10.max17.vcf.gz #250,604 sites
ln -s ../../1_Data/1.5d_genotyping/Stercorarius.RAZObowtie.merged.filteredSNPs.auto.40stringent.min10.max17.vcf.gz #701,494 sites
ln -s ../../1_Data/1.5d_genotyping/Stercorarius.RAZObowtie.merged.filteredSNPs.auto.50stringent.min10.max17.vcf.gz #1,459,144 sites

#dataset for chrZ Dinvestigate
ln -s ../../1_Data/1.5d_genotyping/Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.medstringent.max17.vcf.gz .
ln -s ../../1_Data/1.5d_genotyping/Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.medstringent.min3.max17.vcf.gz .

```

## ABBA BABA Test

We will need a phylogeny if we don't want Dsuite to guess it. It does not need branch lengths, and should be in Newick format. Population names should correspond to those in the sample file.  

```bash
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/AlcaUria_out 
echo "(Outgroup,((parasiticus,longicaudus),(pomarinus,(skua,(maccormicki,(antarcticus,chilensis))))));" > skuas.nwk
```

Now that we have our input files, we could do an exploratory ABBA BABA test which will test every posssible trio in the dataset. Warning, if you include chrZ, be aware that female samples might have chrW sequence mapped to it, which could cause strange results.  
```bash
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/AlcaUria_out 

#run ABBA BABA test
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17 #
```
**Timing**: uses one thread, Stercorariidae took 146m51 (without -t, old CISK) or 125m30. Lepidothrix took 117 min the first dataset, 260m55 with a filtered dataset.  

The results are in `samples_ID_WholeGenome_combine.txt`. Each trio is represented on one line; P1 is in column 1, P2 in column 2, and P3 in column 3. Column 4 is the number of BBAA sites, column 5 is BABA and 6 is ABBA.  

The D values are in `samples_ID_WholeGenome_tree.txt`, along with a p-value (from jackknife). There is also `samples_ID_WholeGenome_Dmin.txt`: those are the miniumum D value that could be obtained for a trio, but it is possible that the topology of which species is P3 could be wrong. We will check this and see if it makes sense with our idea of the phylogeny. Another set of D values is given in `samples_ID_WholeGenome_BBAA.txt` which selects P1 and P2 as the pair that share the most derived alleles. (see more explanation in the [tutorial](https://github.com/mmatschiner/tutorials/tree/master/analysis_of_introgression_with_snp_data#q1)). Hopefully, the topology will make sense in most cases, but this is not exactly a robust phylogenetic method (that is not the purpose of ABBA BABA).  

# Jaeger outgroup

Now we will repeat this process, but use the small jaegers as outgroups (P4) for analyses that do not involve them. The use of a closer P4 may increase accuracy by reducing the frequency of convergent mutations.  
```bash
mkdir -p  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/jaeger_out
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/jaeger_out 
cat > samples_ID.txt
```
```
LTJA_MKP990 Outgroup
PAJA_B20730 Outgroup
PAJA_USNM606730 Outgroup
POJA_MKP1559 pomarinus
POJA_4 pomarinus
POJA_IB2659 pomarinus
GRSK_MKP1592 skua
GRSK_MKP1593 skua
CISK2 antarcticus
CISK55 antarcticus
CISK_3 antarcticus
CHSK_MKP2451 chilensis
ANSK7 maccormicki
ANSK01 maccormicki
ANSK8 maccormicki
Fratercula_arctica xxx
Alca_torda xxx
Uria_lomvia xxx
```
```bash
sed -i 's/ /\t/g' samples_ID.txt
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17 #

```
The relationships amongst the *Catharacta* skuas is somewhat uncertain due to their likely history of extensive hybridization. Repeat the ABBA BABA test with another alternative tree topology for the node that is uncertain. I will report D for all three possible topologies at that node.  
```bash
#topology in which antarcticus is sister to maccormicki
mkdir -p  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/jaeger_out/alternate
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/jaeger_out/alternate
echo "(Outgroup,(pomarinus,(skua,(chilensis,(antarcticus,maccormicki)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17.vcf.gz ../samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17 #

#topology in which chilensis is sister to maccormicki
mkdir -p  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/jaeger_out/alternate2
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/jaeger_out/alternate2
echo "(Outgroup,(pomarinus,(skua,(antarcticus,(chilensis,maccormicki)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17.vcf.gz ../samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17 #
```

# Visualize

Now we will plot some heatmaps. This is a script by [mmatschiner](https://github.com/mmatschiner/tutorials/blob/master/analysis_of_introgression_with_snp_data/src/plot_d.rb). We must supply a sample order, in which the sample names exactly match the species/population names given to Dsuite:  
```bash
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/AlcaUria_out 

cat > samples.txt #without outgroup
```
```
longicaudus
parasiticus
pomarinus
skua
maccormicki
antarcticus
chilensis
```
```bash
cut -f 1,2,3,4,6,7 samples_ID_Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17_Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17_tree.txt > plotting_data.txt

ruby /home/0_PROGRAMS/mmatschiner/plot_d.rb plotting_data.txt samples.txt 0.23 Dsuite.svg

```
Warning, the script rounds the legend down to the nearest 0.1, so 0.15 shows up as 0.1 for example (I will fix this mannually in the figure).  

# Fbranch
Now, we can run Fbranch.  

For more details see: Malinsky, M., Svardal, H., Tyers, A.M. et al. Whole-genome sequences of Malawi cichlids reveal multiple radiations interconnected by gene flow. Nat Ecol Evol 2, 1940–1955 (2018). https://doi.org/10.1038/s41559-018-0717-x  

The input must have f_g values, as generated by Dsuite, and should be generated using the same tree topology as will be plotted.  
```bash
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/AlcaUria_out 
conda activate basic #numpy and pandas installed, so I am using a conda environment with those installed
#Run Fbranch
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Fbranch skuas.nwk samples_ID_Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17_Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17_tree.txt > fbranch.txt

#plot the results!
/home/0_PROGRAMS/Dsuite/utils/dtools.py fbranch.txt ./skuas.nwk

#repeat for the other set of tests with jaeger outgroup
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/jaeger_out
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Fbranch skuas.nwk samples_ID_Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17_Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17_tree.txt > fbranch.txt
/home/0_PROGRAMS/Dsuite/utils/dtools.py fbranch.txt ./skuas.nwk

#repeat for an alternate topology
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/jaeger_out/alternate2
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Fbranch skuas.nwk samples_ID_Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17_Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17_tree.txt > fbranch.txt
/home/0_PROGRAMS/Dsuite/utils/dtools.py fbranch.txt ./skuas.nwk

```
**Timing**: less than 1 second  

Now I will bring these figures onto my personal computer to edit for aesthetics (species names, fonts, etc).  

# Investigate trios

Now, we can choose trios to investigate in further detail with a genome scan of fd or fdM.  
One thing to keep in mind: f_d is NOT symmetrically distributed around 0 in cases of no introgression. It can reach arbitrarily low negative values which are NOT meaningful. In contrast the negative values of f_dM ARE meaningful.  

Make a list of what trio to investigate, and then investigate!  

This dataset is the same as the dataset used for the ABBA BABA test, but was NOT filtered to remove sites with a lot of missing data. The ABBA BABA dataset was filtered to remove sites that had more than 3 samples with missing data, and that gave sufficient power for the analysis, but is not sufficient for a detailed genome scan. The scan will of course only calculate the statistics for which all members of a trio have data, so the missing data filter is not so necessary.  
```bash
#get dataset of SNPs
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA
#I tested a few other datasets to ensure that results were not sensitive to filters
ln -s ../../1_Data/1.5d_genotyping/Stercorarius.RAZObowtie.merged.filteredSNPs.auto.max17.vcf.gz .
ln -s ../../1_Data/1.5d_genotyping/Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.lax.max17.vcf.gz .
ln -s ../../1_Data/1.5d_genotyping/Stercorarius.RAZObowtie.merged.filteredSNPs.auto.min4.max17.vcf.gz .


mkdir -p  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/Dinvestigate_medstringent
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/Dinvestigate_medstringent

#I am investigating introgression betwen GRSK and CISK
echo -e "skua\tantarcticus\tmaccormicki" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.skua.antarcticus.maccormicki
echo -e "maccormicki\tantarcticus\tskua" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.maccormicki.antarcticus.skua
echo -e "chilensis\tantarcticus\tskua" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.chilensis.antarcticus.skua

#I am investigating introgression betwen GRSK and POJA
echo -e "maccormicki\tskua\tpomarinus" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.maccormicki.skua.pomarinus
echo -e "chilensis\tskua\tpomarinus" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.chilensis.skua.pomarinus
echo -e "antarcticus\tskua\tpomarinus" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.antarcticus.skua.pomarinus

#I am investigating introgression betwen ANSK and CISK
echo -e "chilensis\tantarcticus\tmaccormicki" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.chilensis.antarcticus.maccormicki
echo -e "maccormicki\tantarcticus\tchilensis" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.maccormicki.antarcticus.chilensis
```

Here are the (autosomal) estimates of the statistics:  

```
maccormicki	antarcticus	skua
D=0.234049
f_d=0.0908988	3379.75/37181.5
f_dM=0.0609164	3379.75/55481.8
real	27m42.914s
user	27m15.532s
sys	0m27.376s

chilensis	antarcticus	skua
D=0.231778
f_d=0.0914281	2093.59/22898.8
f_dM=0.0569556	2093.59/36758.4
real	27m54.210s
user	27m26.528s
sys	0m27.648s

skua	antarcticus	maccormicki
D=0.278669
f_d=0.131407	4273/32517.4
f_dM=0.0755371	4273/56568.2
real	27m27.184s
user	27m0.392s
sys	0m26.808s

chilensis	antarcticus	maccormicki
D=0.180894
f_d=0.0833637	1721.97/20656.1
f_dM=0.0482157	1721.97/35713.8
real	28m9.684s
user	27m41.692s
sys	0m27.968s

maccormicki	antarcticus	chilensis
D=0.20732
f_d=0.0864235	2039.31/23596.7
f_dM=0.056805	2039.31/35900.2
real	27m33.418s
user	27m6.248s
sys	0m27.176s

maccormicki	skua	pomarinus
D=0.054089
f_d=0.0135432	637.576/47077.4
f_dM=0.00907562	637.576/70251.6
real	27m20.710s
user	26m52.904s
sys	0m27.820s

chilensis	skua	pomarinus
D=0.0681874
f_d=0.0177651	519.052/29217.5
f_dM=0.0111504	519.052/46550.1
real	28m15.448s
user	27m47.800s
sys	0m27.636s

antarcticus	skua	pomarinus
D=0.0358143
f_d=0.0085412	378.813/44351.2
f_dM=0.00552403	378.813/68575.4
real	28m16.756s
user	27m48.268s
sys	0m28.452s
```

In order to visualize the Z chromosome, we need to use more relaxed filters, because our current filters are too stringent for the dataset. Our dataset contains many females, which have only one copy of chrZ. We need to lower genotype depth cutoffs in order to retain enough data for those sites. To compare the effects of lowering depth, I have one version with the same filters as the autosomal data (medstringent) and one with min depth of 3x (medstringent_min3). Note that I am not publishing the chrZ result as it is heavily biased by artifacts caused by mixing males and females, but I still want to look at it to better understand the dataset.  

```bash
#repeat for chrZ
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/Dinvestigate_medstringent
#I am investigating introgression betwen GRSK and CISK
echo -e "skua\tantarcticus\tmaccormicki" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.medstringent.max17.vcf.gz ../samples_ID.txt test_trios.txt -n chrZ.skua.antarcticus.maccormicki
echo -e "maccormicki\tantarcticus\tskua" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.medstringent.max17.vcf.gz ../samples_ID.txt test_trios.txt -n chrZ.maccormicki.antarcticus.skua
echo -e "chilensis\tantarcticus\tskua" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.medstringent.max17.vcf.gz ../samples_ID.txt test_trios.txt -n chrZ.chilensis.antarcticus.skua

#I am investigating introgression betwen GRSK and POJA
echo -e "maccormicki\tskua\tpomarinus" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.medstringent.max17.vcf.gz ../samples_ID.txt test_trios.txt -n chrZ.maccormicki.skua.pomarinus
echo -e "chilensis\tskua\tpomarinus" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.medstringent.max17.vcf.gz ../samples_ID.txt test_trios.txt -n chrZ.chilensis.skua.pomarinus
echo -e "antarcticus\tskua\tpomarinus" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.medstringent.max17.vcf.gz ../samples_ID.txt test_trios.txt -n chrZ.antarcticus.skua.pomarinus

#I am investigating introgression betwen ANSK and CISK
echo -e "chilensis\tantarcticus\tmaccormicki" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.medstringent.max17.vcf.gz ../samples_ID.txt test_trios.txt -n chrZ.chilensis.antarcticus.maccormicki
echo -e "maccormicki\tantarcticus\tchilensis" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.medstringent.max17.vcf.gz ../samples_ID.txt test_trios.txt -n chrZ.maccormicki.antarcticus.chilensis

#combine the autosomal data with the chrZ data
grep -v "chrZ" skua_antarcticus_maccormicki_localFstats_auto.skua.antarcticus.maccormicki_50_50.txt > temp
tail -n +2 skua_antarcticus_maccormicki_localFstats_chrZ.skua.antarcticus.maccormicki_50_50.txt | cat temp - > skua_antarcticus_maccormicki_localFstats_all.skua.antarcticus.maccormicki_50_50.txt

grep -v "chrZ" chilensis_antarcticus_maccormicki_localFstats_auto.chilensis.antarcticus.maccormicki_50_50.txt > temp
tail -n +2 chilensis_antarcticus_maccormicki_localFstats_chrZ.chilensis.antarcticus.maccormicki_50_50.txt | cat temp - > chilensis_antarcticus_maccormicki_localFstats_all.chilensis.antarcticus.maccormicki_50_50.txt

grep -v "chrZ" maccormicki_antarcticus_chilensis_localFstats_auto.maccormicki.antarcticus.chilensis_50_50.txt > temp
tail -n +2 maccormicki_antarcticus_chilensis_localFstats_chrZ.maccormicki.antarcticus.chilensis_50_50.txt | cat temp - > maccormicki_antarcticus_chilensis_localFstats_all.maccormicki.antarcticus.chilensis_50_50.txt

grep -v "chrZ" antarcticus_skua_pomarinus_localFstats_auto.antarcticus.skua.pomarinus_50_50.txt > temp
tail -n +2 antarcticus_skua_pomarinus_localFstats_chrZ.antarcticus.skua.pomarinus_50_50.txt | cat temp - > antarcticus_skua_pomarinus_localFstats_all.antarcticus.skua.pomarinus_50_50.txt

grep -v "chrZ" maccormicki_skua_pomarinus_localFstats_auto.maccormicki.skua.pomarinus_50_50.txt > temp
tail -n +2 maccormicki_skua_pomarinus_localFstats_chrZ.maccormicki.skua.pomarinus_50_50.txt | cat temp - > maccormicki_skua_pomarinus_localFstats_all.maccormicki.skua.pomarinus_50_50.txt

grep -v "chrZ" chilensis_skua_pomarinus_localFstats_auto.chilensis.skua.pomarinus_50_50.txt > temp
tail -n +2 chilensis_skua_pomarinus_localFstats_chrZ.chilensis.skua.pomarinus_50_50.txt | cat temp - > chilensis_skua_pomarinus_localFstats_all.chilensis.skua.pomarinus_50_50.txt

grep -v "chrZ" maccormicki_antarcticus_skua_localFstats_auto.maccormicki.antarcticus.skua_50_50.txt > temp
tail -n +2 maccormicki_antarcticus_skua_localFstats_chrZ.maccormicki.antarcticus.skua_50_50.txt | cat temp - > maccormicki_antarcticus_skua_localFstats_all.maccormicki.antarcticus.skua_50_50.txt

grep -v "chrZ" chilensis_antarcticus_skua_localFstats_auto.chilensis.antarcticus.skua_50_50.txt > temp
tail -n +2 chilensis_antarcticus_skua_localFstats_chrZ.chilensis.antarcticus.skua_50_50.txt | cat temp - > chilensis_antarcticus_skua_localFstats_all.chilensis.antarcticus.skua_50_50.txt





#repeat for chrZ with a min depth of 3 instead of 6 (chrZ has lower depth in females)
mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/Dinvestigate_medstringent_min3
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/Dinvestigate_medstringent_min3
#I am investigating introgression betwen GRSK and CISK
echo -e "skua\tantarcticus\tmaccormicki" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.medstringent.min3.max17.vcf.gz ../samples_ID.txt test_trios.txt -n chrZ.skua.antarcticus.maccormicki
echo -e "maccormicki\tantarcticus\tskua" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.medstringent.min3.max17.vcf.gz ../samples_ID.txt test_trios.txt -n chrZ.maccormicki.antarcticus.skua
echo -e "chilensis\tantarcticus\tskua" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.medstringent.min3.max17.vcf.gz ../samples_ID.txt test_trios.txt -n chrZ.chilensis.antarcticus.skua

#I am investigating introgression betwen GRSK and POJA
echo -e "maccormicki\tskua\tpomarinus" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.medstringent.min3.max17.vcf.gz ../samples_ID.txt test_trios.txt -n chrZ.maccormicki.skua.pomarinus
echo -e "chilensis\tskua\tpomarinus" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.medstringent.min3.max17.vcf.gz ../samples_ID.txt test_trios.txt -n chrZ.chilensis.skua.pomarinus
echo -e "antarcticus\tskua\tpomarinus" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.medstringent.min3.max17.vcf.gz ../samples_ID.txt test_trios.txt -n chrZ.antarcticus.skua.pomarinus

#I am investigating introgression betwen ANSK and CISK
echo -e "chilensis\tantarcticus\tmaccormicki" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.medstringent.min3.max17.vcf.gz ../samples_ID.txt test_trios.txt -n chrZ.chilensis.antarcticus.maccormicki
echo -e "maccormicki\tantarcticus\tchilensis" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.medstringent.min3.max17.vcf.gz ../samples_ID.txt test_trios.txt -n chrZ.maccormicki.antarcticus.chilensis


#combine the autosomal data with the chrZ data
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/Dinvestigate_medstringent

grep -v "chrZ" skua_antarcticus_maccormicki_localFstats_auto.skua.antarcticus.maccormicki_50_50.txt > temp
tail -n +2 ../Dinvestigate_medstringent_min4/skua_antarcticus_maccormicki_localFstats_chrZ.skua.antarcticus.maccormicki_50_50.txt | cat temp - > skua_antarcticus_maccormicki_localFstats_all.skua.antarcticus.maccormicki_50_50.txt

grep -v "chrZ" chilensis_antarcticus_maccormicki_localFstats_auto.chilensis.antarcticus.maccormicki_50_50.txt > temp
tail -n +2 ../Dinvestigate_medstringent_min4/chilensis_antarcticus_maccormicki_localFstats_chrZ.chilensis.antarcticus.maccormicki_50_50.txt | cat temp - > chilensis_antarcticus_maccormicki_localFstats_all.chilensis.antarcticus.maccormicki_50_50.txt

grep -v "chrZ" maccormicki_antarcticus_chilensis_localFstats_auto.maccormicki.antarcticus.chilensis_50_50.txt > temp
tail -n +2 ../Dinvestigate_medstringent_min4/maccormicki_antarcticus_chilensis_localFstats_chrZ.maccormicki.antarcticus.chilensis_50_50.txt | cat temp - > maccormicki_antarcticus_chilensis_localFstats_all.maccormicki.antarcticus.chilensis_50_50.txt

grep -v "chrZ" antarcticus_skua_pomarinus_localFstats_auto.antarcticus.skua.pomarinus_50_50.txt > temp
tail -n +2 ../Dinvestigate_medstringent_min4/antarcticus_skua_pomarinus_localFstats_chrZ.antarcticus.skua.pomarinus_50_50.txt | cat temp - > antarcticus_skua_pomarinus_localFstats_all.antarcticus.skua.pomarinus_50_50.txt

grep -v "chrZ" maccormicki_skua_pomarinus_localFstats_auto.maccormicki.skua.pomarinus_50_50.txt > temp
tail -n +2 ../Dinvestigate_medstringent_min4/maccormicki_skua_pomarinus_localFstats_chrZ.maccormicki.skua.pomarinus_50_50.txt | cat temp - > maccormicki_skua_pomarinus_localFstats_all.maccormicki.skua.pomarinus_50_50.txt

grep -v "chrZ" chilensis_skua_pomarinus_localFstats_auto.chilensis.skua.pomarinus_50_50.txt > temp
tail -n +2 ../Dinvestigate_medstringent_min4/chilensis_skua_pomarinus_localFstats_chrZ.chilensis.skua.pomarinus_50_50.txt | cat temp - > chilensis_skua_pomarinus_localFstats_all.chilensis.skua.pomarinus_50_50.txt

grep -v "chrZ" maccormicki_antarcticus_skua_localFstats_auto.maccormicki.antarcticus.skua_50_50.txt > temp
tail -n +2 ../Dinvestigate_medstringent_min4/maccormicki_antarcticus_skua_localFstats_chrZ.maccormicki.antarcticus.skua_50_50.txt | cat temp - > maccormicki_antarcticus_skua_localFstats_all.maccormicki.antarcticus.skua_50_50.txt

grep -v "chrZ" chilensis_antarcticus_skua_localFstats_auto.chilensis.antarcticus.skua_50_50.txt > temp
tail -n +2 ../Dinvestigate_medstringent_min4/chilensis_antarcticus_skua_localFstats_chrZ.chilensis.antarcticus.skua_50_50.txt | cat temp - > chilensis_antarcticus_skua_localFstats_all.chilensis.antarcticus.skua_50_50.txt

```
* `-w`: window size, step size for the sliding window (measured in SNPs, not in bp.

The output file will have one row per window. It will have the estimate for D, as well as f_d (the proportion of the window influenced by introgression) and f_dM  

```
min3 chrZ
skua	antarcticus	maccormicki
D=0.326249
f_d=0.174629	509.896/2919.88
f_dM=0.104359	509.896/4885.99
real	1m58.117s
user	1m55.860s
sys	0m2.256s

maccormicki	antarcticus	skua
D=0.14364
f_d=0.0608577	176.625/2902.26
f_dM=0.0387263	176.625/4560.85
real	2m0.047s
user	1m58.024s
sys	0m2.008s

chilensis	antarcticus	skua
D=0.413186
f_d=0.202702	344.417/1699.13
f_dM=0.123143	344.417/2796.88
real	1m59.335s
user	1m57.276s
sys	0m2.056s

maccormicki	skua	pomarinus
D=-0.138718
f_d=-0.0300962	-141.708/4708.51
f_dM=-0.0207967	-141.708/6813.98
real	2m3.136s
user	2m0.972s
sys	0m2.160s

chilensis	skua	pomarinus
D=0.212788
f_d=0.0637322	158.771/2491.22
f_dM=0.0408959	158.771/3882.32
real	1m57.733s
user	1m55.776s
sys	0m1.956s

antarcticus	skua	pomarinus
D=-0.0657657
f_d=-0.0150931	-67.3715/4463.73
f_dM=-0.0106671	-67.3715/6315.83
real	1m58.967s
user	1m56.832s
sys	0m2.136s

chilensis	antarcticus	maccormicki
D=0.402236
f_d=0.215709	356.806/1654.11
f_dM=0.126592	356.806/2818.54
real	2m0.156s
user	1m58.132s
sys	0m2.024s

maccormicki	antarcticus	chilensis
D=0.196946
f_d=0.0783966	130.042/1658.77
f_dM=0.0501018	130.042/2595.55
real	1m56.980s
user	1m54.884s
sys	0m2.096s

```

## Visualize genome scan
I am using R to plot the results, and so the code for the plotting step is provided in a .Rmd file (`3.1d_Dsuite.rmd`), which is knit into an .html file (`3.1d_Dsuite.html`).  

# Intergenic vs genic regions  
Do levels of introgression vary between genic and intergenic regions? I will repeat the ABBA BABA test for genic and intergenic regions separately.  

## Intergenic regions  
First, create a folder to hold the results, and specify the sample populations.  
```bash
mkdir -p /home/1_Else/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/jaegerout_notgenes
cd  /home/1_Else/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/jaegerout_notgenes
cat > samples_ID.txt
```
```
LTJA_MKP990 Outgroup
PAJA_B20730 Outgroup
PAJA_USNM606730 Outgroup
POJA_MKP1559 pomarinus
POJA_4 pomarinus
POJA_IB2659 pomarinus
GRSK_MKP1592 skua
GRSK_MKP1593 skua
CISK2 antarcticus
CISK55 antarcticus
CISK_3 antarcticus
CHSK_MKP2451 chilensis
ANSK7 maccormicki
ANSK01 maccormicki
ANSK8 maccormicki
Fratercula_arctica xxx
Alca_torda xxx
Uria_lomvia xxx
```
```bash
sed -i 's/ /\t/g' samples_ID.txt
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk

#filter the VCF file to remove sequences within annotated genes
/home/0_PROGRAMS/bcftools-1.14/bcftools view -T ^/home/1_Else/0_GENOMES5/Stercorarius/0_Reference_genome/0.3_annotation/liftoff/Stercorarius_Stercorarius_liftoff.genes.bed -O z ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17.vcf.gz > ../Stercorarius.RAZObowtie.merged.filteredSNPs.intergenic.stringent.max17.vcf.gz

#run the ABBA BABA test
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.intergenic.stringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.intergenic.stringent.max17 #
```

## Genic regions  
Now we will repeat for genic regions. Note that this includes both introns and exons because exons have too few SNPs, but I did check results with exons only to ensure the patterns were consistent (see appendix below)  
```bash
mkdir -p /home/1_Else/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/jaegerout_genes
cd  /home/1_Else/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/jaegerout_genes
cat > samples_ID.txt
```
```
LTJA_MKP990 Outgroup
PAJA_B20730 Outgroup
PAJA_USNM606730 Outgroup
POJA_MKP1559 pomarinus
POJA_4 pomarinus
POJA_IB2659 pomarinus
GRSK_MKP1592 skua
GRSK_MKP1593 skua
CISK2 antarcticus
CISK55 antarcticus
CISK_3 antarcticus
CHSK_MKP2451 chilensis
ANSK7 maccormicki
ANSK01 maccormicki
ANSK8 maccormicki
Fratercula_arctica xxx
Alca_torda xxx
Uria_lomvia xxx
```
```bash
sed -i 's/ /\t/g' samples_ID.txt
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk

#make a vcf file that contains only genic regions
tabix -R /home/1_Else/0_GENOMES5/Stercorarius/0_Reference_genome/0.3_annotation/liftoff/Stercorarius_Stercorarius_liftoff.genes.bed -h ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17.vcf.gz > ../Stercorarius.RAZObowtie.merged.filteredSNPs.genes.stringent.max17.vcf.gz

#run the ABBA BABA test
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.genes.stringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.genes.stringent.max17 #

```

## Fbranch for genic vs intergenic regions  
Now, we can run Fbranch as above.  
The input must have f_g values, as generated by Dsuite, and should be generated using the same tree topology as will be plotted.  
```bash
cd  /home/1_Else/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/jaegerout_notgenes
conda activate basic #numpy and pandas installed
#Run Fbranch
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Fbranch skuas.nwk samples_ID_Stercorarius.RAZObowtie.merged.filteredSNPs.intergenic.stringent.max17_tree.txt > fbranch.txt

#plot the results!
/home/0_PROGRAMS/Dsuite/utils/dtools.py fbranch.txt ./skuas.nwk

#repeat for the other set of tests
cd /home/1_Else/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/jaegerout_genes
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Fbranch skuas.nwk samples_ID_Stercorarius.RAZObowtie.merged.filteredSNPs.genes.stringent.max17_tree.txt > fbranch.txt
/home/0_PROGRAMS/Dsuite/utils/dtools.py fbranch.txt ./skuas.nwk

#repeat for an alternate topology
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/jaeger_out/alternate2
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Fbranch skuas.nwk samples_ID_Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17_Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17_tree.txt > fbranch.txt
/home/0_PROGRAMS/Dsuite/utils/dtools.py fbranch.txt ./skuas.nwk

```
**Timing**: less than 1 second

# Appendix: testing robustness   
### Readers will probably want to skip this section - I kept it here for completeness

After running the analysis, I want to rerun the analyses with additional datasets with different filters to ensure that the interpretation of the results are not dependant upon choice of bioinformatic filters, sample choice, etc. These results will not be presented in the manuscript (otherwise it would be too long to read with extra details that do not contribute to the key points), but are useful as behind-the-scenes checks before publishing results.  


## Testing robustness to choice of (P4) outgroup  
Now, I want to ensure that the interpretation of the results is not dependant on the choice of outgroup. I will rerun the analysis with *Uria* or *Alca* in P4, and also with *Fratercula* in P4. Note that I am not using *Fratercula* in the published project (since it is low sequencing depth), but I included it in the dataset as a check to ensure that the results are consistent even when using this different outgroup.  

```bash
mkdir -p  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/Alca_out
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/Alca_out 
cat > samples_ID.txt
```
```
LTJA_MKP990 longicaudus
PAJA_B20730 parasiticus
PAJA_USNM606730 parasiticus
POJA_MKP1559 pomarinus
POJA_4 pomarinus
POJA_IB2659 pomarinus
GRSK_MKP1592 skua
GRSK_MKP1593 skua
CISK2 antarcticus
CISK55 antarcticus
CISK_3 antarcticus
CHSK_MKP2451 chilensis
ANSK7 maccormicki
ANSK01 maccormicki
ANSK8 maccormicki
Fratercula_arctica xxx
Alca_torda Outgroup
Uria_lomvia xxx
```
```bash
sed -i 's/ /\t/g' samples_ID.txt
echo "(Outgroup,((parasiticus,longicaudus),(pomarinus,(skua,(maccormicki,(antarcticus,chilensis))))));" > skuas.nwk
#time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.PAJAbowtie.filteredSNPs.auto.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.PAJAbowtie.filteredSNPs.auto.Alcid_out.Hiseq #78m7.014s
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17 #

```
```bash
mkdir -p  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/Uria_out
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/Uria_out 
cat > samples_ID.txt
```
```
LTJA_MKP990 longicaudus
PAJA_B20730 parasiticus
PAJA_USNM606730 parasiticus
POJA_MKP1559 pomarinus
POJA_4 pomarinus
POJA_IB2659 pomarinus
GRSK_MKP1592 skua
GRSK_MKP1593 skua
CISK2 antarcticus
CISK55 antarcticus
CISK_3 antarcticus
CHSK_MKP2451 chilensis
ANSK7 maccormicki
ANSK01 maccormicki
ANSK8 maccormicki
Fratercula_arctica xxx
Alca_torda xxx
Uria_lomvia Outgroup
```
```bash
sed -i 's/ /\t/g' samples_ID.txt
echo "(Outgroup,((parasiticus,longicaudus),(pomarinus,(skua,(maccormicki,(antarcticus,chilensis))))));" > skuas.nwk
#time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.PAJAbowtie.filteredSNPs.auto.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.PAJAbowtie.filteredSNPs.auto.Alcid_out.Hiseq #78m7.014s
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17 #

```
```bash
mkdir -p  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/AlcaUriaFratercula_out
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/AlcaUriaFratercula_out 
cat > samples_ID.txt
```
```
LTJA_MKP990 longicaudus
PAJA_B20730 parasiticus
PAJA_USNM606730 parasiticus
POJA_MKP1559 pomarinus
POJA_4 pomarinus
POJA_IB2659 pomarinus
GRSK_MKP1592 skua
GRSK_MKP1593 skua
CISK2 antarcticus
CISK55 antarcticus
CISK_3 antarcticus
CHSK_MKP2451 chilensis
ANSK7 maccormicki
ANSK01 maccormicki
ANSK8 maccormicki
Fratercula_arctica Outgroup
Alca_torda Outgroup
Uria_lomvia Outgroup
```
```bash
sed -i 's/ /\t/g' samples_ID.txt
echo "(Outgroup,((parasiticus,longicaudus),(pomarinus,(skua,(maccormicki,(antarcticus,chilensis))))));" > skuas.nwk
#time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.PAJAbowtie.filteredSNPs.auto.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.PAJAbowtie.filteredSNPs.auto.Alcid_out.Hiseq #78m7.014s
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17 #

```

Results are extremely similar amongst the choices of P4 and do not change overall interpretations. Adding *Fratercula* has little effect. Using only *Uria* leads to slightly lower D when *pomarinus* is P2 and *parasiticus*/*longicaudus* are P3, while using only *Alca* leads to slightly higher D with those trios, but both are statistically significant with a similarly very small f4 ratio. The three main "introgression" events (*antarcticus*-*skua*, *antarcticus*-*maccormicki*, *skua*-*pomarinus*) are qualitatively identical between choice of outgroup, so we can be more confident that are interpretations are not an artifact of the particular chosen outgroup.

## Testing robustness: Hiseq vs Novaseq  
Next, I want to ensure that the results are not being driven by differences between Hiseq and Novaseq sequences, so I am going to analyze each of the datasets one at a time. Note that I am doing it twice: once with only Novaseq skuas, and again with only Hiseq samples. Note that I am including *parasiticus*, *longicaudus*, and *chilensis* in both even though they are all Hiseq.
```bash
mkdir -p  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/Alcid_out.Hiseq
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/Alcid_out.Hiseq #on node
cat > samples_ID.txt
```
```
LTJA_MKP990 longicaudus
PAJA_B20730 parasiticus
PAJA_USNM606730 parasiticus
POJA_MKP1559 pomarinus
POJA_4 xxx
POJA_IB2659 xxx
GRSK_MKP1592 skua
GRSK_MKP1593 xxx
CISK2 xxx
CISK55 xxx
CISK_3 antarcticus
CHSK_MKP2451 chilensis
ANSK7 maccormicki
ANSK01 xxx
ANSK8 xxx
Fratercula_arctica xxx
Alca_torda Outgroup
Uria_lomvia Outgroup
```
```bash
sed -i 's/ /\t/g' samples_ID.txt
echo "(Outgroup,((parasiticus,longicaudus),(pomarinus,(skua,(maccormicki,(antarcticus,chilensis))))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17.Alcid_out.Hiseq #6m18.740s

```

```bash
mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/Alcid_out.Novaseq
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/Alcid_out.Novaseq 
cat > samples_ID.txt
```
```
LTJA_MKP990 longicaudus
PAJA_B20730 parasiticus
PAJA_USNM606730 parasiticus
POJA_MKP1559 xxx
POJA_4 pomarinus
POJA_IB2659 pomarinus
GRSK_MKP1592 xxx
GRSK_MKP1593 skua
CISK2 antarcticus
CISK55 antarcticus
CISK_3 xxx
CHSK_MKP2451 chilensis
ANSK7 xxx
ANSK01 maccormicki
ANSK8 maccormicki
Fratercula_arctica xxx
Alca_torda Outgroup
Uria_lomvia Outgroup
```
```bash
sed -i 's/ /\t/g' samples_ID.txt
echo "(Outgroup,((parasiticus,longicaudus),(pomarinus,(skua,(maccormicki,(antarcticus,chilensis))))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17.Alcid_out.Novaseq #6m18.740s

```
Results are qualitatively consistent between the two datasets.  

## Testing robustness: relaxed depth and quality filters  

Now I am testing two datasets to ensure patterns remain consistent even with relaxed filters.  

With jaegers as the outgroup:  
```bash
mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/176stringent.min3.jaegerout
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/176stringent.min3.jaegerout
cat > samples_ID.txt
```
```
LTJA_MKP990 Outgroup
PAJA_B20730 Outgroup
PAJA_USNM606730 Outgroup
POJA_MKP1559 pomarinus
POJA_4 pomarinus
POJA_IB2659 pomarinus
GRSK_MKP1592 skua
GRSK_MKP1593 skua
CISK2 antarcticus
CISK55 antarcticus
CISK_3 antarcticus
CHSK_MKP2451 chilensis
ANSK7 maccormicki
ANSK01 maccormicki
ANSK8 maccormicki
Fratercula_arctica xxx
Alca_torda xxx
Uria_lomvia xxx
```
```bash
sed -i 's/ /\t/g' samples_ID.txt
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.filteredSNPs.auto.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.filteredSNPs.auto.jaegerout #78m7.014s

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/176stringent.min6.max15.jaegerout
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/176stringent.min6.max15.jaegerout
cp ../176stringent.min3.jaegerout/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.filteredSNPs.auto.stringent.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.PAJAbowtie.filteredSNPs.auto.stringent.jaegerout #78m7.014s

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/jaeger_out/medstringent && cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/jaeger_out/medstringent
cp ../samples_ID.txt .; cp ../skuas.nwk .
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17 #

#repeat with the two alternative topologies
mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/jaeger_out/alternate/medstringent && cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/jaeger_out/alternate/medstringent
cp ../../samples_ID.txt .; echo "(Outgroup,(pomarinus,(skua,(chilensis,(antarcticus,maccormicki)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../../../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17 #

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/jaeger_out/alternate2/medstringent && cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/jaeger_out/alternate2/medstringent
cp ../../samples_ID.txt .; echo "(Outgroup,(pomarinus,(skua,(antarcticus,(chilensis,maccormicki)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../../../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17 #


```

With *Alca* as the outgroup:  

```bash
mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/176stringent.min3
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/176stringent.min3
cat > samples_ID.txt
```
```
LTJA_MKP990 longicaudus
PAJA_B20730 parasiticus
PAJA_USNM606730 parasiticus
POJA_MKP1559 pomarinus
POJA_4 pomarinus
POJA_IB2659 pomarinus
GRSK_MKP1592 skua
GRSK_MKP1593 skua
CISK2 antarcticus
CISK55 antarcticus
CISK_3 antarcticus
CHSK_MKP2451 chilensis
ANSK7 maccormicki
ANSK01 maccormicki
ANSK8 maccormicki
Fratercula_arctica xxx
Alca_torda Outgroup
Uria_lomvia xxx
```
```bash
sed -i 's/ /\t/g' samples_ID.txt
echo "(Outgroup,((parasiticus,longicaudus),(pomarinus,(skua,(maccormicki,(antarcticus,chilensis))))));" > skuas.nwk

time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios Stercorarius.RAZObowtie.filteredSNPs.auto.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.filteredSNPs.auto #260m55.157s (192m12.003s)

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/176stringent.min6.max15.Alcaout
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/176stringent.min6.max15.Alcaout
cp ../176stringent.min3/samples_ID.txt .
echo "(Outgroup,((parasiticus,longicaudus),(pomarinus,(skua,(maccormicki,(antarcticus,chilensis))))));" > skuas.nwk
#try more stringent settings
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios Stercorarius.RAZObowtie.filteredSNPs.auto.stringent.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.filteredSNPs.auto.stringent #260m55.157s (192m12.003s)

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/AlcaUria_out/medstringent && cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/AlcaUria_out/medstringent
cp ../samples_ID.txt .; cp ../skuas.nwk .
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17 #

```

Patterns remain the same, but are less consistent when allowing 3x coverage in genotype calls - the D statistic is non-significant with P1=*antarcticus* P2=*skua* and P3=*pomarinus* under the relaxed filters. Note that genotype calls with 3x coverage are not reliable, as they have a high chance of missing heterozygous calls and biasing heterozygous sites towards the reference allele.    

## Testing robustness: missing data and genotype depth filters  
Now I am testing combinations of missing data filters (max 100, 50, 40, 30, or 10% missing data) and genotype depths (10x) to ensure results are robust to these filters.  

```bash
mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/30stringent
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/30stringent
cat > samples_ID.txt
```
```
LTJA_MKP990 Outgroup
PAJA_B20730 Outgroup
PAJA_USNM606730 Outgroup
POJA_MKP1559 pomarinus
POJA_4 pomarinus
POJA_IB2659 pomarinus
GRSK_MKP1592 skua
GRSK_MKP1593 skua
CISK2 antarcticus
CISK55 antarcticus
CISK_3 antarcticus
CHSK_MKP2451 chilensis
ANSK7 maccormicki
ANSK01 maccormicki
ANSK8 maccormicki
Fratercula_arctica xxx
Alca_torda xxx
Uria_lomvia xxx
```
```bash
sed -i 's/ /\t/g' samples_ID.txt
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.30stringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.30stringent.max17.jaegerout #78m7.014s

```
```bash
#jaegerout
mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/10stringent
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/10stringent
cp ../30stringent/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.10stringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.10stringent.max17.jaegerout #78m7.014s

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/40stringent
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/40stringent
cp ../30stringent/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.40stringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.40stringent.max17.jaegerout #78m7.014s

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/50stringent
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/50stringent
cp ../30stringent/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.50stringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.50stringent.max17.jaegerout #78m7.014s

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/medstringent.min10
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/medstringent.min10
cp ../30stringent/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.min10.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.min10.max17.jaegerout #78m7.014s

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/40stringent.min10
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/40stringent.min10
cp ../30stringent/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.40stringent.min10.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.40stringent.min10.max17.jaegerout #78m7.014s

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/50stringent.min10
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/50stringent.min10
cp ../30stringent/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.50stringent.min10.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.50stringent.min10.max17.jaegerout #78m7.014s

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/30stringent.min10
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/30stringent.min10
cp ../30stringent/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.30stringent.min10.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.30stringent.min10.max17.jaegerout #78m7.014s

#AlcaUria out
mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/medstringent.min10.AlcaUria_out
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/medstringent.min10.AlcaUria_out
cp ../AlcaUria_out/samples_ID.txt .
echo "(Outgroup,((parasiticus,longicaudus),(pomarinus,(skua,(maccormicki,(antarcticus,chilensis))))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.min10.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.min10.max17.AlcaUria_out #78m7.014s

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/30stringent.AlcaUria_out
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/30stringent.AlcaUria_out
cp ../AlcaUria_out/samples_ID.txt .
echo "(Outgroup,((parasiticus,longicaudus),(pomarinus,(skua,(maccormicki,(antarcticus,chilensis))))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.30stringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.30stringent.max17.AlcaUria_out #78m7.014s


```
The three proposed introgression events are always highly statistically significant regardless of filter and never flip direction. The other tested trios (that do not have statistically significant introgression events) sometimes flip direction, and sometimes become statistically significant at p<0.05, which is not surprising.  

When greatly relaxing genotype depth filters to allow called genotypes with only 3x depth, several additional trios become statistically significantly unbalanced, and these patterns go away when genotype min depth was raised to 6x or 10x. I suspect that this is due to the unreliability of calling heterozygous sites at such low depth. If two reads are alternate and one is reference, it might get called heterozygous, but if two are reference and one is alternate, it might get called as homozygous reference. I have seen genotype calls like this in the data when looking at the raw reads alignment and comparing to genotype calls. This could lead to a bias towards the reference genome for samples with higher heterozygosity and/or lower sequencing depth; if not accounted for, this could affect ABBA BABA pattern counts and lead to a false signal.  


## Testing robustness: choice of different pomarinus samples  
We sequenced three different *S. pomarinus*, so this is an opportunity to ensure that the results are robust to the choice of sample. I want to make sure that the pattern is not driven by any single sample. I extracted the DNA for these samples on three separate days, and they were sequenced at two different facilities with different technologies with different error profiles, so if the same pattern appears for all three, then contamination/chance/sequencing error of *S. pomarinus* is very unlikely to be driving the signal.  

Now, I will test each POJA one at a time with all the different missing data filters. Note: f4 ratios will NOT be useable when there is only one sample in P3 as it uses allele frequencies! I am looking at the values of D, rather than the f4 ratio.  

```bash
mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/medstringent_POJA_4
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/medstringent_POJA_4
cat > samples_ID.txt
```
```
LTJA_MKP990 Outgroup
PAJA_B20730 Outgroup
PAJA_USNM606730 Outgroup
POJA_MKP1559 xxx
POJA_4 pomarinus
POJA_IB2659 xxx
GRSK_MKP1592 skua
GRSK_MKP1593 skua
CISK2 antarcticus
CISK55 antarcticus
CISK_3 antarcticus
CHSK_MKP2451 chilensis
ANSK7 maccormicki
ANSK01 maccormicki
ANSK8 maccormicki
Fratercula_arctica xxx
Alca_torda xxx
Uria_lomvia xxx
```
```bash
sed -i 's/ /\t/g' samples_ID.txt
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.jaegerout #78m7.014s

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/10stringent_POJA_4
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/10stringent_POJA_4
cp ../medstringent_POJA_4/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.10stringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.10stringent.max17.jaegerout_POJA_4 

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/40stringent_POJA_4
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/40stringent_POJA_4
cp ../medstringent_POJA_4/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.40stringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.40stringent.max17.jaegerout_POJA_4 

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/50stringent_POJA_4
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/50stringent_POJA_4
cp ../medstringent_POJA_4/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.50stringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.50stringent.max17.jaegerout_POJA_4 

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/medstringent.min10_POJA_4
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/medstringent.min10_POJA_4
cp ../medstringent_POJA_4/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.min10.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.min10.max17.jaegerout_POJA_4 

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/40stringent.min10_POJA_4
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/40stringent.min10_POJA_4
cp ../medstringent_POJA_4/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.40stringent.min10.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.40stringent.min10.max17.jaegerout_POJA_4 

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/50stringent.min10_POJA_4
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/50stringent.min10_POJA_4
cp ../medstringent_POJA_4/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.50stringent.min10.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.50stringent.min10.max17.jaegerout_POJA_4 

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/30stringent.min10_POJA_4
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/30stringent.min10_POJA_4
cp ../medstringent_POJA_4/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.30stringent.min10.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.30stringent.min10.max17.jaegerout_POJA_4 

```

```bash
mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/medstringent_POJA_IB2659
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/medstringent_POJA_IB2659
cat > samples_ID.txt
```
```
LTJA_MKP990 Outgroup
PAJA_B20730 Outgroup
PAJA_USNM606730 Outgroup
POJA_MKP1559 xxx
POJA_4 xxx
POJA_IB2659 pomarinus
GRSK_MKP1592 skua
GRSK_MKP1593 skua
CISK2 antarcticus
CISK55 antarcticus
CISK_3 antarcticus
CHSK_MKP2451 chilensis
ANSK7 maccormicki
ANSK01 maccormicki
ANSK8 maccormicki
Fratercula_arctica xxx
Alca_torda xxx
Uria_lomvia xxx
```
```bash
sed -i 's/ /\t/g' samples_ID.txt
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.jaegerout

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/10stringent_POJA_IB2659
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/10stringent_POJA_IB2659
cp ../medstringent_POJA_IB2659/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.10stringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.10stringent.max17.jaegerout_POJA_IB2659 

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/40stringent_POJA_IB2659
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/40stringent_POJA_IB2659
cp ../medstringent_POJA_IB2659/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.40stringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.40stringent.max17.jaegerout_POJA_IB2659 

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/50stringent_POJA_IB2659
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/50stringent_POJA_IB2659
cp ../medstringent_POJA_IB2659/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.50stringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.50stringent.max17.jaegerout_POJA_IB2659 

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/medstringent.min10_POJA_IB2659
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/medstringent.min10_POJA_IB2659
cp ../medstringent_POJA_IB2659/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.min10.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.min10.max17.jaegerout_POJA_IB2659 

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/40stringent.min10_POJA_IB2659
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/40stringent.min10_POJA_IB2659
cp ../medstringent_POJA_IB2659/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.40stringent.min10.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.40stringent.min10.max17.jaegerout_POJA_IB2659 

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/50stringent.min10_POJA_IB2659
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/50stringent.min10_POJA_IB2659
cp ../medstringent_POJA_IB2659/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.50stringent.min10.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.50stringent.min10.max17.jaegerout_POJA_IB2659 

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/30stringent.min10_POJA_IB2659
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/30stringent.min10_POJA_IB2659
cp ../medstringent_POJA_IB2659/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.30stringent.min10.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.30stringent.min10.max17.jaegerout_POJA_IB2659 

```

```bash
mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/medstringent_POJA_MKP1559
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/medstringent_POJA_MKP1559
cat > samples_ID.txt
```
```
LTJA_MKP990 Outgroup
PAJA_B20730 Outgroup
PAJA_USNM606730 Outgroup
POJA_MKP1559 pomarinus
POJA_4 xxx
POJA_IB2659 xxx
GRSK_MKP1592 skua
GRSK_MKP1593 skua
CISK2 antarcticus
CISK55 antarcticus
CISK_3 antarcticus
CHSK_MKP2451 chilensis
ANSK7 maccormicki
ANSK01 maccormicki
ANSK8 maccormicki
Fratercula_arctica xxx
Alca_torda xxx
Uria_lomvia xxx
```
```bash
sed -i 's/ /\t/g' samples_ID.txt
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.jaegerout #78m7.014s


mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/10stringent_POJA_MKP1559
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/10stringent_POJA_MKP1559
cp ../medstringent_POJA_MKP1559/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.10stringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.10stringent.max17.jaegerout_POJA_MKP1559 

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/40stringent_POJA_MKP1559
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/40stringent_POJA_MKP1559
cp ../medstringent_POJA_MKP1559/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.40stringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.40stringent.max17.jaegerout_POJA_MKP1559 

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/50stringent_POJA_MKP1559
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/50stringent_POJA_MKP1559
cp ../medstringent_POJA_MKP1559/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.50stringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.50stringent.max17.jaegerout_POJA_MKP1559 

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/medstringent.min10_POJA_MKP1559
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/medstringent.min10_POJA_MKP1559
cp ../medstringent_POJA_MKP1559/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.min10.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.min10.max17.jaegerout_POJA_MKP1559 

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/40stringent.min10_POJA_MKP1559
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/40stringent.min10_POJA_MKP1559
cp ../medstringent_POJA_MKP1559/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.40stringent.min10.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.40stringent.min10.max17.jaegerout_POJA_MKP1559 

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/50stringent.min10_POJA_MKP1559
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/50stringent.min10_POJA_MKP1559
cp ../medstringent_POJA_MKP1559/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.50stringent.min10.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.50stringent.min10.max17.jaegerout_POJA_MKP1559 

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/30stringent.min10_POJA_MKP1559
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/30stringent.min10_POJA_MKP1559
cp ../medstringent_POJA_MKP1559/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.30stringent.min10.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.30stringent.min10.max17.jaegerout_POJA_MKP1559 

```
Results: all combinations of sample, missing data filter, and min genotype depth produce qualitatively similar values. All of them show an excess of shared derived alleles between *S. skua* and *S. pomarinus* (but a few combinations are no longer statistically significant - these are the datasets that were filtered most stringently so that only a couple hundred ABBA and BABA sites remain). The magnitude of *D* is qualitatively similar, but slightly higher with POJA_MKP1559 than POJA_IB2659, which is slightly higher than POJA_4. Since the same pattern was produced by different *S. pomarinus* samples that were kept separate in the lab, I am confident that the signal is not a result of a technical issue with the *S. pomarinus* sample.

```bash
mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/medstringent_POJA_split
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/medstringent_POJA_split
cat > samples_ID.txt
```
```
LTJA_MKP990 Outgroup
PAJA_B20730 Outgroup
PAJA_USNM606730 Outgroup
POJA_MKP1559 pomarinus_E
POJA_4 pomarinus_E
POJA_IB2659 pomarinus_W
GRSK_MKP1592 skua
GRSK_MKP1593 skua
CISK2 antarcticus
CISK55 antarcticus
CISK_3 antarcticus
CHSK_MKP2451 chilensis
ANSK7 maccormicki
ANSK01 maccormicki
ANSK8 maccormicki
Fratercula_arctica xxx
Alca_torda xxx
Uria_lomvia xxx
```
```bash
sed -i 's/ /\t/g' samples_ID.txt
echo "(Outgroup,((pomarinus_E,pomarinus_W),(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.jaegerout #48m40.157s

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/stringent_POJA_split
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/stringent_POJA_split
cp ../medstringent_POJA_split/samples_ID.txt .
echo "(Outgroup,((pomarinus_E,pomarinus_W),(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17.stringent_POJA_split #5m42.650s

```
Neither west nor east *pomarinus* share statistically significantly more derived alleles with any *Catharacta*.

## Testing robustness: choice of different skua samples
We sequenced two different *S. skua*, so this is an opportunity to ensure that the results are robust to the choice of sample. I want to make sure that the pattern is not driven by any single sample. I extracted the DNA for these samples on two separate days (over a year apart), they were stored separately, and they were sequenced at two different facilities with different technologies with different error profiles, so if the same pattern appears for both, then contamination/chance/sequencing error of *S. skua* is very unlikely to be driving the signal.

```bash
mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/medstringent_GRSK_MKP1592
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/medstringent_GRSK_MKP1592
cat > samples_ID.txt
```
```
LTJA_MKP990 Outgroup
PAJA_B20730 Outgroup
PAJA_USNM606730 Outgroup
POJA_MKP1559 pomarinus
POJA_4 pomarinus
POJA_IB2659 pomarinus
GRSK_MKP1592 xxx
GRSK_MKP1593 skua
CISK2 antarcticus
CISK55 antarcticus
CISK_3 antarcticus
CHSK_MKP2451 chilensis
ANSK7 maccormicki
ANSK01 maccormicki
ANSK8 maccormicki
Fratercula_arctica xxx
Alca_torda xxx
Uria_lomvia xxx
```
```bash
sed -i 's/ /\t/g' samples_ID.txt
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.jaegerout #78m7.014s

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/10stringent_GRSK_MKP1592
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/10stringent_GRSK_MKP1592
cp ../medstringent_GRSK_MKP1592/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.10stringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.10stringent.max17.jaegerout_GRSK_MKP1592 

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/40stringent_GRSK_MKP1592
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/40stringent_GRSK_MKP1592
cp ../medstringent_GRSK_MKP1592/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.40stringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.40stringent.max17.jaegerout_GRSK_MKP1592 

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/50stringent_GRSK_MKP1592
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/50stringent_GRSK_MKP1592
cp ../medstringent_GRSK_MKP1592/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.50stringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.50stringent.max17.jaegerout_GRSK_MKP1592 

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/medstringent.min10_GRSK_MKP1592
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/medstringent.min10_GRSK_MKP1592
cp ../medstringent_GRSK_MKP1592/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.min10.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.min10.max17.jaegerout_GRSK_MKP1592 

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/40stringent.min10_GRSK_MKP1592
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/40stringent.min10_GRSK_MKP1592
cp ../medstringent_GRSK_MKP1592/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.40stringent.min10.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.40stringent.min10.max17.jaegerout_GRSK_MKP1592 

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/50stringent.min10_GRSK_MKP1592
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/50stringent.min10_GRSK_MKP1592
cp ../medstringent_GRSK_MKP1592/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.50stringent.min10.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.50stringent.min10.max17.jaegerout_GRSK_MKP1592 

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/30stringent.min10_GRSK_MKP1592
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/30stringent.min10_GRSK_MKP1592
cp ../medstringent_GRSK_MKP1592/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.30stringent.min10.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.30stringent.min10.max17.jaegerout_GRSK_MKP1592 
```

```bash
mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/medstringent_GRSK_MKP1592_ActuallyTheOtherOneWas1593
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/medstringent_GRSK_MKP1592_ActuallyTheOtherOneWas1593
cat > samples_ID.txt
```
```
LTJA_MKP990 Outgroup
PAJA_B20730 Outgroup
PAJA_USNM606730 Outgroup
POJA_MKP1559 pomarinus
POJA_4 pomarinus
POJA_IB2659 pomarinus
GRSK_MKP1592 skua
GRSK_MKP1593 xxx
CISK2 antarcticus
CISK55 antarcticus
CISK_3 antarcticus
CHSK_MKP2451 chilensis
ANSK7 maccormicki
ANSK01 maccormicki
ANSK8 maccormicki
Fratercula_arctica xxx
Alca_torda xxx
Uria_lomvia xxx
```
```bash
sed -i 's/ /\t/g' samples_ID.txt
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.jaegerout 

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/10stringent_GRSK_MKP1592_ActuallyTheOtherOneWas1593
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/10stringent_GRSK_MKP1592_ActuallyTheOtherOneWas1593
cp ../medstringent_GRSK_MKP1592_ActuallyTheOtherOneWas1593/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.10stringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.10stringent.max17.jaegerout_GRSK_MKP1592 

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/40stringent_GRSK_MKP1592_ActuallyTheOtherOneWas1593
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/40stringent_GRSK_MKP1592_ActuallyTheOtherOneWas1593
cp ../medstringent_GRSK_MKP1592_ActuallyTheOtherOneWas1593/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.40stringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.40stringent.max17.jaegerout_GRSK_MKP1592 

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/50stringent_GRSK_MKP1592_ActuallyTheOtherOneWas1593
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/50stringent_GRSK_MKP1592_ActuallyTheOtherOneWas1593
cp ../medstringent_GRSK_MKP1592_ActuallyTheOtherOneWas1593/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.50stringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.50stringent.max17.jaegerout_GRSK_MKP1592 

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/medstringent.min10_GRSK_MKP1592_ActuallyTheOtherOneWas1593
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/medstringent.min10_GRSK_MKP1592_ActuallyTheOtherOneWas1593
cp ../medstringent_GRSK_MKP1592_ActuallyTheOtherOneWas1593/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.min10.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.min10.max17.jaegerout_GRSK_MKP1592 

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/40stringent.min10_GRSK_MKP1592_ActuallyTheOtherOneWas1593
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/40stringent.min10_GRSK_MKP1592_ActuallyTheOtherOneWas1593
cp ../medstringent_GRSK_MKP1592_ActuallyTheOtherOneWas1593/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.40stringent.min10.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.40stringent.min10.max17.jaegerout_GRSK_MKP1592 

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/50stringent.min10_GRSK_MKP1592_ActuallyTheOtherOneWas1593
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/50stringent.min10_GRSK_MKP1592_ActuallyTheOtherOneWas1593
cp ../medstringent_GRSK_MKP1592_ActuallyTheOtherOneWas1593/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.50stringent.min10.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.50stringent.min10.max17.jaegerout_GRSK_MKP1592 

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/30stringent.min10_GRSK_MKP1592_ActuallyTheOtherOneWas1593
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/30stringent.min10_GRSK_MKP1592_ActuallyTheOtherOneWas1593
cp ../medstringent_GRSK_MKP1592_ActuallyTheOtherOneWas1593/samples_ID.txt .
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.30stringent.min10.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.auto.30stringent.min10.max17.jaegerout_GRSK_MKP1592 

```

### Appendix: testing robustness of genome scan to missing data filter
With the dataset `Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17.vcf.gz`, there were not enough sites to get good resolution across the genome, but I still checked the scans to ensure that the overall patterns were not different. This is the dataset used for the ABBA BABA test, which included a filter to remove sites with missing data for more than 3 samples. This filter was removed for the genome scan.  
```bash
mkdir -p  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/Dinvestigate_stringent
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/Dinvestigate_stringent

#I am investigating introgression betwen GRSK and CISK
echo -e "skua\tantarcticus\tmaccormicki" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.skua.antarcticus.maccormicki
echo -e "maccormicki\tantarcticus\tskua" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.maccormicki.antarcticus.skua
echo -e "chilensis\tantarcticus\tskua" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.chilensis.antarcticus.skua

#I am investigating introgression betwen GRSK and POJA
echo -e "maccormicki\tskua\tpomarinus" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.maccormicki.skua.pomarinus
echo -e "chilensis\tskua\tpomarinus" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.chilensis.skua.pomarinus
echo -e "antarcticus\tskua\tpomarinus" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.antarcticus.skua.pomarinus

#I am investigating introgression betwen ANSK and CISK
echo -e "chilensis\tantarcticus\tmaccormicki" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.chilensis.antarcticus.maccormicki
echo -e "maccormicki\tantarcticus\tchilensis" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.maccormicki.antarcticus.chilensis

```
```
maccormicki	skua	pomarinus
D=0.0837264
f_d=0.0217562	355.236/16328
f_dM=0.0145447	355.236/24423.8
real	2m47.020s
user	2m44.736s
sys	0m2.280s

skua	antarcticus	maccormicki
D=0.252017
f_d=0.122065	1398.24/11454.9
f_dM=0.0678238	1398.24/20615.8
real	2m44.652s
user	2m42.076s
sys	0m2.556s

antarcticus	skua	pomarinus
D=0.0820334
f_d=0.0200329	307.031/15326.3
f_dM=0.0128121	307.031/23964.2
real	2m44.596s
user	2m42.108s
sys	0m2.480s

chilensis	skua	pomarinus
D=0.0783769
f_d=0.0206462	291.5/14118.8
f_dM=0.0125042	291.5/23312.1
real	2m46.995s
user	2m44.480s
sys	0m2.516s

chilensis	antarcticus	skua
D=0.218894
f_d=0.0862421	1010.26/11714.2
f_dM=0.0522566	1010.26/19332.7
real	2m45.215s
user	2m42.708s
sys	0m2.492s

maccormicki	antarcticus	skua
D=0.229172
f_d=0.0894643	1233.81/13791.1
f_dM=0.0606163	1233.81/20354.5
real	2m45.119s
user	2m42.428s
sys	0m2.680s

```

The overall pattern produced by the scan was the same (see the rmd file).  

### Appendix: testing robustness of genome scan to genotype depth filter
In this dataset, the filters are the same as used for the ABBA BABA test above (including missing data filter), but I filtered to allow genotype calls with a minimum of 4x depth (rather than 6x depth). This increases the number of sites availabe for the scan, increasing resolution across the genome. The results look very similar (mostly overlapping lines) to the results from the dataset with no missing data filter but a min 6x depth.  
```bash
mkdir -p  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/Dinvestigate_min4
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/Dinvestigate_min4

#I am investigating introgression betwen GRSK and CISK
echo -e "skua\tantarcticus\tmaccormicki" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.min4.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.skua.antarcticus.maccormicki
echo -e "maccormicki\tantarcticus\tskua" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.min4.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.maccormicki.antarcticus.skua
echo -e "chilensis\tantarcticus\tskua" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.min4.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.chilensis.antarcticus.skua

#I am investigating introgression betwen GRSK and POJA
echo -e "maccormicki\tskua\tpomarinus" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.min4.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.maccormicki.skua.pomarinus
echo -e "chilensis\tskua\tpomarinus" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.min4.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.chilensis.skua.pomarinus
echo -e "antarcticus\tskua\tpomarinus" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.min4.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.antarcticus.skua.pomarinus

#I am investigating introgression betwen ANSK and CISK
echo -e "chilensis\tantarcticus\tmaccormicki" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.min4.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.chilensis.antarcticus.maccormicki
echo -e "maccormicki\tantarcticus\tchilensis" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.min4.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.maccormicki.antarcticus.chilensis

```
```
skua	antarcticus	maccormicki
D=0.26576
f_d=0.134096	3814.26/28444.2
f_dM=0.0755446	3814.26/50490.1
real	6m45.545s
user	6m39.572s
sys	0m5.976s

chilensis	antarcticus	maccormicki
D=0.18445
f_d=0.089979	2142.65/23812.7
f_dM=0.0496548	2142.65/43150.8
real	6m46.889s
user	6m40.700s
sys	0m6.192s

maccormicki	antarcticus	chilensis
D=0.214019
f_d=0.0884943	2579.67/29150.7
f_dM=0.0595466	2579.67/43321.8
real	6m48.252s
user	6m42.256s
sys	0m5.984s

maccormicki	antarcticus	skua
D=0.229632
f_d=0.0937318	3141.19/33512.5
f_dM=0.0633498	3141.19/49584.8
real	6m48.599s
user	6m42.464s
sys	0m6.120s

chilensis	antarcticus	skua
D=0.243347
f_d=0.0998981	2888.71/28916.5
f_dM=0.0604796	2888.71/47763.4
real	6m46.808s
user	6m40.804s
sys	0m6.004s

maccormicki	skua	pomarinus
D=0.056815
f_d=0.0154439	612.312/39647.5
f_dM=0.0102188	612.312/59920.3
real	6m47.018s
user	6m40.716s
sys	0m6.268s

antarcticus	skua	pomarinus
D=0.0451363
f_d=0.0117229	434.01/37022.5
f_dM=0.00738581	434.01/58762.7
real	6m46.176s
user	6m40.060s
sys	0m6.116s

chilensis	skua	pomarinus
D=0.0597965
f_d=0.0166014	577.958/34813.9
f_dM=0.00994733	577.958/58101.9
real	6m49.194s
user	6m42.664s
sys	0m6.532s

```

### Appendix: testing robustness of genome scan to relaxed filters  
This dataset (Stercorarius.RAZObowtie.merged.filteredSNPs.auto.max17.vcf.gz) has very relaxed filters: MQ and qual min 20 (rather than 25), no genotype quality filter, and min depth 4 (rather than 6). It gives good resolution across the genome, but some of the relaxed filters may be unnecessary (possibly adding noise instead of adding good data). The overall pattern is qualitatively the same as the more stringent datasets.  
```bash
mkdir -p  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/Dinvestigate_NotStringent
cd  /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/Dinvestigate_NotStringent

#I am investigating introgression betwen GRSK and CISK
echo -e "skua\tantarcticus\tmaccormicki" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.skua.antarcticus.maccormicki
echo -e "maccormicki\tantarcticus\tskua" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.maccormicki.antarcticus.skua
echo -e "chilensis\tantarcticus\tskua" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.chilensis.antarcticus.skua

#I am investigating introgression betwen GRSK and POJA
echo -e "maccormicki\tskua\tpomarinus" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.maccormicki.skua.pomarinus
echo -e "chilensis\tskua\tpomarinus" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.chilensis.skua.pomarinus
echo -e "antarcticus\tskua\tpomarinus" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.antarcticus.skua.pomarinus

#I am investigating introgression betwen ANSK and CISK
echo -e "chilensis\tantarcticus\tmaccormicki" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.chilensis.antarcticus.maccormicki
echo -e "maccormicki\tantarcticus\tchilensis" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.maccormicki.antarcticus.chilensis

#I am investigating introgression betwen PAJA and a skua like POJA
echo -e "longicaudus\tparasiticus\tpomarinus" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.longicaudus.parasiticus.pomarinus

#I am investigating introgression betwen PAJA and a skua like GRSK (verify the POJA result)
echo -e "longicaudus\tparasiticus\tskua" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.longicaudus.parasiticus.skua

#I am investigating introgression betwen PAJA and POJA
echo -e "maccormicki\tpomarinus\tparasiticus" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.maccormicki.pomarinus.parasiticus
echo -e "antarcticus\tpomarinus\tparasiticus" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.antarcticus.pomarinus.parasiticus

#I am investigating between CISK and POJA
echo -e "maccormicki\tantarcticus\tpomarinus" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.maccormicki.antarcticus.pomarinus
echo -e "chilensis\tantarcticus\tpomarinus" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.max17.vcf.gz ../samples_ID.txt test_trios.txt -n auto.chilensis.antarcticus.pomarinus


#repeat for chrZ with a dataset with relaxed filters

#I am investigating introgression betwen GRSK and CISK
echo -e "skua\tantarcticus\tmaccormicki" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.lax.max17.vcf.gz ../samples_ID.txt test_trios.txt -n chrZ.skua.antarcticus.maccormicki
echo -e "maccormicki\tantarcticus\tskua" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.lax.max17.vcf.gz ../samples_ID.txt test_trios.txt -n chrZ.maccormicki.antarcticus.skua
echo -e "chilensis\tantarcticus\tskua" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.lax.max17.vcf.gz ../samples_ID.txt test_trios.txt -n chrZ.chilensis.antarcticus.skua

#I am investigating introgression betwen GRSK and POJA
echo -e "maccormicki\tskua\tpomarinus" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.lax.max17.vcf.gz ../samples_ID.txt test_trios.txt -n chrZ.maccormicki.skua.pomarinus
echo -e "chilensis\tskua\tpomarinus" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.lax.max17.vcf.gz ../samples_ID.txt test_trios.txt -n chrZ.chilensis.skua.pomarinus
echo -e "antarcticus\tskua\tpomarinus" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.lax.max17.vcf.gz ../samples_ID.txt test_trios.txt -n chrZ.antarcticus.skua.pomarinus

#I am investigating introgression betwen ANSK and CISK
echo -e "chilensis\tantarcticus\tmaccormicki" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.lax.max17.vcf.gz ../samples_ID.txt test_trios.txt -n chrZ.chilensis.antarcticus.maccormicki
echo -e "maccormicki\tantarcticus\tchilensis" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.lax.max17.vcf.gz ../samples_ID.txt test_trios.txt -n chrZ.maccormicki.antarcticus.chilensis

#I am investigating introgression betwen PAJA and a skua like POJA
echo -e "longicaudus\tparasiticus\tpomarinus" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.lax.max17.vcf.gz ../samples_ID.txt test_trios.txt -n chrZ.longicaudus.parasiticus.pomarinus

#I am investigating introgression betwen PAJA and a skua like GRSK (verify the POJA result)
echo -e "longicaudus\tparasiticus\tskua" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.lax.max17.vcf.gz ../samples_ID.txt test_trios.txt -n chrZ.longicaudus.parasiticus.skua

#I am investigating introgression betwen PAJA and POJA
echo -e "maccormicki\tpomarinus\tparasiticus" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.lax.max17.vcf.gz ../samples_ID.txt test_trios.txt -n chrZ.maccormicki.pomarinus.parasiticus
echo -e "antarcticus\tpomarinus\tparasiticus" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.lax.max17.vcf.gz ../samples_ID.txt test_trios.txt -n chrZ.antarcticus.pomarinus.parasiticus

#I am investigating between CISK and POJA
echo -e "maccormicki\tantarcticus\tpomarinus" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.lax.max17.vcf.gz ../samples_ID.txt test_trios.txt -n chrZ.maccormicki.antarcticus.pomarinus
echo -e "chilensis\tantarcticus\tpomarinus" > test_trios.txt #The \t means tab, this makes a tab delimited file.
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dinvestigate -w 50,50 ../Stercorarius.RAZObowtie.merged.filteredSNPs.chrZ.lax.max17.vcf.gz ../samples_ID.txt test_trios.txt -n chrZ.chilensis.antarcticus.pomarinus


#combine the autosomal data with the chrZ data
grep -v "chrZ" skua_antarcticus_maccormicki_localFstats_auto.skua.antarcticus.maccormicki_50_50.txt > temp
tail -n +2 skua_antarcticus_maccormicki_localFstats_chrZ.skua.antarcticus.maccormicki_50_50.txt | cat temp - > skua_antarcticus_maccormicki_localFstats_all.skua.antarcticus.maccormicki_50_50.txt

grep -v "chrZ" longicaudus_parasiticus_skua_localFstats_auto.longicaudus.parasiticus.skua_50_50.txt > temp
tail -n +2 longicaudus_parasiticus_skua_localFstats_chrZ.longicaudus.parasiticus.skua_50_50.txt | cat temp - > longicaudus_parasiticus_skua_localFstats_all.longicaudus.parasiticus.skua_50_50.txt

grep -v "chrZ" antarcticus_pomarinus_parasiticus_localFstats_auto.antarcticus.pomarinus.parasiticus_50_50.txt > temp
tail -n +2 antarcticus_pomarinus_parasiticus_localFstats_chrZ.antarcticus.pomarinus.parasiticus_50_50.txt | cat temp - > antarcticus_pomarinus_parasiticus_localFstats_all.antarcticus.pomarinus.parasiticus_50_50.txt

grep -v "chrZ" maccormicki_antarcticus_chilensis_localFstats_auto.maccormicki.antarcticus.chilensis_50_50.txt > temp
tail -n +2 maccormicki_antarcticus_chilensis_localFstats_chrZ.maccormicki.antarcticus.chilensis_50_50.txt | cat temp - > maccormicki_antarcticus_chilensis_localFstats_all.maccormicki.antarcticus.chilensis_50_50.txt

grep -v "chrZ" antarcticus_skua_pomarinus_localFstats_auto.antarcticus.skua.pomarinus_50_50.txt > temp
tail -n +2 antarcticus_skua_pomarinus_localFstats_chrZ.antarcticus.skua.pomarinus_50_50.txt | cat temp - > antarcticus_skua_pomarinus_localFstats_all.antarcticus.skua.pomarinus_50_50.txt

grep -v "chrZ" maccormicki_antarcticus_pomarinus_localFstats_auto.maccormicki.antarcticus.pomarinus_50_50.txt > temp
tail -n +2 maccormicki_antarcticus_pomarinus_localFstats_chrZ.maccormicki.antarcticus.pomarinus_50_50.txt | cat temp - > maccormicki_antarcticus_pomarinus_localFstats_all.maccormicki.antarcticus.pomarinus_50_50.txt

grep -v "chrZ" chilensis_antarcticus_maccormicki_localFstats_auto.chilensis.antarcticus.maccormicki_50_50.txt > temp
tail -n +2 chilensis_antarcticus_maccormicki_localFstats_chrZ.chilensis.antarcticus.maccormicki_50_50.txt | cat temp - > chilensis_antarcticus_maccormicki_localFstats_all.chilensis.antarcticus.maccormicki_50_50.txt

grep -v "chrZ" maccormicki_antarcticus_skua_localFstats_auto.maccormicki.antarcticus.skua_50_50.txt > temp
tail -n +2 maccormicki_antarcticus_skua_localFstats_chrZ.maccormicki.antarcticus.skua_50_50.txt | cat temp - > maccormicki_antarcticus_skua_localFstats_all.maccormicki.antarcticus.skua_50_50.txt

grep -v "chrZ" chilensis_antarcticus_pomarinus_localFstats_auto.chilensis.antarcticus.pomarinus_50_50.txt > temp
tail -n +2 chilensis_antarcticus_pomarinus_localFstats_chrZ.chilensis.antarcticus.pomarinus_50_50.txt | cat temp - > chilensis_antarcticus_pomarinus_localFstats_all.chilensis.antarcticus.pomarinus_50_50.txt

grep -v "chrZ" maccormicki_pomarinus_parasiticus_localFstats_auto.maccormicki.pomarinus.parasiticus_50_50.txt > temp
tail -n +2 maccormicki_pomarinus_parasiticus_localFstats_chrZ.maccormicki.pomarinus.parasiticus_50_50.txt | cat temp - > maccormicki_pomarinus_parasiticus_localFstats_all.maccormicki.pomarinus.parasiticus_50_50.txt

grep -v "chrZ" chilensis_antarcticus_skua_localFstats_auto.chilensis.antarcticus.skua_50_50.txt > temp
tail -n +2 chilensis_antarcticus_skua_localFstats_chrZ.chilensis.antarcticus.skua_50_50.txt | cat temp - > chilensis_antarcticus_skua_localFstats_all.chilensis.antarcticus.skua_50_50.txt

grep -v "chrZ" maccormicki_skua_pomarinus_localFstats_auto.maccormicki.skua.pomarinus_50_50.txt > temp
tail -n +2 maccormicki_skua_pomarinus_localFstats_chrZ.maccormicki.skua.pomarinus_50_50.txt | cat temp - > maccormicki_skua_pomarinus_localFstats_all.maccormicki.skua.pomarinus_50_50.txt

grep -v "chrZ" chilensis_skua_pomarinus_localFstats_auto.chilensis.skua.pomarinus_50_50.txt > temp
tail -n +2 chilensis_skua_pomarinus_localFstats_chrZ.chilensis.skua.pomarinus_50_50.txt | cat temp - > chilensis_skua_pomarinus_localFstats_all.chilensis.skua.pomarinus_50_50.txt

grep -v "chrZ" skua_antarcticus_maccormicki_localFstats_auto.skua.antarcticus.maccormicki_50_50.txt > temp
tail -n +2 skua_antarcticus_maccormicki_localFstats_chrZ.skua.antarcticus.maccormicki_50_50.txt | cat temp - > skua_antarcticus_maccormicki_localFstats_all.skua.antarcticus.maccormicki_50_50.txt

grep -v "chrZ" longicaudus_parasiticus_pomarinus_localFstats_auto.longicaudus.parasiticus.pomarinus_50_50.txt > temp
tail -n +2 longicaudus_parasiticus_pomarinus_localFstats_chrZ.longicaudus.parasiticus.pomarinus_50_50.txt | cat temp - > longicaudus_parasiticus_pomarinus_localFstats_all.longicaudus.parasiticus.pomarinus_50_50.txt
```

```
skua	antarcticus	maccormicki
D=0.252017
f_d=0.122065	1398.24/11454.9
f_dM=0.0678238	1398.24/20615.8

antarcticus	pomarinus	parasiticus
D=0.0915967
f_d=0.00173729	229.281/131976
f_dM=0.00151304	229.281/151537

maccormicki	antarcticus	skua
D=0.229172
f_d=0.0894643	1233.81/13791.1
f_dM=0.0606163	1233.81/20354.5

chilensis	antarcticus	maccormicki
D=0.162656
f_d=0.0769025	738.035/9597.02
f_dM=0.0422605	738.035/17463.9

maccormicki	skua	pomarinus
D=0.0837264
f_d=0.0217562	355.236/16328
f_dM=0.0145447	355.236/24423.8

longicaudus	parasiticus	pomarinus
D=-0.0137241
f_d=-0.000859885	-129.885/151050
f_dM=-0.000603483	-129.885/215226

longicaudus	parasiticus	skua
D=-0.0132369
f_d=-0.000774594	-123.375/159277
f_dM=-0.000554113	-123.375/222653

antarcticus	pomarinus	parasiticus
D=0.0915967
f_d=0.00173729	229.281/131976
f_dM=0.00151304	229.281/151537

maccormicki	pomarinus	parasiticus
D=0.0920555
f_d=0.00174796	230.74/132005
f_dM=0.00154257	230.74/149581

chilensis	skua	pomarinus
D=0.0783769
f_d=0.0206462	291.5/14118.8
f_dM=0.0125042	291.5/23312.1

chilensis	antarcticus	skua
D=0.218894
f_d=0.0862421	1010.26/11714.2
f_dM=0.0522566	1010.26/19332.7

maccormicki	antarcticus	pomarinus
D=0.0145712
f_d=0.00317275	47.9271/15105.9
f_dM=0.0022016	47.9271/21769.2

antarcticus	skua	pomarinus
D=0.0820334
f_d=0.0200329	307.031/15326.3
f_dM=0.0128121	307.031/23964.2

```

These Dinvestigate scans gave qualitatively very similar results to the published dataset, improving confidence that the patterns are not sensitive to the chosen bioinformatic filters.  


## Appendix: Check robustness of genic vs intergenic comparison  

### Exons only  
I am also going to restrict the analysis to just the exons (i.e., exclude introns) to ensure that patterns are consistent. Note that exons make up a rather small portion of the genome and have lower sequence diversity, so there are few SNPs in the exons, giving very little power for the ABBA BABA test. Despite having few SNPs, patterns are very similar to those including introns.  

First, using Alcids as the outgroup: 
```bash
mkdir -p  /home/1_Else/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/Alcid_out_exons
cd  /home/1_Else/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/Alcid_out_exons
cat > samples_ID.txt
```
```
LTJA_MKP990 longicaudus
PAJA_B20730 parasiticus
PAJA_USNM606730 parasiticus
POJA_MKP1559 pomarinus
POJA_4 pomarinus
POJA_IB2659 pomarinus
GRSK_MKP1592 skua
GRSK_MKP1593 skua
CISK2 antarcticus
CISK55 antarcticus
CISK_3 antarcticus
CHSK_MKP2451 chilensis
ANSK7 maccormicki
ANSK01 maccormicki
ANSK8 maccormicki
Fratercula_arctica xxx
Alca_torda Outgroup
Uria_lomvia Outgroup
```
```bash
sed -i 's/ /\t/g' samples_ID.txt
echo "(Outgroup,((parasiticus,longicaudus),(pomarinus,(skua,(maccormicki,(antarcticus,chilensis))))));" > skuas.nwk

#filter to keep sites in exons, then do abba baba
tabix ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17.vcf.gz
tabix -R /home/1_Else/0_GENOMES5/Stercorarius/0_Reference_genome/0.3_annotation/liftoff/Stercorarius_Stercorarius_liftoff.exons.bed -h ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17.vcf.gz > ../Stercorarius.RAZObowtie.merged.filteredSNPs.exons.stringent.max17.vcf.gz
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.exons.stringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.exons.stringent.max17 #
```

Now I am going to test a few different strategies for missing data filters, in which I filter for missing data only amongst the focal quartet (i.e., ignoring all other populations).  
```bash
#filter for missing data only within the focal quartet
/home/0_PROGRAMS/bcftools-1.14/bcftools view --samples ^Fratercula_arctica,LTJA_MKP990,PAJA_B20730,PAJA_USNM606730,CHSK_MKP2451,CISK_3,CISK55,CISK2 ../../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.vcf.gz | /home/0_PROGRAMS/bcftools-1.14/bcftools view -e 'F_MISSING > 0.2' -Oz -o Stercorarius.RAZObowtie.merged.filteredSNPs.pojagrskansk.max17.vcf.gz 
/home/0_PROGRAMS/bcftools-1.14/bcftools view --samples ^Fratercula_arctica,LTJA_MKP990,PAJA_B20730,PAJA_USNM606730,CHSK_MKP2451,ANSK8,ANSK01,ANSK7 ../../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.vcf.gz | /home/0_PROGRAMS/bcftools-1.14/bcftools view -e 'F_MISSING > 0.2' -Oz -o Stercorarius.RAZObowtie.merged.filteredSNPs.pojagrskcisk.max17.vcf.gz 
/home/0_PROGRAMS/bcftools-1.14/bcftools view --samples ^Fratercula_arctica,LTJA_MKP990,PAJA_B20730,PAJA_USNM606730,ANSK7,ANSK01,ANSK8,CISK_3,CISK55,CISK2 ../../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.vcf.gz | /home/0_PROGRAMS/bcftools-1.14/bcftools view -e 'F_MISSING > 0.2' -Oz -o Stercorarius.RAZObowtie.merged.filteredSNPs.pojagrskchsk.max17.vcf.gz 
#index the vcf files
tabix Stercorarius.RAZObowtie.merged.filteredSNPs.pojagrskchsk.max17.vcf.gz
tabix Stercorarius.RAZObowtie.merged.filteredSNPs.pojagrskcisk.max17.vcf.gz
tabix Stercorarius.RAZObowtie.merged.filteredSNPs.pojagrskansk.max17.vcf.gz

#filter VCF files to keep only exons
tabix -R /home/1_Else/0_GENOMES5/Stercorarius/0_Reference_genome/0.3_annotation/liftoff/Stercorarius_Stercorarius_liftoff.exons.bed -h Stercorarius.RAZObowtie.merged.filteredSNPs.pojagrskcisk.max17.vcf.gz > ../../Stercorarius.RAZObowtie.merged.exons.pojagrskcisk.max17.vcf.gz
tabix -R /home/1_Else/0_GENOMES5/Stercorarius/0_Reference_genome/0.3_annotation/liftoff/Stercorarius_Stercorarius_liftoff.exons.bed -h Stercorarius.RAZObowtie.merged.filteredSNPs.pojagrskansk.max17.vcf.gz > ../../Stercorarius.RAZObowtie.merged.exons.pojagrskansk.max17.vcf.gz
tabix -R /home/1_Else/0_GENOMES5/Stercorarius/0_Reference_genome/0.3_annotation/liftoff/Stercorarius_Stercorarius_liftoff.exons.bed -h Stercorarius.RAZObowtie.merged.filteredSNPs.pojagrskchsk.max17.vcf.gz > ../../Stercorarius.RAZObowtie.merged.exons.pojagrskchsk.max17.vcf.gz

#run the ABBA BABA test
grep -e "CISK" -e "GRSK" -e "POJA" -e "Alca" -e "Uria" samples_ID.txt > pojagrskcisk.samples_ID.txt
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../../Stercorarius.RAZObowtie.merged.exons.pojagrskcisk.max17.vcf.gz pojagrskcisk.samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.exons.pojagrskcisk.max17 #
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../../Stercorarius.RAZObowtie.merged.exons.pojagrskansk.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.exons.pojagrskansk.max17 #
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../../Stercorarius.RAZObowtie.merged.exons.pojagrskchsk.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.exons.pojagrskchsk.max17 #


#repeat, but filter for missing data with more stringent missing data filter (max 10% vs 20%)
/home/0_PROGRAMS/bcftools-1.14/bcftools view --samples ^Fratercula_arctica,LTJA_MKP990,PAJA_B20730,PAJA_USNM606730,CHSK_MKP2451,CISK_3,CISK55,CISK2 ../../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.vcf.gz | /home/0_PROGRAMS/bcftools-1.14/bcftools view -e 'F_MISSING > 0.1' -Oz -o Stercorarius.RAZObowtie.merged.filteredSNPs.stringent.pojagrskansk.max17.vcf.gz #max 1 in 10 missing
/home/0_PROGRAMS/bcftools-1.14/bcftools view --samples ^Fratercula_arctica,LTJA_MKP990,PAJA_B20730,PAJA_USNM606730,CHSK_MKP2451,ANSK8,ANSK01,ANSK7 ../../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.vcf.gz | /home/0_PROGRAMS/bcftools-1.14/bcftools view -e 'F_MISSING > 0.1' -Oz -o Stercorarius.RAZObowtie.merged.filteredSNPs.stringent.pojagrskcisk.max17.vcf.gz #max 1 in 10 missing
/home/0_PROGRAMS/bcftools-1.14/bcftools view --samples ^Fratercula_arctica,LTJA_MKP990,PAJA_B20730,PAJA_USNM606730,ANSK7,ANSK01,ANSK8,CISK_3,CISK55,CISK2 ../../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.vcf.gz | /home/0_PROGRAMS/bcftools-1.14/bcftools view -e 'F_MISSING > 0.1' -Oz -o Stercorarius.RAZObowtie.merged.filteredSNPs.stringent.pojagrskchsk.max17.vcf.gz #max 1 in 8 missing
#index the vcf files
tabix Stercorarius.RAZObowtie.merged.filteredSNPs.stringent.pojagrskchsk.max17.vcf.gz
tabix Stercorarius.RAZObowtie.merged.filteredSNPs.stringent.pojagrskcisk.max17.vcf.gz
tabix Stercorarius.RAZObowtie.merged.filteredSNPs.stringent.pojagrskansk.max17.vcf.gz
#filter VCF files to keep only exons
tabix -R /home/1_Else/0_GENOMES5/Stercorarius/0_Reference_genome/0.3_annotation/liftoff/Stercorarius_Stercorarius_liftoff.exons.bed -h Stercorarius.RAZObowtie.merged.filteredSNPs.stringent.pojagrskcisk.max17.vcf.gz > ../../Stercorarius.RAZObowtie.merged.exons.stringent.pojagrskcisk.max17.vcf.gz
tabix -R /home/1_Else/0_GENOMES5/Stercorarius/0_Reference_genome/0.3_annotation/liftoff/Stercorarius_Stercorarius_liftoff.exons.bed -h Stercorarius.RAZObowtie.merged.filteredSNPs.stringent.pojagrskansk.max17.vcf.gz > ../../Stercorarius.RAZObowtie.merged.exons.stringent.pojagrskansk.max17.vcf.gz
tabix -R /home/1_Else/0_GENOMES5/Stercorarius/0_Reference_genome/0.3_annotation/liftoff/Stercorarius_Stercorarius_liftoff.exons.bed -h Stercorarius.RAZObowtie.merged.filteredSNPs.stringent.pojagrskchsk.max17.vcf.gz > ../../Stercorarius.RAZObowtie.merged.exons.stringent.pojagrskchsk.max17.vcf.gz
#run the ABBA BABA test
grep -e "CISK" -e "GRSK" -e "POJA" -e "Alca" -e "Uria" samples_ID.txt > pojagrskcisk.samples_ID.txt
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../../Stercorarius.RAZObowtie.merged.exons.stringent.pojagrskcisk.max17.vcf.gz pojagrskcisk.samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.exons.stringent.pojagrskcisk.max17 #

```

Repeat with jaeger as the outgroup.  
```bash
mkdir -p  /home/1_Else/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/jaegerout_exons
cd  /home/1_Else/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/jaegerout_exons
cat > samples_ID.txt
```
```
LTJA_MKP990 Outgroup
PAJA_B20730 Outgroup
PAJA_USNM606730 Outgroup
POJA_MKP1559 pomarinus
POJA_4 pomarinus
POJA_IB2659 pomarinus
GRSK_MKP1592 skua
GRSK_MKP1593 skua
CISK2 antarcticus
CISK55 antarcticus
CISK_3 antarcticus
CHSK_MKP2451 chilensis
ANSK7 maccormicki
ANSK01 maccormicki
ANSK8 maccormicki
Fratercula_arctica xxx
Alca_torda xxx
Uria_lomvia xxx
```
```bash
sed -i 's/ /\t/g' samples_ID.txt
echo "(Outgroup,(pomarinus,(skua,(maccormicki,(antarcticus,chilensis)))));" > skuas.nwk

time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../Stercorarius.RAZObowtie.merged.filteredSNPs.exons.stringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.exons.stringent.max17 #
```

Patterns are consistent with results that include the introns (i.e., D and f are lower that the estimates from intergenic regions).  

### No missing data filter  
I am going to check the results with the medstringent dataset to ensure that patterns are not sensitive to the missing data filter.  
```bash
#no missing data filter, intergenic
mkdir -p /home/1_Else/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/jaegerout_notgenes/medstringent && cd /home/1_Else/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/jaegerout_notgenes/medstringent
cp ../samples_ID.txt .; cp ../skuas.nwk .
/home/0_PROGRAMS/bcftools-1.14/bcftools view -T ^/home/1_Else/0_GENOMES5/Stercorarius/0_Reference_genome/0.3_annotation/liftoff/Stercorarius_Stercorarius_liftoff.genes.bed -O z ../../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.vcf.gz > ../../Stercorarius.RAZObowtie.merged.filteredSNPs.intergenic.medstringent.max17.vcf.gz
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../../Stercorarius.RAZObowtie.merged.filteredSNPs.intergenic.medstringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.intergenic.medstringent.max17 #

#no missing data filter, exons, Alcid outgroup
mkdir -p /home/1_Else/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/Alcid_out_exons/medstringent && cd /home/1_Else/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/Alcid_out_exons/medstringent
cp ../samples_ID.txt .; cp ../skuas.nwk .
tabix ../../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.vcf.gz
tabix -R /home/1_Else/0_GENOMES5/Stercorarius/0_Reference_genome/0.3_annotation/liftoff/Stercorarius_Stercorarius_liftoff.exons.bed -h ../../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.medstringent.max17.vcf.gz > ../../Stercorarius.RAZObowtie.merged.filteredSNPs.exons.medstringent.max17.vcf.gz
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../../Stercorarius.RAZObowtie.merged.filteredSNPs.exons.medstringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.exons.medstringent.max17 #

#no missing data filter, exons, jaeger outgroup
mkdir -p /home/1_Else/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/jaegerout_exons/medstringent && cd /home/1_Else/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.1d_Dsuite_ABBABABA/jaegerout_exons/medstringent
cp ../samples_ID.txt .; cp ../skuas.nwk .
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios ../../Stercorarius.RAZObowtie.merged.filteredSNPs.exons.medstringent.max17.vcf.gz samples_ID.txt -t skuas.nwk -n Stercorarius.RAZObowtie.merged.filteredSNPs.exons.medstringent.max17 #

#/home/0_PROGRAMS/bedtools intersect -v -a ../Stercorarius.RAZObowtie.merged.filteredSNPs.auto.stringent.max17.vcf.gz -b /home/1_Else/0_GENOMES5/Stercorarius/0_Reference_genome/0.3_annotation/liftoff/Stercorarius_Stercorarius_liftoff.genes.bed -wa > ../Stercorarius.RAZObowtie.merged.filteredSNPs.intergenic.stringent.max17.vcf

```
Results are consistent.  