### Overview
This section details how to generate the dataset of genetic distances estimated in 17-kb blocks across the genome.  

### Required Input
This step requires the set of 17-kb sequence alignments that were generated in step `1.6.2d_phylogenetic_blocks`.  

### Output
The output is a set of genetic distance estimates between every sample in 17-kb windows of the genome. These are analyzed in the file `3.2.2d_mtDNA_vs_nuclear_distances`.    

## Generate distance matrices

Now that we have a set of blocks, we will generate distance matrices for each one, with the same pipeline used in step 3.2.1c.  
```bash
mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.2.2d_mtDNA_vs_nuclear_distances/17kb_blocks
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.2.2d_mtDNA_vs_nuclear_distances/17kb_blocks

#make folders to hold the distance matrices
mkdir -p distance_matrices

#make a list of the chunks to loop through
find /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/1.6.2d_phylogenetic_blocks/strict/17kb_blocks/ | tail -n +2 > 17kbchunks.txt #removes the first line because it is just the directory name

#calculate Kimura 2-parameter distances for all chunks by looping through the list of chunks. Make sure the name of the chunk is in the output name
time cat 17kbchunks.txt | sed 's;/home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/1.6.2d_phylogenetic_blocks/strict/17kb_blocks/;;g' | parallel --jobs 29 /home/0_PROGRAMS/EMBOSS-6.6.0/bin/distmat -sequence /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/1.6.2d_phylogenetic_blocks/strict/17kb_blocks/{1} -nucmethod 2 -stdout -outfile distance_matrices/{1}.distmat

#now loop through the chunks, and make a new file where you will print the name of each chunk folllowed by their distance matrix (just the 17 lines (I have 17 samples) after the first sample, which in my case is ANSK7 (but the N is turned into '-')). I am doing this in serial so that I can control the order in which each chunk finishes, even though it is slower.
time cat 17kbchunks.txt | sed 's;/home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/1.6.2d_phylogenetic_blocks/strict/17kb_blocks/;;g' | while read chunk ; do printf "$chunk" >> 17kbwindows.distmat ; cat distance_matrices/"$chunk".distmat | grep -A 16 "Alca" | paste -s -d ' ' >> 17kbwindows.distmat ; done

#We might want to filter by missingness, so let's count missing data in each window.
#create a blank file to store our data
touch chunk.missingness
#loop through each chunk of the genome, count how many N's are on each line, and record this. All samples must be in the same order in all nexus files for this to work!
time cat 17kbchunks.txt | while read chunk ; do rm -f temp ; echo "$chunk" >> temp ; tail -n +6 "$chunk" | head -n -2 | tr -d -c '\-\n' | awk '{ print length; }' >> temp ; paste temp chunk.missingness > temp3 && mv temp3 chunk.missingness ; done #76m
```

Now I will analyze this data in `R`; I wrote my pipeline in a `.Rmd` file: `3.2.2d_mtDNA_vs_nuclear_distances.Rmd`, which is also rendered into an html file with results (`3.2.2d_mtDNA_vs_nuclear_distances.html`.  







# Appendix: dataset c

I also wanted to repeat the process with the data mapped to *Stercorarius parasiticus* to better understand how reference bias would affect the results. By using the *Stercorarius* reference genome, the two small jaegers (*S. parasiticus* and *S. longicaudus*) may have different error rates than the other samples due to differences in phylogenetic distance. Heterozygous samples will be biased towards the *S. parasiticus* allele, which could have subtle but important affects on the patterns. Using the more distant *Alca torda* genome may add a bit more noise, but at least it will be biased in a way that is less likely to affect interpretation of the data or lead to incorrect comclusions.  

```bash
#enter the folder containing the sequence alignments that we made previously
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/1.6.2c_phylogenetic_blocks/strict
conda activate basic #has ruby installed
#create phylogenetic blocks (as explained in the Phylogenetic Blocks page of this repository). Takes up to 2m45 per scaffold.
cat ../scaff_names | parallel time ruby /home/0_PROGRAMS/mmatschiner/extract_blocks.rb {1}.noN.fa 17kb_blocks 16707 0.2
ls 17kb_blocks | wc -l #count how many blocks were made
#67407

mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.2c_D3_distancematrix/17kb_blocks
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.2c_D3_distancematrix/17kb_blocks

#make folders to hold the distance matrices
mkdir -p distance_matrices

#make a list of the chunks to loop through
find /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/1.6.2c_phylogenetic_blocks/strict/17kb_blocks/ | tail -n +2 > 17kbchunks.txt #removes the first line because it is just the directory name

#calculate Kimura 2-parameter distances for all chunks by looping through the list of chunks. Make sure the name of the chunk is in the output name
time cat 17kbchunks.txt | sed 's;/home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/1.6.2c_phylogenetic_blocks/strict/17kb_blocks/;;g' | parallel --jobs 29 /home/0_PROGRAMS/EMBOSS-6.6.0/bin/distmat -sequence /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/1.6.2c_phylogenetic_blocks/strict/17kb_blocks/{1} -nucmethod 2 -stdout -outfile distance_matrices/{1}.distmat

#now loop through the chunks, and make a new file where you will print the name of each chunk folllowed by their distance matrix (just the 17 lines (I have 17 samples) after the first sample, which in my case is ANSK7 (but the N is turned into '-')). I am doing this in serial so that I can control the order in which each chunk finishes, even though it is slower.
time cat 17kbchunks.txt | sed 's;/home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/1.6.2c_phylogenetic_blocks/strict/17kb_blocks/;;g' | while read chunk ; do printf "$chunk" >> 17kbwindows.distmat ; cat distance_matrices/"$chunk".distmat | grep -A 16 "Alca" | paste -s -d ' ' >> 17kbwindows.distmat ; done

#We might want to filter by missingness, so let's count missing data in each window.
#create a blank file to store our data
touch chunk.missingness
#loop through each chunk of the genome, count how many N's are on each line, and record this. All samples must be in the same order in all nexus files for this to work!
time cat 17kbchunks.txt | while read chunk ; do rm -f temp ; echo "$chunk" >> temp ; tail -n +6 "$chunk" | head -n -2 | tr -d -c '\-\n' | awk '{ print length; }' >> temp ; paste temp chunk.missingness > temp3 && mv temp3 chunk.missingness ; done #76m
```
