# D3 Test of Introgression

**Purpose**: this page describes how to calculate distance matrices from sequences alignments, and then make plots summarizing patterns of divergence across the genome. Then, it calculates the D3 statistic as a test for introgression.  
**Procedure**: break the scaffold-length sequence alignment into chunks (this was done in step 1.6.2c), calculate sequence divergence for each chunk, and then visualize these values. Then, use them to calculate D3.  
**Input**: 5-kb sequence alignment blocks  
**Output**: A distance matrix for each sequence block, and D3 statistics.  

# Calculating Nuclear Genetic Distance  
This section calculates distance matrices from sequence alignments.  

There is a how-to page about distance calculations [here](http://emboss.sourceforge.net/apps/release/6.4/emboss/apps/distmat.html). Many different methods exist for calculating distances between sequences. These may either calculate raw differences (simply % difference between sequences), or correct for different biases or for saturation of mutations. Pure % difference is the "uncorrected (p) distance" between sequences, which is generally reasonable for closely related sequences, but for very divergent sequences will underestimate the divergence as it becomes important to correct for multiple substitutions that may occur at the same site. The Jukes Cantor model accounts for multiple substitutions, and the Kimura 2-parameter distance goes a step farther to account for the fact that transitions are more common than transversions.  

First, you need sequence alignments. We made these in step 1.6.2c. Note that since Emboss incorporates N's into divergence estimation (which we don't want), you will need to use the scaffold alignments where N's were converted to gaps (-).  

```bash
mkdir -p /home/1_Else/0_GENOMES5/Stercorarius/3_Introgression/3.2.1c_D3_distancematrix/
#cd /home/1_Else/0_GENOMES5/Stercorarius/3_Introgression/3.2.1c_D3_distancematrix/ #on new server
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.2.1c_D3_distancematrix/ #on old server

#make folders to hold the distance matrices
mkdir -p distance_matrices

#make a list of the sequence alignment chunks to loop through
find /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/1.6.2c_phylogenetic_blocks/strict/blocks/ | tail -n +2 > chunks.txt #removes the first line because it is just the directory name #fast
#note that there are too many files to list with wildcards, so the following type of command would not work:
#find blocks/*.nex > chunks.txt
wc -l chunks.txt #223915

#calculate Kimura 2-parameter distances for all chunks by looping through the list of chunks. Make sure the name of the chunk is in the output name
time cat chunks.txt | sed 's;/home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/1.6.2c_phylogenetic_blocks/strict/blocks/;;g' | parallel /home/0_PROGRAMS/EMBOSS-6.6.0/bin/distmat -sequence /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/1.6.2c_phylogenetic_blocks/strict/blocks/{1} -nucmethod 2 -stdout -outfile distance_matrices/{1}.distmat #23m15.377s (427m55.772s) 
#This would run it in serial, if you do not have the cores available to do it in parallel
#cat chunks.txt | sed 's;/home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorariidae/2_haplotypes_0a_1a_2ab_strict/blocks/;;g' | while read chunk ; do /home/0_PROGRAMS/EMBOSS-6.6.0/bin/distmat -sequence /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorariidae/2_haplotypes_0a_1a_2ab_strict/blocks/"$chunk" -nucmethod 2 -stdout -outfile distance_matrices/"$chunk".distmat ; done

#now loop through the chunks, and make a new file where you will print the name of each chunk folllowed by their distance matrix (just the 17 lines (I have 17 samples) after the first sample, which in my case is ANSK7 (but the N is turned into '-')). I am doing this in serial so that I can control the order in which each chunk finishes, even though it is slower.
time cat chunks.txt | sed 's;/home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/1.6.2c_phylogenetic_blocks/strict/blocks/;;g' | while read chunk ; do printf "$chunk" >> allwindows.distmat ; cat distance_matrices/"$chunk".distmat | grep -A 16 "Alca" | paste -s -d ' ' >> allwindows.distmat ; done #took 22m7.673s (5m2.116s) #22m7.673s (5m2.116s) on main server

#Count missing data in each window, in order to filter data by missingness
#create a blank file to store our data
touch chunk.missingness
#loop through each chunk of the genome, count how many N's are on each line, and record this. All samples must be in the same order in all nexus files for this to work!
time cat chunks.txt | while read chunk ; do rm -f temp ; echo "$chunk" >> temp ; tail -n +6 "$chunk" | head -n -2 | tr -d -c '\-\n' | awk '{ print length; }' >> temp ; paste temp chunk.missingness > temp3 && mv temp3 chunk.missingness ; done #789m41.586s (251m39.604s) on main server
#this would do in parallel but I think would give issues with accessing the same file by different threads
#time cat chunks.txt | parallel echo {1} >> temp.{1} ; cat {1} | tr -d -c 'N\n' | awk '{ print length; }' >> temp.{1} 
#paste temp.* chunk.missingness > temp3 && mv temp3 chunk.missingness
#took 9740m19.497s (2449m31.508s)




#Now repeat for four additional datasets. These datasets are to investigate the effects of mapping quality filters and angsd -dofasta settings. To have confidence in the interpretation of our results, they should not be sensitive to these filters.

#test another strictness (MQ>15, dofasta 2) (10kb blocks and max 5% missing)
#dataset strict2 was made with -dofasta 2 -minMapQ 15 -minQ 20 -remove_bads 1 -only_proper_pairs 1 -uniqueOnly 1 -doCounts 1 -setMinDepth 3
mkdir -p /home/1_Else/0_GENOMES5/Stercorarius/3_Introgression/3.2.1c_D3_distancematrix/strict2
cd /home/1_Else/0_GENOMES5/Stercorarius/3_Introgression/3.2.1c_D3_distancematrix/strict2

mkdir -p distance_matrices
find /home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.6.2c_phylogenetic_blocks/strict2/blocks/ | tail -n +2 > chunks.txt #removes the first line because it is just the directory name #fast
time cat chunks.txt | sed 's;/home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.6.2c_phylogenetic_blocks/strict2/blocks/;;g' | parallel /home/0_PROGRAMS/EMBOSS-6.6.0/bin/distmat -sequence /home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.6.2c_phylogenetic_blocks/strict2/blocks/{1} -nucmethod 2 -stdout -outfile distance_matrices/{1}.distmat #23m15.377s (427m55.772s)
time cat chunks.txt | sed 's;/home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.6.2c_phylogenetic_blocks/strict2/blocks/;;g' | while read chunk ; do printf "$chunk" >> allwindows.distmat ; cat distance_matrices/"$chunk".distmat | grep -A 16 "A.SK01" | paste -s -d ' ' >> allwindows.distmat ; done #took 629m15.376s (292m18.972s)



#repeat with same dataset as above, but with 5kb blocks and max 20% missing (MQ>15, dofasta2) (5kb blocks and max 20% missing)
mkdir -p /home/1_Else/0_GENOMES5/Stercorarius/3_Introgression/3.2.1c_D3_distancematrix/strict3
cd /home/1_Else/0_GENOMES5/Stercorarius/3_Introgression/3.2.1c_D3_distancematrix/strict3

mkdir -p distance_matrices
find /home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.6.2c_phylogenetic_blocks/strict2/blocksmin20/ | tail -n +2 > chunks.txt #removes the first line because it is just the directory name #fast
time cat chunks.txt | sed 's;/home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.6.2c_phylogenetic_blocks/strict2/blocksmin20/;;g' | parallel /home/0_PROGRAMS/EMBOSS-6.6.0/bin/distmat -sequence /home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.6.2c_phylogenetic_blocks/strict2/blocksmin20/{1} -nucmethod 2 -stdout -outfile distance_matrices/{1}.distmat #23m15.377s (427m55.772s)
time cat chunks.txt | sed 's;/home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.6.2c_phylogenetic_blocks/strict2/blocksmin20/;;g' | while read chunk ; do printf "$chunk" >> allwindows.distmat ; cat distance_matrices/"$chunk".distmat | grep -A 16 "A.SK01" | paste -s -d ' ' >> allwindows.distmat ; done #took 629m15.376s (292m18.972s)

touch chunk.missingness
time cat chunks.txt | while read chunk ; do rm -f temp ; echo "$chunk" >> temp ; tail -n +6 "$chunk" | head -n -2 | tr -d -c '\-\n' | awk '{ print length; }' >> temp ; paste temp chunk.missingness > temp3 && mv temp3 chunk.missingness ; done



#test another strictness (MQ>15, dofasta3) (5kb blocks and max 20% missing)
#dataset strict3 was made with -dofasta 3 -minMapQ 15 -minQ 20 -remove_bads 1 -only_proper_pairs 1 -uniqueOnly 1 -doCounts 1 -setMinDepth 3
mkdir -p /home/1_Else/0_GENOMES5/Stercorarius/3_Introgression/3.2.1c_D3_distancematrix/strict32
cd /home/1_Else/0_GENOMES5/Stercorarius/3_Introgression/3.2.1c_D3_distancematrix/strict32

mkdir -p distance_matrices
find /home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.6.2c_phylogenetic_blocks/strict3/blocks | tail -n +2 > chunks.txt #removes the first line because it is just the directory name #fast
time cat chunks.txt | sed 's;/home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.6.2c_phylogenetic_blocks/strict3/blocks/;;g' | parallel /home/0_PROGRAMS/EMBOSS-6.6.0/bin/distmat -sequence /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/1.6.2c_phylogenetic_blocks/strict3/blocks/{1} -nucmethod 2 -stdout -outfile distance_matrices/{1}.distmat #23m15.377s (427m55.772s)
time cat chunks.txt | sed 's;/home/1_Else/0_GENOMES5/Stercorarius/1_Data/1.6.2c_phylogenetic_blocks/strict3/blocks/;;g' | while read chunk ; do printf "$chunk" >> allwindows.distmat ; cat distance_matrices/"$chunk".distmat | grep -A 16 "A.SK01" | paste -s -d ' ' >> allwindows.distmat ; done #took 629m15.376s (292m18.972s)

touch chunk.missingness
time cat chunks.txt | while read chunk ; do rm -f temp ; echo "$chunk" >> temp ; tail -n +6 "$chunk" | head -n -2 | tr -d -c '\-\n' | awk '{ print length; }' >> temp ; paste temp chunk.missingness > temp3 && mv temp3 chunk.missingness ; done



#test another strictness (MQ>20, dofasta2) (5kb blocks and max 20% missing)
#dataset strict4 was made with -dofasta 2 -minMapQ 20 -minQ 20 -remove_bads 1 -only_proper_pairs 1 -uniqueOnly 1 -doCounts 1 -setMinDepth 3
mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_Introgression/3.2.1c_D3_distancematrix/strict4
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_Introgression/3.2.1c_D3_distancematrix/strict4

mkdir -p distance_matrices
find /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/1.6.2c_phylogenetic_blocks/strict4/blocks | tail -n +2 > chunks.txt #removes the first line because it is just the directory name #fast
time cat chunks.txt | sed 's;/home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/1.6.2c_phylogenetic_blocks/strict4/blocks/;;g' | parallel /home/0_PROGRAMS/EMBOSS-6.6.0/bin/distmat -sequence /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/1.6.2c_phylogenetic_blocks/strict4/blocks/{1} -nucmethod 2 -stdout -outfile distance_matrices/{1}.distmat #23m15.377s (427m55.772s)
time cat chunks.txt | sed 's;/home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/1_Data/1.6.2c_phylogenetic_blocks/strict4/blocks/;;g' | while read chunk ; do printf "$chunk" >> allwindows.distmat ; cat distance_matrices/"$chunk".distmat | grep -A 16 "A.SK01" | paste -s -d ' ' >> allwindows.distmat ; done #took 629m15.376s (292m18.972s)

touch chunk.missingness
time cat chunks.txt | while read chunk ; do rm -f temp ; echo "$chunk" >> temp ; tail -n +6 "$chunk" | head -n -2 | tr -d -c '\-\n' | awk '{ print length; }' >> temp ; paste temp chunk.missingness > temp3 && mv temp3 chunk.missingness ; done
```
Now that we have our distance matrices, we can visualize these results in R to interpret the results. I detailed that pipeline in an `.Rmd` file, `3.2.1c_D3_distancematrix.Rmd`, which also calculates D3.   

# Calculating Mitochondrial Genetic Distance  

So far we have looked at nuclear genetic distances. We should also look at mitochondrial distances. We can calculate distances using the alignment we generated for the phylogeny. This alignment has had one copy of the duplicated control region removed: since the two copies evolve with concerted evolution, it could be considered pseudoreplication to include both copies since they do not carry independant information.  

```bash
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.2.1c_D3_distancematrix #on old server
mkdir -p mitochondrial_distances

#there are no N's in this sequence, otherwise N's should be replaced by gaps.
#kimura 2 parameter
/home/0_PROGRAMS/EMBOSS-6.6.0/bin/distmat -sequence ../../2_Phylogeny/2.1.1_mtDNA_ML_phylogeny/Stercorariidae_MitogenomesCorrectedSinglecopy_withOutgroups.mafft.afa -nucmethod 2 -stdout -outfile mitochondrial_distances/mtDNA.distmat
#uncorrected p distance
/home/0_PROGRAMS/EMBOSS-6.6.0/bin/distmat -sequence /home/0_GENOMES5/Stercorarius/2_Phylogeny/2.1.1_mtDNA_ML_phylogeny/Stercorariidae_MitogenomesCorrectedSinglecopy_withOutgroups.mafft.afa -nucmethod 0 -stdout -outfile mitochondrial_distances/mtDNA.pdistmat

#replace ambiguity codes with gaps so that they do not contribute to genetic distances calculation
sed 's/r/-/g; s/y/-/g; s/m/-/g; s/d/-/g' /home/0_GENOMES5/Stercorarius/2_Phylogeny/2.1.1_mtDNA_ML_phylogeny/Stercorariidae_MitogenomesCorrectedSinglecopy_withOutgroups.mafft.afa > mitochondrial_distances/Stercorariidae_MitogenomesCorrectedSinglecopy_withOutgroups.mafft.noN.afa #keep in mind that it is case sensitive and there are more possible ambiguity codes - these are the only ones that I have in my dataset.
/home/0_PROGRAMS/EMBOSS-6.6.0/bin/distmat -sequence /home/0_GENOMES5/Stercorarius/2_Phylogeny/2.1.1_mtDNA_ML_phylogeny/Stercorariidae_MitogenomesCorrectedSinglecopy_withOutgroups.mafft.noN.afa -nucmethod 2 -stdout -outfile mitochondrial_distances/mtDNA.noN.distmat
#not counting the ambiguity codes made very little difference - this large impact was decreasing divergence between S. antarcticus 2 and S. skua MKP1593 by 0.09, from 1.5 to 1.41. I would not like to count ambiguity codes because in my dataset they seem to be caused by numts or difficulty distinguishing the two copies of the duplication - not real sources of divergence between the samples.
```
* `-nucmethod`: 2 for kimura 2 parameter, 0 for uncorrected p.  

**Timing**: 8m33 (75m5 user time) to create matrices, 14m34 (5m57 user time) to put into the `allwindows.distmat` file.  
