### Overview  
This section details how I extracted the mitonuclear genes and produced their phylogenies.  

### Required Input  
This step requires a dataset of reference proteins to extract from a reference genome, and a bam file of sequences mapped to this reference.  

### Output  
The output is a series of alignments for genes of interest, and phylogenies for these loci.  

## Locate genes of interest  in reference genome  

First, we need to determine where the mitonuclear genes are in the genome.  

To locate our genes of interest we need three things:  
1) A file `samples.txt` listing our sample names to loop through  
2) a file `genes.list` listing the genes to loop through  
3) seperate fasta files with a reference protein for each gene  

First, I make my lists:  
```bash
cat > samples.txt
```
```
LTJA_MKP990
PAJA_B20730
PAJA_USNM606730
POJA_MKP1559
POJA_4
POJA_IB2659
GRSK_MKP1592
GRSK_MKP1593
CISK2
CISK55
CISK3
CHSK_MKP2451
ANSK7
ANSK01
ANSK8
Alca_torda
Uria_lomvia
```

I obtained reference genes from NCBI [Orthologs](https://www.ncbi.nlm.nih.gov/gene/4706/ortholog/?scope=89593&term=NDUFAB1). I saved each one as `"$gene"_refseq_protein.fasta` and put them into a folder called `mitonuclear_refseq_genes`  
```bash
cat > genes.list
```
```
NDUFS7
NDUFS8
NDUFV2
NDUFS3
NDUFS2
NDUFV1
NDUFS1
NDUFS6
NDUFA12
NDUFS4
NDUFA9
NDUFAB1
NDUFA2
NDUFA1
NDUFB3
NDUFA5
NDUFA6
NDUFA11
NDUFB11
NDUFS5
NDUFB4
NDUFA13
NDUFB7
NDUFA8
NDUFB9
NDUFB10
NDUFB8
NDUFC2
NDUFB2
NDUFA7
NDUFA3
NDUFA4
NDUFB5
NDUFB1
NDUFC1
NDUFA10
NDUFV3
NDUFB6
NDUFAF1
NDUFAF2
NDUFAF3
NDUFAF4
NDUFAF5
NDUFAF6
NDUFAF7
FOXRED1
ACAD9
ECSIT
NUBPL
TIMMDC1
NDUFAF8

ATP5F1A
ATP5F1B
ATP5F1C
ATP5F1D
ATP5F1E
ATP5G2
ATP5MC1
ATP5MC3
ATP5ME
ATP5MF
ATP5MG
ATP5PB
ATP5PD
ATP5PF
ATP5PO
ATPAF1
ATPAF2
TMEM70

LOC116807099
LOC106888355
LOC106888755
LOC106884871
LOC106886384
LOC106902040
LOC106891403
LOC106885327
LOC106892097
LOC106895276
LOC106890876
LYRM7
LOC106898212
LOC106894798
LOC106863540
TTC19
PTCD2

APOPT1
LOC106890876
LOC106898051
LOC106891637
LOC106898802
LOC106893136
LOC106886322
LOC106893886
LOC106891146
LOC106898730
LOC106887984
LOC106896845
LOC106885830
LOC106890424
LOC106892914
LOC106888317
LOC106898440
LOC115498214
LOC106899568
LOC104292194
LOC106902292
LOC107050595
LOC106892261
LOC106895441
LOC106885025
LOC106893159
LOC110391373
FASTKD2
LRPPRC
OXA1L
LOC115491649
PET117
LOC106900569
LOC115492592
SURF1
LOC106899605
COX10

POLRMT
TFAM
TFB1M
TFB2M
MTERF1
AARS1
AARS2
DARS1
DARS2
EPRS1
FARS2
FARSA
FARSB
GARS1
HARS1
HARS2
KARS1
NARS1
NARS2
PARS2
SARS1
SARS2
TARS1
TARS2
ELAC2
SSBP1
POLG
TWNK
SLC25A4
TRMT10C
HSD17B10
PRORP
MRPS2
IMP3
MRPS5
MRPS6
MRPS7
MRPS9
MRPS10
MRPS11
MRPS12
MRPS13
MRPS14
MRPS15
MRPS16
MRPS17
MRPS18A
MRPS18B
MRPS18C
MRPS21
MRPS22
MRPS23
MRPS24
MRPS25
MRPS26
MRPS27
DAP3
MRPS30
MRPS31
MRPS33
MRPS34
MRPS35
MRPS36
MRPL1
MRPL2
MRPL3
MRPL4
MRPL5
MRPL6
MRPL7
MRPL8
MRPL9
MRPL10
MRPL11
MRPL12
MRPL13
MRPL14
MRPL15
MRPL16
MRPL17
MRPL18
MRPL19
MRPL20
MRPL21
MRPL22
MRPL23
MRPL24
MRPL25
MRPL26
MRPL27
MRPL28
MRPL29
MRPL30
MRPL31
MRPL32
MRPL33
MRPL34
MRPL35
MRPL36
MRPL37
MRPL38
MRPL39
MRPL40
MRPL41
MRPL42

CARS2
EARS2
IARS2
LARS2
MARS2
RARS2
VARS2
WARS2
YARS2
```
Then, I loop through the genes list, searching for each gene in the reference genome using tBLASTn.  
```bash
mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.5_mitonuclear
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.5_mitonuclear

mkdir -p blast_hits_Stercorarius_parasiticus
#/home/0_PROGRAMS/ncbi-blast-2.12.0+/bin/makeblastdb -dbtype nucl -in /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/0_Reference_genome/0_reference/Stercorarius_parasiticus.ref.fa
cat genes.list | parallel time /home/0_PROGRAMS/ncbi-blast-2.12.0+/bin/tblastn -query mitonuclear_refseq_genes/{1}_refseq_protein.fasta -db /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/0_Reference_genome/0_reference/Stercorarius_parasiticus.ref.fa -outfmt '"6 qseqid sseqid evalue bitscore pident qstart qend sstart send qseq sseq"' ">" blast_hits_Stercorarius_parasiticus/Charadriiformes_{1}_Alctor.tsv

#read the results
cat genes.list | while read gene ; do echo "" ; echo "" ; echo "$gene" ; cat blast_hits_Stercorarius_parasiticus/Charadriiformes_"$gene"_Alctor.tsv | cut -f 1-9 | sort -n -k 8 ; done

#cat genes.list | while read gene ; do echo "" ; echo "" ; echo "$gene" ; cat Charadriiformes_"$gene"_Alctor.tsv | cut -f 1-9 | sort -n -k 8 | awk '{ if ($5 >= 80) { print } }' ; done #filter to min 80% identity if desired, but I prefer to see the whole dataset for context

```
## Interpreting the data  

From the BLAST results, we need to identify 1) The chromosome name containing the gene 2) the start and 3) the stop position of the gene. The chromosome names are given in column 2, and the start and stop positions are given in columns 8 and 9 of the output tsv file. There will often be multiple hits. You can choose the best one based on the E-value in column 3 and the percent identity in column 5. You can safely ignore anything with an E-value near 1 or higher. Usually real hits will have an E-value less than 1E-5 but it depends on the sequence. If the E-value is very low but the percent identity is less than 50% (and it used closely related query species), then I might assume that the sequence is just a paralog. (real orthologs between closely related birds are usually above 90% identity).  

The gene will likely be divided into multiple hits, these may be seperate exons divided by introns. Identify the start and end of the protein by looking at column 6 and 7, which give the start and stop position of the hit relative to the protein sequence. Find the start of the hit closest to the start of the protein, and the end of the hit closest to the end of the protein.  

Once I gathered this data, I just used Excel to quickly calculate `start position - end position` for each protein. This will give the length of the gene. If it is negative, then the gene is in reverse orientation - reverse the start and stop position so that it is oriented correctly for the next step.  

Make a text file with 4 columns: 1) Gene name 2) chromosome name 3) start position 4) end position  

```bash
cat > sequence.list
```
```
NDUFA5	chr1	24376180	24379480	3300
NDUFA12	chr1	49789469	49800474	11005
NDUFA6	chr1	54316427	54318641	2214
NDUFB2	chr1	62728120	62728182	62
NDUFA9	chr1	77515286	77531165	15879
NDUFB4	chr1	87324946	87326602	1656
TIMMDC1	chr1	100711012	100716333	5321
NDUFV3	chr1	119075874	119083803	7929
NDUFA3	chr1	186086557	186086598	41
NDUFC2	chr1	208670918	208671825	907
NDUFB8	chr10	19205477	19208810	3333
NDUFAF3	chr11	5944556	5945325	769
ACAD9	chr11	22825514	22849435	23921
NDUFAF8	chr11	35723135	35723251	116
NDUFA8	chr15	4066261	4067869	1608
NDUFS5	chr17	366205	367013	808
FOXRED1	chr18	496248	497754	1506
NDUFA4	chr2	29301225	29302855	1630
NDUFS6	chr2	93974075	93982691	8616
NDUFV2	chr2	107630446	107642935	12489
NDUFAF6	chr2	137628460	137645237	16777
NDUFB9	chr2	150542857	150546041	3184
NDUFA7	chr21	1135743	1136442	699
NDUFA11	chr21	1300759	1301520	761
NDUFS7	chr21	4015167	4018281	3114
NDUFA4L2	chr24	1867693	1867752	59
NDUFAF5	chr3	20745542	20751622	6080
NDUFAF7	chr3	37597356	37603051	5695
NDUFAF4	chr3	81266277	81270192	3915
NDUFC1	chr4	13025067	13026075	1008
NDUFB1	chr5	5790292	5791289	997
NDUFS3	chr5	23559582	23563177	3595
NDUFS3.2	chr5	23559379	23563704
NDUFS3.3	chr5	23559379	23573704
NDUFAF1	chr5	25656833	25666614	9781
NUBPL	chr5	37535605	37620921	85316
NDUFS8	chr5	68863080	68863223	143
NDUFV1	chr5	68990461	68992040	1579
NDUFB5	chr6	2136766	2141842	5076
NDUFA2	chr6	30685601	30686453	852
NDUFA1	chr7	10519095	10521402	2307
NDUFB10	chr8	6958479	6960235	1756
NDUFAB1	chr8	7966805	7968350	1545
NDUFB6	chr8	17852811	17855943	3132
NDUFB3	chr9	29816709	29819522	2813
NDUFS1	chr9	31159234	31175488	16254
NDUFA10	chr9	38765964	38807025	41061
NDUFS4	chrZ	18485872	18524541	38669
NDUFAF2	chrZ	21480725	21550552	69827
NDUFS2	VWZL01000567.1	832	5105	4273
ECSIT	VWZL01002810.1	58	899	841

CYC1	chr1	202713181	202713384
UQCRFS1	chr7	33325704	33326318
UQCRC1	chr11	2810688	2819164
UQCRC2	chr8	8867768	8878247
UQCRH	chr12	24707879	24708052
UQCRB	chr2	138096670	138101616
UQCRQ	chr6	42420895	42422065		
UQCR10	chr13	19422380	19424109
UQCR11	chr21	498074	499506
BCS1L	chr9	17693634	17698874
COA1	chr2	56340269	56366711
LYRM7	chrZ	48848575	48851782
UQCC1	chr13	17750856	17795891
UQCC2	chr20	3440595	3444146			
TTC19	chr14	7855851	7859330
PTCD2	chrZ	28033084	28046672

APOPT1	chr5	12442345	12449231
COA1	chr2	56340269	56366711
COA3	chr19	4560654	4560988
COA4	chr1	206084268	206084651
COA5	chr1	143797271	143800531
COA6	chr3	43321053	43323027

COA7	chr12	28207716	28209626
COX11	chr11	33510905	33512604
COX14	chr24	323931	324086
COX15	chr10	24500304	24503698
COX16	chr5	29162115	29212725
COX17	chr1	87728082	87730044
COX18	chr4	1334265	1339351
COX19	chr8	1995060	1996615
COX20	chr3	39012435	39015872
COX4I1	chr7	44214392	44217320
COX5A	chr8	19630559	19633417
COX6A1	chr13	22202875	22204189
COX6B1	VMED01000056.1	84990	85845
COX6C	chr2	139711407	139713271
COX7A2	chr3	90229665	90232188
COX7B	chr7	14499909	14500649
COX7C	chrZ	77860056	77861314
COX7A2L	chr3	26846934	26855108
FASTKD2	chr9	30927430	30939274
LRPPRC	chr3	28331606	28413120
PET117	chr3	8541111	8541269
SCO1	chr11	25653925	25659270
SURF1	chr15	1486114	1491007
TACO1	VWZL01009353.1	7	207
COX10	chr11	28195428	28370435

ATP5F1A	chrZ	4249695	4256103	6408
ATP5F1B	VWZL01005027.1	103	1354	1251
ATP5F1C	chr1	5010189	5016088	5899
ATP5F1D	chr21	3723061	3724452	1391
ATP5F1E	chr13	875235	875360	125
ATP5G2	chr24	818767	820026	1259				
ATP5MC1	chr19	7014586	7015772	1186
ATP5MC3	chr9	36234382	36304056	69674				
ATP5ME	chrZ	51210444	51210524	80
ATP5MF	chr8	4686483	4688795	2312
ATP5MG	chr18	7953504	7953831	327
ATP5PB	chr20	6382203	6384698	2495
ATP5PD	chr11	31612689	31615218	2529
ATP5PF	chr1	111391211	111392707	1496
ATP5PO	chr1	114752258	114755497	3239
ATPAF1	chr12	25046214	25056158	9944
ATPAF2	chr8	5458436	5463004	4568
TMEM70	chr2	128635276	128636906	1630

POLRMT	chr21	3286904	3305710
TFAM	chr10	3094203	3103572
TFB1M	chr3	56957158	56983116
TFB2M	chr3	38166523	38177246
MTERF1	chr1	59121390	59122580
AARS1	chr7	25222334	25237917
AARS2	chr3	33831813	33849830
DARS1	chr9	7461964	7502904
DARS2	chr12	2544024	2557454
EPRS1	chr3	21240646	21277922
FARS2	chr2	71396942	71588983
FARSA	VWZL01004773.1	5360	7330
FARSB	chr6	20507146	20546171
GARS1	chr2	4733655	4761276
HARS1	chr6	30709092	30734138
HARS2	chr6	30710179	30734138
KARS1	chr7	47190406	47199030
NARS1	chrZ	538256	549328
NARS2	chr1	208851776	208877330
PARS2	chr12	29143656	29145074
SARS1	chr20	1722009	1725611
TARS1	chrZ	12400939	12416748
TARS2	chr8	26446194	26458557
ELAC2	chr11	26072304	26085423
SSBP1	chr1	63175574	63178652
POLG	chr8	34358018	34368856
TWNK	chr10	25918380	25922566
SLC25A4	chr4	75866445	75868405
TRMT10C	chr1	93323430	93324674
HSD17B10	VWZL01000864.1	205	609
PRORP	chr5	39288918	39339451
MRPS2	chr15	2682584	2685708
IMP3	chr3	117918326	117918562
MRPS5	chr3	16050500	16107320
MRPS6	chr1	114852063	114863858
MRPS7	chr11	31496918	31499711
MRPS9	chr1	146774188	146801510
MRPS10	chr3	25697419	25701183
MRPS11	chr8	34648940	34652014
MRPS12	VMED01000060.1	34666	35001
MRPS14	chr12	44445	47282
MRPS15	chr17	2179046	2184737
MRPS16	chr16	3255	3454
MRPS17	chr14	9973144	9974862
MRPS18A	chr3	34550877	34568495
MRPS18B	VWZL01000873.1	424	624
MRPS18C	chr4	18292504	18293613
MRPS21	chr23	1804933	1805097
MRPS22	chr6	23036547	23042169
MRPS23	chr14	4858156	4860037
MRPS24	chr22	355305	356336
MRPS25	chr11	16885514	16886745
MRPS26	chr4	43992960	43993082
MRPS27	chrZ	28051076	28098509
DAP3	chr23	1305744	1323070
MRPS30	chrZ	16567271	16571788
MRPS31	chr1	185553484	185572259
MRPS33	chr1	62833234	62833449
MRPS34	chr8	18208739	18210343
MRPS35	chr1	79406127	79427116
MRPS36	chrZ	24550980	24555741
MRPL1	chr4	2576322	2589079
MRPL2	chr3	7465566	7467102
MRPL3	chr2	46059493	46089324
MRPL9	VWZL01002321.1	7489	8960
MRPL10	chr19	6502756	6504277
MRPL12	chr11	34853142	34856579
MRPL13	chr2	148826263	148843236
MRPL14	chr3	34077198	34080282
MRPL15	chr2	120341787	120349200
MRPL16	VMED01000072.1	45980	47631
MRPL17	chrZ	11528324	11529060
MRPL18	chr3	51388159	51390513
MRPL19	chr3	116725033	116728343
MRPL20	chr16	6837283	6841165
MRPL21	chr5	69901625	69910452
MRPL22	chr6	48885668	48890856
MRPL23	chr5	52313137	52325111
MRPL24	VWZL01004084.1	75	224
MRPL27	chr11	31929168	31930781
MRPL28	chr8	16194534	16197295
MRPL30	chr1	144019771	144025072
MRPL32	chr2	55952335	55954465
MRPL33	chr3	30962735	30965085
MRPL34	chr21	4380645	4385269
MRPL35	chr4	47419035	47420386
MRPL36	chr2	93973184	93973294
MRPL37	chr12	29036434	29040401
MRPL38	chr11	30933505	30937495
MRPL39	chr1	111334748	111342771
MRPL40	chr13	33686285	33688139
MRPL41	chr15	11281095	11281499
MRPL42	chr1	49299644	49304702

NDUFS2	VWZL01000567.1	832	5105	4273
ECSIT	VWZL01002810.1	58	899	841
COX6B1	VMED01000056.1	84990	85845
TACO1	VWZL01009353.1	7	207
ATP5F1B	VWZL01005027.1	103	1354	1251
FARSA	VWZL01004773.1	5360	7330
HSD17B10	VWZL01000864.1	205	609
MRPS12	VMED01000060.1	34666	35001
MRPS18B	VWZL01000873.1	424	624
MRPL9	VWZL01002321.1	7489	8960
MRPL16	VMED01000072.1	45980	47631
MRPL24	VWZL01004084.1	75	224

NDUFS7_2	chr21	4015146	4018281	3114


CARS2	chr1	150831071	150871129	40058
EARS2	chr8	8022100	8028203	6103
IARS2	chr3	21194241	21223995	29754
LARS2	chr2	47196926	47282423	85497
MARS2	chr7	687056	694756	7700
RARS2	chr3	85307112	85345547	38435
VARS2	VWZL01008462.1	3	241	238
WARS2	chr1	86058191	86080347	22156
YARS2	chr1	64871577	64877465	5888
```
Now we will loop through this list: extracting the sequence from the genome alignment, aligning each sample, and then optionally making a phylogeny.  
```bash
cat samples.txt | while read sample ; do cat sequence.list | while read gene chr start stop ; do /home/0_PROGRAMS/angsd/angsd -dofasta 3 -r "$chr":"$start"-"$stop" -i ../../1_Data/1.3c_mapping/merged/"$sample".PAJAbowtie.marked.bam -out "$sample"_"$gene" ; done ; done
gunzip ./*.fa.gz

#remove N's. This script first removes any lines that are purely N repeating from start to finish (^N*$)
#also put species name in fasta header
cat samples.txt | while read sample ; do cat sequence.list | while read gene chr start stop ; do grep -v "^N*$" "$sample"_"$gene".fa | sed "s/chr.*$/$sample/g" > temp && mv temp "$sample"_"$gene".fa ; done ; done

#for VMED chromosome samples
#cat samples.txt | while read sample ; do cat sequence.list | while read gene chr start stop ; do grep -v "^N*$" "$sample"_"$gene".fa | sed "s/V............1/$sample/g" > temp && mv temp "$sample"_"$gene".fa ; done ; done

#remove Uria
mkdir -p trash
mv Uria*fa trash

#put all samples together
cat sequence.list | while read gene chr start stop ; do cat ./*_"$gene".fa > Stercorariidae."$gene".fa ; done

#align
cat sequence.list | cut -f 1 | parallel 'time /home/0_PROGRAMS/mafft-7.490-with-extensions/bin/mafft --auto --thread 1 Stercorariidae.{1}.fa > Stercorariidae.{1}.mafft.afa'

#phylogeny
cat sequence.list | cut -f 1 | parallel time /home/0_PROGRAMS/iqtree-2.1.3-Linux/bin/iqtree2 -s Stercorariidae.{1}.mafft.afa -o Alca_torda -nt 1 -redo 


```
Took up to 46m27.352s per gene to make the phylogeny.  


# Phylogeny in 5 kb blocks

The phylogeny for some of the genes is poorly resolved, particularly for very short genes that may be less than 1 kb in length. To improve the phylogeny, I will repeat the process for a full 5 kb window starting from the start position of the gene. (Note that this may mean start position plus 5 kb for genes on the plus strand, or starting position minus 5 kb for genes on the minus strand).  

```bash
mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.5_mitonuclear/5kb_blocks
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.5_mitonuclear/5kb_blocks

cat > sequence.list
```
```
NDUFA5	chr1	24376180	24381180
NDUFA12	chr1	49795474	49800474
NDUFA6	chr1	54316427	54321427
NDUFB2	chr1	62728120	62733120
NDUFA9	chr1	77515286	77520286
NDUFB4	chr1	87321602	87326602
TIMMDC1	chr1	100711333	100716333
NDUFV3	chr1	119075874	119080874
NDUFA3	chr1	186081598	186086598
NDUFC2	chr1	208666825	208671825
NDUFB8	chr10	19205477	19210477
NDUFAF3	chr11	5945325	5950325
ACAD9	chr11	22844435	22849435
NDUFAF8	chr11	35723135	35728135
NDUFA8	chr15	4062869	4067869
NDUFS5	chr17	362013	367013
FOXRED1	chr18	492754	497754
NDUFA4	chr2	29297855	29302855
NDUFS6	chr2	93974075	93979075
NDUFV2	chr2	107637935	107642935
NDUFAF6	chr2	137628460	137633460
NDUFB9	chr2	150542857	150547857
NDUFA7	chr21	1135743	1140743
NDUFA11	chr21	1296520	1301520
NDUFS7	chr21	4015167	4020167	
NDUFA4L2	chr24	1862752	1867752
NDUFAF5	chr3	20751622	20756622
NDUFAF7	chr3	37597356	37602356
NDUFAF4	chr3	81266277	81271277
NDUFC1	chr4	13025067	13030067	
NDUFB1	chr5	5786289	5791289
NDUFS3	chr5	23558177	23563177
NDUFAF1	chr5	25666614	25671614
NUBPL	chr5	37535605	37540605
NDUFS8	chr5	68858223	68863223
NDUFV1	chr5	68987040	68992040
NDUFB5	chr6	2136842	2141842
NDUFA2	chr6	30681453	30686453
NDUFA1	chr7	10519095	10524095
NDUFB10	chr8	6958479	6963479
NDUFAB1	chr8	7963350	7968350
NDUFB6	chr8	17850943	17855943
NDUFB3	chr9	29816709	29821709
NDUFS1	chr9	31159234	31164234
NDUFA10	chr9	38765964	38770964
NDUFS4	chrZ	18485872	18490872
NDUFAF2	chrZ	21480725	21485725
NDUFS2	VWZL01000567.1	832	5832
ECSIT	VWZL01002810.1	58	5058


CYC1	chr1	202713181	202718181
UQCRFS1	chr7	33321318	33326318
UQCRC1	chr11	2810688	2815688
UQCRC2	chr8	8873247	8878247
UQCRH	chr12	24707879	24712879
UQCRB	chr2	138096616	138101616
UQCRQ	chr6	42417065	42422065
UQCR10	chr13	19419109	19424109
UQCR11	chr21	498074	503074
BCS1L	chr9	17693634	17698634
COA1	chr2	56361711	56366711
LYRM7	chrZ	48846782	48851782
UQCC1	chr13	17790891	17795891
UQCC2	chr20	3440595	3445595
TTC19	chr14	7854330	7859330
PTCD2	chrZ	28041672	28046672

APOPT1	chr5	12442345	12447345
COA1	chr2	56361711	56366711
COA3	chr19	4560654	4565654
COA4	chr1	206084268	206089268
COA5	chr1	143795531	143800531
COA6	chr3	43318027	43323027
COA7	chr12	28204626	28209626
COX11	chr11	33507604	33512604
COX14	chr24	323931	328931
COX15	chr10	24500304	24505304
COX16	chr5	29162115	29167115
COX17	chr1	87728082	87733082
COX18	chr4	1334265	1339265
COX19	chr8	1991615	1996615
COX20	chr3	39010872	39015872	
COX4I1	chr7	44212320	44217320
COX5A	chr8	19630559	19635559	
COX6A1	chr13	22202875	22207875	
COX6B1	VMED01000056.1	80845	85845	
COX6C	chr2	139708271	139713271	
COX7A2	chr3	90229665	90234665		
COX7B	chr7	14495649	14500649
COX7C	chrZ	77860056	77865056
COX7A2L	chr3	26850108	26855108			
FASTKD2	chr9	30934274	30939274
LRPPRC	chr3	28408120	28413120
PET117	chr3	8541111	8546111
SCO1	chr11	25654270	25659270			
SURF1	chr15	1486007	1491007
TACO1	VWZL01009353.1	7	5007
COX10	chr11	28365435	28370435

ATP5F1A	chrZ	4249695	4254695
ATP5F1B	VWZL01005027.1	-3646	1354
ATP5F1C	chr1	5010189	5015189
ATP5F1D	chr21	3723061	3728061
ATP5F1E	chr13	870360	875360
ATP5G2	chr24	818767	823767
ATP5MC1	chr19	7014586	7019586
ATP5MC3	chr9	36234382	36239382
ATP5ME	chrZ	51210444	51215444
ATP5MF	chr8	4683795	4688795
ATP5MG	chr18	7953504	7958504
ATP5PB	chr20	6382203	6387203
ATP5PD	chr11	31612689	31617689
ATP5PF	chr1	111387707	111392707
ATP5PO	chr1	114750497	114755497
ATPAF1	chr12	25051158	25056158
ATPAF2	chr8	5458004	5463004
TMEM70	chr2	128635276	128640276

POLRMT	chr21	3300710	3305710
TFAM	chr10	3094203	3099203
TFB1M	chr3	56978116	56983116
TFB2M	chr3	38166523	38171523
MTERF1	chr1	59121390	59126390
AARS1	chr7	25222334	25227334
AARS2	chr3	33831813	33836813
DARS1	chr9	7461964	7466964
DARS2	chr12	2552454	2557454
EPRS1	chr3	21240646	21245646
FARS2	chr2	71583983	71588983
FARSA	VWZL01004773.1	2330	7330
FARSB	chr6	20507146	20512146
GARS1	chr2	4756276	4761276
HARS1	chr6	30729138	30734138
HARS2	chr6	30729138	30734138
KARS1	chr7	47190406	47195406
NARS1	chrZ	538256	543256
NARS2	chr1	208872330	208877330
PARS2	chr12	29140074	29145074
SARS1	chr20	1722009	1727009
TARS1	chrZ	12400939	12405939
TARS2	chr8	26446194	26451194
ELAC2	chr11	26072304	26077304
SSBP1	chr1	63175574	63180574
POLG	chr8	34358018	34363018
TWNK	chr10	25917566	25922566
SLC25A4	chr4	75863405	75868405
TRMT10C	chr1	93323430	93328430
HSD17B10	VWZL01000864.1	-4391	609
PRORP	chr5	39288918	39293918
MRPS2	chr15	2682584	2687584
IMP3	chr3	117913562	117918562
MRPS5	chr3	16050500	16055500
MRPS6	chr1	114852063	114857063
MRPS7	chr11	31494711	31499711
MRPS9	chr1	146774188	146779188
MRPS10	chr3	25696183	25701183
MRPS11	chr8	34647014	34652014
MRPS12	VMED01000060.1	34666	39666
MRPS14	chr12	44445	49445
MRPS15	chr17	2179737	2184737
MRPS16	chr16	3255	8255
MRPS17	chr14	9969862	9974862
MRPS18A	chr3	34550877	34555877
MRPS18B	VWZL01000873.1	-4376	624
MRPS18C	chr4	18288613	18293613
MRPS21	chr23	1804933	1809933
MRPS22	chr6	23037169	23042169
MRPS23	chr14	4855037	4860037
MRPS24	chr22	355305	360305
MRPS25	chr11	16881745	16886745
MRPS26	chr4	43988082	43993082
MRPS27	chrZ	28051076	28056076
DAP3	chr23	1305744	1310744
MRPS30	chrZ	16567271	16572271
MRPS31	chr1	185553484	185558484	
MRPS33	chr1	62828449	62833449
MRPS34	chr8	18208739	18213739
MRPS35	chr1	79422116	79427116
MRPS36	chrZ	24550980	24555980
MRPL1	chr4	2576322	2581322
MRPL2	chr3	7465566	7470566
MRPL3	chr2	46084324	46089324
MRPL9	VWZL01002321.1	7489	12489
MRPL10	chr19	6499277	6504277
MRPL12	chr11	34851579	34856579
MRPL13	chr2	148838236	148843236
MRPL14	chr3	34077198	34082198
MRPL15	chr2	120341787	120346787
MRPL16	VMED01000072.1	45980	50980
MRPL17	chrZ	11524060	11529060
MRPL18	chr3	51385513	51390513
MRPL19	chr3	116723343	116728343
MRPL20	chr16	6836165	6841165
MRPL21	chr5	69905452	69910452
MRPL22	chr6	48885668	48890668
MRPL23	chr5	52313137	52318137
MRPL24	VWZL01004084.1	75	5075
MRPL27	chr11	31925781	31930781
MRPL28	chr8	16194534	16199534
MRPL30	chr1	144019771	144024771
MRPL32	chr2	55952335	55957335
MRPL33	chr3	30960085	30965085
MRPL34	chr21	4380645	4385645
MRPL35	chr4	47415386	47420386
MRPL36	chr2	93968294	93973294
MRPL37	chr12	29036434	29041434
MRPL38	chr11	30933505	30938505
MRPL39	chr1	111337771	111342771
MRPL40	chr13	33683139	33688139
MRPL41	chr15	11281095	11286095
MRPL42	chr1	49299644	49304644

COX6B1	VMED01000056.1	80845	85845	
MRPS12	VMED01000060.1	34666	39666
MRPL16	VMED01000072.1	45980	50980
NDUFS2	VWZL01000567.1	832	5832
ECSIT	VWZL01002810.1	58	5058
TACO1	VWZL01009353.1	7	5007
ATP5F1B	VWZL01005027.1	-3646	1354
FARSA	VWZL01004773.1	2330	7330
HSD17B10	VWZL01000864.1	-4391	609
MRPS18B	VWZL01000873.1	-4376	624
MRPL9	VWZL01002321.1	7489	12489
MRPL24	VWZL01004084.1	75	5075

```
Now, as before, we will loop through this list: extracting the sequence from the genome alignment, aligning each sample, and then optionally making a phylogeny.  
```bash
mkdir -p sequences
cat ../samples.txt | while read sample ; do cat sequence.list | while read gene chr start stop ; do /home/0_PROGRAMS/angsd/angsd -dofasta 3 -r "$chr":"$start"-"$stop" -i ../../../1_Data/1.3c_mapping/merged/"$sample".PAJAbowtie.marked.bam -out sequences/"$sample"_"$gene" ; done ; done
gunzip ./sequences/*.fa.gz


#remove N's. This script first removes any lines that are purely N repeating from start to finish (^N*$)
#also put species name in fasta header
cat ../samples.txt | while read sample ; do cat sequence.list | while read gene chr start stop ; do grep -v "^N*$" sequences/"$sample"_"$gene".fa | sed "s/chr.*$/$sample/g" > temp && mv temp sequences/"$sample"_"$gene".fa ; done ; done

#for VMED and VWZL chromosome samples
cat ../samples.txt | while read sample ; do cat sequence.list | while read gene chr start stop ; do grep -v "^N*$" sequences/"$sample"_"$gene".fa | sed "s/V............1/$sample/g" > temp && mv temp sequences/"$sample"_"$gene".fa ; done ; done

#remove Uria
mkdir -p trash
mv sequences/Uria*fa trash

#put all samples together
cat sequence.list | while read gene chr start stop ; do cat ./sequences/*_"$gene".fa > ./sequences/Stercorariidae."$gene".fa ; done

#align
mkdir -p alignments
cat sequence.list | cut -f 1 | parallel 'time /home/0_PROGRAMS/mafft-7.490-with-extensions/bin/mafft --auto --thread 1 sequences/Stercorariidae.{1}.fa > ./alignments/Stercorariidae.{1}.mafft.afa'

#phylogeny
cat sequence.list | cut -f 1 | parallel time /home/0_PROGRAMS/iqtree-2.1.3-Linux/bin/iqtree2 -s alignments/Stercorariidae.{1}.mafft.afa -o Alca_torda -nt 1 -redo 

```
Took up to 46m27.352s per gene to make the phylogeny  

# Phylogeny in 10 kb blocks  

The phylogeny for some of the genes is poorly resolved, particularly for very short genes that may be less than 1 kb in length. To improve the phylogeny, I will repeat the process for a full 10 kb window starting from the start position of the gene. (Note that this may mean start position plus 10 kb for genes on the plus strand, or starting position minus 10 kb for genes on the minus strand).  

```bash
mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.5_mitonuclear/10kb_blocks
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.5_mitonuclear/10kb_blocks

cat > sequence.list
```
```
NDUFA5	chr1	24376180	24386180
NDUFA12	chr1	49790474	49800474
NDUFA6	chr1	54316427	54326427
NDUFB2	chr1	62728120	62738120
NDUFA9	chr1	77515286	77525286
NDUFB4	chr1	87316602	87326602
TIMMDC1	chr1	100706333	100716333
NDUFV3	chr1	119075874	119085874
NDUFA3	chr1	186076598	186086598
NDUFC2	chr1	208661825	208671825
NDUFB8	chr10	19205477	19215477
NDUFAF3	chr11	5945325	5955325
ACAD9	chr11	22839435	22849435
NDUFAF8	chr11	35723135	35733135
NDUFA8	chr15	4057869	4067869
NDUFS5	chr17	357013	367013
FOXRED1	chr18	487754	497754
NDUFA4	chr2	29292855	29302855
NDUFS6	chr2	93974075	93984075
NDUFV2	chr2	107632935	107642935
NDUFAF6	chr2	137628460	137638460
NDUFB9	chr2	150542857	150552857
NDUFA7	chr21	1135743	1145743
NDUFA11	chr21	1291520	1301520
NDUFS7	chr21	4015167	4025167		
NDUFA4L2	chr24	1857752	1867752
NDUFAF5	chr3	20751622	20761622
NDUFAF7	chr3	37597356	37607356
NDUFAF4	chr3	81266277	81276277
NDUFC1	chr4	13025067	13035067	
NDUFB1	chr5	5781289	5791289
NDUFS3	chr5	23553177	23563177
NDUFAF1	chr5	25666614	25676614
NUBPL	chr5	37535605	37545605
NDUFS8	chr5	68853223	68863223
NDUFV1	chr5	68982040	68992040
NDUFB5	chr6	2131842	2141842
NDUFA2	chr6	30676453	30686453
NDUFA1	chr7	10519095	10529095
NDUFB10	chr8	6958479	6968479
NDUFAB1	chr8	7958350	7968350
NDUFB6	chr8	17845943	17855943
NDUFB3	chr9	29816709	29826709
NDUFS1	chr9	31159234	31169234
NDUFA10	chr9	38765964	38775964
NDUFS4	chrZ	18485872	18495872
NDUFAF2	chrZ	21480725	21490725
NDUFS2	VWZL01000567.1	832	10832
ECSIT	VWZL01002810.1	58	10058

CYC1	chr1	202713181	202723181
UQCRFS1	chr7	33316318	33326318
UQCRC1	chr11	2810688	2820688
UQCRC2	chr8	8868247	8878247
UQCRH	chr12	24707879	24717879
UQCRB	chr2	138091616	138101616
UQCRQ	chr6	42412065	42422065
UQCR10	chr13	19414109	19424109
UQCR11	chr21	498074	508074
BCS1L	chr9	17693634	17703634
COA1	chr2	56356711	56366711
LYRM7	chrZ	48841782	48851782
UQCC1	chr13	17785891	17795891
UQCC2	chr20	3440595	3450595
TTC19	chr14	7849330	7859330
PTCD2	chrZ	28036672	28046672

APOPT1	chr5	12442345	12452345
COA1	chr2	56356711	56366711
COA3	chr19	4560654	4570654
COA4	chr1	206084268	206094268
COA5	chr1	143790531	143800531
COA6	chr3	43313027	43323027
COA7	chr12	28199626	28209626		
COX11	chr11	33502604	33512604
COX14	chr24	323931	333931
COX15	chr10	24500304	24510304
COX16	chr5	29162115	29172115
COX17	chr1	87728082	87738082
COX18	chr4	1334265	1344265
COX19	chr8	1986615	1996615
COX20	chr3	39005872	39015872
COX4I1	chr7	44207320	44217320
COX5A	chr8	19630559	19640559
COX6A1	chr13	22202875	22212875
COX6B1	VMED01000056.1	75845	85845
COX6C	chr2	139703271	139713271
COX7A2	chr3	90229665	90239665
COX7B	chr7	14490649	14500649
COX7C	chrZ	77860056	77870056
COX7A2L	chr3	26845108	26855108
FASTKD2	chr9	30929274	30939274
LRPPRC	chr3	28403120	28413120
PET117	chr3	8541111	8551111
SCO1	chr11	25649270	25659270
SURF1	chr15	1481007	1491007
TACO1	VWZL01009353.1	7	10007
COX10	chr11	28360435	28370435

ATP5F1A	chrZ	4249695	4259695
ATP5F1B	VWZL01005027.1	-8646	1354
ATP5F1C	chr1	5010189	5020189
ATP5F1D	chr21	3723061	3733061
ATP5F1E	chr13	865360	875360
ATP5G2	chr24	818767	828767
ATP5MC1	chr19	7014586	7024586
ATP5MC3	chr9	36234382	36244382
ATP5ME	chrZ	51210444	51220444
ATP5MF	chr8	4678795	4688795
ATP5MG	chr18	7953504	7963504
ATP5PB	chr20	6382203	6392203
ATP5PD	chr11	31612689	31622689
ATP5PF	chr1	111382707	111392707
ATP5PO	chr1	114745497	114755497
ATPAF1	chr12	25046158	25056158
ATPAF2	chr8	5453004	5463004
TMEM70	chr2	128635276	128645276

POLRMT	chr21	3295710	3305710
TFAM	chr10	3094203	3104203
TFB1M	chr3	56973116	56983116
TFB2M	chr3	38166523	38176523
MTERF1	chr1	59121390	59131390
AARS1	chr7	25222334	25232334
AARS2	chr3	33831813	33841813
DARS1	chr9	7461964	7471964
DARS2	chr12	2547454	2557454
EPRS1	chr3	21240646	21250646
FARS2	chr2	71578983	71588983
FARSA	VWZL01004773.1	-2670	7330
FARSB	chr6	20507146	20517146
GARS1	chr2	4751276	4761276
HARS1	chr6	30724138	30734138
HARS2	chr6	30724138	30734138
KARS1	chr7	47190406	47200406
NARS1	chrZ	538256	548256
NARS2	chr1	208867330	208877330
PARS2	chr12	29135074	29145074
SARS1	chr20	1722009	1732009
TARS1	chrZ	12400939	12410939
TARS2	chr8	26446194	26456194
ELAC2	chr11	26072304	26082304
SSBP1	chr1	63175574	63185574
POLG	chr8	34358018	34368018
TWNK	chr10	25912566	25922566
SLC25A4	chr4	75858405	75868405
TRMT10C	chr1	93323430	93333430
HSD17B10	VWZL01000864.1	-9391	609
PRORP	chr5	39288918	39298918
MRPS2	chr15	2682584	2692584
IMP3	chr3	117908562	117918562
MRPS5	chr3	16050500	16060500
MRPS6	chr1	114852063	114862063
MRPS7	chr11	31489711	31499711
MRPS9	chr1	146774188	146784188
MRPS10	chr3	25691183	25701183
MRPS11	chr8	34642014	34652014
MRPS12	VMED01000060.1	34666	44666
MRPS14	chr12	44445	54445
MRPS15	chr17	2174737	2184737
MRPS16	chr16	3255	13255
MRPS17	chr14	9964862	9974862
MRPS18A	chr3	34550877	34560877
MRPS18B	VWZL01000873.1	-9376	624
MRPS18C	chr4	18283613	18293613
MRPS21	chr23	1804933	1814933
MRPS22	chr6	23032169	23042169
MRPS23	chr14	4850037	4860037
MRPS24	chr22	355305	365305
MRPS25	chr11	16876745	16886745
MRPS26	chr4	43983082	43993082
MRPS27	chrZ	28051076	28061076
DAP3	chr23	1305744	1315744
MRPS30	chrZ	16567271	16577271
MRPS31	chr1	185553484	185563484
MRPS33	chr1	62823449	62833449
MRPS34	chr8	18208739	18218739
MRPS35	chr1	79417116	79427116
MRPS36	chrZ	24550980	24560980
MRPL1	chr4	2576322	2586322
MRPL2	chr3	7465566	7475566
MRPL3	chr2	46079324	46089324
MRPL9	VWZL01002321.1	7489	17489
MRPL10	chr19	6494277	6504277
MRPL12	chr11	34846579	34856579
MRPL13	chr2	148833236	148843236
MRPL14	chr3	34077198	34087198
MRPL15	chr2	120341787	120351787
MRPL16	VMED01000072.1	45980	55980
MRPL17	chrZ	11519060	11529060
MRPL18	chr3	51380513	51390513
MRPL19	chr3	116718343	116728343
MRPL20	chr16	6831165	6841165
MRPL21	chr5	69900452	69910452
MRPL22	chr6	48885668	48895668
MRPL23	chr5	52313137	52323137
MRPL24	VWZL01004084.1	75	10075
MRPL27	chr11	31920781	31930781
MRPL28	chr8	16194534	16204534
MRPL30	chr1	144019771	144029771
MRPL32	chr2	55952335	55962335
MRPL33	chr3	30955085	30965085
MRPL34	chr21	4380645	4390645
MRPL35	chr4	47410386	47420386
MRPL36	chr2	93963294	93973294
MRPL37	chr12	29036434	29046434
MRPL38	chr11	30933505	30943505
MRPL39	chr1	111332771	111342771
MRPL40	chr13	33678139	33688139
MRPL41	chr15	11281095	11291095
MRPL42	chr1	49299644	49309644



MRPL16	VMED01000072.1	45980	55980
MRPS12	VMED01000060.1	34666	44666
COX6B1	VMED01000056.1	75845	85845



```
```bash
mkdir -p sequences
cat ../samples.txt | while read sample ; do cat sequence.list | while read gene chr start stop ; do /home/0_PROGRAMS/angsd/angsd -dofasta 3 -r "$chr":"$start"-"$stop" -i ../../../1_Data/1.3c_mapping/merged/"$sample".PAJAbowtie.marked.bam -out sequences/"$sample"_"$gene" ; done ; done
gunzip ./sequences/*.fa.gz


#remove N's. This script first removes any lines that are purely N repeating from start to finish (^N*$)
#also put species name in fasta header
cat ../samples.txt | while read sample ; do cat sequence.list | while read gene chr start stop ; do grep -v "^N*$" sequences/"$sample"_"$gene".fa | sed "s/chr.*$/$sample/g" > temp && mv temp sequences/"$sample"_"$gene".fa ; done ; done

#for VMED chromosome samples
#cat ../samples.txt | while read sample ; do cat sequence.list | while read gene chr start stop ; do grep -v "^N*$" "$sample"_"$gene".fa | sed "s/VMED.........1/$sample/g" > temp && mv temp "$sample"_"$gene".fa ; done ; done

#remove Uria
mkdir -p trash
mv sequences/Uria*fa trash

#put all samples together
cat sequence.list | while read gene chr start stop ; do cat ./sequences/*_"$gene".fa > ./sequences/Stercorariidae."$gene".fa ; done

#align
mkdir -p alignments
cat sequence.list | cut -f 1 | parallel 'time /home/0_PROGRAMS/mafft-7.490-with-extensions/bin/mafft --auto --thread 1 sequences/Stercorariidae.{1}.fa > ./alignments/Stercorariidae.{1}.mafft.afa'

#phylogeny
cat sequence.list | cut -f 1 | parallel time /home/0_PROGRAMS/iqtree-2.1.3-Linux/bin/iqtree2 -s alignments/Stercorariidae.{1}.mafft.afa -o Alca_torda -nt 1 -redo 

time /home/0_PROGRAMS/iqtree-2.1.3-Linux/bin/iqtree2 -s alignments/Stercorariidae.MRPL16.mafft.afa -o Alca_torda -nt 1 -redo 
```
Took up to 46m27.352s per gene to make the phylogeny.  


## Examine alignments.

We have 8 genes that show the patterns that would be consistent with introgression from *S. pomarinus* to *S. skua*: complex I: NDUFS2, NDUFS3, NDUFS5, NDUFS7, FOXRED1; complex III: TTC19; complex V: ATP5PD; ribosome: MRPL32.  

Now I will look at the coding sequences for these loci to determine whether they hold any amino acid differences between *S. skua* and the southern hemisphere taxa.  

#### Complex I  
* FOXRED1 is very difficult to identify the intron-exon boundaries in the 5' end as it appears to be poorly conserved in birds. After AA 67, it is more conserved. There are no AA variants in the focal group in AA 67 to the end. In the first section of the transcript, there is a site where one GRSK and one CISK have a mutation and a site where ANSK01 has a mutation (cannot tell whether these are synonymous or nonsynonymous), nevertheless these would not match the mitonuclear pattern expected.  
* NDUFS7: it is difficult to infer the 5' end of the protein since the references for *Calidris* and *Charadrius* start in different places (*Calidris* is shorter and does not start with AUG, but the *Charadrius* sequence would imply frameshifts in the Stercorariids, so it looks like the *Charadrius* 5' end is not right for the *Stercorarius*, but it might be truncated as it does not have a start codon). In the region that I was able to examine, there is only V->L in two of the *S. pomarinus*, not shared with *S. skua*. No evidence for mitonuclear inpompatibilities. Note that the NCBI annotation would place it from chr21:4015146-4018281, and we looked at 4015167-4018281, so we may have missed 21 bp. I looked in this region as well, and there were no variants that matched the mtDNA pattern.  
* NDUFS5: it looks like the *S. parasiticus* reference was not assembled correctly in the middle of this protein - there is an 8 bp deletion (which would induce a frameshift), and the skuas do not align across the deletion, so it looks like it is not a real deletion but rather an assembly error. I am unable to observe the state of 3 amino acids due to this error. There are no AA variants in the focal group except for the 5 AA leading up to the indel, but none of those 5 follow the mtDNA pattern (2 are private to POJA_MKP1559, one is in ANSK7+CISK55, and 2 are segregating in S. skua, pomarinus, and southern hemisphere).  
* NDUFS3: missing the first 51 AA in the first alignment. No variants in the focal group for AA 52-239. It looks like *Calidris* was annotated with a start codon that is not ATG in the Stercorariids, so I redid the search using the sequence of *Charadrius* and was able to extend it so it is missing just 19 AA. Still no Stercorariid variants! The *Stercorarius parasiticus* annotation from NCBI is for chr5:23560769-23563698, which is less than the region we identified, so it is uncertain exactly where the start codon for this gene is in *Stercorarius*, perhaps it was not assembled in *Stercorarius parasiticus*. I blasted the protein to the *Alca torda* genome, and it was also missing the first few amino acids, so perhaps the NCBI annotation is correct and *Stercorarius* really does have a shorter protein than the sandpipers - we would need RNAseq data to know this. Looking farther upstream, I found a region that may be homologous, though it is hard to tell as it has high GC content and relatively low sequence identity. Regardless, it has no variants in *Catharacta*, except a possible private allele in one *S. skua*.  
* NDUFS2: The 5' end of the Stercorariid sequences do not have the ATG start codon that is annotated as the start codon in *Calidris* - without RNA sequences I do not know whether the new start codon is farther upstream or downstream. The NCBI annotation places it farther upstream. The 3' end appears to be truncated in the genome assembly: it is missing the last 10 amino acids annotated in *Calidris* (CAGGACATCGTCTTCGGGGAAGTGGATCGGTGA), but it is unclear whether this is truly missing or if the Stercorariid sequence is different than the *Calidris* prediction. Within the whole sequence, there are no variants in the entire set of *Catharacta*.  

#### Complex III  
* TTC19 has an amico acid variant (A68T) in all three S. antarcticus.  
#### Complex V  
* ATP5PD has an amino acid variant (I5V) in ANSK7 and ANSK8 (but not ANSK01). Since it is polymorphic in S. maccormicki, it is unlikely to be involved in a mitonuclear incompatibility.   
#### Ribosome  
* MRPL32: all the southern skuas have V47A, but so does POJA_MKP1559. POJA_MKP1559 also has L134P.  

## Conclusion
None of the mitonuclear genes show a pattern that could be consistent with mitonuclear co-introgession (i.e., a phylogeny that could be consistent with introgression, and an amino acid substitution that could possibly have been selected on).  
