### Overview
This pipeline describes how to extract the W-chromosome homolog genes on the Z chromosome.  

### Required Input
This step requires a dataset of reference proteins to extract from a reference genome, and a bam file of sequences mapped to this reference.  

### Output
The output is a series of alignments for genes of interest, and phylogenies for these loci.  

I would like to see if the homologs of W-linked genes are more likely to have patterns cosistent with introgression from *S. pomarinus* into *S. skua* compared to the background levels in the genome. The W chromosome is genetically linked to the mitochondrion, since both are inherited maternally. Since the mtDNA has introgressed, the W chromosome must also have introgressed. The W chromosome contains genes, and most of these functional genes contain homologs on chrZ. It is conceivable that the two copies could need to co-evolve, and that incompatibilities could arise between chrW and chrZ alleles in different lineages. If these incompatibilities arose between *S. pomarinus* and *Catharacta*, we might expect that the *S. pomarinus* chrZ allele would have introgressed along with the chrW.  

I acquired the CDS sequences of all the 44 sequences located on the *Ficedula albicollis* Z chromosome with homologs on the W. I downloded them using Biomart (http://uswest.ensembl.org/biomart/martview/9058e5a21ed55ede472426f83d58b1ad) by specifying the Ensembl sequence IDs listed in the paper [ Evolutionary analysis of the female-specific avian W chromosome](https://www.nature.com/articles/ncomms8330/tables/1). Note that this species is fairly divergent from the Stercoraiidae, but few W chromosomes have been analyzed at the time of writing in detail in birds and *Ficedula* is closer than *Gallus*.  

## Identify locations of genes in reference  

First set up with a list of samples and a list of genes. I obtained a file `Ficalb_Wgenes.fasta` from Biomart with all the *Ficedula albicollis* W-chromosome homologs on the Z (contents of this file in the appendix at the bottom of this file).  
```bash
mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.6_W_Z_homologs
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.6_W_Z_homologs

cat > samples.txt
```
```
LTJA_MKP990
PAJA_B20730
PAJA_USNM606730
POJA_MKP1559
POJA_4
POJA_IB2659
GRSK_MKP1592
GRSK_MKP1593
CISK2
CISK55
CISK3
CHSK_MKP2451
ANSK7
ANSK01
ANSK8
Alca_torda
Uria_lomvia
```
```bash
cat > genes.list
```
```
ENSFALG00000000978
ENSFALG00000002006
ENSFALG00000002013
ENSFALG00000002058
ENSFALG00000002143
ENSFALG00000002149
ENSFALG00000002160
ENSFALG00000002587
ENSFALG00000002876
ENSFALG00000003294
ENSFALG00000004845
ENSFALG00000005137
ENSFALG00000006359
ENSFALG00000008875
ENSFALG00000008886
ENSFALG00000008934
ENSFALG00000008943
ENSFALG00000008945
ENSFALG00000009068
ENSFALG00000009780
ENSFALG00000009835
ENSFALG00000009864
ENSFALG00000009866
ENSFALG00000009887
ENSFALG00000009894
ENSFALG00000009935
ENSFALG00000010073
ENSFALG00000010292
ENSFALG00000010312
ENSFALG00000010471
ENSFALG00000010499
ENSFALG00000010515
ENSFALG00000010536
ENSFALG00000010733
ENSFALG00000010790
ENSFALG00000010987
ENSFALG00000011056
ENSFALG00000011073
ENSFALG00000011128
ENSFALG00000012406
ENSFALG00000012478
ENSFALG00000012745
ENSFALG00000014613
ENSFALG00000014649
```
Then, I use tblastn to locate these sequences in the reference genome.  
```bash
mkdir -p /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.6_W_Z_homologs
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.6_W_Z_homologs

conda activate fasplit
mkdir -p W_genes
faSplit byname Ficalb_Wgenes.fasta W_genes/

mkdir -p blast_hits_Stercorarius_parasiticus
cat genes.list | parallel time /home/0_PROGRAMS/ncbi-blast-2.12.0+/bin/tblastn -query W_genes/{1}*.fa -db /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/0_Reference_genome/0_reference/Stercorarius_parasiticus.ref.fa -outfmt '"6 qseqid sseqid evalue bitscore pident qstart qend sstart send qseq sseq"' ">" blast_hits_Stercorarius_parasiticus/{1}.tsv
#read the results (I am filtering to only look at the Z chromosome)
cat genes.list | while read gene ; do echo "" ; echo "" ; echo "$gene" ; cat blast_hits_Stercorarius_parasiticus/"$gene".tsv | cut -f 1-9 | sort -n -k 8 ; done

```

Make a text file with 4 columns: 1) Gene name 2) chromosome name 3) start position 4) end position  

```bash
cat > sequence.list
```
```
RNF38	chrZ	49631168	49725121
RPL37	chrZ	15297090	15299631
PRKAA1	chrZ	15249830	15272366
NIPBL	chrZ	13677084	13773871
SUB1	chrZ	12096704	12104906
ZFR	chrZ	12039942	12084272
GOLPH3	chrZ	11941430	11952193
FEM1C	chrZ	62084858	62099736
ZNF462	chrZ	53301831	53352567
ENSFALG00000003294.1	chrZ	40685751	40769470
HINT1	chrZ	48862027	48863064
KCMF1	chrZ	51552717	51566024
UBQLN1	chrZ	42867961	42881021
ATP5F1A	chrZ	4249695	4256103
C18orf25	chrZ	4183301	4212186
SMAD2	chrZ	3778952	3818086
CTIF	chrZ	3381118	3503065
SMAD7	chrZ	3331904	3352777
ARRDC3	chrZ	79751787	79763458
VCP	chrZ	10647968	10664346
SREK1	chrZ	23470675	23503113
FAM219A	chrZ	9718670	9724795
MRPS36	chrZ	24551831	24555741
UBAP1	chrZ	9602559	9612827
DCAF12	chrZ	9551247	9575111
UBAP2	chrZ	9436256	9505462
CERT1	chrZ	26663820	26729381
TNPO1	chrZ	27823671	27873902
MAP1B	chrZ	28111049	28123305
RASA1	chrZ	78052910	78092863
CHD1	chrZ	55663997	55703455
RFX3	chrZ	29552087	29612402
CDC37L1	chrZ	30154552	30160752
ZFAND5	chrZ	38221718	38232660
SNX18	chrZ	18850117	18869666
MIER3	chrZ	19908350	19924775
ZSWIM6	chrZ	21643956	21754190
KIF2A	chrZ	21998103	22032207
ZNF131	chrZ	16002095	16014688
ENSFALG00000012406.1	chrZ	47146675	47170734
HNRNPK	chrZ	42972219	42986586
CKMT2	chrZ	75282705	75298535
NFIL3	chrZ	48354693	48356066
ENSFALG00000014649.1	chrZ	82795627	82796418
```
I expanded the range for a couple of them to capture missing exons  
```
SMAD2_2	chrZ	3326835	3818086
C18orf25_3	chrZ	4183301	4242186
```

## Extract sequences 

Now we will loop through this list: extracting the sequence from the genome alignment, aligning each sample, and then making a phylogeny.  
```bash
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.6_W_Z_homologs

mkdir -p extracted_seqs
cat samples.txt | while read sample ; do cat sequence.list | while read gene chr start stop ; do /home/0_PROGRAMS/angsd/angsd -dofasta 3 -r "$chr":"$start"-"$stop" -i ../../1_Data/1.3c_mapping/merged/"$sample".PAJAbowtie.marked.bam -out extracted_seqs/"$sample"_"$gene" ; done ; done
gunzip ./extracted_seqs/*.fa.gz

#remove N's. This script first removes any lines that are purely N repeating from start to finish (^N*$)
#also put species name in fasta header
cat samples.txt | while read sample ; do cat sequence.list | while read gene chr start stop ; do grep -v "^N*$" ./extracted_seqs/"$sample"_"$gene".fa | sed "s/chr.*$/$sample/g" > temp && mv temp ./extracted_seqs/"$sample"_"$gene".fa ; done ; done

#remove Uria
mkdir -p trash
mv ./extracted_seqs/Uria*fa trash

#put all samples together
cat sequence.list | while read gene chr start stop ; do cat ./extracted_seqs/*_"$gene".fa > ./extracted_seqs/Stercorariidae."$gene".fa ; done

#align
mkdir alignments
cat sequence.list | cut -f 1 | parallel 'time /home/0_PROGRAMS/mafft-7.490-with-extensions/bin/mafft --auto --thread 1 ./extracted_seqs/Stercorariidae.{1}.fa > alignments/Stercorariidae.{1}.mafft.afa'

#phylogeny
cat sequence.list | cut -f 1 | parallel time /home/0_PROGRAMS/iqtree-2.1.3-Linux/bin/iqtree2 -s alignments/Stercorariidae.{1}.mafft.afa -o Alca_torda -nt 1 -redo 

```

### Analyze results: Might chrW homologs on chrZ have introgressed with the chrW?  

Of all these genes, only 3 (MIER3, SMAD7, and C18orf25) have tree topologies that are consistent with what we would expect to see from introgression (see mitonuclear section). Could this have been driven by selection against incompatibility between chrW and chrZ homologs with different protein sequences? I aligned the sequences with coding sequences from *Ficedula* in order to identify intron/exon boundaries and trimmed out the introns. Then, I looked at the encoded protein sequences. None of the three genes show differences between *S. skua* and the southern hemisphere species; therefore, we do not have evidence for selective introgression of these genes based on protein sequence. Of course, we cannot know what the ancestral protein sequence of the *S. skua* were before they captured the *S. pomarinus* chrW, and we cannot rule out selection for introgression based on features other than protein sequence, so we cannot state that they did *not* selectively introgress - we simply have no evidence of it.  

Note that the isoform of C18orf25 in the dataset corresponds to `ENSFALT00000009308.2`; I was able to recover this isoform in its entirety, but could not recover the 5' exon of the longer isoform `ENSFALT00000034045.1` even when I extended the alignment almost all the way to the next gene (Haus1). Perhaps Stercorariids do not have that alternative isoform.  





# Appendix: *Alca* reference

This would use *Alca torda* as the reference. I am not using this pipeline but have kept it for my records.

```bash
mkdir /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.6d_W_Z_homologs
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.6d_W_Z_homologs
cp ../3.6_W_Z_homologs/samples.txt .
mkdir -p blast_hits_Alca_torda
/home/0_PROGRAMS/ncbi-blast-2.12.0+/bin/makeblastdb -dbtype nucl -in /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/0_Reference_genome/0.1_scaffolding/GCA_008658365.1/data/GCA_008658365.1/GCA_008658365.1_bAlcTor1_genomic.fna
cat genes.list | parallel time /home/0_PROGRAMS/ncbi-blast-2.12.0+/bin/tblastn -query ../3.6_W_Z_homologs/W_genes/{1}*.fa -db /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/0_Reference_genome/0.1_scaffolding/GCA_008658365.1/data/GCA_008658365.1/GCA_008658365.1_bAlcTor1_genomic.fna -outfmt '"6 qseqid sseqid evalue bitscore pident qstart qend sstart send qseq sseq"' ">" blast_hits_Alca_torda/{1}.tsv
cat genes.list | while read gene ; do echo "" ; echo "" ; echo "$gene" ; cat blast_hits_Alca_torda/"$gene".tsv | cut -f 1-9 | sort -n -k 8 ; done
```
```bash
cat > genes.list
```
```bash
cat > sequence.list
```
```bash
cd /home/0_GENOMES1/0_RESEQUENCING_PROJECTS/2_PROJECTS/Stercorarius/3_introgression/3.6d_W_Z_homologs
mkdir -p extracted_seqs
cat samples.txt | while read sample ; do cat sequence.list | while read gene chr start stop ; do /home/0_PROGRAMS/angsd/angsd -dofasta 3 -r "$chr":"$start"-"$stop" -i ../../1_Data/1.3d_mapping/merged/"$sample".RAZObowtie.marked.bam -out extracted_seqs/"$sample"_"$gene" ; done ; done
gunzip ./extracted_seqs/*.fa.gz
cat samples.txt | while read sample ; do cat sequence.list | while read gene chr start stop ; do grep -v "^N*$" ./extracted_seqs/"$sample"_"$gene".fa | sed "s/chr.*$/$sample/g" > temp && mv temp ./extracted_seqs/"$sample"_"$gene".fa ; done ; done
mkdir -p trash
mv ./extracted_seqs/Uria*fa trash
cat sequence.list | while read gene chr start stop ; do cat ./extracted_seqs/*_"$gene".fa > ./extracted_seqs/Stercorariidae."$gene".fa ; done
mkdir alignments
cat sequence.list | cut -f 1 | parallel 'time /home/0_PROGRAMS/mafft-7.490-with-extensions/bin/mafft --auto --thread 1 ./extracted_seqs/Stercorariidae.{1}.fa > alignments/Stercorariidae.{1}.mafft.afa'
cat sequence.list | cut -f 1 | parallel time /home/0_PROGRAMS/iqtree-2.1.3-Linux/bin/iqtree2 -s alignments/Stercorariidae.{1}.mafft.afa -o Alca_torda -nt 1 -redo 
```
